<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>csb.bio.io.fasta</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="csb-module.html">Package&nbsp;csb</a> ::
        <a href="csb.bio-module.html">Package&nbsp;bio</a> ::
        <a href="csb.bio.io-module.html">Package&nbsp;io</a> ::
        Module&nbsp;fasta
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="csb.bio.io.fasta-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="csb.bio.io.fasta-module.html" onclick="show_private();">Module csb.bio.io.fasta</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-docstring">FASTA format sequence I/O.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-docstring">This module provides parsers and writers for sequences and alignments in</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-docstring">FASTA format. The most basic usage is:</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; parser = SequenceParser()</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; parser.parse_file('sequences.fa')</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;SequenceCollection&gt;   # collection of L{AbstractSequence}s</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">This will load all sequences in memory. If you are parsing a huge file,</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">then you could efficiently read the file sequence by sequence:</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; for seq in parser.read('sequences.fa'):</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">            ...            # seq is an L{AbstractSequence}</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">            </tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">L{BaseSequenceParser} is the central class in this module, which defines a</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">common infrastructure for all sequence readers. L{SequenceParser} is a standard</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">implementation, and L{PDBSequenceParser} is specialized to read FASTA sequences</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">with PDB headers.</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">For parsing alignments, have a look at L{SequenceAlignmentReader} and</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">L{StructureAlignmentFactory}.</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">Finally, this module provides a number of L{OutputBuilder}s, which know how to</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">write L{AbstractSequence} and L{AbstractAlignment} objects to FASTA files:</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open('file.fa', 'w') as out:</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">            builder = OutputBuilder.create(AlignmentFormats.FASTA, out)</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">            builder.add_alignment(alignment)</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">            builder.add_sequence(sequence)</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">            ...</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">or you could instantiate any of the L{OutputBuilder}s directly.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Package csb=csb-module.html"><a title="csb" class="py-name" href="#" onclick="return doclink('link-0', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package csb.bio.io=csb.bio.io-module.html,Package csb.io=csb.io-module.html"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-1', 'io', 'link-1');">io</a></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-2" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-2', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package csb.core=csb.core-module.html"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-3', 'core', 'link-3');">core</a></tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">abc</tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Class abc.ABCMeta=abc.ABCMeta-class.html"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-4', 'ABCMeta', 'link-4');">ABCMeta</a></tt><tt class="py-op">,</tt> <tt class="py-name">abstractmethod</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-5', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package csb.bio=csb.bio-module.html"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-6', 'bio', 'link-6');">bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Variable csb.bio.fragments.Assignment.sequence=csb.bio.fragments.Assignment-class.html#sequence,Variable csb.bio.fragments.Target.sequence=csb.bio.fragments.Target-class.html#sequence,Package csb.bio.sequence=csb.bio.sequence-module.html,Variable csb.bio.sequence.AbstractSequence.sequence=csb.bio.sequence.AbstractSequence-class.html#sequence,Variable csb.bio.sequence.Sequence.sequence=csb.bio.sequence.Sequence-class.html#sequence,Variable csb.bio.sequence.SequenceAdapter.sequence=csb.bio.sequence.SequenceAdapter-class.html#sequence,Variable csb.bio.structure.Chain.sequence=csb.bio.structure.Chain-class.html#sequence"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-7', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Class csb.bio.sequence.SequenceTypes=csb.bio.sequence.SequenceTypes-class.html"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-8', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Class csb.bio.sequence.SequenceAlphabets=csb.bio.sequence.SequenceAlphabets-class.html"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-9', 'SequenceAlphabets', 'link-9');">SequenceAlphabets</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class csb.bio.sequence.AlignmentFormats=csb.bio.sequence.AlignmentFormats-class.html"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-10', 'AlignmentFormats', 'link-10');">AlignmentFormats</a></tt><tt class="py-op">,</tt> <tt id="link-11" class="py-name" targets="Class csb.bio.sequence.SequenceError=csb.bio.sequence.SequenceError-class.html"><a title="csb.bio.sequence.SequenceError" class="py-name" href="#" onclick="return doclink('link-11', 'SequenceError', 'link-11');">SequenceError</a></tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-12', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-13', 'bio', 'link-6');">bio</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-14', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Class csb.bio.sequence.SequenceAlignment=csb.bio.sequence.SequenceAlignment-class.html"><a title="csb.bio.sequence.SequenceAlignment" class="py-name" href="#" onclick="return doclink('link-15', 'SequenceAlignment', 'link-15');">SequenceAlignment</a></tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name" targets="Class csb.bio.sequence.StructureAlignment=csb.bio.sequence.StructureAlignment-class.html"><a title="csb.bio.sequence.StructureAlignment" class="py-name" href="#" onclick="return doclink('link-16', 'StructureAlignment', 'link-16');">StructureAlignment</a></tt><tt class="py-op">,</tt> <tt id="link-17" class="py-name" targets="Class csb.bio.sequence.A3MAlignment=csb.bio.sequence.A3MAlignment-class.html"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-17', 'A3MAlignment', 'link-17');">A3MAlignment</a></tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-18" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-18', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-19', 'bio', 'link-6');">bio</a></tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-20', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">import</tt> <tt id="link-21" class="py-name" targets="Class csb.bio.sequence.SequenceCollection=csb.bio.sequence.SequenceCollection-class.html"><a title="csb.bio.sequence.SequenceCollection" class="py-name" href="#" onclick="return doclink('link-21', 'SequenceCollection', 'link-21');">SequenceCollection</a></tt><tt class="py-op">,</tt> <tt id="link-22" class="py-name" targets="Class csb.bio.sequence.AbstractSequence=csb.bio.sequence.AbstractSequence-class.html"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-22', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">,</tt> <tt id="link-23" class="py-name" targets="Class csb.bio.sequence.Sequence=csb.bio.sequence.Sequence-class.html"><a title="csb.bio.sequence.Sequence" class="py-name" href="#" onclick="return doclink('link-23', 'Sequence', 'link-23');">Sequence</a></tt><tt class="py-op">,</tt> <tt id="link-24" class="py-name" targets="Class csb.bio.sequence.RichSequence=csb.bio.sequence.RichSequence-class.html"><a title="csb.bio.sequence.RichSequence" class="py-name" href="#" onclick="return doclink('link-24', 'RichSequence', 'link-24');">RichSequence</a></tt><tt class="py-op">,</tt> <tt id="link-25" class="py-name" targets="Class csb.bio.sequence.ChainSequence=csb.bio.sequence.ChainSequence-class.html"><a title="csb.bio.sequence.ChainSequence" class="py-name" href="#" onclick="return doclink('link-25', 'ChainSequence', 'link-25');">ChainSequence</a></tt> </tt>
<a name="SequenceFormatError"></a><div id="SequenceFormatError-def"><a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt> <a class="py-toggle" href="#" id="SequenceFormatError-toggle" onclick="return toggle('SequenceFormatError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceFormatError-class.html">SequenceFormatError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceFormatError-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SequenceFormatError-expanded"><a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"> </tt>
<a name="BaseSequenceParser"></a><div id="BaseSequenceParser-def"><a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt> <a class="py-toggle" href="#" id="BaseSequenceParser-toggle" onclick="return toggle('BaseSequenceParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html">BaseSequenceParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BaseSequenceParser-expanded"><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">    FASTA parser template. Subclasses must implement the way FASTA strings are</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    handled by overriding C{BaseSequenceParser.read_sequence}.</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">    @param product: sequence product factory (an L{AbstractSequence} subclass)</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">    @type product: type</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">    @param product_type: default L{SequenceTypes} member for the products</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">    @type product_type: L{EnumItem}</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-26" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-26', 'ABCMeta', 'link-4');">ABCMeta</a></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="BaseSequenceParser.__init__"></a><div id="BaseSequenceParser.__init__-def"><a name="L63"></a><tt class="py-lineno"> 63</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.__init__-toggle" onclick="return toggle('BaseSequenceParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">product</tt><tt class="py-op">=</tt><tt id="link-27" class="py-name"><a title="csb.bio.sequence.Sequence" class="py-name" href="#" onclick="return doclink('link-27', 'Sequence', 'link-23');">Sequence</a></tt><tt class="py-op">,</tt> <tt class="py-param">product_type</tt><tt class="py-op">=</tt><tt id="link-28" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-28', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Protein=csb.bio.sequence.SequenceTypes-class.html#Protein"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-29', 'Protein', 'link-29');">Protein</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.__init__-expanded"><a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">         </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_product</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">         </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">issubclass</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name" targets="Variable csb.apps.hhfrag.PredictionBuilder.product=csb.apps.hhfrag.PredictionBuilder-class.html#product,Variable csb.statistics.samplers.mc.ShortTrajectoryBuilder.product=csb.statistics.samplers.mc.ShortTrajectoryBuilder-class.html#product,Variable csb.statistics.samplers.mc.TrajectoryBuilder.product=csb.statistics.samplers.mc.TrajectoryBuilder-class.html#product"><a title="csb.apps.hhfrag.PredictionBuilder.product
csb.statistics.samplers.mc.ShortTrajectoryBuilder.product
csb.statistics.samplers.mc.TrajectoryBuilder.product" class="py-name" href="#" onclick="return doclink('link-30', 'product', 'link-30');">product</a></tt><tt class="py-op">,</tt> <tt id="link-31" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-31', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-32" class="py-name"><a title="csb.apps.hhfrag.PredictionBuilder.product
csb.statistics.samplers.mc.ShortTrajectoryBuilder.product
csb.statistics.samplers.mc.TrajectoryBuilder.product" class="py-name" href="#" onclick="return doclink('link-32', 'product', 'link-30');">product</a></tt><tt class="py-op">)</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">         </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-33" class="py-name" targets="Variable csb.bio.io.fasta.BaseSequenceParser.product_type=csb.bio.io.fasta.BaseSequenceParser-class.html#product_type,Variable csb.bio.io.fasta.SequenceAlignmentReader.product_type=csb.bio.io.fasta.SequenceAlignmentReader-class.html#product_type,Variable csb.bio.io.fasta.StructureAlignmentFactory.product_type=csb.bio.io.fasta.StructureAlignmentFactory-class.html#product_type"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-33', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Variable csb.core.EnumItem.enum=csb.core.EnumItem-class.html#enum,Class csb.core.enum=csb.core.enum-class.html"><a title="csb.core.EnumItem.enum
csb.core.enum" class="py-name" href="#" onclick="return doclink('link-34', 'enum', 'link-34');">enum</a></tt> <tt class="py-keyword">is</tt> <tt id="link-35" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-35', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-36', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">         </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_product</tt> <tt class="py-op">=</tt> <tt id="link-37" class="py-name"><a title="csb.apps.hhfrag.PredictionBuilder.product
csb.statistics.samplers.mc.ShortTrajectoryBuilder.product
csb.statistics.samplers.mc.TrajectoryBuilder.product" class="py-name" href="#" onclick="return doclink('link-37', 'product', 'link-30');">product</a></tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt id="link-38" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-38', 'product_type', 'link-33');">product_type</a></tt> </tt>
</div><a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">         </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="BaseSequenceParser.product_factory"></a><div id="BaseSequenceParser.product_factory-def"><a name="L77"></a><tt class="py-lineno"> 77</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.product_factory-toggle" onclick="return toggle('BaseSequenceParser.product_factory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#product_factory">product_factory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.product_factory-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.product_factory-expanded"><a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">        Factory used to build sequence products</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: class</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_product</tt> </tt>
</div><a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">     </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="BaseSequenceParser.product_type"></a><div id="BaseSequenceParser.product_type-def"><a name="L85"></a><tt class="py-lineno"> 85</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.product_type-toggle" onclick="return toggle('BaseSequenceParser.product_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#product_type">product_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.product_type-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.product_type-expanded"><a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">        Default sequence type of the products - a member of L{SequenceTypes}</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: enum item</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> </tt>
</div><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">         </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="BaseSequenceParser.read_sequence"></a><div id="BaseSequenceParser.read_sequence-def"><a name="L93"></a><tt class="py-lineno"> 93</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.read_sequence-toggle" onclick="return toggle('BaseSequenceParser.read_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#read_sequence">read_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.read_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.read_sequence-expanded"><a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">        Parse a single FASTA string</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new sequence, created with L{BaseSequenceParser.product_factory}</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractSequence}</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceFormatError: on parse error</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"> </tt>
<a name="BaseSequenceParser.parse_string"></a><div id="BaseSequenceParser.parse_string-def"><a name="L104"></a><tt class="py-lineno">104</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.parse_string-toggle" onclick="return toggle('BaseSequenceParser.parse_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#parse_string">parse_string</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">fasta_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.parse_string-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.parse_string-expanded"><a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">        Read FASTA sequences from an (m)FASTA-formatted string</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">        @param fasta_string: FASTA string to parse</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">        @type fasta_string: str</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a list of L{Sequence}s</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceCollection}</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceFormatError: on parse error        </tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt id="link-39" class="py-name" targets="Variable csb.io.EntryWriter.stream=csb.io.EntryWriter-class.html#stream"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-39', 'stream', 'link-39');">stream</a></tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-40', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-41', 'io', 'link-1');">io</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Class csb.io.MemoryStream=csb.io.MemoryStream-class.html"><a title="csb.io.MemoryStream" class="py-name" href="#" onclick="return doclink('link-42', 'MemoryStream', 'link-42');">MemoryStream</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt id="link-43" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-43', 'stream', 'link-39');">stream</a></tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method csb.bio.io.clans.Clans.write()=csb.bio.io.clans.Clans-class.html#write,Method csb.bio.io.clans.ClansFileBuilder.write()=csb.bio.io.clans.ClansFileBuilder-class.html#write,Method csb.bio.io.fasta.OutputBuilder.write()=csb.bio.io.fasta.OutputBuilder-class.html#write,Method csb.bio.io.hhpred.HHMFileBuilder.write()=csb.bio.io.hhpred.HHMFileBuilder-class.html#write,Method csb.bio.io.mrc.DensityMapWriter.write()=csb.bio.io.mrc.DensityMapWriter-class.html#write,Method csb.bio.io.wwpdb.FileBuilder.write()=csb.bio.io.wwpdb.FileBuilder-class.html#write,Method csb.build.RevisionHandler.write()=csb.build.RevisionHandler-class.html#write,Method csb.io.AutoFlushStream.write()=csb.io.AutoFlushStream-class.html#write,Method csb.io.EntryWriter.write()=csb.io.EntryWriter-class.html#write"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-44', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">fasta_string</tt><tt class="py-op">)</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method csb.bio.io.clans.ClansParser.parse_file()=csb.bio.io.clans.ClansParser-class.html#parse_file,Method csb.bio.io.fasta.BaseSequenceParser.parse_file()=csb.bio.io.fasta.BaseSequenceParser-class.html#parse_file,Method csb.bio.io.hhpred.HHOutputParser.parse_file()=csb.bio.io.hhpred.HHOutputParser-class.html#parse_file"><a title="csb.bio.io.clans.ClansParser.parse_file
csb.bio.io.fasta.BaseSequenceParser.parse_file
csb.bio.io.hhpred.HHOutputParser.parse_file" class="py-name" href="#" onclick="return doclink('link-45', 'parse_file', 'link-45');">parse_file</a></tt><tt class="py-op">(</tt><tt id="link-46" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-46', 'stream', 'link-39');">stream</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="BaseSequenceParser.parse_file"></a><div id="BaseSequenceParser.parse_file-def"><a name="L122"></a><tt class="py-lineno">122</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.parse_file-toggle" onclick="return toggle('BaseSequenceParser.parse_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#parse_file">parse_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">fasta_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.parse_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.parse_file-expanded"><a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">        Read FASTA sequences from a (m)FASTA file</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">        @param fasta_file: input FASTA file name or opened stream</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        @type fasta_file: str, file</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a list of L{Sequence}s</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceCollection}</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceFormatError: on parse error        </tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">fasta_file</tt><tt class="py-op">,</tt> <tt id="link-47" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-47', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-48', 'core', 'link-3');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">            <tt id="link-49" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-49', 'stream', 'link-39');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">open</tt><tt class="py-op">(</tt><tt class="py-name">fasta_file</tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">            <tt id="link-50" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-50', 'stream', 'link-39');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">fasta_file</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">        <tt class="py-name">reader</tt> <tt class="py-op">=</tt> <tt id="link-51" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-51', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-52', 'io', 'link-1');">io</a></tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Class csb.io.EntryReader=csb.io.EntryReader-class.html"><a title="csb.io.EntryReader" class="py-name" href="#" onclick="return doclink('link-53', 'EntryReader', 'link-53');">EntryReader</a></tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-54', 'stream', 'link-39');">stream</a></tt><tt class="py-op">,</tt> <tt id="link-55" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-55', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-56" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.DELIMITER=csb.bio.io.wwpdb.FastResidueMapper-class.html#DELIMITER,Variable csb.bio.sequence.AbstractSequence.DELIMITER=csb.bio.sequence.AbstractSequence-class.html#DELIMITER"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-56', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">entry</tt> <tt class="py-keyword">in</tt> <tt class="py-name">reader</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Variable csb.bio.io.clans.Clans.entries=csb.bio.io.clans.Clans-class.html#entries,Method csb.io.EntryReader.entries()=csb.io.EntryReader-class.html#entries"><a title="csb.bio.io.clans.Clans.entries
csb.io.EntryReader.entries" class="py-name" href="#" onclick="return doclink('link-57', 'entries', 'link-57');">entries</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-name">seqs</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Method csb.bio.fragments.rosetta.RosettaFragmentMap.append()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#append,Method csb.bio.hmm.EmissionTable.append()=csb.bio.hmm.EmissionTable-class.html#append,Method csb.bio.hmm.HMMLayer.append()=csb.bio.hmm.HMMLayer-class.html#append,Method csb.bio.hmm.TransitionTable.append()=csb.bio.hmm.TransitionTable-class.html#append,Method csb.bio.structure.ChainResiduesCollection.append()=csb.bio.structure.ChainResiduesCollection-class.html#append,Method csb.bio.structure.DisorderedAtom.append()=csb.bio.structure.DisorderedAtom-class.html#append,Method csb.bio.structure.EnsembleModelsCollection.append()=csb.bio.structure.EnsembleModelsCollection-class.html#append,Method csb.bio.structure.ResidueAtomsTable.append()=csb.bio.structure.ResidueAtomsTable-class.html#append,Method csb.bio.structure.SecondaryStructure.append()=csb.bio.structure.SecondaryStructure-class.html#append,Method csb.bio.structure.StructureChainsTable.append()=csb.bio.structure.StructureChainsTable-class.html#append,Method csb.core.CollectionContainer.append()=csb.core.CollectionContainer-class.html#append,Method csb.core.DictionaryContainer.append()=csb.core.DictionaryContainer-class.html#append"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-58', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method csb.bio.io.fasta.BaseSequenceParser.read_sequence()=csb.bio.io.fasta.BaseSequenceParser-class.html#read_sequence,Method csb.bio.io.fasta.PDBSequenceParser.read_sequence()=csb.bio.io.fasta.PDBSequenceParser-class.html#read_sequence,Method csb.bio.io.fasta.SequenceParser.read_sequence()=csb.bio.io.fasta.SequenceParser-class.html#read_sequence"><a title="csb.bio.io.fasta.BaseSequenceParser.read_sequence
csb.bio.io.fasta.PDBSequenceParser.read_sequence
csb.bio.io.fasta.SequenceParser.read_sequence" class="py-name" href="#" onclick="return doclink('link-59', 'read_sequence', 'link-59');">read_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">entry</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-60" class="py-name"><a title="csb.bio.sequence.SequenceCollection" class="py-name" href="#" onclick="return doclink('link-60', 'SequenceCollection', 'link-21');">SequenceCollection</a></tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"> </tt>
<a name="BaseSequenceParser.read"></a><div id="BaseSequenceParser.read-def"><a name="L147"></a><tt class="py-lineno">147</tt> <a class="py-toggle" href="#" id="BaseSequenceParser.read-toggle" onclick="return toggle('BaseSequenceParser.read');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.BaseSequenceParser-class.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">fasta_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BaseSequenceParser.read-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BaseSequenceParser.read-expanded"><a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">        Read FASTA sequences from an (m)FASTA file.</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"><tt class="py-docstring">        @param fasta_file: input FASTA file name or opened stream</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-docstring">        @type fasta_file: str, file</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"><tt class="py-docstring">        @return: efficient cursor over all L{Sequence}s (parse on demand)</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: iterator</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceFormatError: on parse error        </tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">fasta_file</tt><tt class="py-op">,</tt> <tt id="link-61" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-61', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-62', 'core', 'link-3');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">            <tt id="link-63" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-63', 'stream', 'link-39');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">open</tt><tt class="py-op">(</tt><tt class="py-name">fasta_file</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">            <tt id="link-64" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-64', 'stream', 'link-39');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">fasta_file</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-name">reader</tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-65', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-66', 'io', 'link-1');">io</a></tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="csb.io.EntryReader" class="py-name" href="#" onclick="return doclink('link-67', 'EntryReader', 'link-53');">EntryReader</a></tt><tt class="py-op">(</tt><tt id="link-68" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-68', 'stream', 'link-39');">stream</a></tt><tt class="py-op">,</tt> <tt id="link-69" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-69', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-70', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">entry</tt> <tt class="py-keyword">in</tt> <tt class="py-name">reader</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="csb.bio.io.clans.Clans.entries
csb.io.EntryReader.entries" class="py-name" href="#" onclick="return doclink('link-71', 'entries', 'link-57');">entries</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.read_sequence
csb.bio.io.fasta.PDBSequenceParser.read_sequence
csb.bio.io.fasta.SequenceParser.read_sequence" class="py-name" href="#" onclick="return doclink('link-72', 'read_sequence', 'link-59');">read_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">entry</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">     </tt>
<a name="SequenceParser"></a><div id="SequenceParser-def"><a name="L169"></a><tt class="py-lineno">169</tt> <a class="py-toggle" href="#" id="SequenceParser-toggle" onclick="return toggle('SequenceParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceParser-class.html">SequenceParser</a><tt class="py-op">(</tt><tt class="py-base-class">BaseSequenceParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SequenceParser-expanded"><a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring">    Standard FASTA parser. See L{BaseSequenceParser} for details.</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">     </tt>
<a name="SequenceParser.read_sequence"></a><div id="SequenceParser.read_sequence-def"><a name="L174"></a><tt class="py-lineno">174</tt> <a class="py-toggle" href="#" id="SequenceParser.read_sequence-toggle" onclick="return toggle('SequenceParser.read_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceParser-class.html#read_sequence">read_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceParser.read_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceParser.read_sequence-expanded"><a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name" targets="Method csb.bio.sequence.AbstractSequence.strip()=csb.bio.sequence.AbstractSequence-class.html#strip,Method csb.bio.sequence.Sequence.strip()=csb.bio.sequence.Sequence-class.html#strip,Method csb.bio.sequence.SequenceAdapter.strip()=csb.bio.sequence.SequenceAdapter-class.html#strip"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-73', 'strip', 'link-73');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">splitlines</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">         </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt id="link-74" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-74', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-75', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">lines</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-76" class="py-name" targets="Class csb.bio.io.fasta.SequenceFormatError=csb.bio.io.fasta.SequenceFormatError-class.html"><a title="csb.bio.io.fasta.SequenceFormatError" class="py-name" href="#" onclick="return doclink('link-76', 'SequenceFormatError', 'link-76');">SequenceFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Empty FASTA entry'</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt id="link-77" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.header=csb.bio.sequence.AbstractSequence-class.html#header,Variable csb.bio.sequence.SequenceAdapter.header=csb.bio.sequence.SequenceAdapter-class.html#header,Variable csb.bio.structure.Chain.header=csb.bio.structure.Chain-class.html#header"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-77', 'header', 'link-77');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt id="link-78" class="py-name" targets="Variable csb.bio.fragments.FragmentMatch.id=csb.bio.fragments.FragmentMatch-class.html#id,Variable csb.bio.fragments.Target.id=csb.bio.fragments.Target-class.html#id,Variable csb.bio.fragments.rosetta.RosettaFragment.id=csb.bio.fragments.rosetta.RosettaFragment-class.html#id,Variable csb.bio.hmm.HHpredHit.id=csb.bio.hmm.HHpredHit-class.html#id,Variable csb.bio.hmm.ProfileHMM.id=csb.bio.hmm.ProfileHMM-class.html#id,Variable csb.bio.io.wwpdb.ResidueInfo.id=csb.bio.io.wwpdb.ResidueInfo-class.html#id,Variable csb.bio.sequence.AbstractSequence.id=csb.bio.sequence.AbstractSequence-class.html#id,Variable csb.bio.sequence.SequenceAdapter.id=csb.bio.sequence.SequenceAdapter-class.html#id,Variable csb.bio.structure.Chain.id=csb.bio.structure.Chain-class.html#id,Variable csb.bio.structure.Residue.id=csb.bio.structure.Residue-class.html#id"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-78', 'id', 'link-78');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-79', 'header', 'link-77');">header</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt id="link-80" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-80', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name" targets="Static Method csb.bio.nmr.NOESpectrum.join()=csb.bio.nmr.NOESpectrum-class.html#join"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-81', 'join', 'link-81');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Variable csb.bio.io.fasta.BaseSequenceParser.product_factory=csb.bio.io.fasta.BaseSequenceParser-class.html#product_factory"><a title="csb.bio.io.fasta.BaseSequenceParser.product_factory" class="py-name" href="#" onclick="return doclink('link-82', 'product_factory', 'link-82');">product_factory</a></tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-83', 'id', 'link-78');">id</a></tt><tt class="py-op">,</tt> <tt id="link-84" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-84', 'header', 'link-77');">header</a></tt><tt class="py-op">,</tt> <tt id="link-85" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-85', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-86', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">     </tt>
<a name="PDBSequenceParser"></a><div id="PDBSequenceParser-def"><a name="L189"></a><tt class="py-lineno">189</tt> <a class="py-toggle" href="#" id="PDBSequenceParser-toggle" onclick="return toggle('PDBSequenceParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.PDBSequenceParser-class.html">PDBSequenceParser</a><tt class="py-op">(</tt><tt class="py-base-class">SequenceParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBSequenceParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="PDBSequenceParser-expanded"><a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"><tt class="py-docstring">    PDB FASTA parser. Reads the PDB ID and sequence type from the header.</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"><tt class="py-docstring">    See L{BaseSequenceParser} for more details.</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">         </tt>
<a name="PDBSequenceParser.read_sequence"></a><div id="PDBSequenceParser.read_sequence-def"><a name="L195"></a><tt class="py-lineno">195</tt> <a class="py-toggle" href="#" id="PDBSequenceParser.read_sequence-toggle" onclick="return toggle('PDBSequenceParser.read_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PDBSequenceParser-class.html#read_sequence">read_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBSequenceParser.read_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PDBSequenceParser.read_sequence-expanded"><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt id="link-87" class="py-name" targets="Variable csb.bio.io.clans.ClansEntry.seq=csb.bio.io.clans.ClansEntry-class.html#seq"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-87', 'seq', 'link-87');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-88" class="py-name" targets="Class csb.bio.io.fasta.PDBSequenceParser=csb.bio.io.fasta.PDBSequenceParser-class.html"><a title="csb.bio.io.fasta.PDBSequenceParser" class="py-name" href="#" onclick="return doclink('link-88', 'PDBSequenceParser', 'link-88');">PDBSequenceParser</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.read_sequence
csb.bio.io.fasta.PDBSequenceParser.read_sequence
csb.bio.io.fasta.SequenceParser.read_sequence" class="py-name" href="#" onclick="return doclink('link-89', 'read_sequence', 'link-59');">read_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt id="link-90" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-90', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-91', 'header', 'link-77');">header</a></tt> <tt class="py-keyword">and</tt> <tt id="link-92" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-92', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-93', 'id', 'link-78');">id</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-94', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-95', 'id', 'link-78');">id</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt><tt class="py-op">(</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">6</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt id="link-96" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-96', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-97', 'header', 'link-77');">header</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name" targets="Method csb.bio.io.wwpdb.CustomStructureProvider.find()=csb.bio.io.wwpdb.CustomStructureProvider-class.html#find,Method csb.bio.io.wwpdb.FileSystemStructureProvider.find()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#find,Method csb.bio.io.wwpdb.RemoteStructureProvider.find()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#find,Method csb.bio.io.wwpdb.StructureProvider.find()=csb.bio.io.wwpdb.StructureProvider-class.html#find,Function csb.bio.io.wwpdb.find()=csb.bio.io.wwpdb-module.html#find,Method csb.bio.structure.Chain.find()=csb.bio.structure.Chain-class.html#find,Method csb.bio.structure.DisorderedAtom.find()=csb.bio.structure.DisorderedAtom-class.html#find"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-98', 'find', 'link-98');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">'mol:'</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-99" class="py-name"><a title="csb.bio.io.fasta.SequenceFormatError" class="py-name" href="#" onclick="return doclink('link-99', 'SequenceFormatError', 'link-76');">SequenceFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Does not look like a PDB header: {0}'</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method csb.bio.io.wwpdb.EntryID.format()=csb.bio.io.wwpdb.EntryID-class.html#format,Method csb.bio.io.wwpdb.SeqResID.format()=csb.bio.io.wwpdb.SeqResID-class.html#format,Method csb.bio.sequence.A3MAlignment.format()=csb.bio.sequence.A3MAlignment-class.html#format,Method csb.bio.sequence.AbstractAlignment.format()=csb.bio.sequence.AbstractAlignment-class.html#format"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-100', 'format', 'link-100');">format</a></tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-101', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-102', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt id="link-103" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-103', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-104', 'id', 'link-78');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-105" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-105', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-106', 'id', 'link-78');">id</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'_'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-name">stype</tt> <tt class="py-op">=</tt> <tt id="link-107" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-107', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-108', 'header', 'link-77');">header</a></tt><tt class="py-op">.</tt><tt class="py-name">partition</tt><tt class="py-op">(</tt><tt class="py-string">'mol:'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">partition</tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt id="link-109" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-109', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-110" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.type=csb.bio.fragments.ResidueEventInfo-class.html#type,Variable csb.bio.fragments.TargetResidue.type=csb.bio.fragments.TargetResidue-class.html#type,Variable csb.bio.hmm.State.type=csb.bio.hmm.State-class.html#type,Variable csb.bio.hmm.Transition.type=csb.bio.hmm.Transition-class.html#type,Variable csb.bio.io.clans.ClansSeqgroup.type=csb.bio.io.clans.ClansSeqgroup-class.html#type,Variable csb.bio.io.wwpdb.ResidueInfo.type=csb.bio.io.wwpdb.ResidueInfo-class.html#type,Variable csb.bio.sequence.AbstractSequence.type=csb.bio.sequence.AbstractSequence-class.html#type,Variable csb.bio.sequence.ResidueInfo.type=csb.bio.sequence.ResidueInfo-class.html#type,Variable csb.bio.sequence.SequenceAdapter.type=csb.bio.sequence.SequenceAdapter-class.html#type,Variable csb.bio.structure.Chain.type=csb.bio.structure.Chain-class.html#type,Variable csb.bio.structure.Residue.type=csb.bio.structure.Residue-class.html#type,Variable csb.bio.structure.SecondaryStructureElement.type=csb.bio.structure.SecondaryStructureElement-class.html#type,Variable csb.io.tsv.ColumnInfo.type=csb.io.tsv.ColumnInfo-class.html#type"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-110', 'type', 'link-110');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-111" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-111', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-112', 'core', 'link-3');">core</a></tt><tt class="py-op">.</tt><tt id="link-113" class="py-name" targets="Class csb.core.Enum=csb.core.Enum-class.html"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-113', 'Enum', 'link-113');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method csb.apps.ArgHandler.parse()=csb.apps.ArgHandler-class.html#parse,Method csb.bio.io.dssp.DSSPParser.parse()=csb.bio.io.dssp.DSSPParser-class.html#parse,Method csb.bio.io.dssp.StrideParser.parse()=csb.bio.io.dssp.StrideParser-class.html#parse,Method csb.bio.io.hhpred.HHProfileParser.parse()=csb.bio.io.hhpred.HHProfileParser-class.html#parse,Method csb.bio.io.procheck.ProcheckParser.parse()=csb.bio.io.procheck.ProcheckParser-class.html#parse,Method csb.bio.io.psipred.PSIPredParser.parse()=csb.bio.io.psipred.PSIPredParser-class.html#parse,Method csb.bio.io.vasco.VascoStructureParser.parse()=csb.bio.io.vasco.VascoStructureParser-class.html#parse,Method csb.bio.io.whatif.WhatCheckParser.parse()=csb.bio.io.whatif.WhatCheckParser-class.html#parse,Method csb.bio.io.wwpdb.AbstractStructureParser.parse()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse,Method csb.bio.io.wwpdb.DegenerateID.parse()=csb.bio.io.wwpdb.DegenerateID-class.html#parse,Method csb.bio.io.wwpdb.EntryID.parse()=csb.bio.io.wwpdb.EntryID-class.html#parse,Method csb.bio.io.wwpdb.SeqResID.parse()=csb.bio.io.wwpdb.SeqResID-class.html#parse,Method csb.bio.io.wwpdb.StandardID.parse()=csb.bio.io.wwpdb.StandardID-class.html#parse,Static Method csb.bio.nmr.Label.parse()=csb.bio.nmr.Label-class.html#parse,Static Method csb.bio.sequence.A3MAlignment.parse()=csb.bio.sequence.A3MAlignment-class.html#parse,Static Method csb.bio.sequence.SequenceAlignment.parse()=csb.bio.sequence.SequenceAlignment-class.html#parse,Static Method csb.bio.sequence.StructureAlignment.parse()=csb.bio.sequence.StructureAlignment-class.html#parse,Static Method csb.bio.sequence.alignment.SimilarityMatrix.parse()=csb.bio.sequence.alignment.SimilarityMatrix-class.html#parse,Static Method csb.bio.structure.SecondaryStructure.parse()=csb.bio.structure.SecondaryStructure-class.html#parse,Static Method csb.core.Enum.parse()=csb.core.Enum-class.html#parse"><a title="csb.apps.ArgHandler.parse
csb.bio.io.dssp.DSSPParser.parse
csb.bio.io.dssp.StrideParser.parse
csb.bio.io.hhpred.HHProfileParser.parse
csb.bio.io.procheck.ProcheckParser.parse
csb.bio.io.psipred.PSIPredParser.parse
csb.bio.io.vasco.VascoStructureParser.parse
csb.bio.io.whatif.WhatCheckParser.parse
csb.bio.io.wwpdb.AbstractStructureParser.parse
csb.bio.io.wwpdb.DegenerateID.parse
csb.bio.io.wwpdb.EntryID.parse
csb.bio.io.wwpdb.SeqResID.parse
csb.bio.io.wwpdb.StandardID.parse
csb.bio.nmr.Label.parse
csb.bio.sequence.A3MAlignment.parse
csb.bio.sequence.SequenceAlignment.parse
csb.bio.sequence.StructureAlignment.parse
csb.bio.sequence.alignment.SimilarityMatrix.parse
csb.bio.structure.SecondaryStructure.parse
csb.core.Enum.parse" class="py-name" href="#" onclick="return doclink('link-114', 'parse', 'link-114');">parse</a></tt><tt class="py-op">(</tt><tt id="link-115" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-115', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">,</tt> <tt class="py-name">stype</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-116" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-116', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-117', 'core', 'link-3');">core</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Class csb.core.EnumValueError=csb.core.EnumValueError-class.html"><a title="csb.core.EnumValueError" class="py-name" href="#" onclick="return doclink('link-118', 'EnumValueError', 'link-118');">EnumValueError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt id="link-119" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-119', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-120', 'type', 'link-110');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-121" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-121', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-122" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Unknown=csb.bio.sequence.SequenceTypes-class.html#Unknown"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-122', 'Unknown', 'link-122');">Unknown</a></tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-123" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-123', 'seq', 'link-87');">seq</a></tt> </tt>
</div></div><a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">     </tt>
<a name="A3MSequenceIterator"></a><div id="A3MSequenceIterator-def"><a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt> <a class="py-toggle" href="#" id="A3MSequenceIterator-toggle" onclick="return toggle('A3MSequenceIterator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MSequenceIterator-class.html">A3MSequenceIterator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MSequenceIterator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="A3MSequenceIterator-expanded"><a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">     </tt>
<a name="A3MSequenceIterator.__init__"></a><div id="A3MSequenceIterator.__init__-def"><a name="L214"></a><tt class="py-lineno">214</tt> <a class="py-toggle" href="#" id="A3MSequenceIterator.__init__-toggle" onclick="return toggle('A3MSequenceIterator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MSequenceIterator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">,</tt> <tt class="py-param">insertion</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MSequenceIterator.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MSequenceIterator.__init__-expanded"><a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">         </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_temp</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_insertion</tt> <tt class="py-op">=</tt> <tt class="py-name">insertion</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">         </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-124" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-124', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">             </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-125" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-125', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-126', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">            <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name" targets="Function csb.numeric.reverse()=csb.numeric-module.html#reverse"><a title="csb.numeric.reverse" class="py-name" href="#" onclick="return doclink('link-127', 'reverse', 'link-127');">reverse</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_temp</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-128', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt>   </tt>
</div><a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">     </tt>
<a name="A3MSequenceIterator.next"></a><div id="A3MSequenceIterator.next-def"><a name="L225"></a><tt class="py-lineno">225</tt> <a class="py-toggle" href="#" id="A3MSequenceIterator.next-toggle" onclick="return toggle('A3MSequenceIterator.next');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MSequenceIterator-class.html#next">next</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MSequenceIterator.next-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MSequenceIterator.next-expanded"><a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">         </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_temp</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt id="link-129" class="py-name" targets="Method csb.io.tsv.Table.column()=csb.io.tsv.Table-class.html#column"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-129', 'column', 'link-129');">column</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-op">]</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-name">has_insertion</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-130" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-130', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-131" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-131', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt> <tt class="py-keyword">and</tt> <tt id="link-132" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-132', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">islower</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">                <tt class="py-name">has_insertion</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-133" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-133', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">has_insertion</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt id="link-134" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-134', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">islower</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">                    <tt id="link-135" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-135', 'column', 'link-129');">column</a></tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-136', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_insertion</tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">                    <tt id="link-137" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-137', 'column', 'link-129');">column</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-138', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-139', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt class="py-name">pop</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">                <tt id="link-140" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-140', 'column', 'link-129');">column</a></tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-141', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">any</tt><tt class="py-op">(</tt><tt id="link-142" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-142', 'column', 'link-129');">column</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-143" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-143', 'column', 'link-129');">column</a></tt>   </tt>
</div><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">     </tt>
<a name="A3MSequenceIterator.__iter__"></a><div id="A3MSequenceIterator.__iter__-def"><a name="L250"></a><tt class="py-lineno">250</tt> <a class="py-toggle" href="#" id="A3MSequenceIterator.__iter__-toggle" onclick="return toggle('A3MSequenceIterator.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MSequenceIterator-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MSequenceIterator.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MSequenceIterator.__iter__-expanded"><a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">     </tt>
<a name="A3MSequenceIterator.__next__"></a><div id="A3MSequenceIterator.__next__-def"><a name="L253"></a><tt class="py-lineno">253</tt> <a class="py-toggle" href="#" id="A3MSequenceIterator.__next__-toggle" onclick="return toggle('A3MSequenceIterator.__next__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MSequenceIterator-class.html#__next__">__next__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MSequenceIterator.__next__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MSequenceIterator.__next__-expanded"><a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method csb.bio.io.fasta.A3MSequenceIterator.next()=csb.bio.io.fasta.A3MSequenceIterator-class.html#next,Method csb.bio.structure.CompositeEntityIterator.next()=csb.bio.structure.CompositeEntityIterator-class.html#next,Method csb.statistics.samplers.mc.propagators.Looper.next()=csb.statistics.samplers.mc.propagators.Looper-class.html#next"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-144', 'next', 'link-144');">next</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>     </tt>
</div></div><a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">     </tt>
<a name="SequenceAlignmentReader"></a><div id="SequenceAlignmentReader-def"><a name="L256"></a><tt class="py-lineno">256</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader-toggle" onclick="return toggle('SequenceAlignmentReader');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html">SequenceAlignmentReader</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SequenceAlignmentReader-expanded"><a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence alignment parser.</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"><tt class="py-docstring">    @param product_type: default L{SequenceTypes} member for the sequence products</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"><tt class="py-docstring">    @type product_type: L{EnumItem}</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line"><tt class="py-docstring">    @param strict: if True, raise exception on duplicate sequence identifiers.</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line"><tt class="py-docstring">                   See L{csb.bio.sequence.AbstractAlignment} for details</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"><tt class="py-docstring">    @type strict: bool    </tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">     </tt>
<a name="SequenceAlignmentReader.__init__"></a><div id="SequenceAlignmentReader.__init__-def"><a name="L267"></a><tt class="py-lineno">267</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader.__init__-toggle" onclick="return toggle('SequenceAlignmentReader.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">product_type</tt><tt class="py-op">=</tt><tt id="link-145" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-145', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-146', 'Protein', 'link-29');">Protein</a></tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceAlignmentReader.__init__-expanded"><a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">         </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-147" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-147', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="csb.core.EnumItem.enum
csb.core.enum" class="py-name" href="#" onclick="return doclink('link-148', 'enum', 'link-34');">enum</a></tt> <tt class="py-keyword">is</tt> <tt id="link-149" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-149', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">:</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-150', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">         </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt id="link-151" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-151', 'product_type', 'link-33');">product_type</a></tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> <tt class="py-op">=</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt id="link-152" class="py-name" targets="Variable csb.bio.io.fasta.SequenceAlignmentReader.strict=csb.bio.io.fasta.SequenceAlignmentReader-class.html#strict,Variable csb.bio.io.fasta.StructureAlignmentFactory.strict=csb.bio.io.fasta.StructureAlignmentFactory-class.html#strict"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-152', 'strict', 'link-152');">strict</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAlignmentReader.product_type"></a><div id="SequenceAlignmentReader.product_type-def"><a name="L276"></a><tt class="py-lineno">276</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader.product_type-toggle" onclick="return toggle('SequenceAlignmentReader.product_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html#product_type">product_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader.product_type-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceAlignmentReader.product_type-expanded"><a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring">        Default sequence type of the alignment entries - a member of L{SequenceTypes}</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: enum item</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> </tt>
</div><a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAlignmentReader.strict"></a><div id="SequenceAlignmentReader.strict-def"><a name="L284"></a><tt class="py-lineno">284</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader.strict-toggle" onclick="return toggle('SequenceAlignmentReader.strict');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html#strict">strict</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader.strict-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceAlignmentReader.strict-expanded"><a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"><tt class="py-docstring">        True if strict mode is enabled</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> </tt>
</div><a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">             </tt>
<a name="SequenceAlignmentReader.read_fasta"></a><div id="SequenceAlignmentReader.read_fasta-def"><a name="L291"></a><tt class="py-lineno">291</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader.read_fasta-toggle" onclick="return toggle('SequenceAlignmentReader.read_fasta');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html#read_fasta">read_fasta</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader.read_fasta-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceAlignmentReader.read_fasta-expanded"><a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line"><tt class="py-docstring">        Parse an alignment in multi-FASTA format.</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: alignment string</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceAlignment}</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">         </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt id="link-153" class="py-name" targets="Variable csb.apps.ArgHandler.parser=csb.apps.ArgHandler-class.html#parser,Variable csb.apps.hhsearch.HHsearch.parser=csb.apps.hhsearch.HHsearch-class.html#parser"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-153', 'parser', 'link-153');">parser</a></tt> <tt class="py-op">=</tt> <tt id="link-154" class="py-name" targets="Class csb.bio.io.fasta.SequenceParser=csb.bio.io.fasta.SequenceParser-class.html"><a title="csb.bio.io.fasta.SequenceParser" class="py-name" href="#" onclick="return doclink('link-154', 'SequenceParser', 'link-154');">SequenceParser</a></tt><tt class="py-op">(</tt><tt id="link-155" class="py-name"><a title="csb.bio.sequence.RichSequence" class="py-name" href="#" onclick="return doclink('link-155', 'RichSequence', 'link-24');">RichSequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-156', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt id="link-157" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-157', 'parser', 'link-153');">parser</a></tt><tt class="py-op">.</tt><tt id="link-158" class="py-name" targets="Method csb.bio.io.fasta.BaseSequenceParser.parse_string()=csb.bio.io.fasta.BaseSequenceParser-class.html#parse_string,Method csb.bio.io.hhpred.HHOutputParser.parse_string()=csb.bio.io.hhpred.HHOutputParser-class.html#parse_string"><a title="csb.bio.io.fasta.BaseSequenceParser.parse_string
csb.bio.io.hhpred.HHOutputParser.parse_string" class="py-name" href="#" onclick="return doclink('link-158', 'parse_string', 'link-158');">parse_string</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">         </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-159" class="py-name"><a title="csb.bio.sequence.SequenceAlignment" class="py-name" href="#" onclick="return doclink('link-159', 'SequenceAlignment', 'link-15');">SequenceAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-160" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-160', 'strict', 'link-152');">strict</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-161', 'strict', 'link-152');">strict</a></tt><tt class="py-op">)</tt>  </tt>
</div><a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">     </tt>
<a name="SequenceAlignmentReader.read_a3m"></a><div id="SequenceAlignmentReader.read_a3m-def"><a name="L306"></a><tt class="py-lineno">306</tt> <a class="py-toggle" href="#" id="SequenceAlignmentReader.read_a3m-toggle" onclick="return toggle('SequenceAlignmentReader.read_a3m');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.SequenceAlignmentReader-class.html#read_a3m">read_a3m</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignmentReader.read_a3m-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceAlignmentReader.read_a3m-expanded"><a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"><tt class="py-docstring">        Parse an alignment in A3M format.</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: alignment string</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{A3MAlignment}</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt id="link-162" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.alphabet=csb.bio.sequence.AbstractSequence-class.html#alphabet,Variable csb.bio.sequence.SequenceAdapter.alphabet=csb.bio.sequence.SequenceAdapter-class.html#alphabet,Variable csb.bio.structure.Chain.alphabet=csb.bio.structure.Chain-class.html#alphabet"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-162', 'alphabet', 'link-162');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-163', 'SequenceAlphabets', 'link-9');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-164" class="py-name" targets="Function csb.bio.io.dssp.get()=csb.bio.io.dssp-module.html#get,Method csb.bio.io.wwpdb.RemoteStructureProvider.get()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#get,Method csb.bio.io.wwpdb.StructureProvider.get()=csb.bio.io.wwpdb.StructureProvider-class.html#get,Function csb.bio.io.wwpdb.get()=csb.bio.io.wwpdb-module.html#get,Static Method csb.bio.nmr.AtomConnectivity.get()=csb.bio.nmr.AtomConnectivity-class.html#get,Method csb.bio.nmr.NOEPeak.get()=csb.bio.nmr.NOEPeak-class.html#get,Static Method csb.bio.nmr.RandomCoil.get()=csb.bio.nmr.RandomCoil-class.html#get,Static Method csb.bio.sequence.SequenceAlphabets.get()=csb.bio.sequence.SequenceAlphabets-class.html#get,Static Method csb.build.BuildTypes.get()=csb.build.BuildTypes-class.html#get,Static Method csb.io.plots.Backend.get()=csb.io.plots.Backend-class.html#get,Method csb.statistics.maxent.MaxentModel.get()=csb.statistics.maxent.MaxentModel-class.html#get"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-164', 'get', 'link-164');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-165', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-comment"># parse all "mis-aligned" sequences as case-sensitive strings        </tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt id="link-166" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-166', 'parser', 'link-153');">parser</a></tt> <tt class="py-op">=</tt> <tt id="link-167" class="py-name"><a title="csb.bio.io.fasta.SequenceParser" class="py-name" href="#" onclick="return doclink('link-167', 'SequenceParser', 'link-154');">SequenceParser</a></tt><tt class="py-op">(</tt><tt id="link-168" class="py-name"><a title="csb.bio.sequence.Sequence" class="py-name" href="#" onclick="return doclink('link-168', 'Sequence', 'link-23');">Sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-169', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-170', 'parser', 'link-153');">parser</a></tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.parse_string
csb.bio.io.hhpred.HHOutputParser.parse_string" class="py-name" href="#" onclick="return doclink('link-171', 'parse_string', 'link-158');">parse_string</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">         </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-comment"># storage for expanded sequences</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">         </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">dummy</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">            <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-172', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">              </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-comment"># expand all sequences with insertion characters and make them equal length</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-173" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-173', 'column', 'link-129');">column</a></tt> <tt class="py-keyword">in</tt> <tt id="link-174" class="py-name" targets="Class csb.bio.io.fasta.A3MSequenceIterator=csb.bio.io.fasta.A3MSequenceIterator-class.html"><a title="csb.bio.io.fasta.A3MSequenceIterator" class="py-name" href="#" onclick="return doclink('link-174', 'A3MSequenceIterator', 'link-174');">A3MSequenceIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-175" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-175', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-176" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.INSERTION=csb.bio.sequence.NucleicAlphabet-class.html#INSERTION,Variable csb.bio.sequence.ProteinAlphabet.INSERTION=csb.bio.sequence.ProteinAlphabet-class.html#INSERTION,Variable csb.bio.sequence.UnknownAlphabet.INSERTION=csb.bio.sequence.UnknownAlphabet-class.html#INSERTION"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-176', 'INSERTION', 'link-176');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">sn</tt><tt class="py-op">,</tt> <tt class="py-name">char</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-177', 'column', 'link-129');">column</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">                <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">sn</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-178', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">         </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-comment"># build normal sequence objects from the equalized sequence strings</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-name">aligned_seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">         </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">sn</tt><tt class="py-op">,</tt> <tt id="link-179" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-179', 'seq', 'link-87');">seq</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">         </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt id="link-180" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-180', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-181" class="py-name"><a title="csb.bio.sequence.RichSequence" class="py-name" href="#" onclick="return doclink('link-181', 'RichSequence', 'link-24');">RichSequence</a></tt><tt class="py-op">(</tt><tt id="link-182" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-182', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-183', 'id', 'link-78');">id</a></tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name"><a title="csb.bio.io.clans.ClansEntry.seq" class="py-name" href="#" onclick="return doclink('link-184', 'seq', 'link-87');">seq</a></tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-185', 'header', 'link-77');">header</a></tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-name">sn</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-186', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">            <tt class="py-name">aligned_seqs</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-187', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt id="link-188" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-188', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">             </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-189" class="py-name"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-189', 'A3MAlignment', 'link-17');">A3MAlignment</a></tt><tt class="py-op">(</tt><tt class="py-name">aligned_seqs</tt><tt class="py-op">,</tt> <tt id="link-190" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-190', 'strict', 'link-152');">strict</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-191', 'strict', 'link-152');">strict</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">     </tt>
<a name="StructureAlignmentFactory"></a><div id="StructureAlignmentFactory-def"><a name="L342"></a><tt class="py-lineno">342</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory-toggle" onclick="return toggle('StructureAlignmentFactory');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html">StructureAlignmentFactory</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="StructureAlignmentFactory-expanded"><a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"><tt class="py-docstring">    Protein structure alignment parser.</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"><tt class="py-docstring">    In order to construct the structural alignment, this factory needs a PDB</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"><tt class="py-docstring">    structure provider: an object, whose C{provider.get} method returns a </tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line"><tt class="py-docstring">    L{csb.bio.structute.Structure} for a given sequence identifier. Sequence</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"><tt class="py-docstring">    identifiers on the other hand need to be split into 'accession number'</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line"><tt class="py-docstring">    and 'chain ID'. By default this is done using a standard PDB Entry ID</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"><tt class="py-docstring">    factory, but clients are free to provide custom factories. An C{id_factory}</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"><tt class="py-docstring">    must be a callable, which accepts a single string identifier and returns</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line"><tt class="py-docstring">    an EntryID object.</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line"><tt class="py-docstring">    @param provider: data source for all structures found in the alignment</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line"><tt class="py-docstring">    @type provider: L{csb.bio.io.wwpdb.StructureProvider}</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"><tt class="py-docstring">    @param id_factory: callable factory, which transforms a sequence ID into</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line"><tt class="py-docstring">                       a L{csb.bio.io.wwpdb.EntryID} object. By default</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line"><tt class="py-docstring">                       this is L{csb.bio.io.wwpdb.EntryID.create}.</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line"><tt class="py-docstring">    @type id_factory: callable</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line"><tt class="py-docstring">    @param strict: if True, raise exception on duplicate sequence identifiers.</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"><tt class="py-docstring">                   See L{csb.bio.sequence.AbstractAlignment} for details</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line"><tt class="py-docstring">    @type strict: bool    </tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">     </tt>
<a name="StructureAlignmentFactory.__init__"></a><div id="StructureAlignmentFactory.__init__-def"><a name="L366"></a><tt class="py-lineno">366</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.__init__-toggle" onclick="return toggle('StructureAlignmentFactory.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">provider</tt><tt class="py-op">,</tt> <tt class="py-param">id_factory</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.__init__-expanded"><a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-192" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-192', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-193', 'bio', 'link-6');">bio</a></tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-194', 'io', 'link-1');">io</a></tt><tt class="py-op">.</tt><tt id="link-195" class="py-name" targets="Module csb.bio.io.wwpdb=csb.bio.io.wwpdb-module.html"><a title="csb.bio.io.wwpdb" class="py-name" href="#" onclick="return doclink('link-195', 'wwpdb', 'link-195');">wwpdb</a></tt> <tt class="py-keyword">import</tt> <tt id="link-196" class="py-name" targets="Class csb.bio.io.wwpdb.EntryID=csb.bio.io.wwpdb.EntryID-class.html"><a title="csb.bio.io.wwpdb.EntryID" class="py-name" href="#" onclick="return doclink('link-196', 'EntryID', 'link-196');">EntryID</a></tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">         </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-197" class="py-name" targets="Variable csb.bio.io.fasta.StructureAlignmentFactory.id_factory=csb.bio.io.fasta.StructureAlignmentFactory-class.html#id_factory"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-197', 'id_factory', 'link-197');">id_factory</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt id="link-198" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-198', 'id_factory', 'link-197');">id_factory</a></tt> <tt class="py-op">=</tt> <tt id="link-199" class="py-name"><a title="csb.bio.io.wwpdb.EntryID" class="py-name" href="#" onclick="return doclink('link-199', 'EntryID', 'link-196');">EntryID</a></tt><tt class="py-op">.</tt><tt id="link-200" class="py-name" targets="Static Method csb.apps.buildhmm.ProfileBuilder.create()=csb.apps.buildhmm.ProfileBuilder-class.html#create,Static Method csb.apps.hhfrag.PredictionBuilder.create()=csb.apps.hhfrag.PredictionBuilder-class.html#create,Static Method csb.bio.fragments.ClusterNode.create()=csb.bio.fragments.ClusterNode-class.html#create,Static Method csb.bio.io.cs.ChemShiftReader.create()=csb.bio.io.cs.ChemShiftReader-class.html#create,Static Method csb.bio.io.fasta.OutputBuilder.create()=csb.bio.io.fasta.OutputBuilder-class.html#create,Static Method csb.bio.io.wwpdb.EntryID.create()=csb.bio.io.wwpdb.EntryID-class.html#create,Static Method csb.bio.io.wwpdb.SparseChainSequence.create()=csb.bio.io.wwpdb.SparseChainSequence-class.html#create,Static Method csb.bio.sequence.ChainSequence.create()=csb.bio.sequence.ChainSequence-class.html#create,Static Method csb.bio.sequence.RichSequence.create()=csb.bio.sequence.RichSequence-class.html#create,Static Method csb.bio.structure.CompositeEntityIterator.create()=csb.bio.structure.CompositeEntityIterator-class.html#create,Static Method csb.bio.structure.Residue.create()=csb.bio.structure.Residue-class.html#create,Static Method csb.core.Enum.create()=csb.core.Enum-class.html#create,Method csb.io.tsv.RepositoryImp.create()=csb.io.tsv.RepositoryImp-class.html#create,Method csb.io.tsv.SQLiteRepository.create()=csb.io.tsv.SQLiteRepository-class.html#create,Static Method csb.statistics.samplers.mc.TrajectoryBuilder.create()=csb.statistics.samplers.mc.TrajectoryBuilder-class.html#create"><a title="csb.apps.buildhmm.ProfileBuilder.create
csb.apps.hhfrag.PredictionBuilder.create
csb.bio.fragments.ClusterNode.create
csb.bio.io.cs.ChemShiftReader.create
csb.bio.io.fasta.OutputBuilder.create
csb.bio.io.wwpdb.EntryID.create
csb.bio.io.wwpdb.SparseChainSequence.create
csb.bio.sequence.ChainSequence.create
csb.bio.sequence.RichSequence.create
csb.bio.structure.CompositeEntityIterator.create
csb.bio.structure.Residue.create
csb.core.Enum.create
csb.io.tsv.RepositoryImp.create
csb.io.tsv.SQLiteRepository.create
csb.statistics.samplers.mc.TrajectoryBuilder.create" class="py-name" href="#" onclick="return doclink('link-200', 'create', 'link-200');">create</a></tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-201" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-201', 'id_factory', 'link-197');">id_factory</a></tt><tt class="py-op">,</tt> <tt class="py-string">'__call__'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-202', 'id_factory', 'link-197');">id_factory</a></tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">         </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-203" class="py-name" targets="Variable csb.bio.io.fasta.StructureAlignmentFactory.provider=csb.bio.io.fasta.StructureAlignmentFactory-class.html#provider"><a title="csb.bio.io.fasta.StructureAlignmentFactory.provider" class="py-name" href="#" onclick="return doclink('link-203', 'provider', 'link-203');">provider</a></tt><tt class="py-op">,</tt> <tt class="py-string">'get'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-204" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.provider" class="py-name" href="#" onclick="return doclink('link-204', 'provider', 'link-203');">provider</a></tt><tt class="py-op">)</tt>         </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">                 </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt id="link-205" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-205', 'SequenceTypes', 'link-8');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-206', 'Protein', 'link-29');">Protein</a></tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> <tt class="py-op">=</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt id="link-207" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-207', 'strict', 'link-152');">strict</a></tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_provider</tt> <tt class="py-op">=</tt> <tt id="link-208" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.provider" class="py-name" href="#" onclick="return doclink('link-208', 'provider', 'link-203');">provider</a></tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id_factory</tt> <tt class="py-op">=</tt> <tt id="link-209" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-209', 'id_factory', 'link-197');">id_factory</a></tt> </tt>
</div><a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="StructureAlignmentFactory.product_type"></a><div id="StructureAlignmentFactory.product_type-def"><a name="L384"></a><tt class="py-lineno">384</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.product_type-toggle" onclick="return toggle('StructureAlignmentFactory.product_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#product_type">product_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.product_type-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.product_type-expanded"><a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"><tt class="py-docstring">        Default sequence type of the alignment rows - a member of L{SequenceTypes}</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: enum item</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> </tt>
</div><a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">     </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="StructureAlignmentFactory.provider"></a><div id="StructureAlignmentFactory.provider-def"><a name="L392"></a><tt class="py-lineno">392</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.provider-toggle" onclick="return toggle('StructureAlignmentFactory.provider');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#provider">provider</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.provider-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.provider-expanded"><a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"><tt class="py-docstring">        Current L{csb.bio.io.wwpdb.StructureProvider} instance in use</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{StructureProvider}</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_provider</tt>   </tt>
</div><a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">     </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="StructureAlignmentFactory.id_factory"></a><div id="StructureAlignmentFactory.id_factory-def"><a name="L400"></a><tt class="py-lineno">400</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.id_factory-toggle" onclick="return toggle('StructureAlignmentFactory.id_factory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#id_factory">id_factory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.id_factory-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.id_factory-expanded"><a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"><tt class="py-docstring">        Current L{csb.bio.io.wwpdb.EntryID} factory instance in use</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{EntryID}        </tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id_factory</tt> </tt>
</div><a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line"> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="StructureAlignmentFactory.strict"></a><div id="StructureAlignmentFactory.strict-def"><a name="L408"></a><tt class="py-lineno">408</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.strict-toggle" onclick="return toggle('StructureAlignmentFactory.strict');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#strict">strict</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.strict-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.strict-expanded"><a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line"><tt class="py-docstring">        True if strict mode is enabled</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> </tt>
</div><a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">             </tt>
<a name="StructureAlignmentFactory.make_alignment"></a><div id="StructureAlignmentFactory.make_alignment-def"><a name="L415"></a><tt class="py-lineno">415</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.make_alignment-toggle" onclick="return toggle('StructureAlignmentFactory.make_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#make_alignment">make_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.make_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.make_alignment-expanded"><a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line"><tt class="py-docstring">        Build a protein structure alignment given a multi-FASTA string</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line"><tt class="py-docstring">        and the current structure C{provider}.</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: alignment string</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceAlignment}</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceError: when an aligned sequence is not a proper</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"><tt class="py-docstring">                              subsequence of its respective source PDB chain</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"><tt class="py-docstring">        @raise StructureNotFoundError: if C{provider} can't provide a structure</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"><tt class="py-docstring">                                       for a given sequence ID</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line"><tt class="py-docstring">        @raise InvalidEntryIDError: if a given sequence ID cannot be parsed        </tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line"> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt id="link-210" class="py-name"><a title="csb.bio.io.clans.Clans.entries
csb.io.EntryReader.entries" class="py-name" href="#" onclick="return doclink('link-210', 'entries', 'link-57');">entries</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt id="link-211" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-211', 'parser', 'link-153');">parser</a></tt> <tt class="py-op">=</tt> <tt id="link-212" class="py-name"><a title="csb.bio.io.fasta.SequenceParser" class="py-name" href="#" onclick="return doclink('link-212', 'SequenceParser', 'link-154');">SequenceParser</a></tt><tt class="py-op">(</tt><tt id="link-213" class="py-name"><a title="csb.bio.sequence.Sequence" class="py-name" href="#" onclick="return doclink('link-213', 'Sequence', 'link-23');">Sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.product_type
csb.bio.io.fasta.SequenceAlignmentReader.product_type
csb.bio.io.fasta.StructureAlignmentFactory.product_type" class="py-name" href="#" onclick="return doclink('link-214', 'product_type', 'link-33');">product_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt id="link-215" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-215', 'parser', 'link-153');">parser</a></tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="csb.bio.io.fasta.BaseSequenceParser.parse_string
csb.bio.io.hhpred.HHOutputParser.parse_string" class="py-name" href="#" onclick="return doclink('link-216', 'parse_string', 'link-158');">parse_string</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">         </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">            <tt id="link-217" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-217', 'id', 'link-78');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-218', 'id_factory', 'link-197');">id_factory</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-219', 'id', 'link-78');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">            <tt id="link-220" class="py-name" targets="Variable csb.bio.fragments.rosetta.RosettaFragment.chain=csb.bio.fragments.rosetta.RosettaFragment-class.html#chain,Method csb.bio.hmm.ProfileHMM.chain()=csb.bio.hmm.ProfileHMM-class.html#chain,Variable csb.bio.io.wwpdb.EntryID.chain=csb.bio.io.wwpdb.EntryID-class.html#chain,Variable csb.bio.io.wwpdb.ResidueInfo.chain=csb.bio.io.wwpdb.ResidueInfo-class.html#chain,Variable csb.bio.nmr.ContactMap.chain=csb.bio.nmr.ContactMap-class.html#chain"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-220', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.provider" class="py-name" href="#" onclick="return doclink('link-221', 'provider', 'link-203');">provider</a></tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-222', 'get', 'link-164');">get</a></tt><tt class="py-op">(</tt><tt id="link-223" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-223', 'id', 'link-78');">id</a></tt><tt class="py-op">.</tt><tt id="link-224" class="py-name" targets="Variable csb.bio.fragments.Target.accession=csb.bio.fragments.Target-class.html#accession,Variable csb.bio.fragments.rosetta.RosettaFragment.accession=csb.bio.fragments.rosetta.RosettaFragment-class.html#accession,Variable csb.bio.io.wwpdb.EntryID.accession=csb.bio.io.wwpdb.EntryID-class.html#accession,Variable csb.bio.structure.Chain.accession=csb.bio.structure.Chain-class.html#accession,Variable csb.bio.structure.Structure.accession=csb.bio.structure.Structure-class.html#accession"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-224', 'accession', 'link-224');">accession</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name" targets="Variable csb.bio.structure.Structure.chains=csb.bio.structure.Structure-class.html#chains"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-225', 'chains', 'link-225');">chains</a></tt><tt class="py-op">[</tt><tt id="link-226" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-226', 'id', 'link-78');">id</a></tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-227', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">             </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">            <tt class="py-name">entry</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method csb.bio.io.fasta.StructureAlignmentFactory.make_entry()=csb.bio.io.fasta.StructureAlignmentFactory-class.html#make_entry"><a title="csb.bio.io.fasta.StructureAlignmentFactory.make_entry" class="py-name" href="#" onclick="return doclink('link-228', 'make_entry', 'link-228');">make_entry</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">,</tt> <tt id="link-229" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-229', 'chain', 'link-220');">chain</a></tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">            <tt id="link-230" class="py-name"><a title="csb.bio.io.clans.Clans.entries
csb.io.EntryReader.entries" class="py-name" href="#" onclick="return doclink('link-230', 'entries', 'link-57');">entries</a></tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-231', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">entry</tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">             </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-232" class="py-name"><a title="csb.bio.sequence.StructureAlignment" class="py-name" href="#" onclick="return doclink('link-232', 'StructureAlignment', 'link-16');">StructureAlignment</a></tt><tt class="py-op">(</tt><tt id="link-233" class="py-name"><a title="csb.bio.io.clans.Clans.entries
csb.io.EntryReader.entries" class="py-name" href="#" onclick="return doclink('link-233', 'entries', 'link-57');">entries</a></tt><tt class="py-op">,</tt> <tt id="link-234" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-234', 'strict', 'link-152');">strict</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-235', 'strict', 'link-152');">strict</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">     </tt>
<a name="StructureAlignmentFactory.make_entry"></a><div id="StructureAlignmentFactory.make_entry-def"><a name="L445"></a><tt class="py-lineno">445</tt> <a class="py-toggle" href="#" id="StructureAlignmentFactory.make_entry-toggle" onclick="return toggle('StructureAlignmentFactory.make_entry');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.StructureAlignmentFactory-class.html#make_entry">make_entry</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">row</tt><tt class="py-op">,</tt> <tt class="py-param">chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignmentFactory.make_entry-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="StructureAlignmentFactory.make_entry-expanded"><a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"><tt class="py-docstring">        Build a protein structure alignment entry, given a sequence alignment</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line"><tt class="py-docstring">        entry and its corresponding source PDB chain.</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"><tt class="py-docstring">        @param row: sequence alignment entry (sequence with gaps)</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"><tt class="py-docstring">        @type row: L{AbstractSequence}, L{SequenceAdapter}</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line"><tt class="py-docstring">        @param chain: source PDB chain</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain: L{csb.bio.structure.Chain}</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"><tt class="py-docstring">        @return: gapped chain sequence, containing cloned residues from the</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line"><tt class="py-docstring">                 source chain (except for the gaps)</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ChainSequence}</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceError: when C{row} is not a proper subsequence of C{chain}        </tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-name">offset</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt>         </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt id="link-236" class="py-name" targets="Variable csb.bio.fragments.Target.residues=csb.bio.fragments.Target-class.html#residues,Variable csb.bio.fragments.rosetta.RosettaFragment.residues=csb.bio.fragments.rosetta.RosettaFragment-class.html#residues,Variable csb.bio.hmm.ProfileHMM.residues=csb.bio.hmm.ProfileHMM-class.html#residues,Variable csb.bio.sequence.AbstractSequence.residues=csb.bio.sequence.AbstractSequence-class.html#residues,Variable csb.bio.sequence.AlignedSequenceAdapter.residues=csb.bio.sequence.AlignedSequenceAdapter-class.html#residues,Variable csb.bio.structure.Chain.residues=csb.bio.structure.Chain-class.html#residues"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-236', 'residues', 'link-236');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt id="link-237" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-237', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-238', 'strip', 'link-73');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-239', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">         </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">            <tt id="link-240" class="py-name" targets="Variable csb.bio.fragments.Assignment.start=csb.bio.fragments.Assignment-class.html#start,Variable csb.bio.fragments.FragmentMatch.start=csb.bio.fragments.FragmentMatch-class.html#start,Variable csb.bio.fragments.rosetta.RosettaFragment.start=csb.bio.fragments.rosetta.RosettaFragment-class.html#start,Variable csb.bio.hmm.HHpredHit.start=csb.bio.hmm.HHpredHit-class.html#start,Variable csb.bio.hmm.ProfileHMM.start=csb.bio.hmm.ProfileHMM-class.html#start,Variable csb.bio.sequence.alignment.AlignmentResult.start=csb.bio.sequence.alignment.AlignmentResult-class.html#start,Variable csb.bio.structure.SecondaryStructure.start=csb.bio.structure.SecondaryStructure-class.html#start,Variable csb.bio.structure.SecondaryStructureElement.start=csb.bio.structure.SecondaryStructureElement-class.html#start,Method csb.core.REMatchProxy.start()=csb.core.REMatchProxy-class.html#start,Method csb.io.plots.Backend.start()=csb.io.plots.Backend-class.html#start"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-240', 'start', 'link-240');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-241" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-241', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-242', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt class="py-name">index</tt><tt class="py-op">(</tt><tt id="link-243" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-243', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-244" class="py-name"><a title="csb.bio.sequence.SequenceError" class="py-name" href="#" onclick="return doclink('link-244', 'SequenceError', 'link-11');">SequenceError</a></tt><tt class="py-op">(</tt><tt class="py-string">'{0}: not a subsequence of {1}'</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-245', 'format', 'link-100');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-246', 'id', 'link-78');">id</a></tt><tt class="py-op">,</tt> <tt id="link-247" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-247', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-248" class="py-name" targets="Variable csb.bio.io.wwpdb.EntryID.entry_id=csb.bio.io.wwpdb.EntryID-class.html#entry_id,Variable csb.bio.structure.Chain.entry_id=csb.bio.structure.Chain-class.html#entry_id"><a title="csb.bio.io.wwpdb.EntryID.entry_id
csb.bio.structure.Chain.entry_id" class="py-name" href="#" onclick="return doclink('link-248', 'entry_id', 'link-248');">entry_id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">         </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rinfo</tt> <tt class="py-keyword">in</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-249', 'residues', 'link-236');">residues</a></tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">             </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">rinfo</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-250', 'type', 'link-110');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-251', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-252" class="py-name" targets="Variable csb.bio.hmm.HHpredHitAlignment.GAP=csb.bio.hmm.HHpredHitAlignment-class.html#GAP,Variable csb.bio.sequence.NucleicAlphabet.GAP=csb.bio.sequence.NucleicAlphabet-class.html#GAP,Variable csb.bio.sequence.ProteinAlphabet.GAP=csb.bio.sequence.ProteinAlphabet-class.html#GAP,Variable csb.bio.sequence.UnknownAlphabet.GAP=csb.bio.sequence.UnknownAlphabet-class.html#GAP"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-252', 'GAP', 'link-252');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                <tt id="link-253" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-253', 'residues', 'link-236');">residues</a></tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-254', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">rinfo</tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                <tt id="link-255" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.rank=csb.bio.fragments.ResidueEventInfo-class.html#rank,Variable csb.bio.hmm.HHpredHit.rank=csb.bio.hmm.HHpredHit-class.html#rank,Variable csb.bio.hmm.HMMLayer.rank=csb.bio.hmm.HMMLayer-class.html#rank,Variable csb.bio.hmm.State.rank=csb.bio.hmm.State-class.html#rank,Variable csb.bio.io.wwpdb.ResidueInfo.rank=csb.bio.io.wwpdb.ResidueInfo-class.html#rank,Variable csb.bio.nmr.Label.rank=csb.bio.nmr.Label-class.html#rank,Variable csb.bio.sequence.ResidueInfo.rank=csb.bio.sequence.ResidueInfo-class.html#rank,Variable csb.bio.structure.Residue.rank=csb.bio.structure.Residue-class.html#rank"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-255', 'rank', 'link-255');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-256" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-256', 'start', 'link-240');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt id="link-257" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-257', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-258', 'residues', 'link-236');">residues</a></tt><tt class="py-op">[</tt><tt id="link-259" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-259', 'rank', 'link-255');">rank</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-260', 'type', 'link-110');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">rinfo</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-261', 'type', 'link-110');">type</a></tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                <tt id="link-262" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-262', 'residues', 'link-236');">residues</a></tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-263', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt id="link-264" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-264', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-265', 'residues', 'link-236');">residues</a></tt><tt class="py-op">[</tt><tt id="link-266" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-266', 'rank', 'link-255');">rank</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name" targets="Method csb.bio.fragments.ChemShiftTarget.clone()=csb.bio.fragments.ChemShiftTarget-class.html#clone,Method csb.bio.fragments.Target.clone()=csb.bio.fragments.Target-class.html#clone,Method csb.bio.nmr.ChemShiftInfo.clone()=csb.bio.nmr.ChemShiftInfo-class.html#clone,Method csb.bio.structure.Atom.clone()=csb.bio.structure.Atom-class.html#clone,Method csb.bio.structure.Chain.clone()=csb.bio.structure.Chain-class.html#clone,Method csb.bio.structure.Residue.clone()=csb.bio.structure.Residue-class.html#clone,Method csb.bio.structure.SecondaryStructure.clone()=csb.bio.structure.SecondaryStructure-class.html#clone,Method csb.statistics.samplers.State.clone()=csb.statistics.samplers.State-class.html#clone"><a title="csb.bio.fragments.ChemShiftTarget.clone
csb.bio.fragments.Target.clone
csb.bio.nmr.ChemShiftInfo.clone
csb.bio.structure.Atom.clone
csb.bio.structure.Chain.clone
csb.bio.structure.Residue.clone
csb.bio.structure.SecondaryStructure.clone
csb.statistics.samplers.State.clone" class="py-name" href="#" onclick="return doclink('link-267', 'clone', 'link-267');">clone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                <tt class="py-name">offset</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">             </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-268" class="py-name"><a title="csb.bio.sequence.ChainSequence" class="py-name" href="#" onclick="return doclink('link-268', 'ChainSequence', 'link-25');">ChainSequence</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-269', 'id', 'link-78');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-270', 'header', 'link-77');">header</a></tt><tt class="py-op">,</tt> <tt id="link-271" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-271', 'residues', 'link-236');">residues</a></tt><tt class="py-op">,</tt> <tt id="link-272" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-272', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-273', 'type', 'link-110');">type</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">         </tt>
<a name="OutputBuilder"></a><div id="OutputBuilder-def"><a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">     </tt>
<a name="L484"></a><tt class="py-lineno">484</tt> <a class="py-toggle" href="#" id="OutputBuilder-toggle" onclick="return toggle('OutputBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html">OutputBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="OutputBuilder-expanded"><a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line"><tt class="py-docstring">    Base sequence/alignment string format builder.</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line"><tt class="py-docstring">    @param output: destination stream, where the product is written.</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line"><tt class="py-docstring">    @type output: file</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line"><tt class="py-docstring">    @param headers: if False, omit headers</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"><tt class="py-docstring">    @type headers: bool</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line"><tt class="py-docstring">    @note: File builders are not guaranteed to check the correctness of the</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"><tt class="py-docstring">           product. It is assumed that the client of the builder knows what</tt> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"><tt class="py-docstring">           it is doing.   </tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-274" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-274', 'ABCMeta', 'link-4');">ABCMeta</a></tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">    <tt id="link-275" class="py-name" targets="Variable csb.bio.io.fasta.OutputBuilder._registry=csb.bio.io.fasta.OutputBuilder-class.html#_registry"><a title="csb.bio.io.fasta.OutputBuilder._registry" class="py-name" href="#" onclick="return doclink('link-275', '_registry', 'link-275');">_registry</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>     </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">         </tt>
<a name="OutputBuilder.__init__"></a><div id="OutputBuilder.__init__-def"><a name="L500"></a><tt class="py-lineno">500</tt> <a class="py-toggle" href="#" id="OutputBuilder.__init__-toggle" onclick="return toggle('OutputBuilder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output</tt><tt class="py-op">,</tt> <tt class="py-param">headers</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.__init__-expanded"><a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-276" class="py-name" targets="Variable csb.bio.fragments.rosetta.OutputBuilder.output=csb.bio.fragments.rosetta.OutputBuilder-class.html#output,Variable csb.bio.io.clans.ClansFileBuilder.output=csb.bio.io.clans.ClansFileBuilder-class.html#output,Variable csb.bio.io.fasta.OutputBuilder.output=csb.bio.io.fasta.OutputBuilder-class.html#output,Variable csb.bio.io.hhpred.HHMFileBuilder.output=csb.bio.io.hhpred.HHMFileBuilder-class.html#output,Variable csb.bio.io.wwpdb.FileBuilder.output=csb.bio.io.wwpdb.FileBuilder-class.html#output,Variable csb.build.Console.output=csb.build.Console-class.html#output"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-276', 'output', 'link-276');">output</a></tt><tt class="py-op">,</tt> <tt class="py-string">'write'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-277" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-277', 'output', 'link-276');">output</a></tt><tt class="py-op">)</tt>      </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line"> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt> <tt class="py-op">=</tt> <tt id="link-278" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-278', 'output', 'link-276');">output</a></tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_headers</tt> <tt class="py-op">=</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt id="link-279" class="py-name" targets="Variable csb.bio.io.fasta.OutputBuilder.headers=csb.bio.io.fasta.OutputBuilder-class.html#headers"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-279', 'headers', 'link-279');">headers</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">         </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="OutputBuilder.create"></a><div id="OutputBuilder.create-def"><a name="L509"></a><tt class="py-lineno">509</tt> <a class="py-toggle" href="#" id="OutputBuilder.create-toggle" onclick="return toggle('OutputBuilder.create');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#create">create</a><tt class="py-op">(</tt><tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">a</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.create-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.create-expanded"><a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"><tt class="py-docstring">        Create an output builder, which knows how to handle the specified</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"><tt class="py-docstring">        sequence/alignment C{format}. Additional arguments are passed to the</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line"><tt class="py-docstring">        builder's constructor.</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line"><tt class="py-docstring">        @param format: L{AlignmentFormats} member</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"><tt class="py-docstring">        @type format: L{EnumItem}</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{OutputBuilder}</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-280" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-280', 'format', 'link-100');">format</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-281" class="py-name" targets="Class csb.bio.fragments.rosetta.OutputBuilder=csb.bio.fragments.rosetta.OutputBuilder-class.html,Class csb.bio.io.fasta.OutputBuilder=csb.bio.io.fasta.OutputBuilder-class.html"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-281', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder._registry" class="py-name" href="#" onclick="return doclink('link-282', '_registry', 'link-275');">_registry</a></tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Unhandled format: {0}'</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-283', 'format', 'link-100');">format</a></tt><tt class="py-op">(</tt><tt id="link-284" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-284', 'format', 'link-100');">format</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">         </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">        <tt class="py-name">klass</tt> <tt class="py-op">=</tt> <tt id="link-285" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-285', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder._registry" class="py-name" href="#" onclick="return doclink('link-286', '_registry', 'link-275');">_registry</a></tt><tt class="py-op">[</tt><tt id="link-287" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-287', 'format', 'link-100');">format</a></tt><tt class="py-op">]</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">klass</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-288" class="py-name" targets="Variable csb.bio.io.clans.Color.a=csb.bio.io.clans.Color-class.html#a,Variable csb.statistics.pdf.GeneralizedInverseGaussian.a=csb.statistics.pdf.GeneralizedInverseGaussian-class.html#a"><a title="csb.bio.io.clans.Color.a
csb.statistics.pdf.GeneralizedInverseGaussian.a" class="py-name" href="#" onclick="return doclink('link-288', 'a', 'link-288');">a</a></tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> </tt>
</div><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="OutputBuilder.register"></a><div id="OutputBuilder.register-def"><a name="L526"></a><tt class="py-lineno">526</tt> <a class="py-toggle" href="#" id="OutputBuilder.register-toggle" onclick="return toggle('OutputBuilder.register');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#register">register</a><tt class="py-op">(</tt><tt class="py-param">format</tt><tt class="py-op">,</tt> <tt class="py-param">klass</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.register-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.register-expanded"><a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"><tt class="py-docstring">        Register a new output builder.</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"><tt class="py-docstring">        @param format: L{AlignmentFormats} member</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"><tt class="py-docstring">        @type format: L{EnumItem}</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line"><tt class="py-docstring">        @param klass: builder class (L{OutputBuilder} sub-class)</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line"><tt class="py-docstring">        @type klass: type</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>     </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-289" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-289', 'format', 'link-100');">format</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-290" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-290', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder._registry" class="py-name" href="#" onclick="return doclink('link-291', '_registry', 'link-275');">_registry</a></tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">issubclass</tt><tt class="py-op">(</tt><tt class="py-name">klass</tt><tt class="py-op">,</tt> <tt id="link-292" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-292', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">)</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">         </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt id="link-293" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-293', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder._registry" class="py-name" href="#" onclick="return doclink('link-294', '_registry', 'link-275');">_registry</a></tt><tt class="py-op">[</tt><tt id="link-295" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-295', 'format', 'link-100');">format</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">klass</tt> </tt>
</div><a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="OutputBuilder.output"></a><div id="OutputBuilder.output-def"><a name="L541"></a><tt class="py-lineno">541</tt> <a class="py-toggle" href="#" id="OutputBuilder.output-toggle" onclick="return toggle('OutputBuilder.output');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#output">output</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.output-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.output-expanded"><a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line"><tt class="py-docstring">        Destination stream</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: stream</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt> </tt>
</div><a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">     </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="OutputBuilder.headers"></a><div id="OutputBuilder.headers-def"><a name="L549"></a><tt class="py-lineno">549</tt> <a class="py-toggle" href="#" id="OutputBuilder.headers-toggle" onclick="return toggle('OutputBuilder.headers');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#headers">headers</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.headers-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.headers-expanded"><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"><tt class="py-docstring">        True if sequence headers will be written to the destination</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_headers</tt> </tt>
</div><a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line"> </tt>
<a name="OutputBuilder.write"></a><div id="OutputBuilder.write-def"><a name="L556"></a><tt class="py-lineno">556</tt> <a class="py-toggle" href="#" id="OutputBuilder.write-toggle" onclick="return toggle('OutputBuilder.write');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.write-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.write-expanded"><a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line"><tt class="py-docstring">        Write a chunk of C{text} to the output stream.</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-296', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt>    </tt>
</div><a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line"> </tt>
<a name="OutputBuilder.writeline"></a><div id="OutputBuilder.writeline-def"><a name="L562"></a><tt class="py-lineno">562</tt> <a class="py-toggle" href="#" id="OutputBuilder.writeline-toggle" onclick="return toggle('OutputBuilder.writeline');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#writeline">writeline</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.writeline-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.writeline-expanded"><a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"><tt class="py-docstring">        Write a chunk of C{text}, followed by a newline terminator.</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-297', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-298', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">)</tt>             </tt>
</div><a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">     </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="OutputBuilder.add_sequence"></a><div id="OutputBuilder.add_sequence-def"><a name="L570"></a><tt class="py-lineno">570</tt> <a class="py-toggle" href="#" id="OutputBuilder.add_sequence-toggle" onclick="return toggle('OutputBuilder.add_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#add_sequence">add_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.add_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.add_sequence-expanded"><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line"><tt class="py-docstring">        Format and append a new sequence to the product.</tt> </tt>
<a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line"><tt class="py-docstring">        @type sequence: L{AbstractSequence}    </tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">     </tt>
<a name="OutputBuilder.add_many"></a><div id="OutputBuilder.add_many-def"><a name="L577"></a><tt class="py-lineno">577</tt> <a class="py-toggle" href="#" id="OutputBuilder.add_many-toggle" onclick="return toggle('OutputBuilder.add_many');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#add_many">add_many</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.add_many-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.add_many-expanded"><a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line"><tt class="py-docstring">        Format and append a collection of L{AbstractSequence}s to the product.</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line"><tt class="py-docstring">        @type sequences: iterable of L{AbstractSequence}s</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name" targets="Method csb.bio.io.fasta.A3MOutputBuilder.add_sequence()=csb.bio.io.fasta.A3MOutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.FASTAOutputBuilder.add_sequence()=csb.bio.io.fasta.FASTAOutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.OutputBuilder.add_sequence()=csb.bio.io.fasta.OutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.PIROutputBuilder.add_sequence()=csb.bio.io.fasta.PIROutputBuilder-class.html#add_sequence"><a title="csb.bio.io.fasta.A3MOutputBuilder.add_sequence
csb.bio.io.fasta.FASTAOutputBuilder.add_sequence
csb.bio.io.fasta.OutputBuilder.add_sequence
csb.bio.io.fasta.PIROutputBuilder.add_sequence" class="py-name" href="#" onclick="return doclink('link-299', 'add_sequence', 'link-299');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
</div><a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">             </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt>             </tt>
<a name="OutputBuilder.add_alignment"></a><div id="OutputBuilder.add_alignment-def"><a name="L586"></a><tt class="py-lineno">586</tt> <a class="py-toggle" href="#" id="OutputBuilder.add_alignment-toggle" onclick="return toggle('OutputBuilder.add_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#add_alignment">add_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.add_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.add_alignment-expanded"><a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line"><tt class="py-docstring">        Format and append an alignment to the product.</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line"><tt class="py-docstring">        @type alignment: L{AbstractAlignment} </tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">     </tt>
<a name="OutputBuilder.add_separator"></a><div id="OutputBuilder.add_separator-def"><a name="L593"></a><tt class="py-lineno">593</tt> <a class="py-toggle" href="#" id="OutputBuilder.add_separator-toggle" onclick="return toggle('OutputBuilder.add_separator');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#add_separator">add_separator</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">separator</tt><tt class="py-op">=</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.add_separator-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.add_separator-expanded"><a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line"><tt class="py-docstring">        Append a sequence separator to the product.</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name" targets="Method csb.bio.io.clans.ClansFileBuilder.writeline()=csb.bio.io.clans.ClansFileBuilder-class.html#writeline,Method csb.bio.io.fasta.OutputBuilder.writeline()=csb.bio.io.fasta.OutputBuilder-class.html#writeline,Method csb.bio.io.hhpred.HHMFileBuilder.writeline()=csb.bio.io.hhpred.HHMFileBuilder-class.html#writeline,Method csb.bio.io.wwpdb.FileBuilder.writeline()=csb.bio.io.wwpdb.FileBuilder-class.html#writeline,Method csb.bio.io.wwpdb.PDBFileBuilder.writeline()=csb.bio.io.wwpdb.PDBFileBuilder-class.html#writeline,Method csb.io.EntryWriter.writeline()=csb.io.EntryWriter-class.html#writeline"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-300', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-name">separator</tt><tt class="py-op">)</tt>     </tt>
</div><a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line"> </tt>
<a name="OutputBuilder.add_comment"></a><div id="OutputBuilder.add_comment-def"><a name="L599"></a><tt class="py-lineno">599</tt> <a class="py-toggle" href="#" id="OutputBuilder.add_comment-toggle" onclick="return toggle('OutputBuilder.add_comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.OutputBuilder-class.html#add_comment">add_comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">,</tt> <tt class="py-param">comment</tt><tt class="py-op">=</tt><tt class="py-string">'#'</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">=</tt><tt class="py-number">120</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="OutputBuilder.add_comment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="OutputBuilder.add_comment-expanded"><a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line"><tt class="py-docstring">        Append a comment line to the product.</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line"><tt class="py-docstring">        @param text: comment text</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line"><tt class="py-docstring">        @type text: str</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line"><tt class="py-docstring">        @param comment: comment prefix</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line"><tt class="py-docstring">        @type comment: str</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line"><tt class="py-docstring">        @param length: maximal comment length</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line"><tt class="py-docstring">        @type length: int</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-301" class="py-name" targets="Variable csb.apps.hhfrag.SliceContext.length=csb.apps.hhfrag.SliceContext-class.html#length,Variable csb.bio.fragments.FragmentMatch.length=csb.bio.fragments.FragmentMatch-class.html#length,Variable csb.bio.fragments.Target.length=csb.bio.fragments.Target-class.html#length,Variable csb.bio.fragments.rosetta.RosettaFragment.length=csb.bio.fragments.rosetta.RosettaFragment-class.html#length,Variable csb.bio.hmm.HHpredHit.length=csb.bio.hmm.HHpredHit-class.html#length,Variable csb.bio.hmm.ProfileHMM.length=csb.bio.hmm.ProfileHMM-class.html#length,Variable csb.bio.sequence.AbstractAlignment.length=csb.bio.sequence.AbstractAlignment-class.html#length,Variable csb.bio.sequence.AbstractSequence.length=csb.bio.sequence.AbstractSequence-class.html#length,Variable csb.bio.sequence.SequenceAdapter.length=csb.bio.sequence.SequenceAdapter-class.html#length,Variable csb.bio.sequence.alignment.AlignmentResult.length=csb.bio.sequence.alignment.AlignmentResult-class.html#length,Variable csb.bio.structure.Chain.length=csb.bio.structure.Chain-class.html#length,Variable csb.bio.structure.SecondaryStructureElement.length=csb.bio.structure.SecondaryStructureElement-class.html#length,Variable csb.core.BaseCollectionContainer.length=csb.core.BaseCollectionContainer-class.html#length,Variable csb.core.BaseDictionaryContainer.length=csb.core.BaseDictionaryContainer-class.html#length"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-301', 'length', 'link-301');">length</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-302', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">comment</tt><tt class="py-op">)</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-303', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt>             </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-304', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">[</tt><tt class="py-name">i</tt> <tt class="py-op">:</tt> <tt class="py-name">i</tt> <tt class="py-op">+</tt> <tt id="link-305" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-305', 'length', 'link-301');">length</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-306', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">             </tt>
<a name="FASTAOutputBuilder"></a><div id="FASTAOutputBuilder-def"><a name="L616"></a><tt class="py-lineno">616</tt> <a class="py-toggle" href="#" id="FASTAOutputBuilder-toggle" onclick="return toggle('FASTAOutputBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.FASTAOutputBuilder-class.html">FASTAOutputBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">OutputBuilder</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FASTAOutputBuilder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="FASTAOutputBuilder-expanded"><a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"><tt class="py-docstring">    Formats sequences as standard FASTA strings. See L{OutputBuilder}.</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">    <tt id="link-307" class="py-name" targets="Variable csb.bio.io.fasta.A3MOutputBuilder.FORMAT=csb.bio.io.fasta.A3MOutputBuilder-class.html#FORMAT,Variable csb.bio.io.fasta.FASTAOutputBuilder.FORMAT=csb.bio.io.fasta.FASTAOutputBuilder-class.html#FORMAT,Variable csb.bio.io.fasta.PIROutputBuilder.FORMAT=csb.bio.io.fasta.PIROutputBuilder-class.html#FORMAT"><a title="csb.bio.io.fasta.A3MOutputBuilder.FORMAT
csb.bio.io.fasta.FASTAOutputBuilder.FORMAT
csb.bio.io.fasta.PIROutputBuilder.FORMAT" class="py-name" href="#" onclick="return doclink('link-307', 'FORMAT', 'link-307');">FORMAT</a></tt> <tt class="py-op">=</tt> <tt id="link-308" class="py-name"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-308', 'AlignmentFormats', 'link-10');">AlignmentFormats</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.FASTA=csb.bio.sequence.AlignmentFormats-class.html#FASTA"><a title="csb.bio.sequence.AlignmentFormats.FASTA" class="py-name" href="#" onclick="return doclink('link-309', 'FASTA', 'link-309');">FASTA</a></tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line">     </tt>
<a name="FASTAOutputBuilder.add_sequence"></a><div id="FASTAOutputBuilder.add_sequence-def"><a name="L622"></a><tt class="py-lineno">622</tt> <a class="py-toggle" href="#" id="FASTAOutputBuilder.add_sequence-toggle" onclick="return toggle('FASTAOutputBuilder.add_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.FASTAOutputBuilder-class.html#add_sequence">add_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FASTAOutputBuilder.add_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="FASTAOutputBuilder.add_sequence-expanded"><a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">         </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-310', 'headers', 'link-279');">headers</a></tt><tt class="py-op">:</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-311', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-312" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-312', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-313" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-313', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-314', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-315" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-315', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-316', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">        <tt class="py-name">insertion</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-317', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-318', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-319', 'INSERTION', 'link-176');">INSERTION</a></tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">        <tt id="link-320" class="py-name" targets="Variable csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap=csb.bio.sequence.alignment.AbstractAlignmentAlgorithm-class.html#gap"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-320', 'gap', 'link-320');">gap</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-321" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-321', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-322', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-323', 'GAP', 'link-252');">GAP</a></tt><tt class="py-op">)</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">         </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-324', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-325" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-325', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-326', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">insertion</tt><tt class="py-op">,</tt> <tt id="link-327" class="py-name"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-327', 'gap', 'link-320');">gap</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">         </tt>
<a name="FASTAOutputBuilder.add_alignment"></a><div id="FASTAOutputBuilder.add_alignment-def"><a name="L633"></a><tt class="py-lineno">633</tt> <a class="py-toggle" href="#" id="FASTAOutputBuilder.add_alignment-toggle" onclick="return toggle('FASTAOutputBuilder.add_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.FASTAOutputBuilder-class.html#add_alignment">add_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FASTAOutputBuilder.add_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="FASTAOutputBuilder.add_alignment-expanded"><a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">         </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name" targets="Method csb.bio.io.fasta.OutputBuilder.add_many()=csb.bio.io.fasta.OutputBuilder-class.html#add_many"><a title="csb.bio.io.fasta.OutputBuilder.add_many" class="py-name" href="#" onclick="return doclink('link-328', 'add_many', 'link-328');">add_many</a></tt><tt class="py-op">(</tt><tt id="link-329" class="py-name" targets="Variable csb.bio.hmm.HHpredHit.alignment=csb.bio.hmm.HHpredHit-class.html#alignment,Variable csb.bio.hmm.ProfileHMM.alignment=csb.bio.hmm.ProfileHMM-class.html#alignment,Module csb.bio.sequence.alignment=csb.bio.sequence.alignment-module.html,Method csb.bio.sequence.alignment.AlignmentResult.alignment()=csb.bio.sequence.alignment.AlignmentResult-class.html#alignment"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-329', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-330" class="py-name" targets="Variable csb.bio.sequence.AbstractAlignment.rows=csb.bio.sequence.AbstractAlignment-class.html#rows,Variable csb.io.plots.Chart.rows=csb.io.plots.Chart-class.html#rows"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-330', 'rows', 'link-330');">rows</a></tt><tt class="py-op">)</tt>         </tt>
</div></div><a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">         </tt>
<a name="A3MOutputBuilder"></a><div id="A3MOutputBuilder-def"><a name="L637"></a><tt class="py-lineno">637</tt> <a class="py-toggle" href="#" id="A3MOutputBuilder-toggle" onclick="return toggle('A3MOutputBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MOutputBuilder-class.html">A3MOutputBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">OutputBuilder</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MOutputBuilder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="A3MOutputBuilder-expanded"><a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line"><tt class="py-docstring">    Formats sequences as A3M strings. When appending an alignment, this builder</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line"><tt class="py-docstring">    will write all insertion-containing columns in lower case. Also, gap symbols</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line"><tt class="py-docstring">    are omitted if the respective columns contain insertions. </tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line"><tt class="py-docstring">    See L{OutputBuilder}.</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">    <tt id="link-331" class="py-name"><a title="csb.bio.io.fasta.A3MOutputBuilder.FORMAT
csb.bio.io.fasta.FASTAOutputBuilder.FORMAT
csb.bio.io.fasta.PIROutputBuilder.FORMAT" class="py-name" href="#" onclick="return doclink('link-331', 'FORMAT', 'link-307');">FORMAT</a></tt> <tt class="py-op">=</tt> <tt id="link-332" class="py-name"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-332', 'AlignmentFormats', 'link-10');">AlignmentFormats</a></tt><tt class="py-op">.</tt><tt id="link-333" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.A3M=csb.bio.sequence.AlignmentFormats-class.html#A3M"><a title="csb.bio.sequence.AlignmentFormats.A3M" class="py-name" href="#" onclick="return doclink('link-333', 'A3M', 'link-333');">A3M</a></tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">     </tt>
<a name="A3MOutputBuilder.add_sequence"></a><div id="A3MOutputBuilder.add_sequence-def"><a name="L647"></a><tt class="py-lineno">647</tt> <a class="py-toggle" href="#" id="A3MOutputBuilder.add_sequence-toggle" onclick="return toggle('A3MOutputBuilder.add_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MOutputBuilder-class.html#add_sequence">add_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MOutputBuilder.add_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MOutputBuilder.add_sequence-expanded"><a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">         </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-334', 'headers', 'link-279');">headers</a></tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-335', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-336" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-336', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-337', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-338', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-339', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-340', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line"> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-341', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-342" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-342', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-343', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">         </tt>
<a name="A3MOutputBuilder.add_alignment"></a><div id="A3MOutputBuilder.add_alignment-def"><a name="L655"></a><tt class="py-lineno">655</tt> <a class="py-toggle" href="#" id="A3MOutputBuilder.add_alignment-toggle" onclick="return toggle('A3MOutputBuilder.add_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MOutputBuilder-class.html#add_alignment">add_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MOutputBuilder.add_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MOutputBuilder.add_alignment-expanded"><a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-344" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-344', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">,</tt> <tt id="link-345" class="py-name"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-345', 'A3MAlignment', 'link-17');">A3MAlignment</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_a3m</tt><tt class="py-op">(</tt><tt id="link-346" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-346', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_proper</tt><tt class="py-op">(</tt><tt id="link-347" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-347', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">             </tt>
<a name="A3MOutputBuilder._add_a3m"></a><div id="A3MOutputBuilder._add_a3m-def"><a name="L662"></a><tt class="py-lineno">662</tt> <a class="py-toggle" href="#" id="A3MOutputBuilder._add_a3m-toggle" onclick="return toggle('A3MOutputBuilder._add_a3m');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MOutputBuilder-class.html#_add_a3m">_add_a3m</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MOutputBuilder._add_a3m-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MOutputBuilder._add_a3m-expanded"><a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">         </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt id="link-348" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-348', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-349', 'rows', 'link-330');">rows</a></tt><tt class="py-op">:</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">             </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-350', 'headers', 'link-279');">headers</a></tt><tt class="py-op">:</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-351', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-352" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-352', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-353', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-354', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-355', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">                 </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">            <tt id="link-356" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-356', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">             </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">ci</tt> <tt class="py-keyword">in</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name" targets="Variable csb.bio.sequence.AbstractAlignment.columns=csb.bio.sequence.AbstractAlignment-class.html#columns,Variable csb.bio.sequence.AlignedSequenceAdapter.columns=csb.bio.sequence.AlignedSequenceAdapter-class.html#columns,Variable csb.io.plots.Chart.columns=csb.io.plots.Chart-class.html#columns,Variable csb.io.tsv.DataRow.columns=csb.io.tsv.DataRow-class.html#columns,Variable csb.io.tsv.Table.columns=csb.io.tsv.Table-class.html#columns"><a title="csb.bio.sequence.AbstractAlignment.columns
csb.bio.sequence.AlignedSequenceAdapter.columns
csb.io.plots.Chart.columns
csb.io.tsv.DataRow.columns
csb.io.tsv.Table.columns" class="py-name" href="#" onclick="return doclink('link-357', 'columns', 'link-357');">columns</a></tt><tt class="py-op">:</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">                 </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name" targets="Method csb.bio.fragments.AssignmentFactory.residue()=csb.bio.fragments.AssignmentFactory-class.html#residue,Method csb.bio.fragments.ChemShiftAssignmentFactory.residue()=csb.bio.fragments.ChemShiftAssignmentFactory-class.html#residue,Variable csb.bio.hmm.HMMLayer.residue=csb.bio.hmm.HMMLayer-class.html#residue,Variable csb.bio.nmr.Label.residue=csb.bio.nmr.Label-class.html#residue,Variable csb.bio.structure.Atom.residue=csb.bio.structure.Atom-class.html#residue"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-358', 'residue', 'link-358');">residue</a></tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-359', 'type', 'link-110');">type</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-360', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-361" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-361', 'INSERTION', 'link-176');">INSERTION</a></tt><tt class="py-op">:</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">                    <tt class="py-name">char</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-362', 'residue', 'link-358');">residue</a></tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-363', 'type', 'link-110');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line">                     </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-364" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-364', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-365" class="py-name" targets="Method csb.bio.sequence.A3MAlignment.insertion_at()=csb.bio.sequence.A3MAlignment-class.html#insertion_at"><a title="csb.bio.sequence.A3MAlignment.insertion_at" class="py-name" href="#" onclick="return doclink('link-365', 'insertion_at', 'link-365');">insertion_at</a></tt><tt class="py-op">(</tt><tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-366', 'column', 'link-129');">column</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">                        <tt id="link-367" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-367', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-368', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name" targets="Method csb.io.tsv.Where.lower()=csb.io.tsv.Where-class.html#lower"><a title="csb.io.tsv.Where.lower" class="py-name" href="#" onclick="return doclink('link-369', 'lower', 'link-369');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">                        <tt id="link-370" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-370', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-371', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">)</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">                         </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-372', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-373', 'join', 'link-81');">join</a></tt><tt class="py-op">(</tt><tt id="link-374" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-374', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">             </tt>
<a name="A3MOutputBuilder._add_proper"></a><div id="A3MOutputBuilder._add_proper-def"><a name="L686"></a><tt class="py-lineno">686</tt> <a class="py-toggle" href="#" id="A3MOutputBuilder._add_proper-toggle" onclick="return toggle('A3MOutputBuilder._add_proper');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.A3MOutputBuilder-class.html#_add_proper">_add_proper</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MOutputBuilder._add_proper-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="A3MOutputBuilder._add_proper-expanded"><a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line">         </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt id="link-375" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-375', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-376', 'rows', 'link-330');">rows</a></tt><tt class="py-op">:</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">             </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-377', 'headers', 'link-279');">headers</a></tt><tt class="py-op">:</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-378', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-379" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-379', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-380', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-381', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-382', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">                 </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">            <tt id="link-383" class="py-name" targets="Variable csb.bio.sequence.A3MAlignment.master=csb.bio.sequence.A3MAlignment-class.html#master"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-383', 'master', 'link-383');">master</a></tt> <tt class="py-op">=</tt> <tt id="link-384" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-384', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-385', 'rows', 'link-330');">rows</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">            <tt id="link-386" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-386', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">             </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">ci</tt> <tt class="py-keyword">in</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-387" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.columns
csb.bio.sequence.AlignedSequenceAdapter.columns
csb.io.plots.Chart.columns
csb.io.tsv.DataRow.columns
csb.io.tsv.Table.columns" class="py-name" href="#" onclick="return doclink('link-387', 'columns', 'link-357');">columns</a></tt><tt class="py-op">:</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">                <tt class="py-name">char</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-388', 'residue', 'link-358');">residue</a></tt><tt class="py-op">.</tt><tt id="link-389" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-389', 'type', 'link-110');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">                 </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-390" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-390', 'master', 'link-383');">master</a></tt><tt class="py-op">.</tt><tt id="link-391" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.columns
csb.bio.sequence.AlignedSequenceAdapter.columns
csb.io.plots.Chart.columns
csb.io.tsv.DataRow.columns
csb.io.tsv.Table.columns" class="py-name" href="#" onclick="return doclink('link-391', 'columns', 'link-357');">columns</a></tt><tt class="py-op">[</tt><tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-392', 'column', 'link-129');">column</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-393" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-393', 'residue', 'link-358');">residue</a></tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-394', 'type', 'link-110');">type</a></tt> <tt class="py-op">==</tt> <tt id="link-395" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-395', 'master', 'link-383');">master</a></tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-396', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-397" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-397', 'GAP', 'link-252');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">ci</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-398', 'residue', 'link-358');">residue</a></tt><tt class="py-op">.</tt><tt id="link-399" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-399', 'type', 'link-110');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-400', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-401" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-401', 'GAP', 'link-252');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">                        <tt id="link-402" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-402', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-403', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">.</tt><tt id="link-404" class="py-name"><a title="csb.io.tsv.Where.lower" class="py-name" href="#" onclick="return doclink('link-404', 'lower', 'link-369');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                    <tt id="link-405" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-405', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-406" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-406', 'append', 'link-58');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">)</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line">                         </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-407', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-408', 'join', 'link-81');">join</a></tt><tt class="py-op">(</tt><tt id="link-409" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-409', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt>             </tt>
</div></div><a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line">             </tt>
<a name="PIROutputBuilder"></a><div id="PIROutputBuilder-def"><a name="L710"></a><tt class="py-lineno">710</tt> <a class="py-toggle" href="#" id="PIROutputBuilder-toggle" onclick="return toggle('PIROutputBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html">PIROutputBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">OutputBuilder</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="PIROutputBuilder-expanded"><a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line"><tt class="py-docstring">    Formats sequences as PIR FASTA strings, recognized by Modeller.</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line"><tt class="py-docstring">    See L{OutputBuilder} for general alignment documentation.</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">    <tt id="link-410" class="py-name"><a title="csb.bio.io.fasta.A3MOutputBuilder.FORMAT
csb.bio.io.fasta.FASTAOutputBuilder.FORMAT
csb.bio.io.fasta.PIROutputBuilder.FORMAT" class="py-name" href="#" onclick="return doclink('link-410', 'FORMAT', 'link-307');">FORMAT</a></tt> <tt class="py-op">=</tt> <tt id="link-411" class="py-name"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-411', 'AlignmentFormats', 'link-10');">AlignmentFormats</a></tt><tt class="py-op">.</tt><tt id="link-412" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.PIR=csb.bio.sequence.AlignmentFormats-class.html#PIR"><a title="csb.bio.sequence.AlignmentFormats.PIR" class="py-name" href="#" onclick="return doclink('link-412', 'PIR', 'link-412');">PIR</a></tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">         </tt>
<a name="PIROutputBuilder.add_sequence"></a><div id="PIROutputBuilder.add_sequence-def"><a name="L717"></a><tt class="py-lineno">717</tt> <a class="py-toggle" href="#" id="PIROutputBuilder.add_sequence-toggle" onclick="return toggle('PIROutputBuilder.add_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html#add_sequence">add_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder.add_sequence-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PIROutputBuilder.add_sequence-expanded"><a name="L718"></a><tt class="py-lineno">718</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add</tt><tt class="py-op">(</tt><tt id="link-413" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-413', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">template</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">         </tt>
<a name="PIROutputBuilder.add_alignment"></a><div id="PIROutputBuilder.add_alignment-def"><a name="L720"></a><tt class="py-lineno">720</tt> <a class="py-toggle" href="#" id="PIROutputBuilder.add_alignment-toggle" onclick="return toggle('PIROutputBuilder.add_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html#add_alignment">add_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder.add_alignment-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PIROutputBuilder.add_alignment-expanded"><a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line">         </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-414" class="py-name" targets="Variable csb.statistics.maxent.MaxentModel.n=csb.statistics.maxent.MaxentModel-class.html#n"><a title="csb.statistics.maxent.MaxentModel.n" class="py-name" href="#" onclick="return doclink('link-414', 'n', 'link-414');">n</a></tt><tt class="py-op">,</tt> <tt id="link-415" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-415', 'sequence', 'link-7');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-416" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-416', 'alignment', 'link-329');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-417', 'rows', 'link-330');">rows</a></tt><tt class="py-op">,</tt> <tt id="link-418" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-418', 'start', 'link-240');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-419" class="py-name"><a title="csb.statistics.maxent.MaxentModel.n" class="py-name" href="#" onclick="return doclink('link-419', 'n', 'link-414');">n</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-420" class="py-name" targets="Method csb.bio.io.fasta.PIROutputBuilder.add_target()=csb.bio.io.fasta.PIROutputBuilder-class.html#add_target"><a title="csb.bio.io.fasta.PIROutputBuilder.add_target" class="py-name" href="#" onclick="return doclink('link-420', 'add_target', 'link-420');">add_target</a></tt><tt class="py-op">(</tt><tt id="link-421" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-421', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-422" class="py-name" targets="Method csb.bio.io.fasta.PIROutputBuilder.add_template()=csb.bio.io.fasta.PIROutputBuilder-class.html#add_template,Method csb.bio.io.wwpdb.FileSystemStructureProvider.add_template()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#add_template"><a title="csb.bio.io.fasta.PIROutputBuilder.add_template
csb.bio.io.wwpdb.FileSystemStructureProvider.add_template" class="py-name" href="#" onclick="return doclink('link-422', 'add_template', 'link-422');">add_template</a></tt><tt class="py-op">(</tt><tt id="link-423" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-423', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">             </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">         </tt>
<a name="PIROutputBuilder.add_target"></a><div id="PIROutputBuilder.add_target-def"><a name="L729"></a><tt class="py-lineno">729</tt> <a class="py-toggle" href="#" id="PIROutputBuilder.add_target-toggle" onclick="return toggle('PIROutputBuilder.add_target');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html#add_target">add_target</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder.add_target-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PIROutputBuilder.add_target-expanded"><a name="L730"></a><tt class="py-lineno">730</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add</tt><tt class="py-op">(</tt><tt id="link-424" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-424', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">template</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
</div><a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">     </tt>
<a name="PIROutputBuilder.add_template"></a><div id="PIROutputBuilder.add_template-def"><a name="L732"></a><tt class="py-lineno">732</tt> <a class="py-toggle" href="#" id="PIROutputBuilder.add_template-toggle" onclick="return toggle('PIROutputBuilder.add_template');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html#add_template">add_template</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder.add_template-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PIROutputBuilder.add_template-expanded"><a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add</tt><tt class="py-op">(</tt><tt id="link-425" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-425', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">template</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line">     </tt>
<a name="PIROutputBuilder._add"></a><div id="PIROutputBuilder._add-def"><a name="L735"></a><tt class="py-lineno">735</tt> <a class="py-toggle" href="#" id="PIROutputBuilder._add-toggle" onclick="return toggle('PIROutputBuilder._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.fasta.PIROutputBuilder-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">,</tt> <tt class="py-param">template</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PIROutputBuilder._add-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="PIROutputBuilder._add-expanded"><a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line">                     </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-426', 'headers', 'link-279');">headers</a></tt><tt class="py-op">:</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line">             </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">template</tt><tt class="py-op">:</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">                <tt id="link-427" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-427', 'type', 'link-110');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'structure'</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">                <tt id="link-428" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-428', 'type', 'link-110');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'sequence'</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">             </tt>
<a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line">            <tt id="link-429" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-429', 'id', 'link-78');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-430" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-430', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-431" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-431', 'id', 'link-78');">id</a></tt> </tt>
<a name="L745"></a><tt class="py-lineno">745</tt>  <tt class="py-line">            <tt id="link-432" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-432', 'start', 'link-240');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-433" class="py-name" targets="Variable csb.bio.fragments.Assignment.end=csb.bio.fragments.Assignment-class.html#end,Variable csb.bio.fragments.FragmentMatch.end=csb.bio.fragments.FragmentMatch-class.html#end,Variable csb.bio.fragments.rosetta.RosettaFragment.end=csb.bio.fragments.rosetta.RosettaFragment-class.html#end,Variable csb.bio.hmm.HHpredHit.end=csb.bio.hmm.HHpredHit-class.html#end,Variable csb.bio.hmm.ProfileHMM.end=csb.bio.hmm.ProfileHMM-class.html#end,Variable csb.bio.sequence.alignment.AlignmentResult.end=csb.bio.sequence.alignment.AlignmentResult-class.html#end,Variable csb.bio.structure.SecondaryStructure.end=csb.bio.structure.SecondaryStructure-class.html#end,Variable csb.bio.structure.SecondaryStructureElement.end=csb.bio.structure.SecondaryStructureElement-class.html#end,Method csb.core.REMatchProxy.end()=csb.core.REMatchProxy-class.html#end"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-433', 'end', 'link-433');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'.'</tt> </tt>
<a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line"> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-434" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-434', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-string">'start'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-435" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-435', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-string">'end'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line">                <tt id="link-436" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-436', 'start', 'link-240');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-437" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-437', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-438', 'start', 'link-240');">start</a></tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line">                <tt id="link-439" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-439', 'end', 'link-433');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-440" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-440', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-441', 'end', 'link-433');">end</a></tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line">                             </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line">            <tt id="link-442" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-442', 'header', 'link-77');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'P1;{0}\n{2}:{0}:{3}:{1}:{4}:{1}::::'</tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-443', 'format', 'link-100');">format</a></tt><tt class="py-op">(</tt><tt id="link-444" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-444', 'id', 'link-78');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-445" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-445', 'id', 'link-78');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-446" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-446', 'type', 'link-110');">type</a></tt><tt class="py-op">,</tt> <tt id="link-447" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-447', 'start', 'link-240');">start</a></tt><tt class="py-op">,</tt> <tt id="link-448" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-448', 'end', 'link-433');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line">                             </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-449" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-449', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-450" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-450', 'AbstractSequence', 'link-22');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-451', 'DELIMITER', 'link-56');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-452', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-453" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-453', 'header', 'link-77');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line"> </tt>
<a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line">        <tt class="py-name">insertion</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-454" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-454', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-455', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-456', 'INSERTION', 'link-176');">INSERTION</a></tt><tt class="py-op">)</tt> </tt>
<a name="L757"></a><tt class="py-lineno">757</tt>  <tt class="py-line">        <tt id="link-457" class="py-name"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-457', 'gap', 'link-320');">gap</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-458" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-458', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-459" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-459', 'alphabet', 'link-162');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-460', 'GAP', 'link-252');">GAP</a></tt><tt class="py-op">)</tt> </tt>
<a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">         </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-461" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-461', 'write', 'link-44');">write</a></tt><tt class="py-op">(</tt><tt id="link-462" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-462', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-463" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-463', 'sequence', 'link-7');">sequence</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">insertion</tt><tt class="py-op">,</tt> <tt id="link-464" class="py-name"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-464', 'gap', 'link-320');">gap</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-465', 'writeline', 'link-300');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'*'</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line"> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line"> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line"><tt class="py-comment"># register builders</tt> </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line"><tt class="py-keyword">for</tt> <tt class="py-name">klass</tt> <tt class="py-keyword">in</tt> <tt id="link-466" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-466', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt class="py-name">__subclasses__</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">    <tt id="link-467" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-467', 'OutputBuilder', 'link-281');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-468" class="py-name" targets="Static Method csb.bio.io.fasta.OutputBuilder.register()=csb.bio.io.fasta.OutputBuilder-class.html#register"><a title="csb.bio.io.fasta.OutputBuilder.register" class="py-name" href="#" onclick="return doclink('link-468', 'register', 'link-468');">register</a></tt><tt class="py-op">(</tt><tt class="py-name">klass</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="csb.bio.io.fasta.A3MOutputBuilder.FORMAT
csb.bio.io.fasta.FASTAOutputBuilder.FORMAT
csb.bio.io.fasta.PIROutputBuilder.FORMAT" class="py-name" href="#" onclick="return doclink('link-469', 'FORMAT', 'link-307');">FORMAT</a></tt><tt class="py-op">,</tt> <tt class="py-name">klass</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul  4 20:19:12 2017
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
