<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>csb.bio.sequence</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="csb-module.html">Package&nbsp;csb</a> ::
        <a href="csb.bio-module.html">Package&nbsp;bio</a> ::
        Package&nbsp;sequence
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="csb.bio.sequence-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="csb.bio.sequence-module.html">Package csb.bio.sequence</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-docstring">Sequence and sequence alignment APIs.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-docstring">This module defines the base interfaces for biological sequences and alignments:</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-docstring">L{AbstractSequence} and L{AbstractAlignment}. These are the central abstractions</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-docstring">here. This module provides also a number of useful enumerations, like L{SequenceTypes}</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring">and L{SequenceAlphabets}.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">Sequences</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">=========</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">L{AbstractSequence} has a number of implementations. These are of course interchangeable,</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">but have different intents and may differ significantly in performance. The standard</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">L{Sequence} implementation is what you are after if all you need is high performance</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">and efficient storage (e.g. when you are parsing big files). L{Sequence} objects store</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">their underlying sequences as strings. L{RichSequence}s on the other hand will store</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">their residues as L{ResidueInfo} objects, which have the same basic interface as the </tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">L{csb.bio.structure.Residue} objects. This of course comes at the expense of degraded</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">performance. A L{ChainSequence} is a special case of a rich sequence, whose residue</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">objects are I{actually} real L{csb.bio.structure.Residue}s.</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">Basic usage:</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seq = RichSequence('id', 'desc', 'sequence', SequenceTypes.Protein)</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seq.residues[1]</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;ResidueInfo [1]: SER&gt;</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; seq.dump(sys.stdout)</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;desc</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">    SEQUENCE</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">See L{AbstractSequence} for details.    </tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">Alignments</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">==========</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">L{AbstractAlignment} defines a table-like interface to access the data in an</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">alignment:</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ali = SequenceAlignment.parse("&gt;a\\nABC\\n&gt;b\\nA-C")</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ali[0, 0]</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;SequenceAlignment&gt;   # a new alignment, constructed from row #1, column #1</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ali[0, 1:3]</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;SequenceAlignment&gt;   # a new alignment, constructed from row #1, columns #2..#3</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">which is just a shorthand for using the standard 1-based interface:</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ali.rows[1]</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;AlignedSequenceAdapter: a, 3&gt;                        # row #1 (first sequence)</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; ali.columns[1]</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">    (&lt;ColumnInfo a [1]: ALA&gt;, &lt;ColumnInfo b [1]: ALA&gt;)    # residues at column #1</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">See L{AbstractAlignment} for all details and more examples.</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">There are a number of L{AbstractAlignment} implementations defined here.</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">L{SequenceAlignment} is the default one, nothing surprising. L{A3MAlignment}</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">is a more special one: the first sequence in the alignment is a master sequence.</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">This alignment is usually used in the context of HHpred. More important is the</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">L{StructureAlignment}, which is an alignment of L{csb.bio.structure.Chain} objects.</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">The residues in every aligned sequence are really the L{csb.bio.structure.Residue}</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">objects taken from those chains.</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Package csb=csb-module.html"><a title="csb" class="py-name" href="#" onclick="return doclink('link-0', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package csb.core=csb.core-module.html"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-1', 'core', 'link-1');">core</a></tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-2" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-2', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package csb.bio.io=csb.bio.io-module.html,Package csb.io=csb.io-module.html"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-3', 'io', 'link-3');">io</a></tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">abc</tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name" targets="Class abc.ABCMeta=abc.ABCMeta-class.html"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-4', 'ABCMeta', 'link-4');">ABCMeta</a></tt><tt class="py-op">,</tt> <tt class="py-name">abstractmethod</tt><tt class="py-op">,</tt> <tt class="py-name">abstractproperty</tt> </tt>
<a name="AlignmentFormats"></a><div id="AlignmentFormats-def"><a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt> <a class="py-toggle" href="#" id="AlignmentFormats-toggle" onclick="return toggle('AlignmentFormats');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentFormats-class.html">AlignmentFormats</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentFormats-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AlignmentFormats-expanded"><a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-docstring">    Enumeration of multiple sequence alignment formats</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line">    <tt id="link-5" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.A3M=csb.bio.sequence.AlignmentFormats-class.html#A3M"><a title="csb.bio.sequence.AlignmentFormats.A3M" class="py-name" href="#" onclick="return doclink('link-5', 'A3M', 'link-5');">A3M</a></tt><tt class="py-op">=</tt><tt class="py-string">'a3m'</tt><tt class="py-op">;</tt> <tt id="link-6" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.FASTA=csb.bio.sequence.AlignmentFormats-class.html#FASTA"><a title="csb.bio.sequence.AlignmentFormats.FASTA" class="py-name" href="#" onclick="return doclink('link-6', 'FASTA', 'link-6');">FASTA</a></tt><tt class="py-op">=</tt><tt class="py-string">'fa'</tt><tt class="py-op">;</tt> <tt id="link-7" class="py-name" targets="Variable csb.bio.sequence.AlignmentFormats.PIR=csb.bio.sequence.AlignmentFormats-class.html#PIR"><a title="csb.bio.sequence.AlignmentFormats.PIR" class="py-name" href="#" onclick="return doclink('link-7', 'PIR', 'link-7');">PIR</a></tt><tt class="py-op">=</tt><tt class="py-string">'pir'</tt> </tt>
</div><a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"> </tt>
<a name="SequenceTypes"></a><div id="SequenceTypes-def"><a name="L74"></a><tt class="py-lineno">  74</tt> <a class="py-toggle" href="#" id="SequenceTypes-toggle" onclick="return toggle('SequenceTypes');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceTypes-class.html">SequenceTypes</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceTypes-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceTypes-expanded"><a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">    Enumeration of sequence types</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">    <tt id="link-8" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.NucleicAcid=csb.bio.sequence.SequenceTypes-class.html#NucleicAcid"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-8', 'NucleicAcid', 'link-8');">NucleicAcid</a></tt><tt class="py-op">=</tt><tt class="py-string">'NA'</tt><tt class="py-op">;</tt> <tt id="link-9" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.DNA=csb.bio.sequence.SequenceTypes-class.html#DNA"><a title="csb.bio.sequence.SequenceTypes.DNA" class="py-name" href="#" onclick="return doclink('link-9', 'DNA', 'link-9');">DNA</a></tt><tt class="py-op">=</tt><tt class="py-string">'DNA'</tt><tt class="py-op">;</tt> <tt id="link-10" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.RNA=csb.bio.sequence.SequenceTypes-class.html#RNA"><a title="csb.bio.sequence.SequenceTypes.RNA" class="py-name" href="#" onclick="return doclink('link-10', 'RNA', 'link-10');">RNA</a></tt><tt class="py-op">=</tt><tt class="py-string">'RNA'</tt><tt class="py-op">;</tt> <tt id="link-11" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Protein=csb.bio.sequence.SequenceTypes-class.html#Protein"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-11', 'Protein', 'link-11');">Protein</a></tt><tt class="py-op">=</tt><tt class="py-string">'Protein'</tt><tt class="py-op">;</tt> <tt id="link-12" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Unknown=csb.bio.sequence.SequenceTypes-class.html#Unknown"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-12', 'Unknown', 'link-12');">Unknown</a></tt><tt class="py-op">=</tt><tt class="py-string">'Unknown'</tt>     </tt>
</div><a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"> </tt>
<a name="AlignmentTypes"></a><div id="AlignmentTypes-def"><a name="L80"></a><tt class="py-lineno">  80</tt> <a class="py-toggle" href="#" id="AlignmentTypes-toggle" onclick="return toggle('AlignmentTypes');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentTypes-class.html">AlignmentTypes</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentTypes-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AlignmentTypes-expanded"><a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">    Enumeration of alignment strategies</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line">    <tt id="link-13" class="py-name" targets="Variable csb.bio.sequence.AlignmentTypes.Global=csb.bio.sequence.AlignmentTypes-class.html#Global"><a title="csb.bio.sequence.AlignmentTypes.Global" class="py-name" href="#" onclick="return doclink('link-13', 'Global', 'link-13');">Global</a></tt><tt class="py-op">=</tt><tt class="py-string">'global'</tt><tt class="py-op">;</tt> <tt id="link-14" class="py-name" targets="Variable csb.bio.sequence.AlignmentTypes.Local=csb.bio.sequence.AlignmentTypes-class.html#Local"><a title="csb.bio.sequence.AlignmentTypes.Local" class="py-name" href="#" onclick="return doclink('link-14', 'Local', 'link-14');">Local</a></tt><tt class="py-op">=</tt><tt class="py-string">'local'</tt> </tt>
</div><a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"> </tt>
<a name="NucleicAlphabet"></a><div id="NucleicAlphabet-def"><a name="L86"></a><tt class="py-lineno">  86</tt> <a class="py-toggle" href="#" id="NucleicAlphabet-toggle" onclick="return toggle('NucleicAlphabet');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.NucleicAlphabet-class.html">NucleicAlphabet</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleicAlphabet-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="NucleicAlphabet-expanded"><a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">    Nucleic sequence alphabet</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">    <tt id="link-15" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Adenine=csb.bio.sequence.NucleicAlphabet-class.html#Adenine"><a title="csb.bio.sequence.NucleicAlphabet.Adenine" class="py-name" href="#" onclick="return doclink('link-15', 'Adenine', 'link-15');">Adenine</a></tt><tt class="py-op">=</tt><tt class="py-string">'A'</tt><tt class="py-op">;</tt> <tt id="link-16" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Cytosine=csb.bio.sequence.NucleicAlphabet-class.html#Cytosine"><a title="csb.bio.sequence.NucleicAlphabet.Cytosine" class="py-name" href="#" onclick="return doclink('link-16', 'Cytosine', 'link-16');">Cytosine</a></tt><tt class="py-op">=</tt><tt class="py-string">'C'</tt><tt class="py-op">;</tt> <tt id="link-17" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Guanine=csb.bio.sequence.NucleicAlphabet-class.html#Guanine"><a title="csb.bio.sequence.NucleicAlphabet.Guanine" class="py-name" href="#" onclick="return doclink('link-17', 'Guanine', 'link-17');">Guanine</a></tt><tt class="py-op">=</tt><tt class="py-string">'G'</tt><tt class="py-op">;</tt> <tt id="link-18" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Thymine=csb.bio.sequence.NucleicAlphabet-class.html#Thymine"><a title="csb.bio.sequence.NucleicAlphabet.Thymine" class="py-name" href="#" onclick="return doclink('link-18', 'Thymine', 'link-18');">Thymine</a></tt><tt class="py-op">=</tt><tt class="py-string">'T'</tt><tt class="py-op">;</tt> <tt id="link-19" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Uracil=csb.bio.sequence.NucleicAlphabet-class.html#Uracil"><a title="csb.bio.sequence.NucleicAlphabet.Uracil" class="py-name" href="#" onclick="return doclink('link-19', 'Uracil', 'link-19');">Uracil</a></tt><tt class="py-op">=</tt><tt class="py-string">'U'</tt><tt class="py-op">;</tt> <tt id="link-20" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Purine=csb.bio.sequence.NucleicAlphabet-class.html#Purine"><a title="csb.bio.sequence.NucleicAlphabet.Purine" class="py-name" href="#" onclick="return doclink('link-20', 'Purine', 'link-20');">Purine</a></tt><tt class="py-op">=</tt><tt class="py-string">'R'</tt><tt class="py-op">;</tt> <tt id="link-21" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Pyrimidine=csb.bio.sequence.NucleicAlphabet-class.html#Pyrimidine"><a title="csb.bio.sequence.NucleicAlphabet.Pyrimidine" class="py-name" href="#" onclick="return doclink('link-21', 'Pyrimidine', 'link-21');">Pyrimidine</a></tt><tt class="py-op">=</tt><tt class="py-string">'Y'</tt><tt class="py-op">;</tt> <tt id="link-22" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Ketone=csb.bio.sequence.NucleicAlphabet-class.html#Ketone"><a title="csb.bio.sequence.NucleicAlphabet.Ketone" class="py-name" href="#" onclick="return doclink('link-22', 'Ketone', 'link-22');">Ketone</a></tt><tt class="py-op">=</tt><tt class="py-string">'K'</tt><tt class="py-op">;</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">    <tt id="link-23" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Amino=csb.bio.sequence.NucleicAlphabet-class.html#Amino"><a title="csb.bio.sequence.NucleicAlphabet.Amino" class="py-name" href="#" onclick="return doclink('link-23', 'Amino', 'link-23');">Amino</a></tt><tt class="py-op">=</tt><tt class="py-string">'M'</tt><tt class="py-op">;</tt> <tt id="link-24" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Strong=csb.bio.sequence.NucleicAlphabet-class.html#Strong"><a title="csb.bio.sequence.NucleicAlphabet.Strong" class="py-name" href="#" onclick="return doclink('link-24', 'Strong', 'link-24');">Strong</a></tt><tt class="py-op">=</tt><tt class="py-string">'S'</tt><tt class="py-op">;</tt> <tt id="link-25" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Weak=csb.bio.sequence.NucleicAlphabet-class.html#Weak"><a title="csb.bio.sequence.NucleicAlphabet.Weak" class="py-name" href="#" onclick="return doclink('link-25', 'Weak', 'link-25');">Weak</a></tt><tt class="py-op">=</tt><tt class="py-string">'W'</tt><tt class="py-op">;</tt> <tt id="link-26" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.NotA=csb.bio.sequence.NucleicAlphabet-class.html#NotA"><a title="csb.bio.sequence.NucleicAlphabet.NotA" class="py-name" href="#" onclick="return doclink('link-26', 'NotA', 'link-26');">NotA</a></tt><tt class="py-op">=</tt><tt class="py-string">'B'</tt><tt class="py-op">;</tt> <tt id="link-27" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.NotC=csb.bio.sequence.NucleicAlphabet-class.html#NotC"><a title="csb.bio.sequence.NucleicAlphabet.NotC" class="py-name" href="#" onclick="return doclink('link-27', 'NotC', 'link-27');">NotC</a></tt><tt class="py-op">=</tt><tt class="py-string">'D'</tt><tt class="py-op">;</tt> <tt id="link-28" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.NotG=csb.bio.sequence.NucleicAlphabet-class.html#NotG"><a title="csb.bio.sequence.NucleicAlphabet.NotG" class="py-name" href="#" onclick="return doclink('link-28', 'NotG', 'link-28');">NotG</a></tt><tt class="py-op">=</tt><tt class="py-string">'H'</tt><tt class="py-op">;</tt> <tt id="link-29" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.NotT=csb.bio.sequence.NucleicAlphabet-class.html#NotT"><a title="csb.bio.sequence.NucleicAlphabet.NotT" class="py-name" href="#" onclick="return doclink('link-29', 'NotT', 'link-29');">NotT</a></tt><tt class="py-op">=</tt><tt class="py-string">'V'</tt><tt class="py-op">;</tt> <tt id="link-30" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Any=csb.bio.sequence.NucleicAlphabet-class.html#Any"><a title="csb.bio.sequence.NucleicAlphabet.Any" class="py-name" href="#" onclick="return doclink('link-30', 'Any', 'link-30');">Any</a></tt><tt class="py-op">=</tt><tt class="py-string">'N'</tt><tt class="py-op">;</tt> <tt id="link-31" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Masked=csb.bio.sequence.NucleicAlphabet-class.html#Masked"><a title="csb.bio.sequence.NucleicAlphabet.Masked" class="py-name" href="#" onclick="return doclink('link-31', 'Masked', 'link-31');">Masked</a></tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">;</tt> <tt id="link-32" class="py-name" targets="Variable csb.bio.hmm.HHpredHitAlignment.GAP=csb.bio.hmm.HHpredHitAlignment-class.html#GAP,Variable csb.bio.sequence.NucleicAlphabet.GAP=csb.bio.sequence.NucleicAlphabet-class.html#GAP,Variable csb.bio.sequence.ProteinAlphabet.GAP=csb.bio.sequence.ProteinAlphabet-class.html#GAP,Variable csb.bio.sequence.UnknownAlphabet.GAP=csb.bio.sequence.UnknownAlphabet-class.html#GAP"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-32', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt><tt class="py-op">;</tt> <tt id="link-33" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.INSERTION=csb.bio.sequence.NucleicAlphabet-class.html#INSERTION,Variable csb.bio.sequence.ProteinAlphabet.INSERTION=csb.bio.sequence.ProteinAlphabet-class.html#INSERTION,Variable csb.bio.sequence.UnknownAlphabet.INSERTION=csb.bio.sequence.UnknownAlphabet-class.html#INSERTION"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-33', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">=</tt><tt class="py-string">'.'</tt><tt class="py-op">;</tt> </tt>
</div><a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line">     </tt>
<a name="ProteinAlphabet"></a><div id="ProteinAlphabet-def"><a name="L93"></a><tt class="py-lineno">  93</tt> <a class="py-toggle" href="#" id="ProteinAlphabet-toggle" onclick="return toggle('ProteinAlphabet');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ProteinAlphabet-class.html">ProteinAlphabet</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProteinAlphabet-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ProteinAlphabet-expanded"><a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">    Protein sequence alphabet</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line">    <tt id="link-34" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ALA=csb.bio.sequence.ProteinAlphabet-class.html#ALA,Variable csb.bio.sequence.StdProteinAlphabet.ALA=csb.bio.sequence.StdProteinAlphabet-class.html#ALA"><a title="csb.bio.sequence.ProteinAlphabet.ALA
csb.bio.sequence.StdProteinAlphabet.ALA" class="py-name" href="#" onclick="return doclink('link-34', 'ALA', 'link-34');">ALA</a></tt><tt class="py-op">=</tt><tt class="py-string">'A'</tt><tt class="py-op">;</tt> <tt id="link-35" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ASX=csb.bio.sequence.ProteinAlphabet-class.html#ASX"><a title="csb.bio.sequence.ProteinAlphabet.ASX" class="py-name" href="#" onclick="return doclink('link-35', 'ASX', 'link-35');">ASX</a></tt><tt class="py-op">=</tt><tt class="py-string">'B'</tt><tt class="py-op">;</tt> <tt id="link-36" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.CYS=csb.bio.sequence.ProteinAlphabet-class.html#CYS,Variable csb.bio.sequence.StdProteinAlphabet.CYS=csb.bio.sequence.StdProteinAlphabet-class.html#CYS"><a title="csb.bio.sequence.ProteinAlphabet.CYS
csb.bio.sequence.StdProteinAlphabet.CYS" class="py-name" href="#" onclick="return doclink('link-36', 'CYS', 'link-36');">CYS</a></tt><tt class="py-op">=</tt><tt class="py-string">'C'</tt><tt class="py-op">;</tt> <tt id="link-37" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ASP=csb.bio.sequence.ProteinAlphabet-class.html#ASP,Variable csb.bio.sequence.StdProteinAlphabet.ASP=csb.bio.sequence.StdProteinAlphabet-class.html#ASP"><a title="csb.bio.sequence.ProteinAlphabet.ASP
csb.bio.sequence.StdProteinAlphabet.ASP" class="py-name" href="#" onclick="return doclink('link-37', 'ASP', 'link-37');">ASP</a></tt><tt class="py-op">=</tt><tt class="py-string">'D'</tt><tt class="py-op">;</tt> <tt id="link-38" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.GLU=csb.bio.sequence.ProteinAlphabet-class.html#GLU,Variable csb.bio.sequence.StdProteinAlphabet.GLU=csb.bio.sequence.StdProteinAlphabet-class.html#GLU"><a title="csb.bio.sequence.ProteinAlphabet.GLU
csb.bio.sequence.StdProteinAlphabet.GLU" class="py-name" href="#" onclick="return doclink('link-38', 'GLU', 'link-38');">GLU</a></tt><tt class="py-op">=</tt><tt class="py-string">'E'</tt><tt class="py-op">;</tt> <tt id="link-39" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.PHE=csb.bio.sequence.ProteinAlphabet-class.html#PHE,Variable csb.bio.sequence.StdProteinAlphabet.PHE=csb.bio.sequence.StdProteinAlphabet-class.html#PHE"><a title="csb.bio.sequence.ProteinAlphabet.PHE
csb.bio.sequence.StdProteinAlphabet.PHE" class="py-name" href="#" onclick="return doclink('link-39', 'PHE', 'link-39');">PHE</a></tt><tt class="py-op">=</tt><tt class="py-string">'F'</tt><tt class="py-op">;</tt> <tt id="link-40" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.GLY=csb.bio.sequence.ProteinAlphabet-class.html#GLY,Variable csb.bio.sequence.StdProteinAlphabet.GLY=csb.bio.sequence.StdProteinAlphabet-class.html#GLY"><a title="csb.bio.sequence.ProteinAlphabet.GLY
csb.bio.sequence.StdProteinAlphabet.GLY" class="py-name" href="#" onclick="return doclink('link-40', 'GLY', 'link-40');">GLY</a></tt><tt class="py-op">=</tt><tt class="py-string">'G'</tt><tt class="py-op">;</tt> <tt id="link-41" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.HIS=csb.bio.sequence.ProteinAlphabet-class.html#HIS,Variable csb.bio.sequence.StdProteinAlphabet.HIS=csb.bio.sequence.StdProteinAlphabet-class.html#HIS"><a title="csb.bio.sequence.ProteinAlphabet.HIS
csb.bio.sequence.StdProteinAlphabet.HIS" class="py-name" href="#" onclick="return doclink('link-41', 'HIS', 'link-41');">HIS</a></tt><tt class="py-op">=</tt><tt class="py-string">'H'</tt><tt class="py-op">;</tt> <tt id="link-42" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ILE=csb.bio.sequence.ProteinAlphabet-class.html#ILE,Variable csb.bio.sequence.StdProteinAlphabet.ILE=csb.bio.sequence.StdProteinAlphabet-class.html#ILE"><a title="csb.bio.sequence.ProteinAlphabet.ILE
csb.bio.sequence.StdProteinAlphabet.ILE" class="py-name" href="#" onclick="return doclink('link-42', 'ILE', 'link-42');">ILE</a></tt><tt class="py-op">=</tt><tt class="py-string">'I'</tt><tt class="py-op">;</tt> <tt id="link-43" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.LYS=csb.bio.sequence.ProteinAlphabet-class.html#LYS,Variable csb.bio.sequence.StdProteinAlphabet.LYS=csb.bio.sequence.StdProteinAlphabet-class.html#LYS"><a title="csb.bio.sequence.ProteinAlphabet.LYS
csb.bio.sequence.StdProteinAlphabet.LYS" class="py-name" href="#" onclick="return doclink('link-43', 'LYS', 'link-43');">LYS</a></tt><tt class="py-op">=</tt><tt class="py-string">'K'</tt><tt class="py-op">;</tt> <tt id="link-44" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.LEU=csb.bio.sequence.ProteinAlphabet-class.html#LEU,Variable csb.bio.sequence.StdProteinAlphabet.LEU=csb.bio.sequence.StdProteinAlphabet-class.html#LEU"><a title="csb.bio.sequence.ProteinAlphabet.LEU
csb.bio.sequence.StdProteinAlphabet.LEU" class="py-name" href="#" onclick="return doclink('link-44', 'LEU', 'link-44');">LEU</a></tt><tt class="py-op">=</tt><tt class="py-string">'L'</tt><tt class="py-op">;</tt> <tt id="link-45" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.MET=csb.bio.sequence.ProteinAlphabet-class.html#MET,Variable csb.bio.sequence.StdProteinAlphabet.MET=csb.bio.sequence.StdProteinAlphabet-class.html#MET"><a title="csb.bio.sequence.ProteinAlphabet.MET
csb.bio.sequence.StdProteinAlphabet.MET" class="py-name" href="#" onclick="return doclink('link-45', 'MET', 'link-45');">MET</a></tt><tt class="py-op">=</tt><tt class="py-string">'M'</tt><tt class="py-op">;</tt> <tt id="link-46" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ASN=csb.bio.sequence.ProteinAlphabet-class.html#ASN,Variable csb.bio.sequence.StdProteinAlphabet.ASN=csb.bio.sequence.StdProteinAlphabet-class.html#ASN"><a title="csb.bio.sequence.ProteinAlphabet.ASN
csb.bio.sequence.StdProteinAlphabet.ASN" class="py-name" href="#" onclick="return doclink('link-46', 'ASN', 'link-46');">ASN</a></tt><tt class="py-op">=</tt><tt class="py-string">'N'</tt><tt class="py-op">;</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">    <tt id="link-47" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.PYL=csb.bio.sequence.ProteinAlphabet-class.html#PYL"><a title="csb.bio.sequence.ProteinAlphabet.PYL" class="py-name" href="#" onclick="return doclink('link-47', 'PYL', 'link-47');">PYL</a></tt><tt class="py-op">=</tt><tt class="py-string">'O'</tt><tt class="py-op">;</tt> <tt id="link-48" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.PRO=csb.bio.sequence.ProteinAlphabet-class.html#PRO,Variable csb.bio.sequence.StdProteinAlphabet.PRO=csb.bio.sequence.StdProteinAlphabet-class.html#PRO"><a title="csb.bio.sequence.ProteinAlphabet.PRO
csb.bio.sequence.StdProteinAlphabet.PRO" class="py-name" href="#" onclick="return doclink('link-48', 'PRO', 'link-48');">PRO</a></tt><tt class="py-op">=</tt><tt class="py-string">'P'</tt><tt class="py-op">;</tt> <tt id="link-49" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.GLN=csb.bio.sequence.ProteinAlphabet-class.html#GLN,Variable csb.bio.sequence.StdProteinAlphabet.GLN=csb.bio.sequence.StdProteinAlphabet-class.html#GLN"><a title="csb.bio.sequence.ProteinAlphabet.GLN
csb.bio.sequence.StdProteinAlphabet.GLN" class="py-name" href="#" onclick="return doclink('link-49', 'GLN', 'link-49');">GLN</a></tt><tt class="py-op">=</tt><tt class="py-string">'Q'</tt><tt class="py-op">;</tt> <tt id="link-50" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.ARG=csb.bio.sequence.ProteinAlphabet-class.html#ARG,Variable csb.bio.sequence.StdProteinAlphabet.ARG=csb.bio.sequence.StdProteinAlphabet-class.html#ARG"><a title="csb.bio.sequence.ProteinAlphabet.ARG
csb.bio.sequence.StdProteinAlphabet.ARG" class="py-name" href="#" onclick="return doclink('link-50', 'ARG', 'link-50');">ARG</a></tt><tt class="py-op">=</tt><tt class="py-string">'R'</tt><tt class="py-op">;</tt> <tt id="link-51" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.SER=csb.bio.sequence.ProteinAlphabet-class.html#SER,Variable csb.bio.sequence.StdProteinAlphabet.SER=csb.bio.sequence.StdProteinAlphabet-class.html#SER"><a title="csb.bio.sequence.ProteinAlphabet.SER
csb.bio.sequence.StdProteinAlphabet.SER" class="py-name" href="#" onclick="return doclink('link-51', 'SER', 'link-51');">SER</a></tt><tt class="py-op">=</tt><tt class="py-string">'S'</tt><tt class="py-op">;</tt> <tt id="link-52" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.THR=csb.bio.sequence.ProteinAlphabet-class.html#THR,Variable csb.bio.sequence.StdProteinAlphabet.THR=csb.bio.sequence.StdProteinAlphabet-class.html#THR"><a title="csb.bio.sequence.ProteinAlphabet.THR
csb.bio.sequence.StdProteinAlphabet.THR" class="py-name" href="#" onclick="return doclink('link-52', 'THR', 'link-52');">THR</a></tt><tt class="py-op">=</tt><tt class="py-string">'T'</tt><tt class="py-op">;</tt> <tt id="link-53" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.SEC=csb.bio.sequence.ProteinAlphabet-class.html#SEC"><a title="csb.bio.sequence.ProteinAlphabet.SEC" class="py-name" href="#" onclick="return doclink('link-53', 'SEC', 'link-53');">SEC</a></tt><tt class="py-op">=</tt><tt class="py-string">'U'</tt><tt class="py-op">;</tt> <tt id="link-54" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.VAL=csb.bio.sequence.ProteinAlphabet-class.html#VAL,Variable csb.bio.sequence.StdProteinAlphabet.VAL=csb.bio.sequence.StdProteinAlphabet-class.html#VAL"><a title="csb.bio.sequence.ProteinAlphabet.VAL
csb.bio.sequence.StdProteinAlphabet.VAL" class="py-name" href="#" onclick="return doclink('link-54', 'VAL', 'link-54');">VAL</a></tt><tt class="py-op">=</tt><tt class="py-string">'V'</tt><tt class="py-op">;</tt> <tt id="link-55" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.TRP=csb.bio.sequence.ProteinAlphabet-class.html#TRP,Variable csb.bio.sequence.StdProteinAlphabet.TRP=csb.bio.sequence.StdProteinAlphabet-class.html#TRP"><a title="csb.bio.sequence.ProteinAlphabet.TRP
csb.bio.sequence.StdProteinAlphabet.TRP" class="py-name" href="#" onclick="return doclink('link-55', 'TRP', 'link-55');">TRP</a></tt><tt class="py-op">=</tt><tt class="py-string">'W'</tt><tt class="py-op">;</tt> <tt id="link-56" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.TYR=csb.bio.sequence.ProteinAlphabet-class.html#TYR,Variable csb.bio.sequence.StdProteinAlphabet.TYR=csb.bio.sequence.StdProteinAlphabet-class.html#TYR"><a title="csb.bio.sequence.ProteinAlphabet.TYR
csb.bio.sequence.StdProteinAlphabet.TYR" class="py-name" href="#" onclick="return doclink('link-56', 'TYR', 'link-56');">TYR</a></tt><tt class="py-op">=</tt><tt class="py-string">'Y'</tt><tt class="py-op">;</tt> <tt id="link-57" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.GLX=csb.bio.sequence.ProteinAlphabet-class.html#GLX"><a title="csb.bio.sequence.ProteinAlphabet.GLX" class="py-name" href="#" onclick="return doclink('link-57', 'GLX', 'link-57');">GLX</a></tt><tt class="py-op">=</tt><tt class="py-string">'Z'</tt><tt class="py-op">;</tt> <tt id="link-58" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.UNK=csb.bio.sequence.ProteinAlphabet-class.html#UNK,Variable csb.bio.sequence.UnknownAlphabet.UNK=csb.bio.sequence.UnknownAlphabet-class.html#UNK"><a title="csb.bio.sequence.ProteinAlphabet.UNK
csb.bio.sequence.UnknownAlphabet.UNK" class="py-name" href="#" onclick="return doclink('link-58', 'UNK', 'link-58');">UNK</a></tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">;</tt> <tt id="link-59" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-59', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt><tt class="py-op">;</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">    <tt id="link-60" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-60', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">=</tt><tt class="py-string">'.'</tt><tt class="py-op">;</tt> <tt id="link-61" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.STOP=csb.bio.sequence.ProteinAlphabet-class.html#STOP"><a title="csb.bio.sequence.ProteinAlphabet.STOP" class="py-name" href="#" onclick="return doclink('link-61', 'STOP', 'link-61');">STOP</a></tt><tt class="py-op">=</tt><tt class="py-string">'*'</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line">                                     </tt>
<a name="StdProteinAlphabet"></a><div id="StdProteinAlphabet-def"><a name="L101"></a><tt class="py-lineno"> 101</tt> <a class="py-toggle" href="#" id="StdProteinAlphabet-toggle" onclick="return toggle('StdProteinAlphabet');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.StdProteinAlphabet-class.html">StdProteinAlphabet</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StdProteinAlphabet-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StdProteinAlphabet-expanded"><a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">    Standard protein sequence alphabet</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>       </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">    <tt id="link-62" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.ALA
csb.bio.sequence.StdProteinAlphabet.ALA" class="py-name" href="#" onclick="return doclink('link-62', 'ALA', 'link-34');">ALA</a></tt><tt class="py-op">=</tt><tt class="py-string">'A'</tt><tt class="py-op">;</tt> <tt id="link-63" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.CYS
csb.bio.sequence.StdProteinAlphabet.CYS" class="py-name" href="#" onclick="return doclink('link-63', 'CYS', 'link-36');">CYS</a></tt><tt class="py-op">=</tt><tt class="py-string">'C'</tt><tt class="py-op">;</tt> <tt id="link-64" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.ASP
csb.bio.sequence.StdProteinAlphabet.ASP" class="py-name" href="#" onclick="return doclink('link-64', 'ASP', 'link-37');">ASP</a></tt><tt class="py-op">=</tt><tt class="py-string">'D'</tt><tt class="py-op">;</tt> <tt id="link-65" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.GLU
csb.bio.sequence.StdProteinAlphabet.GLU" class="py-name" href="#" onclick="return doclink('link-65', 'GLU', 'link-38');">GLU</a></tt><tt class="py-op">=</tt><tt class="py-string">'E'</tt><tt class="py-op">;</tt> <tt id="link-66" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.PHE
csb.bio.sequence.StdProteinAlphabet.PHE" class="py-name" href="#" onclick="return doclink('link-66', 'PHE', 'link-39');">PHE</a></tt><tt class="py-op">=</tt><tt class="py-string">'F'</tt><tt class="py-op">;</tt> <tt id="link-67" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.GLY
csb.bio.sequence.StdProteinAlphabet.GLY" class="py-name" href="#" onclick="return doclink('link-67', 'GLY', 'link-40');">GLY</a></tt><tt class="py-op">=</tt><tt class="py-string">'G'</tt><tt class="py-op">;</tt> <tt id="link-68" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.HIS
csb.bio.sequence.StdProteinAlphabet.HIS" class="py-name" href="#" onclick="return doclink('link-68', 'HIS', 'link-41');">HIS</a></tt><tt class="py-op">=</tt><tt class="py-string">'H'</tt><tt class="py-op">;</tt> <tt id="link-69" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.ILE
csb.bio.sequence.StdProteinAlphabet.ILE" class="py-name" href="#" onclick="return doclink('link-69', 'ILE', 'link-42');">ILE</a></tt><tt class="py-op">=</tt><tt class="py-string">'I'</tt><tt class="py-op">;</tt> <tt id="link-70" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.LYS
csb.bio.sequence.StdProteinAlphabet.LYS" class="py-name" href="#" onclick="return doclink('link-70', 'LYS', 'link-43');">LYS</a></tt><tt class="py-op">=</tt><tt class="py-string">'K'</tt><tt class="py-op">;</tt> <tt id="link-71" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.LEU
csb.bio.sequence.StdProteinAlphabet.LEU" class="py-name" href="#" onclick="return doclink('link-71', 'LEU', 'link-44');">LEU</a></tt><tt class="py-op">=</tt><tt class="py-string">'L'</tt><tt class="py-op">;</tt> <tt id="link-72" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.MET
csb.bio.sequence.StdProteinAlphabet.MET" class="py-name" href="#" onclick="return doclink('link-72', 'MET', 'link-45');">MET</a></tt><tt class="py-op">=</tt><tt class="py-string">'M'</tt><tt class="py-op">;</tt> <tt id="link-73" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.ASN
csb.bio.sequence.StdProteinAlphabet.ASN" class="py-name" href="#" onclick="return doclink('link-73', 'ASN', 'link-46');">ASN</a></tt><tt class="py-op">=</tt><tt class="py-string">'N'</tt><tt class="py-op">;</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">    <tt id="link-74" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.PRO
csb.bio.sequence.StdProteinAlphabet.PRO" class="py-name" href="#" onclick="return doclink('link-74', 'PRO', 'link-48');">PRO</a></tt><tt class="py-op">=</tt><tt class="py-string">'P'</tt><tt class="py-op">;</tt> <tt id="link-75" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.GLN
csb.bio.sequence.StdProteinAlphabet.GLN" class="py-name" href="#" onclick="return doclink('link-75', 'GLN', 'link-49');">GLN</a></tt><tt class="py-op">=</tt><tt class="py-string">'Q'</tt><tt class="py-op">;</tt> <tt id="link-76" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.ARG
csb.bio.sequence.StdProteinAlphabet.ARG" class="py-name" href="#" onclick="return doclink('link-76', 'ARG', 'link-50');">ARG</a></tt><tt class="py-op">=</tt><tt class="py-string">'R'</tt><tt class="py-op">;</tt> <tt id="link-77" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.SER
csb.bio.sequence.StdProteinAlphabet.SER" class="py-name" href="#" onclick="return doclink('link-77', 'SER', 'link-51');">SER</a></tt><tt class="py-op">=</tt><tt class="py-string">'S'</tt><tt class="py-op">;</tt> <tt id="link-78" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.THR
csb.bio.sequence.StdProteinAlphabet.THR" class="py-name" href="#" onclick="return doclink('link-78', 'THR', 'link-52');">THR</a></tt><tt class="py-op">=</tt><tt class="py-string">'T'</tt><tt class="py-op">;</tt>  <tt id="link-79" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.VAL
csb.bio.sequence.StdProteinAlphabet.VAL" class="py-name" href="#" onclick="return doclink('link-79', 'VAL', 'link-54');">VAL</a></tt><tt class="py-op">=</tt><tt class="py-string">'V'</tt><tt class="py-op">;</tt> <tt id="link-80" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.TRP
csb.bio.sequence.StdProteinAlphabet.TRP" class="py-name" href="#" onclick="return doclink('link-80', 'TRP', 'link-55');">TRP</a></tt><tt class="py-op">=</tt><tt class="py-string">'W'</tt><tt class="py-op">;</tt> <tt id="link-81" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.TYR
csb.bio.sequence.StdProteinAlphabet.TYR" class="py-name" href="#" onclick="return doclink('link-81', 'TYR', 'link-56');">TYR</a></tt><tt class="py-op">=</tt><tt class="py-string">'Y'</tt> </tt>
</div><a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">     </tt>
<a name="UnknownAlphabet"></a><div id="UnknownAlphabet-def"><a name="L108"></a><tt class="py-lineno"> 108</tt> <a class="py-toggle" href="#" id="UnknownAlphabet-toggle" onclick="return toggle('UnknownAlphabet');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.UnknownAlphabet-class.html">UnknownAlphabet</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">enum</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownAlphabet-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="UnknownAlphabet-expanded"><a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">    Unknown sequence alphabet</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>   </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">    <tt id="link-82" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.UNK
csb.bio.sequence.UnknownAlphabet.UNK" class="py-name" href="#" onclick="return doclink('link-82', 'UNK', 'link-58');">UNK</a></tt><tt class="py-op">=</tt><tt class="py-string">'X'</tt><tt class="py-op">;</tt> <tt id="link-83" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-83', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">=</tt><tt class="py-string">'-'</tt><tt class="py-op">;</tt> <tt id="link-84" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-84', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">=</tt><tt class="py-string">'.'</tt> </tt>
</div><a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">    </tt>
<a name="SequenceAlphabets"></a><div id="SequenceAlphabets-def"><a name="L114"></a><tt class="py-lineno"> 114</tt> <a class="py-toggle" href="#" id="SequenceAlphabets-toggle" onclick="return toggle('SequenceAlphabets');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlphabets-class.html">SequenceAlphabets</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlphabets-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceAlphabets-expanded"><a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence alphabet enumerations.</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">    @note: This class is kept for backwards compatibility. The individual</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">           alphabet classes must be defined in the top level namespace,</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">           otherwise the new enum types cannot be pickled properly. </tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">    <tt class="py-name">Nucleic</tt> <tt class="py-op">=</tt> <tt id="link-85" class="py-name" targets="Class csb.bio.sequence.NucleicAlphabet=csb.bio.sequence.NucleicAlphabet-class.html"><a title="csb.bio.sequence.NucleicAlphabet" class="py-name" href="#" onclick="return doclink('link-85', 'NucleicAlphabet', 'link-85');">NucleicAlphabet</a></tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">    <tt id="link-86" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-86', 'Protein', 'link-11');">Protein</a></tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name" targets="Class csb.bio.sequence.ProteinAlphabet=csb.bio.sequence.ProteinAlphabet-class.html"><a title="csb.bio.sequence.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-87', 'ProteinAlphabet', 'link-87');">ProteinAlphabet</a></tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">    <tt class="py-name">StdProtein</tt> <tt class="py-op">=</tt> <tt id="link-88" class="py-name" targets="Class csb.bio.sequence.StdProteinAlphabet=csb.bio.sequence.StdProteinAlphabet-class.html"><a title="csb.bio.sequence.StdProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-88', 'StdProteinAlphabet', 'link-88');">StdProteinAlphabet</a></tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">    <tt id="link-89" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-89', 'Unknown', 'link-12');">Unknown</a></tt> <tt class="py-op">=</tt> <tt id="link-90" class="py-name" targets="Class csb.bio.sequence.UnknownAlphabet=csb.bio.sequence.UnknownAlphabet-class.html"><a title="csb.bio.sequence.UnknownAlphabet" class="py-name" href="#" onclick="return doclink('link-90', 'UnknownAlphabet', 'link-90');">UnknownAlphabet</a></tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">     </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">    <tt id="link-91" class="py-name" targets="Variable csb.bio.sequence.SequenceAlphabets.MAP=csb.bio.sequence.SequenceAlphabets-class.html#MAP"><a title="csb.bio.sequence.SequenceAlphabets.MAP" class="py-name" href="#" onclick="return doclink('link-91', 'MAP', 'link-91');">MAP</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> <tt id="link-92" class="py-name" targets="Class csb.bio.sequence.SequenceTypes=csb.bio.sequence.SequenceTypes-class.html"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-92', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-93', 'Protein', 'link-11');">Protein</a></tt><tt class="py-op">:</tt> <tt id="link-94" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-94', 'ProteinAlphabet', 'link-87');">ProteinAlphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">            <tt id="link-95" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-95', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-96', 'NucleicAcid', 'link-8');">NucleicAcid</a></tt><tt class="py-op">:</tt> <tt id="link-97" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet" class="py-name" href="#" onclick="return doclink('link-97', 'NucleicAlphabet', 'link-85');">NucleicAlphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">            <tt id="link-98" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-98', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="csb.bio.sequence.SequenceTypes.DNA" class="py-name" href="#" onclick="return doclink('link-99', 'DNA', 'link-9');">DNA</a></tt><tt class="py-op">:</tt> <tt id="link-100" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet" class="py-name" href="#" onclick="return doclink('link-100', 'NucleicAlphabet', 'link-85');">NucleicAlphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">            <tt id="link-101" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-101', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="csb.bio.sequence.SequenceTypes.RNA" class="py-name" href="#" onclick="return doclink('link-102', 'RNA', 'link-10');">RNA</a></tt><tt class="py-op">:</tt> <tt id="link-103" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet" class="py-name" href="#" onclick="return doclink('link-103', 'NucleicAlphabet', 'link-85');">NucleicAlphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">            <tt id="link-104" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-104', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-105', 'Unknown', 'link-12');">Unknown</a></tt><tt class="py-op">:</tt> <tt id="link-106" class="py-name"><a title="csb.bio.sequence.UnknownAlphabet" class="py-name" href="#" onclick="return doclink('link-106', 'UnknownAlphabet', 'link-90');">UnknownAlphabet</a></tt> <tt class="py-op">}</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">     </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">    <tt id="link-107" class="py-name" targets="Variable csb.bio.sequence.SequenceAlphabets.ALL_ALPHABETS=csb.bio.sequence.SequenceAlphabets-class.html#ALL_ALPHABETS"><a title="csb.bio.sequence.SequenceAlphabets.ALL_ALPHABETS" class="py-name" href="#" onclick="return doclink('link-107', 'ALL_ALPHABETS', 'link-107');">ALL_ALPHABETS</a></tt> <tt class="py-op">=</tt> <tt id="link-108" class="py-name" targets="Method csb.bio.hmm.EmissionTable.set()=csb.bio.hmm.EmissionTable-class.html#set,Method csb.bio.hmm.TransitionTable.set()=csb.bio.hmm.TransitionTable-class.html#set,Method csb.statistics.maxent.MaxentModel.set()=csb.statistics.maxent.MaxentModel-class.html#set,Method csb.statistics.pdf.parameterized.AbstractParameter.set()=csb.statistics.pdf.parameterized.AbstractParameter-class.html#set"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-108', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-109" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-109', 'ProteinAlphabet', 'link-87');">ProteinAlphabet</a></tt><tt class="py-op">,</tt> <tt id="link-110" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet" class="py-name" href="#" onclick="return doclink('link-110', 'NucleicAlphabet', 'link-85');">NucleicAlphabet</a></tt><tt class="py-op">,</tt> <tt id="link-111" class="py-name"><a title="csb.bio.sequence.UnknownAlphabet" class="py-name" href="#" onclick="return doclink('link-111', 'UnknownAlphabet', 'link-90');">UnknownAlphabet</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-112" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-112', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt id="link-113" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets.MAP" class="py-name" href="#" onclick="return doclink('link-113', 'MAP', 'link-91');">MAP</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-114" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-114', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-115', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Class csb.core.Enum=csb.core.Enum-class.html"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-116', 'Enum', 'link-116');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name" targets="Variable csb.bio.io.clans.ClansSeqgroup.members=csb.bio.io.clans.ClansSeqgroup-class.html#members,Static Method csb.core.Enum.members()=csb.core.Enum-class.html#members"><a title="csb.bio.io.clans.ClansSeqgroup.members
csb.core.Enum.members" class="py-name" href="#" onclick="return doclink('link-117', 'members', 'link-117');">members</a></tt><tt class="py-op">(</tt><tt id="link-118" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-118', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">     </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="SequenceAlphabets.get"></a><div id="SequenceAlphabets.get-def"><a name="L138"></a><tt class="py-lineno"> 138</tt> <a class="py-toggle" href="#" id="SequenceAlphabets.get-toggle" onclick="return toggle('SequenceAlphabets.get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlphabets-class.html#get">get</a><tt class="py-op">(</tt><tt class="py-param">type</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlphabets.get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAlphabets.get-expanded"><a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring">        Get the alphabet corresponding to the specified sequence C{type}</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">        @param type: a member of L{SequenceTypes}</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">        @type type: L{csb.core.EnumItem}</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.core.enum} </tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-119" class="py-name" targets="Class csb.bio.sequence.SequenceAlphabets=csb.bio.sequence.SequenceAlphabets-class.html"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-119', 'SequenceAlphabets', 'link-119');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets.MAP" class="py-name" href="#" onclick="return doclink('link-120', 'MAP', 'link-91');">MAP</a></tt><tt class="py-op">[</tt><tt id="link-121" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.type=csb.bio.fragments.ResidueEventInfo-class.html#type,Variable csb.bio.fragments.TargetResidue.type=csb.bio.fragments.TargetResidue-class.html#type,Variable csb.bio.hmm.State.type=csb.bio.hmm.State-class.html#type,Variable csb.bio.hmm.Transition.type=csb.bio.hmm.Transition-class.html#type,Variable csb.bio.io.clans.ClansSeqgroup.type=csb.bio.io.clans.ClansSeqgroup-class.html#type,Variable csb.bio.io.wwpdb.ResidueInfo.type=csb.bio.io.wwpdb.ResidueInfo-class.html#type,Variable csb.bio.sequence.AbstractSequence.type=csb.bio.sequence.AbstractSequence-class.html#type,Variable csb.bio.sequence.ResidueInfo.type=csb.bio.sequence.ResidueInfo-class.html#type,Variable csb.bio.sequence.SequenceAdapter.type=csb.bio.sequence.SequenceAdapter-class.html#type,Variable csb.bio.structure.Chain.type=csb.bio.structure.Chain-class.html#type,Variable csb.bio.structure.Residue.type=csb.bio.structure.Residue-class.html#type,Variable csb.bio.structure.SecondaryStructureElement.type=csb.bio.structure.SecondaryStructureElement-class.html#type,Variable csb.io.tsv.ColumnInfo.type=csb.io.tsv.ColumnInfo-class.html#type"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-121', 'type', 'link-121');">type</a></tt><tt class="py-op">]</tt>   </tt>
</div><a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">     </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="SequenceAlphabets.contains"></a><div id="SequenceAlphabets.contains-def"><a name="L148"></a><tt class="py-lineno"> 148</tt> <a class="py-toggle" href="#" id="SequenceAlphabets.contains-toggle" onclick="return toggle('SequenceAlphabets.contains');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlphabets-class.html#contains">contains</a><tt class="py-op">(</tt><tt class="py-param">alphabet</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlphabets.contains-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAlphabets.contains-expanded"><a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">        Return True if C{alphabet} is a sequence alphabet</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">        @type alphabet: L{csb.core.enum}</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-122" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.alphabet=csb.bio.sequence.AbstractSequence-class.html#alphabet,Variable csb.bio.sequence.SequenceAdapter.alphabet=csb.bio.sequence.SequenceAdapter-class.html#alphabet,Variable csb.bio.structure.Chain.alphabet=csb.bio.structure.Chain-class.html#alphabet"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-122', 'alphabet', 'link-122');">alphabet</a></tt> <tt class="py-keyword">in</tt> <tt id="link-123" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-123', 'SequenceAlphabets', 'link-119');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets.ALL_ALPHABETS" class="py-name" href="#" onclick="return doclink('link-124', 'ALL_ALPHABETS', 'link-107');">ALL_ALPHABETS</a></tt>         </tt>
</div></div><a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"> </tt>
<a name="SequenceError"></a><div id="SequenceError-def"><a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt> <a class="py-toggle" href="#" id="SequenceError-toggle" onclick="return toggle('SequenceError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceError-class.html">SequenceError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceError-expanded"><a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"> </tt>
<a name="PositionError"></a><div id="PositionError-def"><a name="L160"></a><tt class="py-lineno"> 160</tt> <a class="py-toggle" href="#" id="PositionError-toggle" onclick="return toggle('PositionError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.PositionError-class.html">PositionError</a><tt class="py-op">(</tt><tt class="py-base-class">IndexError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PositionError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PositionError-expanded"><a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">     </tt>
<a name="PositionError.__init__"></a><div id="PositionError.__init__-def"><a name="L162"></a><tt class="py-lineno"> 162</tt> <a class="py-toggle" href="#" id="PositionError.__init__-toggle" onclick="return toggle('PositionError.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.PositionError-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PositionError.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PositionError.__init__-expanded"><a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">         </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-125" class="py-name" targets="Variable csb.bio.fragments.Assignment.end=csb.bio.fragments.Assignment-class.html#end,Variable csb.bio.fragments.FragmentMatch.end=csb.bio.fragments.FragmentMatch-class.html#end,Variable csb.bio.fragments.rosetta.RosettaFragment.end=csb.bio.fragments.rosetta.RosettaFragment-class.html#end,Variable csb.bio.hmm.HHpredHit.end=csb.bio.hmm.HHpredHit-class.html#end,Variable csb.bio.hmm.ProfileHMM.end=csb.bio.hmm.ProfileHMM-class.html#end,Variable csb.bio.sequence.alignment.AlignmentResult.end=csb.bio.sequence.alignment.AlignmentResult-class.html#end,Variable csb.bio.structure.SecondaryStructure.end=csb.bio.structure.SecondaryStructure-class.html#end,Variable csb.bio.structure.SecondaryStructureElement.end=csb.bio.structure.SecondaryStructureElement-class.html#end,Method csb.core.REMatchProxy.end()=csb.core.REMatchProxy-class.html#end"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-125', 'end', 'link-125');">end</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">            <tt id="link-126" class="py-name" targets="Variable csb.bio.fragments.Assignment.start=csb.bio.fragments.Assignment-class.html#start,Variable csb.bio.fragments.FragmentMatch.start=csb.bio.fragments.FragmentMatch-class.html#start,Variable csb.bio.fragments.rosetta.RosettaFragment.start=csb.bio.fragments.rosetta.RosettaFragment-class.html#start,Variable csb.bio.hmm.HHpredHit.start=csb.bio.hmm.HHpredHit-class.html#start,Variable csb.bio.hmm.ProfileHMM.start=csb.bio.hmm.ProfileHMM-class.html#start,Variable csb.bio.sequence.alignment.AlignmentResult.start=csb.bio.sequence.alignment.AlignmentResult-class.html#start,Variable csb.bio.structure.SecondaryStructure.start=csb.bio.structure.SecondaryStructure-class.html#start,Variable csb.bio.structure.SecondaryStructureElement.start=csb.bio.structure.SecondaryStructureElement-class.html#start,Method csb.core.REMatchProxy.start()=csb.core.REMatchProxy-class.html#start,Method csb.io.plots.Backend.start()=csb.io.plots.Backend-class.html#start"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-126', 'start', 'link-126');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">             </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">index</tt> <tt class="py-op">=</tt> <tt class="py-name">index</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-127', 'start', 'link-126');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-128" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-128', 'start', 'link-126');">start</a></tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-129', 'end', 'link-125');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-130" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-130', 'end', 'link-125');">end</a></tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">         </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-131" class="py-name" targets="Class csb.bio.sequence.PositionError=csb.bio.sequence.PositionError-class.html"><a title="csb.bio.sequence.PositionError" class="py-name" href="#" onclick="return doclink('link-131', 'PositionError', 'link-131');">PositionError</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name" targets="Method csb.Version.__init__()=csb.Version-class.html#__init__,Method csb.apps.AppExit.__init__()=csb.apps.AppExit-class.html#__init__,Method csb.apps.AppRunner.__init__()=csb.apps.AppRunner-class.html#__init__,Method csb.apps.Application.__init__()=csb.apps.Application-class.html#__init__,Method csb.apps.ArgHandler.__init__()=csb.apps.ArgHandler-class.html#__init__,Method csb.apps.buildhmm.NoOutputError.__init__()=csb.apps.buildhmm.NoOutputError-class.html#__init__,Method csb.apps.buildhmm.ProfileBuilder.__init__()=csb.apps.buildhmm.ProfileBuilder-class.html#__init__,Method csb.apps.csfrag.CSfrag.__init__()=csb.apps.csfrag.CSfrag-class.html#__init__,Method csb.apps.csfrag.MatchInfo.__init__()=csb.apps.csfrag.MatchInfo-class.html#__init__,Method csb.apps.csfrag.MatchTable.__init__()=csb.apps.csfrag.MatchTable-class.html#__init__,Method csb.apps.csfrag.ScoringHelper.__init__()=csb.apps.csfrag.ScoringHelper-class.html#__init__,Method csb.apps.embd.Deconvolution.__init__()=csb.apps.embd.Deconvolution-class.html#__init__,Method csb.apps.embd.OutputPathBuilder.__init__()=csb.apps.embd.OutputPathBuilder-class.html#__init__,Method csb.apps.hhfrag.HHfrag.__init__()=csb.apps.hhfrag.HHfrag-class.html#__init__,Method csb.apps.hhfrag.PredictionBuilder.__init__()=csb.apps.hhfrag.PredictionBuilder-class.html#__init__,Method csb.apps.hhfrag.SliceContext.__init__()=csb.apps.hhfrag.SliceContext-class.html#__init__,Method csb.apps.hhsearch.Context.__init__()=csb.apps.hhsearch.Context-class.html#__init__,Method csb.apps.hhsearch.HHTask.__init__()=csb.apps.hhsearch.HHTask-class.html#__init__,Method csb.apps.hhsearch.HHsearch.__init__()=csb.apps.hhsearch.HHsearch-class.html#__init__,Method csb.apps.precision.GlobalInfo.__init__()=csb.apps.precision.GlobalInfo-class.html#__init__,Method csb.apps.precision.LibrarySuperimposer.__init__()=csb.apps.precision.LibrarySuperimposer-class.html#__init__,Method csb.bio.fragments.Assignment.__init__()=csb.bio.fragments.Assignment-class.html#__init__,Method csb.bio.fragments.BenchmarkAdapter.Connection.__init__()=csb.bio.fragments.BenchmarkAdapter.Connection-class.html#__init__,Method csb.bio.fragments.BenchmarkAdapter.__init__()=csb.bio.fragments.BenchmarkAdapter-class.html#__init__,Method csb.bio.fragments.ChemShiftAssignment.__init__()=csb.bio.fragments.ChemShiftAssignment-class.html#__init__,Method csb.bio.fragments.ChemShiftTarget.__init__()=csb.bio.fragments.ChemShiftTarget-class.html#__init__,Method csb.bio.fragments.ClusterNode.__init__()=csb.bio.fragments.ClusterNode-class.html#__init__,Method csb.bio.fragments.ClusterRep.__init__()=csb.bio.fragments.ClusterRep-class.html#__init__,Method csb.bio.fragments.FragmentCluster.__init__()=csb.bio.fragments.FragmentCluster-class.html#__init__,Method csb.bio.fragments.FragmentMatch.__init__()=csb.bio.fragments.FragmentMatch-class.html#__init__,Method csb.bio.fragments.ResidueAssignmentInfo.__init__()=csb.bio.fragments.ResidueAssignmentInfo-class.html#__init__,Method csb.bio.fragments.ResidueEventInfo.__init__()=csb.bio.fragments.ResidueEventInfo-class.html#__init__,Method csb.bio.fragments.RosettaFragsetFactory.__init__()=csb.bio.fragments.RosettaFragsetFactory-class.html#__init__,Method csb.bio.fragments.Target.__init__()=csb.bio.fragments.Target-class.html#__init__,Method csb.bio.fragments.TargetResidue.__init__()=csb.bio.fragments.TargetResidue-class.html#__init__,Method csb.bio.fragments.TorsionAnglePredictor.__init__()=csb.bio.fragments.TorsionAnglePredictor-class.html#__init__,Method csb.bio.fragments.TorsionPredictionInfo.__init__()=csb.bio.fragments.TorsionPredictionInfo-class.html#__init__,Method csb.bio.fragments.rosetta.OutputBuilder.__init__()=csb.bio.fragments.rosetta.OutputBuilder-class.html#__init__,Method csb.bio.fragments.rosetta.ResidueInfo.__init__()=csb.bio.fragments.rosetta.ResidueInfo-class.html#__init__,Method csb.bio.fragments.rosetta.RosettaFragment.__init__()=csb.bio.fragments.rosetta.RosettaFragment-class.html#__init__,Method csb.bio.fragments.rosetta.RosettaFragmentMap.__init__()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#__init__,Method csb.bio.hmm.EVDParameters.__init__()=csb.bio.hmm.EVDParameters-class.html#__init__,Method csb.bio.hmm.EmissionTable.__init__()=csb.bio.hmm.EmissionTable-class.html#__init__,Method csb.bio.hmm.HHpredHit.__init__()=csb.bio.hmm.HHpredHit-class.html#__init__,Method csb.bio.hmm.HHpredHitAlignment.__init__()=csb.bio.hmm.HHpredHitAlignment-class.html#__init__,Method csb.bio.hmm.HHpredHitList.__init__()=csb.bio.hmm.HHpredHitList-class.html#__init__,Method csb.bio.hmm.HMMLayer.__init__()=csb.bio.hmm.HMMLayer-class.html#__init__,Method csb.bio.hmm.HMMLayersCollection.__init__()=csb.bio.hmm.HMMLayersCollection-class.html#__init__,Method csb.bio.hmm.ProfileHMM.__init__()=csb.bio.hmm.ProfileHMM-class.html#__init__,Method csb.bio.hmm.ProfileHMMRegion.__init__()=csb.bio.hmm.ProfileHMMRegion-class.html#__init__,Method csb.bio.hmm.ProfileHMMSegment.__init__()=csb.bio.hmm.ProfileHMMSegment-class.html#__init__,Method csb.bio.hmm.ProfileLength.__init__()=csb.bio.hmm.ProfileLength-class.html#__init__,Method csb.bio.hmm.State.__init__()=csb.bio.hmm.State-class.html#__init__,Method csb.bio.hmm.StateFactory.__init__()=csb.bio.hmm.StateFactory-class.html#__init__,Method csb.bio.hmm.Transition.__init__()=csb.bio.hmm.Transition-class.html#__init__,Method csb.bio.hmm.TransitionTable.__init__()=csb.bio.hmm.TransitionTable-class.html#__init__,Method csb.bio.hmm.TransitionType.__init__()=csb.bio.hmm.TransitionType-class.html#__init__,Method csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__()=csb.bio.hmm.pseudocounts.PseudocountBuilder-class.html#__init__,Method csb.bio.io.clans.Clans.__init__()=csb.bio.io.clans.Clans-class.html#__init__,Method csb.bio.io.clans.ClansEntry.__init__()=csb.bio.io.clans.ClansEntry-class.html#__init__,Method csb.bio.io.clans.ClansEntryCollection.__init__()=csb.bio.io.clans.ClansEntryCollection-class.html#__init__,Method csb.bio.io.clans.ClansFileBuilder.__init__()=csb.bio.io.clans.ClansFileBuilder-class.html#__init__,Method csb.bio.io.clans.ClansFileWriter.__init__()=csb.bio.io.clans.ClansFileWriter-class.html#__init__,Method csb.bio.io.clans.ClansParams.__init__()=csb.bio.io.clans.ClansParams-class.html#__init__,Method csb.bio.io.clans.ClansParser.__init__()=csb.bio.io.clans.ClansParser-class.html#__init__,Method csb.bio.io.clans.ClansSeqgroup.__init__()=csb.bio.io.clans.ClansSeqgroup-class.html#__init__,Method csb.bio.io.clans.ClansSeqgroupCollection.__init__()=csb.bio.io.clans.ClansSeqgroupCollection-class.html#__init__,Method csb.bio.io.clans.Color.__init__()=csb.bio.io.clans.Color-class.html#__init__,Method csb.bio.io.cs.ChemShiftReader.__init__()=csb.bio.io.cs.ChemShiftReader-class.html#__init__,Method csb.bio.io.dssp.ResidueAssignmentInfo.__init__()=csb.bio.io.dssp.ResidueAssignmentInfo-class.html#__init__,Method csb.bio.io.fasta.A3MSequenceIterator.__init__()=csb.bio.io.fasta.A3MSequenceIterator-class.html#__init__,Method csb.bio.io.fasta.BaseSequenceParser.__init__()=csb.bio.io.fasta.BaseSequenceParser-class.html#__init__,Method csb.bio.io.fasta.OutputBuilder.__init__()=csb.bio.io.fasta.OutputBuilder-class.html#__init__,Method csb.bio.io.fasta.SequenceAlignmentReader.__init__()=csb.bio.io.fasta.SequenceAlignmentReader-class.html#__init__,Method csb.bio.io.fasta.StructureAlignmentFactory.__init__()=csb.bio.io.fasta.StructureAlignmentFactory-class.html#__init__,Method csb.bio.io.hhpred.HHMFileBuilder.__init__()=csb.bio.io.hhpred.HHMFileBuilder-class.html#__init__,Method csb.bio.io.hhpred.HHOutputParser.__init__()=csb.bio.io.hhpred.HHOutputParser-class.html#__init__,Method csb.bio.io.hhpred.HHProfileParser.__init__()=csb.bio.io.hhpred.HHProfileParser-class.html#__init__,Method csb.bio.io.mrc.DensityInfo.__init__()=csb.bio.io.mrc.DensityInfo-class.html#__init__,Method csb.bio.io.mrc.DensityMapReader.__init__()=csb.bio.io.mrc.DensityMapReader-class.html#__init__,Method csb.bio.io.mrc.HeaderInfo.__init__()=csb.bio.io.mrc.HeaderInfo-class.html#__init__,Method csb.bio.io.noe.SparkyPeakListReader.__init__()=csb.bio.io.noe.SparkyPeakListReader-class.html#__init__,Method csb.bio.io.noe.XeasyFileBuilder.__init__()=csb.bio.io.noe.XeasyFileBuilder-class.html#__init__,Method csb.bio.io.noe.XeasyPeakListReader.__init__()=csb.bio.io.noe.XeasyPeakListReader-class.html#__init__,Method csb.bio.io.procheck.ProcheckParser.__init__()=csb.bio.io.procheck.ProcheckParser-class.html#__init__,Method csb.bio.io.psipred.PSIPredResidueInfo.__init__()=csb.bio.io.psipred.PSIPredResidueInfo-class.html#__init__,Method csb.bio.io.svg.SSCartoonBuilder.__init__()=csb.bio.io.svg.SSCartoonBuilder-class.html#__init__,Method csb.bio.io.vasco.ChemicalShiftContainer.__init__()=csb.bio.io.vasco.ChemicalShiftContainer-class.html#__init__,Method csb.bio.io.vasco.ShiftInfo.__init__()=csb.bio.io.vasco.ShiftInfo-class.html#__init__,Method csb.bio.io.vasco.VascoStructureParser.__init__()=csb.bio.io.vasco.VascoStructureParser-class.html#__init__,Method csb.bio.io.whatif.WhatCheckParser.__init__()=csb.bio.io.whatif.WhatCheckParser-class.html#__init__,Method csb.bio.io.wwpdb.AbstractStructureParser.__init__()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#__init__,Method csb.bio.io.wwpdb.AsyncParseResult.__init__()=csb.bio.io.wwpdb.AsyncParseResult-class.html#__init__,Method csb.bio.io.wwpdb.AsyncStructureParser.__init__()=csb.bio.io.wwpdb.AsyncStructureParser-class.html#__init__,Method csb.bio.io.wwpdb.CustomStructureProvider.__init__()=csb.bio.io.wwpdb.CustomStructureProvider-class.html#__init__,Method csb.bio.io.wwpdb.EntryID.__init__()=csb.bio.io.wwpdb.EntryID-class.html#__init__,Method csb.bio.io.wwpdb.FastResidueMapper.__init__()=csb.bio.io.wwpdb.FastResidueMapper-class.html#__init__,Method csb.bio.io.wwpdb.FileBuilder.__init__()=csb.bio.io.wwpdb.FileBuilder-class.html#__init__,Method csb.bio.io.wwpdb.FileSystemStructureProvider.__init__()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#__init__,Method csb.bio.io.wwpdb.RemoteStructureProvider.__init__()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#__init__,Method csb.bio.io.wwpdb.ResidueInfo.__init__()=csb.bio.io.wwpdb.ResidueInfo-class.html#__init__,Method csb.bio.io.wwpdb.RobustResidueMapper.__init__()=csb.bio.io.wwpdb.RobustResidueMapper-class.html#__init__,Method csb.bio.nmr.AtomConnectivity.__init__()=csb.bio.nmr.AtomConnectivity-class.html#__init__,Method csb.bio.nmr.ChemShiftInfo.__init__()=csb.bio.nmr.ChemShiftInfo-class.html#__init__,Method csb.bio.nmr.ChemShiftScoringModel.__init__()=csb.bio.nmr.ChemShiftScoringModel-class.html#__init__,Method csb.bio.nmr.ChemicalShiftNetwork.__init__()=csb.bio.nmr.ChemicalShiftNetwork-class.html#__init__,Method csb.bio.nmr.ContactMap.__init__()=csb.bio.nmr.ContactMap-class.html#__init__,Method csb.bio.nmr.ContactMapComparisonInfo.__init__()=csb.bio.nmr.ContactMapComparisonInfo-class.html#__init__,Method csb.bio.nmr.Label.__init__()=csb.bio.nmr.Label-class.html#__init__,Method csb.bio.nmr.NOEPeak.__init__()=csb.bio.nmr.NOEPeak-class.html#__init__,Method csb.bio.nmr.NOESpectrum.__init__()=csb.bio.nmr.NOESpectrum-class.html#__init__,Method csb.bio.nmr.RandomCoil.__init__()=csb.bio.nmr.RandomCoil-class.html#__init__,Method csb.bio.sequence.A3MAlignment.__init__()=csb.bio.sequence.A3MAlignment-class.html#__init__,Method csb.bio.sequence.AbstractAlignment.__init__()=csb.bio.sequence.AbstractAlignment-class.html#__init__,Method csb.bio.sequence.AbstractSequence.__init__()=csb.bio.sequence.AbstractSequence-class.html#__init__,Method csb.bio.sequence.AlignedSequenceAdapter.__init__()=csb.bio.sequence.AlignedSequenceAdapter-class.html#__init__,Method csb.bio.sequence.AlignmentRowsTable.__init__()=csb.bio.sequence.AlignmentRowsTable-class.html#__init__,Method csb.bio.sequence.ColumnInfo.__init__()=csb.bio.sequence.ColumnInfo-class.html#__init__,Method csb.bio.sequence.PositionError.__init__()=csb.bio.sequence.PositionError-class.html#__init__,Method csb.bio.sequence.ResidueInfo.__init__()=csb.bio.sequence.ResidueInfo-class.html#__init__,Method csb.bio.sequence.Sequence.__init__()=csb.bio.sequence.Sequence-class.html#__init__,Method csb.bio.sequence.SequenceAdapter.__init__()=csb.bio.sequence.SequenceAdapter-class.html#__init__,Method csb.bio.sequence.SequenceCollection.__init__()=csb.bio.sequence.SequenceCollection-class.html#__init__,Method csb.bio.sequence.SequenceIndexer.__init__()=csb.bio.sequence.SequenceIndexer-class.html#__init__,Method csb.bio.sequence.SliceHelper.__init__()=csb.bio.sequence.SliceHelper-class.html#__init__,Method csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__()=csb.bio.sequence.alignment.AbstractAlignmentAlgorithm-class.html#__init__,Method csb.bio.sequence.alignment.AlignmentResult.__init__()=csb.bio.sequence.alignment.AlignmentResult-class.html#__init__,Method csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__()=csb.bio.sequence.alignment.GlobalAlignmentAlgorithm-class.html#__init__,Method csb.bio.sequence.alignment.IdentityMatrix.__init__()=csb.bio.sequence.alignment.IdentityMatrix-class.html#__init__,Method csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__()=csb.bio.sequence.alignment.LocalAlignmentAlgorithm-class.html#__init__,Method csb.bio.sequence.alignment.SimilarityMatrix.__init__()=csb.bio.sequence.alignment.SimilarityMatrix-class.html#__init__,Method csb.bio.structure.Atom.__init__()=csb.bio.structure.Atom-class.html#__init__,Method csb.bio.structure.Chain.__init__()=csb.bio.structure.Chain-class.html#__init__,Method csb.bio.structure.ChainResiduesCollection.__init__()=csb.bio.structure.ChainResiduesCollection-class.html#__init__,Method csb.bio.structure.CompositeEntityIterator.__init__()=csb.bio.structure.CompositeEntityIterator-class.html#__init__,Method csb.bio.structure.ConfinedEntityIterator.__init__()=csb.bio.structure.ConfinedEntityIterator-class.html#__init__,Method csb.bio.structure.DisorderedAtom.__init__()=csb.bio.structure.DisorderedAtom-class.html#__init__,Method csb.bio.structure.Ensemble.__init__()=csb.bio.structure.Ensemble-class.html#__init__,Method csb.bio.structure.EnsembleModelsCollection.__init__()=csb.bio.structure.EnsembleModelsCollection-class.html#__init__,Method csb.bio.structure.NucleicResidue.__init__()=csb.bio.structure.NucleicResidue-class.html#__init__,Method csb.bio.structure.ProteinResidue.__init__()=csb.bio.structure.ProteinResidue-class.html#__init__,Method csb.bio.structure.Residue.__init__()=csb.bio.structure.Residue-class.html#__init__,Method csb.bio.structure.ResidueAtomsTable.__init__()=csb.bio.structure.ResidueAtomsTable-class.html#__init__,Method csb.bio.structure.SecondaryStructure.__init__()=csb.bio.structure.SecondaryStructure-class.html#__init__,Method csb.bio.structure.SecondaryStructureElement.__init__()=csb.bio.structure.SecondaryStructureElement-class.html#__init__,Method csb.bio.structure.Structure.__init__()=csb.bio.structure.Structure-class.html#__init__,Method csb.bio.structure.StructureChainsTable.__init__()=csb.bio.structure.StructureChainsTable-class.html#__init__,Method csb.bio.structure.SuperimposeInfo.__init__()=csb.bio.structure.SuperimposeInfo-class.html#__init__,Method csb.bio.structure.TorsionAngles.__init__()=csb.bio.structure.TorsionAngles-class.html#__init__,Method csb.bio.structure.TorsionAnglesCollection.__init__()=csb.bio.structure.TorsionAnglesCollection-class.html#__init__,Method csb.bio.structure.UnknownResidue.__init__()=csb.bio.structure.UnknownResidue-class.html#__init__,Method csb.build.Console.__init__()=csb.build.Console-class.html#__init__,Method csb.build.GitHandler.__init__()=csb.build.GitHandler-class.html#__init__,Method csb.build.MercurialHandler.__init__()=csb.build.MercurialHandler-class.html#__init__,Method csb.build.RevisionError.__init__()=csb.build.RevisionError-class.html#__init__,Method csb.build.RevisionHandler.__init__()=csb.build.RevisionHandler-class.html#__init__,Method csb.build.RevisionInfo.__init__()=csb.build.RevisionInfo-class.html#__init__,Method csb.build.SubversionHandler.__init__()=csb.build.SubversionHandler-class.html#__init__,Method csb.core.BaseCollectionContainer.__init__()=csb.core.BaseCollectionContainer-class.html#__init__,Method csb.core.BaseDictionaryContainer.__init__()=csb.core.BaseDictionaryContainer-class.html#__init__,Method csb.core.CollectionContainer.__init__()=csb.core.CollectionContainer-class.html#__init__,Method csb.core.Container.__init__()=csb.core.Container-class.html#__init__,Method csb.core.DictionaryContainer.__init__()=csb.core.DictionaryContainer-class.html#__init__,Method csb.core.EnumItem.__init__()=csb.core.EnumItem-class.html#__init__,Method csb.core.InterruptibleThread.__init__()=csb.core.InterruptibleThread-class.html#__init__,Method csb.core.Proxy.__init__()=csb.core.Proxy-class.html#__init__,Method csb.core.REMatchProxy.__init__()=csb.core.REMatchProxy-class.html#__init__,Method csb.core.enum.__init__()=csb.core.enum-class.html#__init__,Method csb.core.typedproperty.__init__()=csb.core.typedproperty-class.html#__init__,Method csb.core.validatedproperty.__init__()=csb.core.validatedproperty-class.html#__init__,Method csb.io.AutoFlushStream.__init__()=csb.io.AutoFlushStream-class.html#__init__,Method csb.io.EntryReader.__init__()=csb.io.EntryReader-class.html#__init__,Method csb.io.EntryWriter.__init__()=csb.io.EntryWriter-class.html#__init__,Method csb.io.InvalidCommandError.__init__()=csb.io.InvalidCommandError-class.html#__init__,Method csb.io.ProcessError.__init__()=csb.io.ProcessError-class.html#__init__,Method csb.io.ShellInfo.__init__()=csb.io.ShellInfo-class.html#__init__,Method csb.io.TempFile.__init__()=csb.io.TempFile-class.html#__init__,Method csb.io.TempFolder.__init__()=csb.io.TempFolder-class.html#__init__,Method csb.io.TimeoutError.__init__()=csb.io.TimeoutError-class.html#__init__,Method csb.io.plots.Backend.__init__()=csb.io.plots.Backend-class.html#__init__,Method csb.io.plots.Chart.__init__()=csb.io.plots.Chart-class.html#__init__,Method csb.io.plots.PlotsCollection.__init__()=csb.io.plots.PlotsCollection-class.html#__init__,Method csb.io.plots.WxBackendImpl.__init__()=csb.io.plots.WxBackendImpl-class.html#__init__,Method csb.io.tsv.Between.__init__()=csb.io.tsv.Between-class.html#__init__,Method csb.io.tsv.ColumnInfo.__init__()=csb.io.tsv.ColumnInfo-class.html#__init__,Method csb.io.tsv.DataRow.__init__()=csb.io.tsv.DataRow-class.html#__init__,Method csb.io.tsv.Expression.__init__()=csb.io.tsv.Expression-class.html#__init__,Method csb.io.tsv.Operator.__init__()=csb.io.tsv.Operator-class.html#__init__,Method csb.io.tsv.Predicate.__init__()=csb.io.tsv.Predicate-class.html#__init__,Method csb.io.tsv.RepositoryImp.__init__()=csb.io.tsv.RepositoryImp-class.html#__init__,Method csb.io.tsv.SQLiteRepository.ChunkedReader.__init__()=csb.io.tsv.SQLiteRepository.ChunkedReader-class.html#__init__,Method csb.io.tsv.SQLiteRepository.__init__()=csb.io.tsv.SQLiteRepository-class.html#__init__,Method csb.io.tsv.Step.__init__()=csb.io.tsv.Step-class.html#__init__,Method csb.io.tsv.Table.__init__()=csb.io.tsv.Table-class.html#__init__,Method csb.io.tsv.Where.__init__()=csb.io.tsv.Where-class.html#__init__,Method csb.numeric.InvertibleMatrix.__init__()=csb.numeric.InvertibleMatrix-class.html#__init__,Method csb.numeric.integrators.AbstractIntegrator.__init__()=csb.numeric.integrators.AbstractIntegrator-class.html#__init__,Method csb.statistics.Cumulative.__init__()=csb.statistics.Cumulative-class.html#__init__,Method csb.statistics.ars.ARS.__init__()=csb.statistics.ars.ARS-class.html#__init__,Method csb.statistics.ars.Envelope.__init__()=csb.statistics.ars.Envelope-class.html#__init__,Method csb.statistics.ars.Gauss.__init__()=csb.statistics.ars.Gauss-class.html#__init__,Method csb.statistics.maxent.MaxentModel.__init__()=csb.statistics.maxent.MaxentModel-class.html#__init__,Method csb.statistics.maxent.MaxentPosterior.__init__()=csb.statistics.maxent.MaxentPosterior-class.html#__init__,Method csb.statistics.mixtures.AbstractStructureMixture.__init__()=csb.statistics.mixtures.AbstractStructureMixture-class.html#__init__,Method csb.statistics.mixtures.GaussianMixture.__init__()=csb.statistics.mixtures.GaussianMixture-class.html#__init__,Method csb.statistics.pdf.AbstractDensity.__init__()=csb.statistics.pdf.AbstractDensity-class.html#__init__,Method csb.statistics.pdf.Dirichlet.__init__()=csb.statistics.pdf.Dirichlet-class.html#__init__,Method csb.statistics.pdf.DirichletEstimator.__init__()=csb.statistics.pdf.DirichletEstimator-class.html#__init__,Method csb.statistics.pdf.Gamma.__init__()=csb.statistics.pdf.Gamma-class.html#__init__,Method csb.statistics.pdf.GammaMLEstimator.__init__()=csb.statistics.pdf.GammaMLEstimator-class.html#__init__,Method csb.statistics.pdf.GenNormalBruteForceEstimator.__init__()=csb.statistics.pdf.GenNormalBruteForceEstimator-class.html#__init__,Method csb.statistics.pdf.GeneralizedInverseGaussian.__init__()=csb.statistics.pdf.GeneralizedInverseGaussian-class.html#__init__,Method csb.statistics.pdf.GeneralizedNormal.__init__()=csb.statistics.pdf.GeneralizedNormal-class.html#__init__,Method csb.statistics.pdf.GumbelMaximum.__init__()=csb.statistics.pdf.GumbelMaximum-class.html#__init__,Method csb.statistics.pdf.GumbelMinimum.__init__()=csb.statistics.pdf.GumbelMinimum-class.html#__init__,Method csb.statistics.pdf.InverseGamma.__init__()=csb.statistics.pdf.InverseGamma-class.html#__init__,Method csb.statistics.pdf.InverseGaussian.__init__()=csb.statistics.pdf.InverseGaussian-class.html#__init__,Method csb.statistics.pdf.Laplace.__init__()=csb.statistics.pdf.Laplace-class.html#__init__,Method csb.statistics.pdf.MultivariateGaussian.__init__()=csb.statistics.pdf.MultivariateGaussian-class.html#__init__,Method csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__()=csb.statistics.pdf.MultivariateGaussianMLEstimator-class.html#__init__,Method csb.statistics.pdf.Normal.__init__()=csb.statistics.pdf.Normal-class.html#__init__,Method csb.statistics.pdf.ParameterValueError.__init__()=csb.statistics.pdf.ParameterValueError-class.html#__init__,Method csb.statistics.pdf.parameterized.AbstractParameter.__init__()=csb.statistics.pdf.parameterized.AbstractParameter-class.html#__init__,Method csb.statistics.pdf.parameterized.Parameter.__init__()=csb.statistics.pdf.parameterized.Parameter-class.html#__init__,Method csb.statistics.samplers.EnsembleState.__init__()=csb.statistics.samplers.EnsembleState-class.html#__init__,Method csb.statistics.samplers.State.__init__()=csb.statistics.samplers.State-class.html#__init__,Method csb.statistics.samplers.mc.AbstractMC.__init__()=csb.statistics.samplers.mc.AbstractMC-class.html#__init__,Method csb.statistics.samplers.mc.MCCollection.__init__()=csb.statistics.samplers.mc.MCCollection-class.html#__init__,Method csb.statistics.samplers.mc.PropagationResult.__init__()=csb.statistics.samplers.mc.PropagationResult-class.html#__init__,Method csb.statistics.samplers.mc.Trajectory.__init__()=csb.statistics.samplers.mc.Trajectory-class.html#__init__,Method csb.statistics.samplers.mc.TrajectoryBuilder.__init__()=csb.statistics.samplers.mc.TrajectoryBuilder-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__()=csb.statistics.samplers.mc.multichain.AbstractEnsembleMC-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__()=csb.statistics.samplers.mc.multichain.AbstractExchangeMC-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__()=csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__()=csb.statistics.samplers.mc.multichain.AbstractStepRENS-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__()=csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__()=csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__()=csb.statistics.samplers.mc.multichain.AbstractSwapScheme-class.html#__init__,Method csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__()=csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme-class.html#__init__,Method csb.statistics.samplers.mc.multichain.Gradient.__init__()=csb.statistics.samplers.mc.multichain.Gradient-class.html#__init__,Method csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__()=csb.statistics.samplers.mc.multichain.HMCStepRENS-class.html#__init__,Method csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__()=csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo-class.html#__init__,Method csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__()=csb.statistics.samplers.mc.multichain.InterpolationFactory-class.html#__init__,Method csb.statistics.samplers.mc.multichain.MDRENS.__init__()=csb.statistics.samplers.mc.multichain.MDRENS-class.html#__init__,Method csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__()=csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo-class.html#__init__,Method csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__()=csb.statistics.samplers.mc.multichain.RENSTrajInfo-class.html#__init__,Method csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__()=csb.statistics.samplers.mc.multichain.ReplicaHistory-class.html#__init__,Method csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__()=csb.statistics.samplers.mc.multichain.SingleSwapStatistics-class.html#__init__,Method csb.statistics.samplers.mc.multichain.SwapStatistics.__init__()=csb.statistics.samplers.mc.multichain.SwapStatistics-class.html#__init__,Method csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__()=csb.statistics.samplers.mc.multichain.ThermostattedMDRENS-class.html#__init__,Method csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__()=csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__()=csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__()=csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__()=csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__()=csb.statistics.samplers.mc.neqsteppropagator.MDParam-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__()=csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__()=csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__()=csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__()=csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__()=csb.statistics.samplers.mc.neqsteppropagator.Protocol-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__()=csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__()=csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation-class.html#__init__,Method csb.statistics.samplers.mc.neqsteppropagator.Step.__init__()=csb.statistics.samplers.mc.neqsteppropagator.Step-class.html#__init__,Method csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__()=csb.statistics.samplers.mc.propagators.AbstractMCPropagator-class.html#__init__,Method csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__()=csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator-class.html#__init__,Method csb.statistics.samplers.mc.propagators.HMCPropagator.__init__()=csb.statistics.samplers.mc.propagators.HMCPropagator-class.html#__init__,Method csb.statistics.samplers.mc.propagators.Looper.__init__()=csb.statistics.samplers.mc.propagators.Looper-class.html#__init__,Method csb.statistics.samplers.mc.propagators.MDPropagator.__init__()=csb.statistics.samplers.mc.propagators.MDPropagator-class.html#__init__,Method csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__()=csb.statistics.samplers.mc.propagators.RWMCPropagator-class.html#__init__,Method csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__()=csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__()=csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__()=csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.HMCSampler.__init__()=csb.statistics.samplers.mc.singlechain.HMCSampler-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__()=csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__()=csb.statistics.samplers.mc.singlechain.RWMCSampler-class.html#__init__,Method csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__()=csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator-class.html#__init__,Method csb.statistics.scalemixture.ARSPosteriorAlpha.__init__()=csb.statistics.scalemixture.ARSPosteriorAlpha-class.html#__init__,Method csb.statistics.scalemixture.GammaPosteriorMAP.__init__()=csb.statistics.scalemixture.GammaPosteriorMAP-class.html#__init__,Method csb.statistics.scalemixture.GammaPosteriorSampler.__init__()=csb.statistics.scalemixture.GammaPosteriorSampler-class.html#__init__,Method csb.statistics.scalemixture.GammaPrior.__init__()=csb.statistics.scalemixture.GammaPrior-class.html#__init__,Method csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__()=csb.statistics.scalemixture.InvGammaPosteriorMAP-class.html#__init__,Method csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__()=csb.statistics.scalemixture.InvGammaPosteriorSampler-class.html#__init__,Method csb.statistics.scalemixture.InvGammaPrior.__init__()=csb.statistics.scalemixture.InvGammaPrior-class.html#__init__,Method csb.statistics.scalemixture.ScaleMixture.__init__()=csb.statistics.scalemixture.ScaleMixture-class.html#__init__,Method csb.statistics.scalemixture.ScaleMixturePrior.__init__()=csb.statistics.scalemixture.ScaleMixturePrior-class.html#__init__,Method csb.statmech.ensembles.BoltzmannEnsemble.__init__()=csb.statmech.ensembles.BoltzmannEnsemble-class.html#__init__,Method csb.statmech.ensembles.CompositeEnsemble.__init__()=csb.statmech.ensembles.CompositeEnsemble-class.html#__init__,Method csb.statmech.ensembles.FermiEnsemble.__init__()=csb.statmech.ensembles.FermiEnsemble-class.html#__init__,Method csb.statmech.ensembles.TsallisEnsemble.__init__()=csb.statmech.ensembles.TsallisEnsemble-class.html#__init__,Method csb.statmech.wham.AbstractWHAM.__init__()=csb.statmech.wham.AbstractWHAM-class.html#__init__,Method csb.statmech.wham.WHAM.__init__()=csb.statmech.wham.WHAM-class.html#__init__,Method csb.test.Console.__init__()=csb.test.Console-class.html#__init__"><a title="csb.Version.__init__
csb.apps.AppExit.__init__
csb.apps.AppRunner.__init__
csb.apps.Application.__init__
csb.apps.ArgHandler.__init__
csb.apps.buildhmm.NoOutputError.__init__
csb.apps.buildhmm.ProfileBuilder.__init__
csb.apps.csfrag.CSfrag.__init__
csb.apps.csfrag.MatchInfo.__init__
csb.apps.csfrag.MatchTable.__init__
csb.apps.csfrag.ScoringHelper.__init__
csb.apps.embd.Deconvolution.__init__
csb.apps.embd.OutputPathBuilder.__init__
csb.apps.hhfrag.HHfrag.__init__
csb.apps.hhfrag.PredictionBuilder.__init__
csb.apps.hhfrag.SliceContext.__init__
csb.apps.hhsearch.Context.__init__
csb.apps.hhsearch.HHTask.__init__
csb.apps.hhsearch.HHsearch.__init__
csb.apps.precision.GlobalInfo.__init__
csb.apps.precision.LibrarySuperimposer.__init__
csb.bio.fragments.Assignment.__init__
csb.bio.fragments.BenchmarkAdapter.Connection.__init__
csb.bio.fragments.BenchmarkAdapter.__init__
csb.bio.fragments.ChemShiftAssignment.__init__
csb.bio.fragments.ChemShiftTarget.__init__
csb.bio.fragments.ClusterNode.__init__
csb.bio.fragments.ClusterRep.__init__
csb.bio.fragments.FragmentCluster.__init__
csb.bio.fragments.FragmentMatch.__init__
csb.bio.fragments.ResidueAssignmentInfo.__init__
csb.bio.fragments.ResidueEventInfo.__init__
csb.bio.fragments.RosettaFragsetFactory.__init__
csb.bio.fragments.Target.__init__
csb.bio.fragments.TargetResidue.__init__
csb.bio.fragments.TorsionAnglePredictor.__init__
csb.bio.fragments.TorsionPredictionInfo.__init__
csb.bio.fragments.rosetta.OutputBuilder.__init__
csb.bio.fragments.rosetta.ResidueInfo.__init__
csb.bio.fragments.rosetta.RosettaFragment.__init__
csb.bio.fragments.rosetta.RosettaFragmentMap.__init__
csb.bio.hmm.EVDParameters.__init__
csb.bio.hmm.EmissionTable.__init__
csb.bio.hmm.HHpredHit.__init__
csb.bio.hmm.HHpredHitAlignment.__init__
csb.bio.hmm.HHpredHitList.__init__
csb.bio.hmm.HMMLayer.__init__
csb.bio.hmm.HMMLayersCollection.__init__
csb.bio.hmm.ProfileHMM.__init__
csb.bio.hmm.ProfileHMMRegion.__init__
csb.bio.hmm.ProfileHMMSegment.__init__
csb.bio.hmm.ProfileLength.__init__
csb.bio.hmm.State.__init__
csb.bio.hmm.StateFactory.__init__
csb.bio.hmm.Transition.__init__
csb.bio.hmm.TransitionTable.__init__
csb.bio.hmm.TransitionType.__init__
csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__
csb.bio.io.clans.Clans.__init__
csb.bio.io.clans.ClansEntry.__init__
csb.bio.io.clans.ClansEntryCollection.__init__
csb.bio.io.clans.ClansFileBuilder.__init__
csb.bio.io.clans.ClansFileWriter.__init__
csb.bio.io.clans.ClansParams.__init__
csb.bio.io.clans.ClansParser.__init__
csb.bio.io.clans.ClansSeqgroup.__init__
csb.bio.io.clans.ClansSeqgroupCollection.__init__
csb.bio.io.clans.Color.__init__
csb.bio.io.cs.ChemShiftReader.__init__
csb.bio.io.dssp.ResidueAssignmentInfo.__init__
csb.bio.io.fasta.A3MSequenceIterator.__init__
csb.bio.io.fasta.BaseSequenceParser.__init__
csb.bio.io.fasta.OutputBuilder.__init__
csb.bio.io.fasta.SequenceAlignmentReader.__init__
csb.bio.io.fasta.StructureAlignmentFactory.__init__
csb.bio.io.hhpred.HHMFileBuilder.__init__
csb.bio.io.hhpred.HHOutputParser.__init__
csb.bio.io.hhpred.HHProfileParser.__init__
csb.bio.io.mrc.DensityInfo.__init__
csb.bio.io.mrc.DensityMapReader.__init__
csb.bio.io.mrc.HeaderInfo.__init__
csb.bio.io.noe.SparkyPeakListReader.__init__
csb.bio.io.noe.XeasyFileBuilder.__init__
csb.bio.io.noe.XeasyPeakListReader.__init__
csb.bio.io.procheck.ProcheckParser.__init__
csb.bio.io.psipred.PSIPredResidueInfo.__init__
csb.bio.io.svg.SSCartoonBuilder.__init__
csb.bio.io.vasco.ChemicalShiftContainer.__init__
csb.bio.io.vasco.ShiftInfo.__init__
csb.bio.io.vasco.VascoStructureParser.__init__
csb.bio.io.whatif.WhatCheckParser.__init__
csb.bio.io.wwpdb.AbstractStructureParser.__init__
csb.bio.io.wwpdb.AsyncParseResult.__init__
csb.bio.io.wwpdb.AsyncStructureParser.__init__
csb.bio.io.wwpdb.CustomStructureProvider.__init__
csb.bio.io.wwpdb.EntryID.__init__
csb.bio.io.wwpdb.FastResidueMapper.__init__
csb.bio.io.wwpdb.FileBuilder.__init__
csb.bio.io.wwpdb.FileSystemStructureProvider.__init__
csb.bio.io.wwpdb.RemoteStructureProvider.__init__
csb.bio.io.wwpdb.ResidueInfo.__init__
csb.bio.io.wwpdb.RobustResidueMapper.__init__
csb.bio.nmr.AtomConnectivity.__init__
csb.bio.nmr.ChemShiftInfo.__init__
csb.bio.nmr.ChemShiftScoringModel.__init__
csb.bio.nmr.ChemicalShiftNetwork.__init__
csb.bio.nmr.ContactMap.__init__
csb.bio.nmr.ContactMapComparisonInfo.__init__
csb.bio.nmr.Label.__init__
csb.bio.nmr.NOEPeak.__init__
csb.bio.nmr.NOESpectrum.__init__
csb.bio.nmr.RandomCoil.__init__
csb.bio.sequence.A3MAlignment.__init__
csb.bio.sequence.AbstractAlignment.__init__
csb.bio.sequence.AbstractSequence.__init__
csb.bio.sequence.AlignedSequenceAdapter.__init__
csb.bio.sequence.AlignmentRowsTable.__init__
csb.bio.sequence.ColumnInfo.__init__
csb.bio.sequence.PositionError.__init__
csb.bio.sequence.ResidueInfo.__init__
csb.bio.sequence.Sequence.__init__
csb.bio.sequence.SequenceAdapter.__init__
csb.bio.sequence.SequenceCollection.__init__
csb.bio.sequence.SequenceIndexer.__init__
csb.bio.sequence.SliceHelper.__init__
csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.AlignmentResult.__init__
csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.IdentityMatrix.__init__
csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.SimilarityMatrix.__init__
csb.bio.structure.Atom.__init__
csb.bio.structure.Chain.__init__
csb.bio.structure.ChainResiduesCollection.__init__
csb.bio.structure.CompositeEntityIterator.__init__
csb.bio.structure.ConfinedEntityIterator.__init__
csb.bio.structure.DisorderedAtom.__init__
csb.bio.structure.Ensemble.__init__
csb.bio.structure.EnsembleModelsCollection.__init__
csb.bio.structure.NucleicResidue.__init__
csb.bio.structure.ProteinResidue.__init__
csb.bio.structure.Residue.__init__
csb.bio.structure.ResidueAtomsTable.__init__
csb.bio.structure.SecondaryStructure.__init__
csb.bio.structure.SecondaryStructureElement.__init__
csb.bio.structure.Structure.__init__
csb.bio.structure.StructureChainsTable.__init__
csb.bio.structure.SuperimposeInfo.__init__
csb.bio.structure.TorsionAngles.__init__
csb.bio.structure.TorsionAnglesCollection.__init__
csb.bio.structure.UnknownResidue.__init__
csb.build.Console.__init__
csb.build.GitHandler.__init__
csb.build.MercurialHandler.__init__
csb.build.RevisionError.__init__
csb.build.RevisionHandler.__init__
csb.build.RevisionInfo.__init__
csb.build.SubversionHandler.__init__
csb.core.BaseCollectionContainer.__init__
csb.core.BaseDictionaryContainer.__init__
csb.core.CollectionContainer.__init__
csb.core.Container.__init__
csb.core.DictionaryContainer.__init__
csb.core.EnumItem.__init__
csb.core.InterruptibleThread.__init__
csb.core.Proxy.__init__
csb.core.REMatchProxy.__init__
csb.core.enum.__init__
csb.core.typedproperty.__init__
csb.core.validatedproperty.__init__
csb.io.AutoFlushStream.__init__
csb.io.EntryReader.__init__
csb.io.EntryWriter.__init__
csb.io.InvalidCommandError.__init__
csb.io.ProcessError.__init__
csb.io.ShellInfo.__init__
csb.io.TempFile.__init__
csb.io.TempFolder.__init__
csb.io.TimeoutError.__init__
csb.io.plots.Backend.__init__
csb.io.plots.Chart.__init__
csb.io.plots.PlotsCollection.__init__
csb.io.plots.WxBackendImpl.__init__
csb.io.tsv.Between.__init__
csb.io.tsv.ColumnInfo.__init__
csb.io.tsv.DataRow.__init__
csb.io.tsv.Expression.__init__
csb.io.tsv.Operator.__init__
csb.io.tsv.Predicate.__init__
csb.io.tsv.RepositoryImp.__init__
csb.io.tsv.SQLiteRepository.ChunkedReader.__init__
csb.io.tsv.SQLiteRepository.__init__
csb.io.tsv.Step.__init__
csb.io.tsv.Table.__init__
csb.io.tsv.Where.__init__
csb.numeric.InvertibleMatrix.__init__
csb.numeric.integrators.AbstractIntegrator.__init__
csb.statistics.Cumulative.__init__
csb.statistics.ars.ARS.__init__
csb.statistics.ars.Envelope.__init__
csb.statistics.ars.Gauss.__init__
csb.statistics.maxent.MaxentModel.__init__
csb.statistics.maxent.MaxentPosterior.__init__
csb.statistics.mixtures.AbstractStructureMixture.__init__
csb.statistics.mixtures.GaussianMixture.__init__
csb.statistics.pdf.AbstractDensity.__init__
csb.statistics.pdf.Dirichlet.__init__
csb.statistics.pdf.DirichletEstimator.__init__
csb.statistics.pdf.Gamma.__init__
csb.statistics.pdf.GammaMLEstimator.__init__
csb.statistics.pdf.GenNormalBruteForceEstimator.__init__
csb.statistics.pdf.GeneralizedInverseGaussian.__init__
csb.statistics.pdf.GeneralizedNormal.__init__
csb.statistics.pdf.GumbelMaximum.__init__
csb.statistics.pdf.GumbelMinimum.__init__
csb.statistics.pdf.InverseGamma.__init__
csb.statistics.pdf.InverseGaussian.__init__
csb.statistics.pdf.Laplace.__init__
csb.statistics.pdf.MultivariateGaussian.__init__
csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__
csb.statistics.pdf.Normal.__init__
csb.statistics.pdf.ParameterValueError.__init__
csb.statistics.pdf.parameterized.AbstractParameter.__init__
csb.statistics.pdf.parameterized.Parameter.__init__
csb.statistics.samplers.EnsembleState.__init__
csb.statistics.samplers.State.__init__
csb.statistics.samplers.mc.AbstractMC.__init__
csb.statistics.samplers.mc.MCCollection.__init__
csb.statistics.samplers.mc.PropagationResult.__init__
csb.statistics.samplers.mc.Trajectory.__init__
csb.statistics.samplers.mc.TrajectoryBuilder.__init__
csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__
csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__
csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__
csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__
csb.statistics.samplers.mc.multichain.Gradient.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__
csb.statistics.samplers.mc.multichain.MDRENS.__init__
csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__
csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__
csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__
csb.statistics.samplers.mc.multichain.SwapStatistics.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__
csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.Step.__init__
csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__
csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__
csb.statistics.samplers.mc.propagators.HMCPropagator.__init__
csb.statistics.samplers.mc.propagators.Looper.__init__
csb.statistics.samplers.mc.propagators.MDPropagator.__init__
csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__
csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__
csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__
csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__
csb.statistics.samplers.mc.singlechain.HMCSampler.__init__
csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__
csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__
csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__
csb.statistics.scalemixture.ARSPosteriorAlpha.__init__
csb.statistics.scalemixture.GammaPosteriorMAP.__init__
csb.statistics.scalemixture.GammaPosteriorSampler.__init__
csb.statistics.scalemixture.GammaPrior.__init__
csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__
csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__
csb.statistics.scalemixture.InvGammaPrior.__init__
csb.statistics.scalemixture.ScaleMixture.__init__
csb.statistics.scalemixture.ScaleMixturePrior.__init__
csb.statmech.ensembles.BoltzmannEnsemble.__init__
csb.statmech.ensembles.CompositeEnsemble.__init__
csb.statmech.ensembles.FermiEnsemble.__init__
csb.statmech.ensembles.TsallisEnsemble.__init__
csb.statmech.wham.AbstractWHAM.__init__
csb.statmech.wham.WHAM.__init__
csb.test.Console.__init__" class="py-name" href="#" onclick="return doclink('link-132', '__init__', 'link-132');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">index</tt><tt class="py-op">,</tt> <tt id="link-133" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-133', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-134" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-134', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">         </tt>
<a name="PositionError.__str__"></a><div id="PositionError.__str__-def"><a name="L173"></a><tt class="py-lineno"> 173</tt> <a class="py-toggle" href="#" id="PositionError.__str__-toggle" onclick="return toggle('PositionError.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.PositionError-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PositionError.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PositionError.__str__-expanded"><a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">         </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">index</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-string">'Position {0.index} is out of range [{0.start}, {0.end}]'</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-string">'Out of range [{0.start}, {0.end}]'</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">             </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name" targets="Method csb.bio.io.wwpdb.EntryID.format()=csb.bio.io.wwpdb.EntryID-class.html#format,Method csb.bio.io.wwpdb.SeqResID.format()=csb.bio.io.wwpdb.SeqResID-class.html#format,Method csb.bio.sequence.A3MAlignment.format()=csb.bio.sequence.A3MAlignment-class.html#format,Method csb.bio.sequence.AbstractAlignment.format()=csb.bio.sequence.AbstractAlignment-class.html#format"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-135', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt>             </tt>
</div></div><a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">         </tt>
<a name="SequencePositionError"></a><div id="SequencePositionError-def"><a name="L182"></a><tt class="py-lineno"> 182</tt> <a class="py-toggle" href="#" id="SequencePositionError-toggle" onclick="return toggle('SequencePositionError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequencePositionError-class.html">SequencePositionError</a><tt class="py-op">(</tt><tt class="py-base-class">PositionError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequencePositionError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequencePositionError-expanded"><a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"> </tt>
<a name="ColumnPositionError"></a><div id="ColumnPositionError-def"><a name="L185"></a><tt class="py-lineno"> 185</tt> <a class="py-toggle" href="#" id="ColumnPositionError-toggle" onclick="return toggle('ColumnPositionError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnPositionError-class.html">ColumnPositionError</a><tt class="py-op">(</tt><tt class="py-base-class">PositionError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnPositionError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ColumnPositionError-expanded"><a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">      </tt>
<a name="SequenceNotFoundError"></a><div id="SequenceNotFoundError-def"><a name="L188"></a><tt class="py-lineno"> 188</tt> <a class="py-toggle" href="#" id="SequenceNotFoundError-toggle" onclick="return toggle('SequenceNotFoundError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceNotFoundError-class.html">SequenceNotFoundError</a><tt class="py-op">(</tt><tt class="py-base-class">KeyError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceNotFoundError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceNotFoundError-expanded"><a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"> </tt>
<a name="DuplicateSequenceError"></a><div id="DuplicateSequenceError-def"><a name="L191"></a><tt class="py-lineno"> 191</tt> <a class="py-toggle" href="#" id="DuplicateSequenceError-toggle" onclick="return toggle('DuplicateSequenceError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.DuplicateSequenceError-class.html">DuplicateSequenceError</a><tt class="py-op">(</tt><tt class="py-base-class">KeyError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DuplicateSequenceError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DuplicateSequenceError-expanded"><a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt>            </tt>
</div><a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">                 </tt>
<a name="ResidueInfo"></a><div id="ResidueInfo-def"><a name="L194"></a><tt class="py-lineno"> 194</tt> <a class="py-toggle" href="#" id="ResidueInfo-toggle" onclick="return toggle('ResidueInfo');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html">ResidueInfo</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ResidueInfo-expanded"><a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">         </tt>
<a name="ResidueInfo.__init__"></a><div id="ResidueInfo.__init__-def"><a name="L196"></a><tt class="py-lineno"> 196</tt> <a class="py-toggle" href="#" id="ResidueInfo.__init__-toggle" onclick="return toggle('ResidueInfo.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.__init__-expanded"><a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">         </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt>     </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rank</tt> <tt class="py-op">=</tt> <tt id="link-136" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.rank=csb.bio.fragments.ResidueEventInfo-class.html#rank,Variable csb.bio.hmm.HHpredHit.rank=csb.bio.hmm.HHpredHit-class.html#rank,Variable csb.bio.hmm.HMMLayer.rank=csb.bio.hmm.HMMLayer-class.html#rank,Variable csb.bio.hmm.State.rank=csb.bio.hmm.State-class.html#rank,Variable csb.bio.io.wwpdb.ResidueInfo.rank=csb.bio.io.wwpdb.ResidueInfo-class.html#rank,Variable csb.bio.nmr.Label.rank=csb.bio.nmr.Label-class.html#rank,Variable csb.bio.sequence.ResidueInfo.rank=csb.bio.sequence.ResidueInfo-class.html#rank,Variable csb.bio.structure.Residue.rank=csb.bio.structure.Residue-class.html#rank"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-136', 'rank', 'link-136');">rank</a></tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">         </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-137', 'type', 'link-121');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-138" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-138', 'type', 'link-121');">type</a></tt> </tt>
</div><a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">                 </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="ResidueInfo.type"></a><div id="ResidueInfo.type-def"><a name="L204"></a><tt class="py-lineno"> 204</tt> <a class="py-toggle" href="#" id="ResidueInfo.type-toggle" onclick="return toggle('ResidueInfo.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.type-expanded"><a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">        Residue type - a member of any sequence alphabet</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: enum item</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> </tt>
</div><a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">type</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="ResidueInfo.type"></a><div id="ResidueInfo.type-def"><a name="L211"></a><tt class="py-lineno"> 211</tt> <a class="py-toggle" href="#" id="ResidueInfo.type-toggle" onclick="return toggle('ResidueInfo.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.type-expanded"><a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-139" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-139', 'SequenceAlphabets', 'link-119');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-140" class="py-name" targets="Method csb.bio.nmr.AtomConnectivity.contains()=csb.bio.nmr.AtomConnectivity-class.html#contains,Static Method csb.bio.sequence.SequenceAlphabets.contains()=csb.bio.sequence.SequenceAlphabets-class.html#contains"><a title="csb.bio.nmr.AtomConnectivity.contains
csb.bio.sequence.SequenceAlphabets.contains" class="py-name" href="#" onclick="return doclink('link-140', 'contains', 'link-140');">contains</a></tt><tt class="py-op">(</tt><tt id="link-141" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-141', 'type', 'link-121');">type</a></tt><tt class="py-op">.</tt><tt id="link-142" class="py-name" targets="Variable csb.core.EnumItem.enum=csb.core.EnumItem-class.html#enum,Class csb.core.enum=csb.core.enum-class.html"><a title="csb.core.EnumItem.enum
csb.core.enum" class="py-name" href="#" onclick="return doclink('link-142', 'enum', 'link-142');">enum</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-143" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-143', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt id="link-144" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-144', 'type', 'link-121');">type</a></tt> </tt>
</div><a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">         </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="ResidueInfo.rank"></a><div id="ResidueInfo.rank-def"><a name="L217"></a><tt class="py-lineno"> 217</tt> <a class="py-toggle" href="#" id="ResidueInfo.rank-toggle" onclick="return toggle('ResidueInfo.rank');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html#rank">rank</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.rank-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.rank-expanded"><a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring">        Residue position (1-based)</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rank</tt> </tt>
</div><a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">     </tt>
<a name="ResidueInfo.__repr__"></a><div id="ResidueInfo.__repr__-def"><a name="L224"></a><tt class="py-lineno"> 224</tt> <a class="py-toggle" href="#" id="ResidueInfo.__repr__-toggle" onclick="return toggle('ResidueInfo.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ResidueInfo-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.__repr__-expanded"><a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'&lt;{1} [{0.rank}]: {0.type!r}&gt;'</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-145', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"> </tt>
<a name="ColumnInfo"></a><div id="ColumnInfo-def"><a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">     </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt> <a class="py-toggle" href="#" id="ColumnInfo-toggle" onclick="return toggle('ColumnInfo');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnInfo-class.html">ColumnInfo</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnInfo-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ColumnInfo-expanded"><a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">     </tt>
<a name="ColumnInfo.__init__"></a><div id="ColumnInfo.__init__-def"><a name="L230"></a><tt class="py-lineno"> 230</tt> <a class="py-toggle" href="#" id="ColumnInfo.__init__-toggle" onclick="return toggle('ColumnInfo.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnInfo-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnInfo.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ColumnInfo.__init__-expanded"><a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">         </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method csb.io.tsv.Table.column()=csb.io.tsv.Table-class.html#column"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-146', 'column', 'link-146');">column</a></tt> <tt class="py-op">=</tt> <tt id="link-147" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-147', 'column', 'link-146');">column</a></tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Variable csb.bio.fragments.FragmentMatch.id=csb.bio.fragments.FragmentMatch-class.html#id,Variable csb.bio.fragments.Target.id=csb.bio.fragments.Target-class.html#id,Variable csb.bio.fragments.rosetta.RosettaFragment.id=csb.bio.fragments.rosetta.RosettaFragment-class.html#id,Variable csb.bio.hmm.HHpredHit.id=csb.bio.hmm.HHpredHit-class.html#id,Variable csb.bio.hmm.ProfileHMM.id=csb.bio.hmm.ProfileHMM-class.html#id,Variable csb.bio.io.wwpdb.ResidueInfo.id=csb.bio.io.wwpdb.ResidueInfo-class.html#id,Variable csb.bio.sequence.AbstractSequence.id=csb.bio.sequence.AbstractSequence-class.html#id,Variable csb.bio.sequence.SequenceAdapter.id=csb.bio.sequence.SequenceAdapter-class.html#id,Variable csb.bio.structure.Chain.id=csb.bio.structure.Chain-class.html#id,Variable csb.bio.structure.Residue.id=csb.bio.structure.Residue-class.html#id"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-148', 'id', 'link-148');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-149" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-149', 'id', 'link-148');">id</a></tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-150', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-151" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-151', 'rank', 'link-136');">rank</a></tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name" targets="Method csb.bio.fragments.AssignmentFactory.residue()=csb.bio.fragments.AssignmentFactory-class.html#residue,Method csb.bio.fragments.ChemShiftAssignmentFactory.residue()=csb.bio.fragments.ChemShiftAssignmentFactory-class.html#residue,Variable csb.bio.hmm.HMMLayer.residue=csb.bio.hmm.HMMLayer-class.html#residue,Variable csb.bio.nmr.Label.residue=csb.bio.nmr.Label-class.html#residue,Variable csb.bio.structure.Atom.residue=csb.bio.structure.Atom-class.html#residue"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-152', 'residue', 'link-152');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-153" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-153', 'residue', 'link-152');">residue</a></tt> </tt>
</div><a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"> </tt>
<a name="ColumnInfo.__repr__"></a><div id="ColumnInfo.__repr__-def"><a name="L237"></a><tt class="py-lineno"> 237</tt> <a class="py-toggle" href="#" id="ColumnInfo.__repr__-toggle" onclick="return toggle('ColumnInfo.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnInfo-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnInfo.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ColumnInfo.__repr__-expanded"><a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'&lt;{0.__class__.__name__} {0.id} [{0.column}]: {0.residue.type!r}&gt;'</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-154', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt>                 </tt>
</div></div><a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">     </tt>
<a name="SequenceIndexer"></a><div id="SequenceIndexer-def"><a name="L240"></a><tt class="py-lineno"> 240</tt> <a class="py-toggle" href="#" id="SequenceIndexer-toggle" onclick="return toggle('SequenceIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceIndexer-class.html">SequenceIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceIndexer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceIndexer-expanded"><a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">     </tt>
<a name="SequenceIndexer.__init__"></a><div id="SequenceIndexer.__init__-def"><a name="L242"></a><tt class="py-lineno"> 242</tt> <a class="py-toggle" href="#" id="SequenceIndexer.__init__-toggle" onclick="return toggle('SequenceIndexer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceIndexer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">container</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceIndexer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceIndexer.__init__-expanded"><a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt> <tt class="py-op">=</tt> <tt class="py-name">container</tt> </tt>
</div><a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"> </tt>
<a name="SequenceIndexer.__getitem__"></a><div id="SequenceIndexer.__getitem__-def"><a name="L245"></a><tt class="py-lineno"> 245</tt> <a class="py-toggle" href="#" id="SequenceIndexer.__getitem__-toggle" onclick="return toggle('SequenceIndexer.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceIndexer-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceIndexer.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceIndexer.__getitem__-expanded"><a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">         </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-155" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-155', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name" targets="Variable csb.apps.hhfrag.SliceContext.length=csb.apps.hhfrag.SliceContext-class.html#length,Variable csb.bio.fragments.FragmentMatch.length=csb.bio.fragments.FragmentMatch-class.html#length,Variable csb.bio.fragments.Target.length=csb.bio.fragments.Target-class.html#length,Variable csb.bio.fragments.rosetta.RosettaFragment.length=csb.bio.fragments.rosetta.RosettaFragment-class.html#length,Variable csb.bio.hmm.HHpredHit.length=csb.bio.hmm.HHpredHit-class.html#length,Variable csb.bio.hmm.ProfileHMM.length=csb.bio.hmm.ProfileHMM-class.html#length,Variable csb.bio.sequence.AbstractAlignment.length=csb.bio.sequence.AbstractAlignment-class.html#length,Variable csb.bio.sequence.AbstractSequence.length=csb.bio.sequence.AbstractSequence-class.html#length,Variable csb.bio.sequence.SequenceAdapter.length=csb.bio.sequence.SequenceAdapter-class.html#length,Variable csb.bio.sequence.alignment.AlignmentResult.length=csb.bio.sequence.alignment.AlignmentResult-class.html#length,Variable csb.bio.structure.Chain.length=csb.bio.structure.Chain-class.html#length,Variable csb.bio.structure.SecondaryStructureElement.length=csb.bio.structure.SecondaryStructureElement-class.html#length,Variable csb.core.BaseCollectionContainer.length=csb.core.BaseCollectionContainer-class.html#length,Variable csb.core.BaseDictionaryContainer.length=csb.core.BaseDictionaryContainer-class.html#length"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-156', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-157" class="py-name" targets="Class csb.bio.sequence.SequencePositionError=csb.bio.sequence.SequencePositionError-class.html"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-157', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">(</tt><tt id="link-158" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-158', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-159', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">               </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt class="py-name">_get</tt><tt class="py-op">(</tt><tt id="link-160" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-160', 'rank', 'link-136');">rank</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">     </tt>
<a name="SequenceIndexer.__iter__"></a><div id="SequenceIndexer.__iter__-def"><a name="L252"></a><tt class="py-lineno"> 252</tt> <a class="py-toggle" href="#" id="SequenceIndexer.__iter__-toggle" onclick="return toggle('SequenceIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceIndexer.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceIndexer.__iter__-expanded"><a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">             </tt>
<a name="UngappedSequenceIndexer"></a><div id="UngappedSequenceIndexer-def"><a name="L255"></a><tt class="py-lineno"> 255</tt> <a class="py-toggle" href="#" id="UngappedSequenceIndexer-toggle" onclick="return toggle('UngappedSequenceIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.UngappedSequenceIndexer-class.html">UngappedSequenceIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">SequenceIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UngappedSequenceIndexer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="UngappedSequenceIndexer-expanded"><a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"> </tt>
<a name="UngappedSequenceIndexer.__getitem__"></a><div id="UngappedSequenceIndexer.__getitem__-def"><a name="L257"></a><tt class="py-lineno"> 257</tt> <a class="py-toggle" href="#" id="UngappedSequenceIndexer.__getitem__-toggle" onclick="return toggle('UngappedSequenceIndexer.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.UngappedSequenceIndexer-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UngappedSequenceIndexer.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UngappedSequenceIndexer.__getitem__-expanded"><a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt>  </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt class="py-name">_get_ungapped</tt><tt class="py-op">(</tt><tt id="link-161" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-161', 'rank', 'link-136');">rank</a></tt><tt class="py-op">)</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-162" class="py-name"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-162', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-163" class="py-name"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-163', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-164', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">     </tt>
<a name="UngappedSequenceIndexer.__iter__"></a><div id="UngappedSequenceIndexer.__iter__-def"><a name="L263"></a><tt class="py-lineno"> 263</tt> <a class="py-toggle" href="#" id="UngappedSequenceIndexer.__iter__-toggle" onclick="return toggle('UngappedSequenceIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.UngappedSequenceIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UngappedSequenceIndexer.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="UngappedSequenceIndexer.__iter__-expanded"><a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-165', 'residue', 'link-152');">residue</a></tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-166', 'type', 'link-121');">type</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-167', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-168', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-169', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-170', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-171', 'residue', 'link-152');">residue</a></tt> </tt>
</div></div><a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"> </tt>
<a name="ColumnIndexer"></a><div id="ColumnIndexer-def"><a name="L268"></a><tt class="py-lineno"> 268</tt> <a class="py-toggle" href="#" id="ColumnIndexer-toggle" onclick="return toggle('ColumnIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnIndexer-class.html">ColumnIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">SequenceIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnIndexer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ColumnIndexer-expanded"><a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">     </tt>
<a name="ColumnIndexer.__getitem__"></a><div id="ColumnIndexer.__getitem__-def"><a name="L270"></a><tt class="py-lineno"> 270</tt> <a class="py-toggle" href="#" id="ColumnIndexer.__getitem__-toggle" onclick="return toggle('ColumnIndexer.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ColumnIndexer-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColumnIndexer.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ColumnIndexer.__getitem__-expanded"><a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">         </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-172" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-172', 'column', 'link-146');">column</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-173', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-174" class="py-name" targets="Class csb.bio.sequence.ColumnPositionError=csb.bio.sequence.ColumnPositionError-class.html"><a title="csb.bio.sequence.ColumnPositionError" class="py-name" href="#" onclick="return doclink('link-174', 'ColumnPositionError', 'link-174');">ColumnPositionError</a></tt><tt class="py-op">(</tt><tt id="link-175" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-175', 'column', 'link-146');">column</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-176', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">                 </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt><tt class="py-op">.</tt><tt class="py-name">_get_column</tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-177', 'column', 'link-146');">column</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">     </tt>
<a name="SequenceCollection"></a><div id="SequenceCollection-def"><a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line"> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt> <a class="py-toggle" href="#" id="SequenceCollection-toggle" onclick="return toggle('SequenceCollection');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceCollection-class.html">SequenceCollection</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">ReadOnlyCollectionContainer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceCollection-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceCollection-expanded"><a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line"><tt class="py-docstring">    Represents a list of L{AbstractSequence}s.</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line"> </tt>
<a name="SequenceCollection.__init__"></a><div id="SequenceCollection.__init__-def"><a name="L283"></a><tt class="py-lineno"> 283</tt> <a class="py-toggle" href="#" id="SequenceCollection.__init__-toggle" onclick="return toggle('SequenceCollection.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceCollection-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceCollection.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceCollection.__init__-expanded"><a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-178" class="py-name" targets="Class csb.bio.sequence.SequenceCollection=csb.bio.sequence.SequenceCollection-class.html"><a title="csb.bio.sequence.SequenceCollection" class="py-name" href="#" onclick="return doclink('link-178', 'SequenceCollection', 'link-178');">SequenceCollection</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="csb.Version.__init__
csb.apps.AppExit.__init__
csb.apps.AppRunner.__init__
csb.apps.Application.__init__
csb.apps.ArgHandler.__init__
csb.apps.buildhmm.NoOutputError.__init__
csb.apps.buildhmm.ProfileBuilder.__init__
csb.apps.csfrag.CSfrag.__init__
csb.apps.csfrag.MatchInfo.__init__
csb.apps.csfrag.MatchTable.__init__
csb.apps.csfrag.ScoringHelper.__init__
csb.apps.embd.Deconvolution.__init__
csb.apps.embd.OutputPathBuilder.__init__
csb.apps.hhfrag.HHfrag.__init__
csb.apps.hhfrag.PredictionBuilder.__init__
csb.apps.hhfrag.SliceContext.__init__
csb.apps.hhsearch.Context.__init__
csb.apps.hhsearch.HHTask.__init__
csb.apps.hhsearch.HHsearch.__init__
csb.apps.precision.GlobalInfo.__init__
csb.apps.precision.LibrarySuperimposer.__init__
csb.bio.fragments.Assignment.__init__
csb.bio.fragments.BenchmarkAdapter.Connection.__init__
csb.bio.fragments.BenchmarkAdapter.__init__
csb.bio.fragments.ChemShiftAssignment.__init__
csb.bio.fragments.ChemShiftTarget.__init__
csb.bio.fragments.ClusterNode.__init__
csb.bio.fragments.ClusterRep.__init__
csb.bio.fragments.FragmentCluster.__init__
csb.bio.fragments.FragmentMatch.__init__
csb.bio.fragments.ResidueAssignmentInfo.__init__
csb.bio.fragments.ResidueEventInfo.__init__
csb.bio.fragments.RosettaFragsetFactory.__init__
csb.bio.fragments.Target.__init__
csb.bio.fragments.TargetResidue.__init__
csb.bio.fragments.TorsionAnglePredictor.__init__
csb.bio.fragments.TorsionPredictionInfo.__init__
csb.bio.fragments.rosetta.OutputBuilder.__init__
csb.bio.fragments.rosetta.ResidueInfo.__init__
csb.bio.fragments.rosetta.RosettaFragment.__init__
csb.bio.fragments.rosetta.RosettaFragmentMap.__init__
csb.bio.hmm.EVDParameters.__init__
csb.bio.hmm.EmissionTable.__init__
csb.bio.hmm.HHpredHit.__init__
csb.bio.hmm.HHpredHitAlignment.__init__
csb.bio.hmm.HHpredHitList.__init__
csb.bio.hmm.HMMLayer.__init__
csb.bio.hmm.HMMLayersCollection.__init__
csb.bio.hmm.ProfileHMM.__init__
csb.bio.hmm.ProfileHMMRegion.__init__
csb.bio.hmm.ProfileHMMSegment.__init__
csb.bio.hmm.ProfileLength.__init__
csb.bio.hmm.State.__init__
csb.bio.hmm.StateFactory.__init__
csb.bio.hmm.Transition.__init__
csb.bio.hmm.TransitionTable.__init__
csb.bio.hmm.TransitionType.__init__
csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__
csb.bio.io.clans.Clans.__init__
csb.bio.io.clans.ClansEntry.__init__
csb.bio.io.clans.ClansEntryCollection.__init__
csb.bio.io.clans.ClansFileBuilder.__init__
csb.bio.io.clans.ClansFileWriter.__init__
csb.bio.io.clans.ClansParams.__init__
csb.bio.io.clans.ClansParser.__init__
csb.bio.io.clans.ClansSeqgroup.__init__
csb.bio.io.clans.ClansSeqgroupCollection.__init__
csb.bio.io.clans.Color.__init__
csb.bio.io.cs.ChemShiftReader.__init__
csb.bio.io.dssp.ResidueAssignmentInfo.__init__
csb.bio.io.fasta.A3MSequenceIterator.__init__
csb.bio.io.fasta.BaseSequenceParser.__init__
csb.bio.io.fasta.OutputBuilder.__init__
csb.bio.io.fasta.SequenceAlignmentReader.__init__
csb.bio.io.fasta.StructureAlignmentFactory.__init__
csb.bio.io.hhpred.HHMFileBuilder.__init__
csb.bio.io.hhpred.HHOutputParser.__init__
csb.bio.io.hhpred.HHProfileParser.__init__
csb.bio.io.mrc.DensityInfo.__init__
csb.bio.io.mrc.DensityMapReader.__init__
csb.bio.io.mrc.HeaderInfo.__init__
csb.bio.io.noe.SparkyPeakListReader.__init__
csb.bio.io.noe.XeasyFileBuilder.__init__
csb.bio.io.noe.XeasyPeakListReader.__init__
csb.bio.io.procheck.ProcheckParser.__init__
csb.bio.io.psipred.PSIPredResidueInfo.__init__
csb.bio.io.svg.SSCartoonBuilder.__init__
csb.bio.io.vasco.ChemicalShiftContainer.__init__
csb.bio.io.vasco.ShiftInfo.__init__
csb.bio.io.vasco.VascoStructureParser.__init__
csb.bio.io.whatif.WhatCheckParser.__init__
csb.bio.io.wwpdb.AbstractStructureParser.__init__
csb.bio.io.wwpdb.AsyncParseResult.__init__
csb.bio.io.wwpdb.AsyncStructureParser.__init__
csb.bio.io.wwpdb.CustomStructureProvider.__init__
csb.bio.io.wwpdb.EntryID.__init__
csb.bio.io.wwpdb.FastResidueMapper.__init__
csb.bio.io.wwpdb.FileBuilder.__init__
csb.bio.io.wwpdb.FileSystemStructureProvider.__init__
csb.bio.io.wwpdb.RemoteStructureProvider.__init__
csb.bio.io.wwpdb.ResidueInfo.__init__
csb.bio.io.wwpdb.RobustResidueMapper.__init__
csb.bio.nmr.AtomConnectivity.__init__
csb.bio.nmr.ChemShiftInfo.__init__
csb.bio.nmr.ChemShiftScoringModel.__init__
csb.bio.nmr.ChemicalShiftNetwork.__init__
csb.bio.nmr.ContactMap.__init__
csb.bio.nmr.ContactMapComparisonInfo.__init__
csb.bio.nmr.Label.__init__
csb.bio.nmr.NOEPeak.__init__
csb.bio.nmr.NOESpectrum.__init__
csb.bio.nmr.RandomCoil.__init__
csb.bio.sequence.A3MAlignment.__init__
csb.bio.sequence.AbstractAlignment.__init__
csb.bio.sequence.AbstractSequence.__init__
csb.bio.sequence.AlignedSequenceAdapter.__init__
csb.bio.sequence.AlignmentRowsTable.__init__
csb.bio.sequence.ColumnInfo.__init__
csb.bio.sequence.PositionError.__init__
csb.bio.sequence.ResidueInfo.__init__
csb.bio.sequence.Sequence.__init__
csb.bio.sequence.SequenceAdapter.__init__
csb.bio.sequence.SequenceCollection.__init__
csb.bio.sequence.SequenceIndexer.__init__
csb.bio.sequence.SliceHelper.__init__
csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.AlignmentResult.__init__
csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.IdentityMatrix.__init__
csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.SimilarityMatrix.__init__
csb.bio.structure.Atom.__init__
csb.bio.structure.Chain.__init__
csb.bio.structure.ChainResiduesCollection.__init__
csb.bio.structure.CompositeEntityIterator.__init__
csb.bio.structure.ConfinedEntityIterator.__init__
csb.bio.structure.DisorderedAtom.__init__
csb.bio.structure.Ensemble.__init__
csb.bio.structure.EnsembleModelsCollection.__init__
csb.bio.structure.NucleicResidue.__init__
csb.bio.structure.ProteinResidue.__init__
csb.bio.structure.Residue.__init__
csb.bio.structure.ResidueAtomsTable.__init__
csb.bio.structure.SecondaryStructure.__init__
csb.bio.structure.SecondaryStructureElement.__init__
csb.bio.structure.Structure.__init__
csb.bio.structure.StructureChainsTable.__init__
csb.bio.structure.SuperimposeInfo.__init__
csb.bio.structure.TorsionAngles.__init__
csb.bio.structure.TorsionAnglesCollection.__init__
csb.bio.structure.UnknownResidue.__init__
csb.build.Console.__init__
csb.build.GitHandler.__init__
csb.build.MercurialHandler.__init__
csb.build.RevisionError.__init__
csb.build.RevisionHandler.__init__
csb.build.RevisionInfo.__init__
csb.build.SubversionHandler.__init__
csb.core.BaseCollectionContainer.__init__
csb.core.BaseDictionaryContainer.__init__
csb.core.CollectionContainer.__init__
csb.core.Container.__init__
csb.core.DictionaryContainer.__init__
csb.core.EnumItem.__init__
csb.core.InterruptibleThread.__init__
csb.core.Proxy.__init__
csb.core.REMatchProxy.__init__
csb.core.enum.__init__
csb.core.typedproperty.__init__
csb.core.validatedproperty.__init__
csb.io.AutoFlushStream.__init__
csb.io.EntryReader.__init__
csb.io.EntryWriter.__init__
csb.io.InvalidCommandError.__init__
csb.io.ProcessError.__init__
csb.io.ShellInfo.__init__
csb.io.TempFile.__init__
csb.io.TempFolder.__init__
csb.io.TimeoutError.__init__
csb.io.plots.Backend.__init__
csb.io.plots.Chart.__init__
csb.io.plots.PlotsCollection.__init__
csb.io.plots.WxBackendImpl.__init__
csb.io.tsv.Between.__init__
csb.io.tsv.ColumnInfo.__init__
csb.io.tsv.DataRow.__init__
csb.io.tsv.Expression.__init__
csb.io.tsv.Operator.__init__
csb.io.tsv.Predicate.__init__
csb.io.tsv.RepositoryImp.__init__
csb.io.tsv.SQLiteRepository.ChunkedReader.__init__
csb.io.tsv.SQLiteRepository.__init__
csb.io.tsv.Step.__init__
csb.io.tsv.Table.__init__
csb.io.tsv.Where.__init__
csb.numeric.InvertibleMatrix.__init__
csb.numeric.integrators.AbstractIntegrator.__init__
csb.statistics.Cumulative.__init__
csb.statistics.ars.ARS.__init__
csb.statistics.ars.Envelope.__init__
csb.statistics.ars.Gauss.__init__
csb.statistics.maxent.MaxentModel.__init__
csb.statistics.maxent.MaxentPosterior.__init__
csb.statistics.mixtures.AbstractStructureMixture.__init__
csb.statistics.mixtures.GaussianMixture.__init__
csb.statistics.pdf.AbstractDensity.__init__
csb.statistics.pdf.Dirichlet.__init__
csb.statistics.pdf.DirichletEstimator.__init__
csb.statistics.pdf.Gamma.__init__
csb.statistics.pdf.GammaMLEstimator.__init__
csb.statistics.pdf.GenNormalBruteForceEstimator.__init__
csb.statistics.pdf.GeneralizedInverseGaussian.__init__
csb.statistics.pdf.GeneralizedNormal.__init__
csb.statistics.pdf.GumbelMaximum.__init__
csb.statistics.pdf.GumbelMinimum.__init__
csb.statistics.pdf.InverseGamma.__init__
csb.statistics.pdf.InverseGaussian.__init__
csb.statistics.pdf.Laplace.__init__
csb.statistics.pdf.MultivariateGaussian.__init__
csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__
csb.statistics.pdf.Normal.__init__
csb.statistics.pdf.ParameterValueError.__init__
csb.statistics.pdf.parameterized.AbstractParameter.__init__
csb.statistics.pdf.parameterized.Parameter.__init__
csb.statistics.samplers.EnsembleState.__init__
csb.statistics.samplers.State.__init__
csb.statistics.samplers.mc.AbstractMC.__init__
csb.statistics.samplers.mc.MCCollection.__init__
csb.statistics.samplers.mc.PropagationResult.__init__
csb.statistics.samplers.mc.Trajectory.__init__
csb.statistics.samplers.mc.TrajectoryBuilder.__init__
csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__
csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__
csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__
csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__
csb.statistics.samplers.mc.multichain.Gradient.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__
csb.statistics.samplers.mc.multichain.MDRENS.__init__
csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__
csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__
csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__
csb.statistics.samplers.mc.multichain.SwapStatistics.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__
csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.Step.__init__
csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__
csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__
csb.statistics.samplers.mc.propagators.HMCPropagator.__init__
csb.statistics.samplers.mc.propagators.Looper.__init__
csb.statistics.samplers.mc.propagators.MDPropagator.__init__
csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__
csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__
csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__
csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__
csb.statistics.samplers.mc.singlechain.HMCSampler.__init__
csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__
csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__
csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__
csb.statistics.scalemixture.ARSPosteriorAlpha.__init__
csb.statistics.scalemixture.GammaPosteriorMAP.__init__
csb.statistics.scalemixture.GammaPosteriorSampler.__init__
csb.statistics.scalemixture.GammaPrior.__init__
csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__
csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__
csb.statistics.scalemixture.InvGammaPrior.__init__
csb.statistics.scalemixture.ScaleMixture.__init__
csb.statistics.scalemixture.ScaleMixturePrior.__init__
csb.statmech.ensembles.BoltzmannEnsemble.__init__
csb.statmech.ensembles.CompositeEnsemble.__init__
csb.statmech.ensembles.FermiEnsemble.__init__
csb.statmech.ensembles.TsallisEnsemble.__init__
csb.statmech.wham.AbstractWHAM.__init__
csb.statmech.wham.WHAM.__init__
csb.test.Console.__init__" class="py-name" href="#" onclick="return doclink('link-179', '__init__', 'link-132');">__init__</a></tt><tt class="py-op">(</tt><tt id="link-180" class="py-name" targets="Variable csb.bio.fragments.FragmentCluster.items=csb.bio.fragments.FragmentCluster-class.html#items,Variable csb.bio.structure.AbstractEntity.items=csb.bio.structure.AbstractEntity-class.html#items,Variable csb.bio.structure.Atom.items=csb.bio.structure.Atom-class.html#items,Variable csb.bio.structure.Chain.items=csb.bio.structure.Chain-class.html#items,Variable csb.bio.structure.Ensemble.items=csb.bio.structure.Ensemble-class.html#items,Variable csb.bio.structure.Residue.items=csb.bio.structure.Residue-class.html#items,Variable csb.bio.structure.Structure.items=csb.bio.structure.Structure-class.html#items"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-180', 'items', 'link-180');">items</a></tt><tt class="py-op">=</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-181" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-181', 'type', 'link-121');">type</a></tt><tt class="py-op">=</tt><tt id="link-182" class="py-name" targets="Class csb.bio.sequence.AbstractSequence=csb.bio.sequence.AbstractSequence-class.html"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-182', 'AbstractSequence', 'link-182');">AbstractSequence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"> </tt>
<a name="SequenceCollection.to_fasta"></a><div id="SequenceCollection.to_fasta-def"><a name="L286"></a><tt class="py-lineno"> 286</tt> <a class="py-toggle" href="#" id="SequenceCollection.to_fasta-toggle" onclick="return toggle('SequenceCollection.to_fasta');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceCollection-class.html#to_fasta">to_fasta</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceCollection.to_fasta-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceCollection.to_fasta-expanded"><a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line"><tt class="py-docstring">        Dump the whole collection in mFASTA format.</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">        @param output_file: write the output to this file or stream</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">        @type output_file: str or stream</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-183" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-183', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Package csb.bio=csb.bio-module.html"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-184', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-185', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-186" class="py-name" targets="Module csb.bio.io.fasta=csb.bio.io.fasta-module.html"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-186', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-187" class="py-name" targets="Class csb.bio.io.fasta.FASTAOutputBuilder=csb.bio.io.fasta.FASTAOutputBuilder-class.html"><a title="csb.bio.io.fasta.FASTAOutputBuilder" class="py-name" href="#" onclick="return doclink('link-187', 'FASTAOutputBuilder', 'link-187');">FASTAOutputBuilder</a></tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">             </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-188" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-188', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-189', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-190" class="py-name" targets="Class csb.io.EntryWriter=csb.io.EntryWriter-class.html"><a title="csb.io.EntryWriter" class="py-name" href="#" onclick="return doclink('link-190', 'EntryWriter', 'link-190');">EntryWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">output_file</tt><tt class="py-op">,</tt> <tt id="link-191" class="py-name" targets="Method csb.io.EntryWriter.close()=csb.io.EntryWriter-class.html#close,Method csb.io.TempFile.close()=csb.io.TempFile-class.html#close,Method csb.io.TempFolder.close()=csb.io.TempFolder-class.html#close,Method csb.io.tsv.RepositoryImp.close()=csb.io.tsv.RepositoryImp-class.html#close,Method csb.io.tsv.SQLiteRepository.close()=csb.io.tsv.SQLiteRepository-class.html#close"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-191', 'close', 'link-191');">close</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">out</tt><tt class="py-op">:</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">            <tt id="link-192" class="py-name" targets="Variable csb.test.Console.builder=csb.test.Console-class.html#builder"><a title="csb.test.Console.builder" class="py-name" href="#" onclick="return doclink('link-192', 'builder', 'link-192');">builder</a></tt> <tt class="py-op">=</tt> <tt id="link-193" class="py-name"><a title="csb.bio.io.fasta.FASTAOutputBuilder" class="py-name" href="#" onclick="return doclink('link-193', 'FASTAOutputBuilder', 'link-187');">FASTAOutputBuilder</a></tt><tt class="py-op">(</tt><tt class="py-name">out</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Variable csb.io.EntryWriter.stream=csb.io.EntryWriter-class.html#stream"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-194', 'stream', 'link-194');">stream</a></tt><tt class="py-op">,</tt> <tt id="link-195" class="py-name" targets="Variable csb.bio.io.fasta.OutputBuilder.headers=csb.bio.io.fasta.OutputBuilder-class.html#headers"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-195', 'headers', 'link-195');">headers</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">             </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">                <tt id="link-196" class="py-name"><a title="csb.test.Console.builder" class="py-name" href="#" onclick="return doclink('link-196', 'builder', 'link-192');">builder</a></tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Method csb.bio.io.fasta.A3MOutputBuilder.add_sequence()=csb.bio.io.fasta.A3MOutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.FASTAOutputBuilder.add_sequence()=csb.bio.io.fasta.FASTAOutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.OutputBuilder.add_sequence()=csb.bio.io.fasta.OutputBuilder-class.html#add_sequence,Method csb.bio.io.fasta.PIROutputBuilder.add_sequence()=csb.bio.io.fasta.PIROutputBuilder-class.html#add_sequence"><a title="csb.bio.io.fasta.A3MOutputBuilder.add_sequence
csb.bio.io.fasta.FASTAOutputBuilder.add_sequence
csb.bio.io.fasta.OutputBuilder.add_sequence
csb.bio.io.fasta.PIROutputBuilder.add_sequence" class="py-name" href="#" onclick="return doclink('link-197', 'add_sequence', 'link-197');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt>         </tt>
</div></div><a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"> </tt>
<a name="AbstractSequence"></a><div id="AbstractSequence-def"><a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">         </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt> <a class="py-toggle" href="#" id="AbstractSequence-toggle" onclick="return toggle('AbstractSequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html">AbstractSequence</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AbstractSequence-expanded"><a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring">    Base abstract class for all Sequence objects.</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">    Provides 1-based access to the residues in the sequence via the</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">    sequence.residues property. The sequence object itself also behaves like</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">    a collection and provides 0-based access to its elements (residues).   </tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">    @param id: FASTA ID of this sequence (e.g. accession number)</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">    @type id: str</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">    @param header: FASTA sequence header</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">    @type header: str</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">    @param residues: sequence residues</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">    @type residues: str or collection of L{ResidueInfo}</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">    @param type: a L{SequenceTypes} member (defaults to protein)</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">    @type type: L{EnumItem}</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">      </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-198" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-198', 'ABCMeta', 'link-4');">ABCMeta</a></tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">     </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">    <tt id="link-199" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.DELIMITER=csb.bio.io.wwpdb.FastResidueMapper-class.html#DELIMITER,Variable csb.bio.sequence.AbstractSequence.DELIMITER=csb.bio.sequence.AbstractSequence-class.html#DELIMITER"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-199', 'DELIMITER', 'link-199');">DELIMITER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'&gt;'</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"> </tt>
<a name="AbstractSequence.__init__"></a><div id="AbstractSequence.__init__-def"><a name="L324"></a><tt class="py-lineno"> 324</tt> <a class="py-toggle" href="#" id="AbstractSequence.__init__-toggle" onclick="return toggle('AbstractSequence.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">header</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt id="link-200" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-200', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-201', 'Unknown', 'link-12');">Unknown</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__init__-expanded"><a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">           </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-202', 'id', 'link-148');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-203" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-203', 'id', 'link-148');">id</a></tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.header=csb.bio.sequence.AbstractSequence-class.html#header,Variable csb.bio.sequence.SequenceAdapter.header=csb.bio.sequence.SequenceAdapter-class.html#header,Variable csb.bio.structure.Chain.header=csb.bio.structure.Chain-class.html#header"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-204', 'header', 'link-204');">header</a></tt> <tt class="py-op">=</tt> <tt id="link-205" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-205', 'header', 'link-204');">header</a></tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-206', 'type', 'link-121');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-207" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-207', 'type', 'link-121');">type</a></tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">         </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-208" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-208', 'residue', 'link-152');">residue</a></tt> <tt class="py-keyword">in</tt> <tt id="link-209" class="py-name" targets="Variable csb.bio.fragments.Target.residues=csb.bio.fragments.Target-class.html#residues,Variable csb.bio.fragments.rosetta.RosettaFragment.residues=csb.bio.fragments.rosetta.RosettaFragment-class.html#residues,Variable csb.bio.hmm.ProfileHMM.residues=csb.bio.hmm.ProfileHMM-class.html#residues,Variable csb.bio.sequence.AbstractSequence.residues=csb.bio.sequence.AbstractSequence-class.html#residues,Variable csb.bio.sequence.AlignedSequenceAdapter.residues=csb.bio.sequence.AlignedSequenceAdapter-class.html#residues,Variable csb.bio.structure.Chain.residues=csb.bio.structure.Chain-class.html#residues"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-209', 'residues', 'link-209');">residues</a></tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add</tt><tt class="py-op">(</tt><tt id="link-210" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-210', 'residue', 'link-152');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.__getitem__"></a><div id="AbstractSequence.__getitem__-def"><a name="L338"></a><tt class="py-lineno"> 338</tt> <a class="py-toggle" href="#" id="AbstractSequence.__getitem__-toggle" onclick="return toggle('AbstractSequence.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">spec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__getitem__-expanded"><a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">         </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">            <tt class="py-name">spec</tt> <tt class="py-op">=</tt> <tt id="link-211" class="py-name" targets="Class csb.bio.sequence.SliceHelper=csb.bio.sequence.SliceHelper-class.html"><a title="csb.bio.sequence.SliceHelper" class="py-name" href="#" onclick="return doclink('link-211', 'SliceHelper', 'link-211');">SliceHelper</a></tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-212', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name" targets="Method csb.bio.fragments.rosetta.RosettaFragment.subregion()=csb.bio.fragments.rosetta.RosettaFragment-class.html#subregion,Method csb.bio.hmm.ProfileHMM.subregion()=csb.bio.hmm.ProfileHMM-class.html#subregion,Method csb.bio.sequence.AbstractAlignment.subregion()=csb.bio.sequence.AbstractAlignment-class.html#subregion,Method csb.bio.sequence.AbstractSequence.subregion()=csb.bio.sequence.AbstractSequence-class.html#subregion,Method csb.bio.sequence.Sequence.subregion()=csb.bio.sequence.Sequence-class.html#subregion,Method csb.bio.sequence.SequenceAdapter.subregion()=csb.bio.sequence.SequenceAdapter-class.html#subregion,Method csb.bio.structure.Chain.subregion()=csb.bio.structure.Chain-class.html#subregion,Method csb.bio.structure.SecondaryStructure.subregion()=csb.bio.structure.SecondaryStructure-class.html#subregion"><a title="csb.bio.fragments.rosetta.RosettaFragment.subregion
csb.bio.hmm.ProfileHMM.subregion
csb.bio.sequence.AbstractAlignment.subregion
csb.bio.sequence.AbstractSequence.subregion
csb.bio.sequence.Sequence.subregion
csb.bio.sequence.SequenceAdapter.subregion
csb.bio.structure.Chain.subregion
csb.bio.structure.SecondaryStructure.subregion" class="py-name" href="#" onclick="return doclink('link-213', 'subregion', 'link-213');">subregion</a></tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-214', 'start', 'link-126');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">spec</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name" targets="Method csb.io.plots.Backend.stop()=csb.io.plots.Backend-class.html#stop"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-215', 'stop', 'link-215');">stop</a></tt><tt class="py-op">)</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">0</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">spec</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-216', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">)</tt>             </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.__iter__"></a><div id="AbstractSequence.__iter__-def"><a name="L348"></a><tt class="py-lineno"> 348</tt> <a class="py-toggle" href="#" id="AbstractSequence.__iter__-toggle" onclick="return toggle('AbstractSequence.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__iter__-expanded"><a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">index</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-217', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">index</tt><tt class="py-op">]</tt> </tt>
</div><a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">         </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="AbstractSequence._add"></a><div id="AbstractSequence._add-def"><a name="L353"></a><tt class="py-lineno"> 353</tt> <a class="py-toggle" href="#" id="AbstractSequence._add-toggle" onclick="return toggle('AbstractSequence._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence._add-expanded"><a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-docstring">        Append a C{residue} to the sequence.</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"><tt class="py-docstring">        This is a hook method invoked internally for each residue during object</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line"><tt class="py-docstring">        construction. By implementing this method, sub-classes define how</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line"><tt class="py-docstring">        residues are attached to the sequence object.   </tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="AbstractSequence._get"></a><div id="AbstractSequence._get-def"><a name="L364"></a><tt class="py-lineno"> 364</tt> <a class="py-toggle" href="#" id="AbstractSequence._get-toggle" onclick="return toggle('AbstractSequence._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence._get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence._get-expanded"><a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"><tt class="py-docstring">        Retrieve the sequence residue at the specified position (1-based, positive).</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-docstring">        This is a hook method which defines the actual behavior of the sequence</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-docstring">        residue indexer.</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-docstring">          </tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ResidueInfo}</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequencePositionError: when the supplied rank is out of range</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence._factory"></a><div id="AbstractSequence._factory-def"><a name="L376"></a><tt class="py-lineno"> 376</tt> <a class="py-toggle" href="#" id="AbstractSequence._factory-toggle" onclick="return toggle('AbstractSequence._factory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#_factory">_factory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">a</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence._factory-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence._factory-expanded"><a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-docstring">        Return a new sequence of the current L{AbstractSequence} sub-class.</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-218" class="py-name" targets="Variable csb.bio.io.clans.Color.a=csb.bio.io.clans.Color-class.html#a,Variable csb.statistics.pdf.GeneralizedInverseGaussian.a=csb.statistics.pdf.GeneralizedInverseGaussian-class.html#a"><a title="csb.bio.io.clans.Color.a
csb.statistics.pdf.GeneralizedInverseGaussian.a" class="py-name" href="#" onclick="return doclink('link-218', 'a', 'link-218');">a</a></tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">k</tt><tt class="py-op">)</tt>     </tt>
</div><a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"> </tt>
<a name="AbstractSequence.strip"></a><div id="AbstractSequence.strip-def"><a name="L382"></a><tt class="py-lineno"> 382</tt> <a class="py-toggle" href="#" id="AbstractSequence.strip-toggle" onclick="return toggle('AbstractSequence.strip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#strip">strip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.strip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.strip-expanded"><a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"><tt class="py-docstring">        Remove all gaps and insertions from the sequence.</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new sequence instance, containing no gaps</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractSequence}</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">        <tt id="link-219" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-219', 'residues', 'link-209');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-220" class="py-name" targets="Variable csb.bio.io.clans.Color.r=csb.bio.io.clans.Color-class.html#r"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-220', 'r', 'link-220');">r</a></tt> <tt class="py-keyword">for</tt> <tt id="link-221" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-221', 'r', 'link-220');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt>  </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-222" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-222', 'r', 'link-220');">r</a></tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-223', 'type', 'link-121');">type</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-224', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-225', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-226', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-227', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">         </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-228', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-229', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-230" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-230', 'residues', 'link-209');">residues</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-231', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">             </tt>
<a name="AbstractSequence.subregion"></a><div id="AbstractSequence.subregion-def"><a name="L394"></a><tt class="py-lineno"> 394</tt> <a class="py-toggle" href="#" id="AbstractSequence.subregion-toggle" onclick="return toggle('AbstractSequence.subregion');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#subregion">subregion</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.subregion-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.subregion-expanded"><a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line"><tt class="py-docstring">        Extract a subsequence, defined by [start, end]. The start and end</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line"><tt class="py-docstring">        positions are 1-based, inclusive.</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line"><tt class="py-docstring">        @param start: start position</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line"><tt class="py-docstring">        @type start: int</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line"><tt class="py-docstring">        @param end: end position</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line"><tt class="py-docstring">        @type end: int</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new sequence</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractSequence}</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequencePositionError: if start/end positions are out of range</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">        <tt class="py-name">positions</tt> <tt class="py-op">=</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt id="link-232" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-232', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-233" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-233', 'end', 'link-125');">end</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name" targets="Method csb.bio.sequence.AbstractSequence.extract()=csb.bio.sequence.AbstractSequence-class.html#extract,Method csb.bio.sequence.Sequence.extract()=csb.bio.sequence.Sequence-class.html#extract,Method csb.bio.sequence.SequenceAdapter.extract()=csb.bio.sequence.SequenceAdapter-class.html#extract"><a title="csb.bio.sequence.AbstractSequence.extract
csb.bio.sequence.Sequence.extract
csb.bio.sequence.SequenceAdapter.extract" class="py-name" href="#" onclick="return doclink('link-234', 'extract', 'link-234');">extract</a></tt><tt class="py-op">(</tt><tt class="py-name">positions</tt><tt class="py-op">)</tt> </tt>
</div><a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.extract"></a><div id="AbstractSequence.extract-def"><a name="L412"></a><tt class="py-lineno"> 412</tt> <a class="py-toggle" href="#" id="AbstractSequence.extract-toggle" onclick="return toggle('AbstractSequence.extract');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#extract">extract</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">positions</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.extract-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.extract-expanded"><a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-docstring">        Extract a subsequence, defined by a list of 1-based positions.</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">        @param positions: positions to extract</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line"><tt class="py-docstring">        @type positions: tuple of int</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new sequence</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractSequence}</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequencePositionError: if any position is out of range</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">        <tt id="link-235" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-235', 'end', 'link-125');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-236', 'length', 'link-156');">length</a></tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">        <tt id="link-237" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-237', 'residues', 'link-209');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">         </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-238" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-238', 'rank', 'link-136');">rank</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-239" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-239', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">positions</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-240" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-240', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-241" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-241', 'end', 'link-125');">end</a></tt><tt class="py-op">:</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">                <tt id="link-242" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-242', 'residues', 'link-209');">residues</a></tt><tt class="py-op">.</tt><tt id="link-243" class="py-name" targets="Method csb.bio.fragments.rosetta.RosettaFragmentMap.append()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#append,Method csb.bio.hmm.EmissionTable.append()=csb.bio.hmm.EmissionTable-class.html#append,Method csb.bio.hmm.HMMLayer.append()=csb.bio.hmm.HMMLayer-class.html#append,Method csb.bio.hmm.TransitionTable.append()=csb.bio.hmm.TransitionTable-class.html#append,Method csb.bio.structure.ChainResiduesCollection.append()=csb.bio.structure.ChainResiduesCollection-class.html#append,Method csb.bio.structure.DisorderedAtom.append()=csb.bio.structure.DisorderedAtom-class.html#append,Method csb.bio.structure.EnsembleModelsCollection.append()=csb.bio.structure.EnsembleModelsCollection-class.html#append,Method csb.bio.structure.ResidueAtomsTable.append()=csb.bio.structure.ResidueAtomsTable-class.html#append,Method csb.bio.structure.SecondaryStructure.append()=csb.bio.structure.SecondaryStructure-class.html#append,Method csb.bio.structure.StructureChainsTable.append()=csb.bio.structure.StructureChainsTable-class.html#append,Method csb.core.CollectionContainer.append()=csb.core.CollectionContainer-class.html#append,Method csb.core.DictionaryContainer.append()=csb.core.DictionaryContainer-class.html#append"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-243', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get</tt><tt class="py-op">(</tt><tt id="link-244" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-244', 'rank', 'link-136');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-245" class="py-name"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-245', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">(</tt><tt id="link-246" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-246', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt id="link-247" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-247', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">             </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-248', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-249', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-250" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-250', 'residues', 'link-209');">residues</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-251', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.dump"></a><div id="AbstractSequence.dump-def"><a name="L436"></a><tt class="py-lineno"> 436</tt> <a class="py-toggle" href="#" id="AbstractSequence.dump-toggle" onclick="return toggle('AbstractSequence.dump');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#dump">dump</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.dump-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.dump-expanded"><a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">        Dump the sequence in FASTA format.</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring">        @param output_file: write the output to this file or stream</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">        @type output_file: str or stream</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-252" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-252', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-253', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-254', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-255', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-256" class="py-name"><a title="csb.bio.io.fasta.FASTAOutputBuilder" class="py-name" href="#" onclick="return doclink('link-256', 'FASTAOutputBuilder', 'link-187');">FASTAOutputBuilder</a></tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">             </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-257" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-257', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-258', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="csb.io.EntryWriter" class="py-name" href="#" onclick="return doclink('link-259', 'EntryWriter', 'link-190');">EntryWriter</a></tt><tt class="py-op">(</tt><tt class="py-name">output_file</tt><tt class="py-op">,</tt> <tt id="link-260" class="py-name"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-260', 'close', 'link-191');">close</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">out</tt><tt class="py-op">:</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">            <tt id="link-261" class="py-name"><a title="csb.bio.io.fasta.FASTAOutputBuilder" class="py-name" href="#" onclick="return doclink('link-261', 'FASTAOutputBuilder', 'link-187');">FASTAOutputBuilder</a></tt><tt class="py-op">(</tt><tt class="py-name">out</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-262', 'stream', 'link-194');">stream</a></tt><tt class="py-op">,</tt> <tt id="link-263" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-263', 'headers', 'link-195');">headers</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="csb.bio.io.fasta.A3MOutputBuilder.add_sequence
csb.bio.io.fasta.FASTAOutputBuilder.add_sequence
csb.bio.io.fasta.OutputBuilder.add_sequence
csb.bio.io.fasta.PIROutputBuilder.add_sequence" class="py-name" href="#" onclick="return doclink('link-264', 'add_sequence', 'link-197');">add_sequence</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">         </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.length"></a><div id="AbstractSequence.length-def"><a name="L449"></a><tt class="py-lineno"> 449</tt> <a class="py-toggle" href="#" id="AbstractSequence.length-toggle" onclick="return toggle('AbstractSequence.length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#length">length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.length-expanded"><a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"><tt class="py-docstring">        Number of residues</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">)</tt> </tt>
</div><a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">     </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.id"></a><div id="AbstractSequence.id-def"><a name="L457"></a><tt class="py-lineno"> 457</tt> <a class="py-toggle" href="#" id="AbstractSequence.id-toggle" onclick="return toggle('AbstractSequence.id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#id">id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.id-expanded"><a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring">        Sequence identifier</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> </tt>
</div><a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">id</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="AbstractSequence.id"></a><div id="AbstractSequence.id-def"><a name="L464"></a><tt class="py-lineno"> 464</tt> <a class="py-toggle" href="#" id="AbstractSequence.id-toggle" onclick="return toggle('AbstractSequence.id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#id">id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.id-expanded"><a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-265" class="py-name" targets="Variable csb.core.EnumItem.value=csb.core.EnumItem-class.html#value,Variable csb.statistics.pdf.parameterized.AbstractParameter.value=csb.statistics.pdf.parameterized.AbstractParameter-class.html#value"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-265', 'value', 'link-265');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">            <tt id="link-266" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-266', 'value', 'link-265');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-267" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-267', 'value', 'link-265');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name" targets="Method csb.bio.sequence.AbstractSequence.strip()=csb.bio.sequence.AbstractSequence-class.html#strip,Method csb.bio.sequence.Sequence.strip()=csb.bio.sequence.Sequence-class.html#strip,Method csb.bio.sequence.SequenceAdapter.strip()=csb.bio.sequence.SequenceAdapter-class.html#strip"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-268', 'strip', 'link-268');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt id="link-269" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-269', 'value', 'link-265');">value</a></tt> </tt>
</div><a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">             </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.header"></a><div id="AbstractSequence.header-def"><a name="L470"></a><tt class="py-lineno"> 470</tt> <a class="py-toggle" href="#" id="AbstractSequence.header-toggle" onclick="return toggle('AbstractSequence.header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#header">header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.header-expanded"><a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"><tt class="py-docstring">        Sequence description</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header</tt> </tt>
</div><a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">header</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="AbstractSequence.header"></a><div id="AbstractSequence.header-def"><a name="L477"></a><tt class="py-lineno"> 477</tt> <a class="py-toggle" href="#" id="AbstractSequence.header-toggle" onclick="return toggle('AbstractSequence.header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#header">header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.header-expanded"><a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-270" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-270', 'value', 'link-265');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">            <tt id="link-271" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-271', 'value', 'link-265');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'sequence'</tt>        </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">            <tt id="link-272" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-272', 'value', 'link-265');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-273" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-273', 'value', 'link-265');">value</a></tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-274', 'strip', 'link-268');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">lstrip</tt><tt class="py-op">(</tt><tt id="link-275" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-275', 'AbstractSequence', 'link-182');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-276', 'DELIMITER', 'link-199');">DELIMITER</a></tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header</tt> <tt class="py-op">=</tt> <tt id="link-277" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-277', 'value', 'link-265');">value</a></tt> </tt>
</div><a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">     </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt>   </tt>
<a name="AbstractSequence.type"></a><div id="AbstractSequence.type-def"><a name="L485"></a><tt class="py-lineno"> 485</tt> <a class="py-toggle" href="#" id="AbstractSequence.type-toggle" onclick="return toggle('AbstractSequence.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.type-expanded"><a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring">        Sequence type - a member of L{SequenceTypes}</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: enum item</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> </tt>
</div><a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">type</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="AbstractSequence.type"></a><div id="AbstractSequence.type-def"><a name="L492"></a><tt class="py-lineno"> 492</tt> <a class="py-toggle" href="#" id="AbstractSequence.type-toggle" onclick="return toggle('AbstractSequence.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.type-expanded"><a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-278" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-278', 'value', 'link-265');">value</a></tt><tt class="py-op">,</tt> <tt id="link-279" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-279', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-280', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">            <tt id="link-281" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-281', 'value', 'link-265');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-282" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-282', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-283', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-284', 'Enum', 'link-116');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-285" class="py-name" targets="Method csb.apps.ArgHandler.parse()=csb.apps.ArgHandler-class.html#parse,Method csb.bio.io.dssp.DSSPParser.parse()=csb.bio.io.dssp.DSSPParser-class.html#parse,Method csb.bio.io.dssp.StrideParser.parse()=csb.bio.io.dssp.StrideParser-class.html#parse,Method csb.bio.io.hhpred.HHProfileParser.parse()=csb.bio.io.hhpred.HHProfileParser-class.html#parse,Method csb.bio.io.procheck.ProcheckParser.parse()=csb.bio.io.procheck.ProcheckParser-class.html#parse,Method csb.bio.io.psipred.PSIPredParser.parse()=csb.bio.io.psipred.PSIPredParser-class.html#parse,Method csb.bio.io.vasco.VascoStructureParser.parse()=csb.bio.io.vasco.VascoStructureParser-class.html#parse,Method csb.bio.io.whatif.WhatCheckParser.parse()=csb.bio.io.whatif.WhatCheckParser-class.html#parse,Method csb.bio.io.wwpdb.AbstractStructureParser.parse()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse,Method csb.bio.io.wwpdb.DegenerateID.parse()=csb.bio.io.wwpdb.DegenerateID-class.html#parse,Method csb.bio.io.wwpdb.EntryID.parse()=csb.bio.io.wwpdb.EntryID-class.html#parse,Method csb.bio.io.wwpdb.SeqResID.parse()=csb.bio.io.wwpdb.SeqResID-class.html#parse,Method csb.bio.io.wwpdb.StandardID.parse()=csb.bio.io.wwpdb.StandardID-class.html#parse,Static Method csb.bio.nmr.Label.parse()=csb.bio.nmr.Label-class.html#parse,Static Method csb.bio.sequence.A3MAlignment.parse()=csb.bio.sequence.A3MAlignment-class.html#parse,Static Method csb.bio.sequence.SequenceAlignment.parse()=csb.bio.sequence.SequenceAlignment-class.html#parse,Static Method csb.bio.sequence.StructureAlignment.parse()=csb.bio.sequence.StructureAlignment-class.html#parse,Static Method csb.bio.sequence.alignment.SimilarityMatrix.parse()=csb.bio.sequence.alignment.SimilarityMatrix-class.html#parse,Static Method csb.bio.structure.SecondaryStructure.parse()=csb.bio.structure.SecondaryStructure-class.html#parse,Static Method csb.core.Enum.parse()=csb.core.Enum-class.html#parse"><a title="csb.apps.ArgHandler.parse
csb.bio.io.dssp.DSSPParser.parse
csb.bio.io.dssp.StrideParser.parse
csb.bio.io.hhpred.HHProfileParser.parse
csb.bio.io.procheck.ProcheckParser.parse
csb.bio.io.psipred.PSIPredParser.parse
csb.bio.io.vasco.VascoStructureParser.parse
csb.bio.io.whatif.WhatCheckParser.parse
csb.bio.io.wwpdb.AbstractStructureParser.parse
csb.bio.io.wwpdb.DegenerateID.parse
csb.bio.io.wwpdb.EntryID.parse
csb.bio.io.wwpdb.SeqResID.parse
csb.bio.io.wwpdb.StandardID.parse
csb.bio.nmr.Label.parse
csb.bio.sequence.A3MAlignment.parse
csb.bio.sequence.SequenceAlignment.parse
csb.bio.sequence.StructureAlignment.parse
csb.bio.sequence.alignment.SimilarityMatrix.parse
csb.bio.structure.SecondaryStructure.parse
csb.core.Enum.parse" class="py-name" href="#" onclick="return doclink('link-285', 'parse', 'link-285');">parse</a></tt><tt class="py-op">(</tt><tt id="link-286" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-286', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">,</tt> <tt id="link-287" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-287', 'value', 'link-265');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-288" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-288', 'value', 'link-265');">value</a></tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="csb.core.EnumItem.enum
csb.core.enum" class="py-name" href="#" onclick="return doclink('link-289', 'enum', 'link-142');">enum</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt id="link-290" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-290', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">:</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-291" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-291', 'value', 'link-265');">value</a></tt><tt class="py-op">)</tt>  </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt id="link-292" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-292', 'value', 'link-265');">value</a></tt>  </tt>
</div><a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.sequence"></a><div id="AbstractSequence.sequence-def"><a name="L500"></a><tt class="py-lineno"> 500</tt> <a class="py-toggle" href="#" id="AbstractSequence.sequence-toggle" onclick="return toggle('AbstractSequence.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  </tt>
</div><a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring">        The actual sequence</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name" targets="Static Method csb.bio.nmr.NOESpectrum.join()=csb.bio.nmr.NOESpectrum-class.html#join"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-293', 'join', 'link-293');">join</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-294" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-294', 'r', 'link-220');">r</a></tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-295', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-296" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-296', 'r', 'link-220');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">     </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.alphabet"></a><div id="AbstractSequence.alphabet-def"><a name="L508"></a><tt class="py-lineno"> 508</tt> <a class="py-toggle" href="#" id="AbstractSequence.alphabet-toggle" onclick="return toggle('AbstractSequence.alphabet');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#alphabet">alphabet</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.alphabet-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.alphabet-expanded"><a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line"><tt class="py-docstring">        The sequence alphabet corresponding to the current sequence type</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.core.enum}</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-297" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-297', 'SequenceAlphabets', 'link-119');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-298" class="py-name" targets="Function csb.bio.io.dssp.get()=csb.bio.io.dssp-module.html#get,Method csb.bio.io.wwpdb.RemoteStructureProvider.get()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#get,Method csb.bio.io.wwpdb.StructureProvider.get()=csb.bio.io.wwpdb.StructureProvider-class.html#get,Function csb.bio.io.wwpdb.get()=csb.bio.io.wwpdb-module.html#get,Static Method csb.bio.nmr.AtomConnectivity.get()=csb.bio.nmr.AtomConnectivity-class.html#get,Method csb.bio.nmr.NOEPeak.get()=csb.bio.nmr.NOEPeak-class.html#get,Static Method csb.bio.nmr.RandomCoil.get()=csb.bio.nmr.RandomCoil-class.html#get,Static Method csb.bio.sequence.SequenceAlphabets.get()=csb.bio.sequence.SequenceAlphabets-class.html#get,Static Method csb.build.BuildTypes.get()=csb.build.BuildTypes-class.html#get,Static Method csb.io.plots.Backend.get()=csb.io.plots.Backend-class.html#get,Method csb.statistics.maxent.MaxentModel.get()=csb.statistics.maxent.MaxentModel-class.html#get"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-298', 'get', 'link-298');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt><tt class="py-op">)</tt> </tt>
</div><a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">     </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractSequence.residues"></a><div id="AbstractSequence.residues-def"><a name="L516"></a><tt class="py-lineno"> 516</tt> <a class="py-toggle" href="#" id="AbstractSequence.residues-toggle" onclick="return toggle('AbstractSequence.residues');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#residues">residues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.residues-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.residues-expanded"><a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-docstring">        Rank-based access to the underlying L{residues&lt;csb.bio.sequence.ResidueInfo&gt;}</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceIndexer}</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-299" class="py-name" targets="Class csb.bio.sequence.SequenceIndexer=csb.bio.sequence.SequenceIndexer-class.html"><a title="csb.bio.sequence.SequenceIndexer" class="py-name" href="#" onclick="return doclink('link-299', 'SequenceIndexer', 'link-299');">SequenceIndexer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"> </tt>
<a name="AbstractSequence.__len__"></a><div id="AbstractSequence.__len__-def"><a name="L523"></a><tt class="py-lineno"> 523</tt> <a class="py-toggle" href="#" id="AbstractSequence.__len__-toggle" onclick="return toggle('AbstractSequence.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__len__-expanded"><a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-300', 'length', 'link-156');">length</a></tt> </tt>
</div><a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.__repr__"></a><div id="AbstractSequence.__repr__-def"><a name="L526"></a><tt class="py-lineno"> 526</tt> <a class="py-toggle" href="#" id="AbstractSequence.__repr__-toggle" onclick="return toggle('AbstractSequence.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__repr__-expanded"><a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'&lt;{0.__class__.__name__}: {0.id}, {0.length} residues&gt;'</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-301', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">     </tt>
<a name="AbstractSequence.__str__"></a><div id="AbstractSequence.__str__-def"><a name="L529"></a><tt class="py-lineno"> 529</tt> <a class="py-toggle" href="#" id="AbstractSequence.__str__-toggle" onclick="return toggle('AbstractSequence.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractSequence-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractSequence.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractSequence.__str__-expanded"><a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'{0}{1.header}\n{1.sequence}'</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-302', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt id="link-303" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-303', 'AbstractSequence', 'link-182');">AbstractSequence</a></tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-304', 'DELIMITER', 'link-199');">DELIMITER</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">         </tt>
<a name="Sequence"></a><div id="Sequence-def"><a name="L532"></a><tt class="py-lineno"> 532</tt> <a class="py-toggle" href="#" id="Sequence-toggle" onclick="return toggle('Sequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html">Sequence</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractSequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Sequence-expanded"><a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line"><tt class="py-docstring">    High-performance sequence object. The actual sequence is stored internally</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"><tt class="py-docstring">    as a string. The indexer acts as a residue factory, which creates a new</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line"><tt class="py-docstring">    L{ResidueInfo} instance each time. </tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line"><tt class="py-docstring">    @note: This class was created with parsing large volumes of data in mind. This</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line"><tt class="py-docstring">           comes at the expense of degraded performance of the sequence indexer.</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line"><tt class="py-docstring">    @param id: FASTA ID of this sequence (e.g. accession number)</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line"><tt class="py-docstring">    @type id: str</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"><tt class="py-docstring">    @param header: FASTA sequence header</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-docstring">    @type header: str</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt class="py-docstring">    @param residues: sequence string</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-docstring">    @type residues: str</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-docstring">    @param type: a L{SequenceTypes} member (defaults to protein)</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-docstring">    @type type: L{EnumItem}</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">     </tt>
<a name="Sequence.__init__"></a><div id="Sequence.__init__-def"><a name="L551"></a><tt class="py-lineno"> 551</tt> <a class="py-toggle" href="#" id="Sequence.__init__-toggle" onclick="return toggle('Sequence.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">header</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">=</tt><tt id="link-305" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-305', 'SequenceTypes', 'link-92');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-306', 'Unknown', 'link-12');">Unknown</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.__init__-expanded"><a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">           </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-307', 'id', 'link-148');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-308" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-308', 'id', 'link-148');">id</a></tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-309', 'header', 'link-204');">header</a></tt> <tt class="py-op">=</tt> <tt id="link-310" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-310', 'header', 'link-204');">header</a></tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-311', 'type', 'link-121');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-312" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-312', 'type', 'link-121');">type</a></tt>         </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_append</tt><tt class="py-op">(</tt><tt id="link-313" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-313', 'residues', 'link-209');">residues</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">     </tt>
<a name="Sequence._append"></a><div id="Sequence._append-def"><a name="L564"></a><tt class="py-lineno"> 564</tt> <a class="py-toggle" href="#" id="Sequence._append-toggle" onclick="return toggle('Sequence._append');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#_append">_append</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence._append-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence._append-expanded"><a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">        <tt class="py-comment"># this seems to be the fastest method for sanitization and storage        </tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt> <tt class="py-op">+=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">sub</tt><tt class="py-op">(</tt><tt class="py-string">'([^\w\-\.])+'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> <tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">         </tt>
<a name="Sequence._add"></a><div id="Sequence._add-def"><a name="L568"></a><tt class="py-lineno"> 568</tt> <a class="py-toggle" href="#" id="Sequence._add-toggle" onclick="return toggle('Sequence._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">char</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence._add-expanded"><a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_append</tt><tt class="py-op">(</tt><tt class="py-name">char</tt><tt class="py-op">)</tt> </tt>
</div><a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">             </tt>
<a name="Sequence._get"></a><div id="Sequence._get-def"><a name="L571"></a><tt class="py-lineno"> 571</tt> <a class="py-toggle" href="#" id="Sequence._get-toggle" onclick="return toggle('Sequence._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>         </tt>
</div><a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">         </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">        <tt id="link-314" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-314', 'type', 'link-121');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-315" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-315', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-316', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-317', 'Enum', 'link-116');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="csb.apps.ArgHandler.parse
csb.bio.io.dssp.DSSPParser.parse
csb.bio.io.dssp.StrideParser.parse
csb.bio.io.hhpred.HHProfileParser.parse
csb.bio.io.procheck.ProcheckParser.parse
csb.bio.io.psipred.PSIPredParser.parse
csb.bio.io.vasco.VascoStructureParser.parse
csb.bio.io.whatif.WhatCheckParser.parse
csb.bio.io.wwpdb.AbstractStructureParser.parse
csb.bio.io.wwpdb.DegenerateID.parse
csb.bio.io.wwpdb.EntryID.parse
csb.bio.io.wwpdb.SeqResID.parse
csb.bio.io.wwpdb.StandardID.parse
csb.bio.nmr.Label.parse
csb.bio.sequence.A3MAlignment.parse
csb.bio.sequence.SequenceAlignment.parse
csb.bio.sequence.StructureAlignment.parse
csb.bio.sequence.alignment.SimilarityMatrix.parse
csb.bio.structure.SecondaryStructure.parse
csb.core.Enum.parse" class="py-name" href="#" onclick="return doclink('link-318', 'parse', 'link-285');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-319', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-320" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-320', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-321" class="py-name" targets="Class csb.bio.fragments.rosetta.ResidueInfo=csb.bio.fragments.rosetta.ResidueInfo-class.html,Class csb.bio.io.wwpdb.ResidueInfo=csb.bio.io.wwpdb.ResidueInfo-class.html,Class csb.bio.sequence.ResidueInfo=csb.bio.sequence.ResidueInfo-class.html"><a title="csb.bio.fragments.rosetta.ResidueInfo
csb.bio.io.wwpdb.ResidueInfo
csb.bio.sequence.ResidueInfo" class="py-name" href="#" onclick="return doclink('link-321', 'ResidueInfo', 'link-321');">ResidueInfo</a></tt><tt class="py-op">(</tt><tt id="link-322" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-322', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-323" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-323', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">     </tt>
<a name="Sequence.strip"></a><div id="Sequence.strip-def"><a name="L576"></a><tt class="py-lineno"> 576</tt> <a class="py-toggle" href="#" id="Sequence.strip-toggle" onclick="return toggle('Sequence.strip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#strip">strip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.strip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.strip-expanded"><a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">        <tt id="link-324" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-324', 'residues', 'link-209');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">                        <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-325', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-326', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">                                        <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-327', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-328" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-328', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-329', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-330', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-331" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-331', 'residues', 'link-209');">residues</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-332', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt>         </tt>
</div><a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"> </tt>
<a name="Sequence.subregion"></a><div id="Sequence.subregion-def"><a name="L582"></a><tt class="py-lineno"> 582</tt> <a class="py-toggle" href="#" id="Sequence.subregion-toggle" onclick="return toggle('Sequence.subregion');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#subregion">subregion</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.subregion-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.subregion-expanded"><a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-333" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-333', 'start', 'link-126');">start</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-334" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-334', 'end', 'link-125');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-335', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-336" class="py-name"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-336', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-337', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">                        </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">        <tt id="link-338" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-338', 'residues', 'link-209');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-339" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-339', 'start', 'link-126');">start</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> <tt class="py-op">:</tt> <tt id="link-340" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-340', 'end', 'link-125');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-341', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-342', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-343" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-343', 'residues', 'link-209');">residues</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-344', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt>                 </tt>
</div><a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"> </tt>
<a name="Sequence.extract"></a><div id="Sequence.extract-def"><a name="L590"></a><tt class="py-lineno"> 590</tt> <a class="py-toggle" href="#" id="Sequence.extract-toggle" onclick="return toggle('Sequence.extract');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#extract">extract</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">positions</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.extract-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.extract-expanded"><a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">        <tt id="link-345" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-345', 'end', 'link-125');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-346', 'length', 'link-156');">length</a></tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt id="link-347" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-347', 'residues', 'link-209');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">         </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-348" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-348', 'rank', 'link-136');">rank</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt id="link-349" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-349', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">positions</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-350" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-350', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-351" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-351', 'end', 'link-125');">end</a></tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">                <tt id="link-352" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-352', 'residues', 'link-209');">residues</a></tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-353', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-354" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-354', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-355" class="py-name"><a title="csb.bio.sequence.SequencePositionError" class="py-name" href="#" onclick="return doclink('link-355', 'SequencePositionError', 'link-157');">SequencePositionError</a></tt><tt class="py-op">(</tt><tt id="link-356" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-356', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt id="link-357" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-357', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">             </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-358', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-359', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-360', 'join', 'link-293');">join</a></tt><tt class="py-op">(</tt><tt id="link-361" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-361', 'residues', 'link-209');">residues</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-362', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">             </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Sequence.sequence"></a><div id="Sequence.sequence-def"><a name="L604"></a><tt class="py-lineno"> 604</tt> <a class="py-toggle" href="#" id="Sequence.sequence-toggle" onclick="return toggle('Sequence.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.Sequence-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Sequence.sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Sequence.sequence-expanded"><a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt> </tt>
</div></div><a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"> </tt>
<a name="RichSequence"></a><div id="RichSequence-def"><a name="L607"></a><tt class="py-lineno"> 607</tt> <a class="py-toggle" href="#" id="RichSequence-toggle" onclick="return toggle('RichSequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.RichSequence-class.html">RichSequence</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractSequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RichSequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="RichSequence-expanded"><a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence implementation, which converts the sequence into a list of</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"><tt class="py-docstring">    L{ResidueInfo} objects. See L{AbstractSequence} for details.</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">         </tt>
<a name="RichSequence._add"></a><div id="RichSequence._add-def"><a name="L613"></a><tt class="py-lineno"> 613</tt> <a class="py-toggle" href="#" id="RichSequence._add-toggle" onclick="return toggle('RichSequence._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.RichSequence-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RichSequence._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RichSequence._add-expanded"><a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">         </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-363" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-363', 'residue', 'link-152');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'rank'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-364" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-364', 'residue', 'link-152');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'type'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>             </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-365', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt id="link-366" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-366', 'residue', 'link-152');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">             </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-367" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-367', 'residue', 'link-152');">residue</a></tt><tt class="py-op">.</tt><tt class="py-name">isalpha</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt id="link-368" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-368', 'residue', 'link-152');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-369', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-370', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-371', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-372', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">                 </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">                <tt id="link-373" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-373', 'type', 'link-121');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-374" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-374', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-375', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-376', 'Enum', 'link-116');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="csb.apps.ArgHandler.parse
csb.bio.io.dssp.DSSPParser.parse
csb.bio.io.dssp.StrideParser.parse
csb.bio.io.hhpred.HHProfileParser.parse
csb.bio.io.procheck.ProcheckParser.parse
csb.bio.io.psipred.PSIPredParser.parse
csb.bio.io.vasco.VascoStructureParser.parse
csb.bio.io.whatif.WhatCheckParser.parse
csb.bio.io.wwpdb.AbstractStructureParser.parse
csb.bio.io.wwpdb.DegenerateID.parse
csb.bio.io.wwpdb.EntryID.parse
csb.bio.io.wwpdb.SeqResID.parse
csb.bio.io.wwpdb.StandardID.parse
csb.bio.nmr.Label.parse
csb.bio.sequence.A3MAlignment.parse
csb.bio.sequence.SequenceAlignment.parse
csb.bio.sequence.StructureAlignment.parse
csb.bio.sequence.alignment.SimilarityMatrix.parse
csb.bio.structure.SecondaryStructure.parse
csb.core.Enum.parse" class="py-name" href="#" onclick="return doclink('link-377', 'parse', 'link-285');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-378', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-379" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-379', 'residue', 'link-152');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">                <tt id="link-380" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-380', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-381', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt id="link-382" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo
csb.bio.io.wwpdb.ResidueInfo
csb.bio.sequence.ResidueInfo" class="py-name" href="#" onclick="return doclink('link-382', 'ResidueInfo', 'link-321');">ResidueInfo</a></tt><tt class="py-op">(</tt><tt id="link-383" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-383', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-384" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-384', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">             </tt>
<a name="RichSequence._get"></a><div id="RichSequence._get-def"><a name="L625"></a><tt class="py-lineno"> 625</tt> <a class="py-toggle" href="#" id="RichSequence._get-toggle" onclick="return toggle('RichSequence._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.RichSequence-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RichSequence._get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RichSequence._get-expanded"><a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-385" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-385', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="RichSequence.create"></a><div id="RichSequence.create-def"><a name="L629"></a><tt class="py-lineno"> 629</tt> <a class="py-toggle" href="#" id="RichSequence.create-toggle" onclick="return toggle('RichSequence.create');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.RichSequence-class.html#create">create</a><tt class="py-op">(</tt><tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RichSequence.create-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RichSequence.create-expanded"><a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{RichSequence} from existing L{AbstractSequence}.</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"><tt class="py-docstring">        @type sequence: L{AbstractSequence}</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{RichSequence}</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-386" class="py-name" targets="Class csb.bio.sequence.RichSequence=csb.bio.sequence.RichSequence-class.html"><a title="csb.bio.sequence.RichSequence" class="py-name" href="#" onclick="return doclink('link-386', 'RichSequence', 'link-386');">RichSequence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">                <tt id="link-387" class="py-name" targets="Variable csb.bio.fragments.Assignment.sequence=csb.bio.fragments.Assignment-class.html#sequence,Variable csb.bio.fragments.Target.sequence=csb.bio.fragments.Target-class.html#sequence,Package csb.bio.sequence=csb.bio.sequence-module.html,Variable csb.bio.sequence.AbstractSequence.sequence=csb.bio.sequence.AbstractSequence-class.html#sequence,Variable csb.bio.sequence.Sequence.sequence=csb.bio.sequence.Sequence-class.html#sequence,Variable csb.bio.sequence.SequenceAdapter.sequence=csb.bio.sequence.SequenceAdapter-class.html#sequence,Variable csb.bio.structure.Chain.sequence=csb.bio.structure.Chain-class.html#sequence"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-387', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-388', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt id="link-389" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-389', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-390', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-391" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-391', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-392', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-393" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-393', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-394', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt>     </tt>
</div></div><a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line"> </tt>
<a name="ChainSequence"></a><div id="ChainSequence-def"><a name="L639"></a><tt class="py-lineno"> 639</tt> <a class="py-toggle" href="#" id="ChainSequence-toggle" onclick="return toggle('ChainSequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.ChainSequence-class.html">ChainSequence</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractSequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ChainSequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ChainSequence-expanded"><a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence view for L{csb.bio.structure.Chain} objects.</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line"><tt class="py-docstring">    See L{AbstractSequence} for details.</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">             </tt>
<a name="ChainSequence._add"></a><div id="ChainSequence._add-def"><a name="L645"></a><tt class="py-lineno"> 645</tt> <a class="py-toggle" href="#" id="ChainSequence._add-toggle" onclick="return toggle('ChainSequence._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ChainSequence-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ChainSequence._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ChainSequence._add-expanded"><a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">         </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-395" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-395', 'residue', 'link-152');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'rank'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-396" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-396', 'residue', 'link-152');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'type'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-397" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-397', 'residue', 'link-152');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-398', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt id="link-399" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-399', 'residue', 'link-152');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">             </tt>
<a name="ChainSequence._get"></a><div id="ChainSequence._get-def"><a name="L652"></a><tt class="py-lineno"> 652</tt> <a class="py-toggle" href="#" id="ChainSequence._get-toggle" onclick="return toggle('ChainSequence._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ChainSequence-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ChainSequence._get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ChainSequence._get-expanded"><a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-400" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-400', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">     </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="ChainSequence.create"></a><div id="ChainSequence.create-def"><a name="L656"></a><tt class="py-lineno"> 656</tt> <a class="py-toggle" href="#" id="ChainSequence.create-toggle" onclick="return toggle('ChainSequence.create');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.ChainSequence-class.html#create">create</a><tt class="py-op">(</tt><tt class="py-param">chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ChainSequence.create-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ChainSequence.create-expanded"><a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{ChainSequence} from existing L{Chain} instance.</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain: L{csb.bio.structure.Chain}</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ChainSequence}</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-401" class="py-name" targets="Class csb.bio.sequence.ChainSequence=csb.bio.sequence.ChainSequence-class.html"><a title="csb.bio.sequence.ChainSequence" class="py-name" href="#" onclick="return doclink('link-401', 'ChainSequence', 'link-401');">ChainSequence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">                <tt id="link-402" class="py-name" targets="Variable csb.bio.fragments.rosetta.RosettaFragment.chain=csb.bio.fragments.rosetta.RosettaFragment-class.html#chain,Method csb.bio.hmm.ProfileHMM.chain()=csb.bio.hmm.ProfileHMM-class.html#chain,Variable csb.bio.io.wwpdb.EntryID.chain=csb.bio.io.wwpdb.EntryID-class.html#chain,Variable csb.bio.io.wwpdb.ResidueInfo.chain=csb.bio.io.wwpdb.ResidueInfo-class.html#chain,Variable csb.bio.nmr.ContactMap.chain=csb.bio.nmr.ContactMap-class.html#chain"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-402', 'chain', 'link-402');">chain</a></tt><tt class="py-op">.</tt><tt id="link-403" class="py-name" targets="Variable csb.bio.io.wwpdb.EntryID.entry_id=csb.bio.io.wwpdb.EntryID-class.html#entry_id,Variable csb.bio.structure.Chain.entry_id=csb.bio.structure.Chain-class.html#entry_id"><a title="csb.bio.io.wwpdb.EntryID.entry_id
csb.bio.structure.Chain.entry_id" class="py-name" href="#" onclick="return doclink('link-403', 'entry_id', 'link-403');">entry_id</a></tt><tt class="py-op">,</tt> <tt id="link-404" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-404', 'chain', 'link-402');">chain</a></tt><tt class="py-op">.</tt><tt id="link-405" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-405', 'header', 'link-204');">header</a></tt><tt class="py-op">,</tt> <tt id="link-406" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-406', 'chain', 'link-402');">chain</a></tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-407', 'residues', 'link-209');">residues</a></tt><tt class="py-op">,</tt> <tt id="link-408" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-408', 'chain', 'link-402');">chain</a></tt><tt class="py-op">.</tt><tt id="link-409" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-409', 'type', 'link-121');">type</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"> </tt>
<a name="SequenceAdapter"></a><div id="SequenceAdapter-def"><a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">     </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt> <a class="py-toggle" href="#" id="SequenceAdapter-toggle" onclick="return toggle('SequenceAdapter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html">SequenceAdapter</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceAdapter-expanded"><a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"><tt class="py-docstring">    Base wrapper class for L{AbstractSequence} objects.</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line"><tt class="py-docstring">    Needs to be sub-classed (does not do anything special on its own).</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"><tt class="py-docstring">    @param sequence: adaptee</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line"><tt class="py-docstring">    @type sequence: L{AbstractSequence}</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter.__init__"></a><div id="SequenceAdapter.__init__-def"><a name="L676"></a><tt class="py-lineno"> 676</tt> <a class="py-toggle" href="#" id="SequenceAdapter.__init__-toggle" onclick="return toggle('SequenceAdapter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.__init__-expanded"><a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">         </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-410" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-410', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-411" class="py-name"><a title="csb.bio.sequence.AbstractSequence" class="py-name" href="#" onclick="return doclink('link-411', 'AbstractSequence', 'link-182');">AbstractSequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-412" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-412', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">         </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt> <tt class="py-op">=</tt> <tt id="link-413" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-413', 'sequence', 'link-387');">sequence</a></tt> </tt>
</div><a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"> </tt>
<a name="SequenceAdapter.__getitem__"></a><div id="SequenceAdapter.__getitem__-def"><a name="L683"></a><tt class="py-lineno"> 683</tt> <a class="py-toggle" href="#" id="SequenceAdapter.__getitem__-toggle" onclick="return toggle('SequenceAdapter.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">i</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.__getitem__-expanded"><a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> </tt>
</div><a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter.__iter__"></a><div id="SequenceAdapter.__iter__-def"><a name="L686"></a><tt class="py-lineno"> 686</tt> <a class="py-toggle" href="#" id="SequenceAdapter.__iter__-toggle" onclick="return toggle('SequenceAdapter.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.__iter__-expanded"><a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">)</tt> </tt>
</div><a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">                 </tt>
<a name="SequenceAdapter.__repr__"></a><div id="SequenceAdapter.__repr__-def"><a name="L689"></a><tt class="py-lineno"> 689</tt> <a class="py-toggle" href="#" id="SequenceAdapter.__repr__-toggle" onclick="return toggle('SequenceAdapter.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.__repr__-expanded"><a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'&lt;{0.__class__.__name__}: {0.id}, {0.length}&gt;'</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-414', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt>         </tt>
</div><a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter.__str__"></a><div id="SequenceAdapter.__str__-def"><a name="L692"></a><tt class="py-lineno"> 692</tt> <a class="py-toggle" href="#" id="SequenceAdapter.__str__-toggle" onclick="return toggle('SequenceAdapter.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.__str__-expanded"><a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">)</tt> </tt>
</div><a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter._add"></a><div id="SequenceAdapter._add-def"><a name="L695"></a><tt class="py-lineno"> 695</tt> <a class="py-toggle" href="#" id="SequenceAdapter._add-toggle" onclick="return toggle('SequenceAdapter._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter._add-expanded"><a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter._get"></a><div id="SequenceAdapter._get-def"><a name="L698"></a><tt class="py-lineno"> 698</tt> <a class="py-toggle" href="#" id="SequenceAdapter._get-toggle" onclick="return toggle('SequenceAdapter._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter._get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter._get-expanded"><a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt class="py-name">_get</tt><tt class="py-op">(</tt><tt id="link-415" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-415', 'rank', 'link-136');">rank</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter._factory"></a><div id="SequenceAdapter._factory-def"><a name="L701"></a><tt class="py-lineno"> 701</tt> <a class="py-toggle" href="#" id="SequenceAdapter._factory-toggle" onclick="return toggle('SequenceAdapter._factory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#_factory">_factory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">a</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>         </tt>
</div><a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-416" class="py-name"><a title="csb.bio.io.clans.Color.a
csb.statistics.pdf.GeneralizedInverseGaussian.a" class="py-name" href="#" onclick="return doclink('link-416', 'a', 'link-218');">a</a></tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter.strip"></a><div id="SequenceAdapter.strip-def"><a name="L704"></a><tt class="py-lineno"> 704</tt> <a class="py-toggle" href="#" id="SequenceAdapter.strip-toggle" onclick="return toggle('SequenceAdapter.strip');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#strip">strip</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.strip-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.strip-expanded"><a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-417', 'strip', 'link-268');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">             </tt>
<a name="SequenceAdapter.subregion"></a><div id="SequenceAdapter.subregion-def"><a name="L707"></a><tt class="py-lineno"> 707</tt> <a class="py-toggle" href="#" id="SequenceAdapter.subregion-toggle" onclick="return toggle('SequenceAdapter.subregion');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#subregion">subregion</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.subregion-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.subregion-expanded"><a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.subregion
csb.bio.hmm.ProfileHMM.subregion
csb.bio.sequence.AbstractAlignment.subregion
csb.bio.sequence.AbstractSequence.subregion
csb.bio.sequence.Sequence.subregion
csb.bio.sequence.SequenceAdapter.subregion
csb.bio.structure.Chain.subregion
csb.bio.structure.SecondaryStructure.subregion" class="py-name" href="#" onclick="return doclink('link-418', 'subregion', 'link-213');">subregion</a></tt><tt class="py-op">(</tt><tt id="link-419" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-419', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-420" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-420', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">     </tt>
<a name="SequenceAdapter.extract"></a><div id="SequenceAdapter.extract-def"><a name="L710"></a><tt class="py-lineno"> 710</tt> <a class="py-toggle" href="#" id="SequenceAdapter.extract-toggle" onclick="return toggle('SequenceAdapter.extract');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#extract">extract</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">positions</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.extract-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.extract-expanded"><a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="csb.bio.sequence.AbstractSequence.extract
csb.bio.sequence.Sequence.extract
csb.bio.sequence.SequenceAdapter.extract" class="py-name" href="#" onclick="return doclink('link-421', 'extract', 'link-234');">extract</a></tt><tt class="py-op">(</tt><tt class="py-name">positions</tt><tt class="py-op">)</tt>     </tt>
</div><a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line"> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.id"></a><div id="SequenceAdapter.id-def"><a name="L714"></a><tt class="py-lineno"> 714</tt> <a class="py-toggle" href="#" id="SequenceAdapter.id-toggle" onclick="return toggle('SequenceAdapter.id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#id">id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.id-expanded"><a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-422" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-422', 'id', 'link-148');">id</a></tt> </tt>
</div><a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line"> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.length"></a><div id="SequenceAdapter.length-def"><a name="L718"></a><tt class="py-lineno"> 718</tt> <a class="py-toggle" href="#" id="SequenceAdapter.length-toggle" onclick="return toggle('SequenceAdapter.length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#length">length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.length-expanded"><a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-423', 'length', 'link-156');">length</a></tt> </tt>
</div><a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line"> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.type"></a><div id="SequenceAdapter.type-def"><a name="L722"></a><tt class="py-lineno"> 722</tt> <a class="py-toggle" href="#" id="SequenceAdapter.type-toggle" onclick="return toggle('SequenceAdapter.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.type-expanded"><a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-424', 'type', 'link-121');">type</a></tt> </tt>
</div><a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line"> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.header"></a><div id="SequenceAdapter.header-def"><a name="L726"></a><tt class="py-lineno"> 726</tt> <a class="py-toggle" href="#" id="SequenceAdapter.header-toggle" onclick="return toggle('SequenceAdapter.header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#header">header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.header-expanded"><a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-425" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-425', 'header', 'link-204');">header</a></tt> </tt>
</div><a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">     </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.sequence"></a><div id="SequenceAdapter.sequence-def"><a name="L730"></a><tt class="py-lineno"> 730</tt> <a class="py-toggle" href="#" id="SequenceAdapter.sequence-toggle" onclick="return toggle('SequenceAdapter.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.sequence-expanded"><a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-426', 'sequence', 'link-387');">sequence</a></tt> </tt>
</div><a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">                     </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="SequenceAdapter.alphabet"></a><div id="SequenceAdapter.alphabet-def"><a name="L734"></a><tt class="py-lineno"> 734</tt> <a class="py-toggle" href="#" id="SequenceAdapter.alphabet-toggle" onclick="return toggle('SequenceAdapter.alphabet');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAdapter-class.html#alphabet">alphabet</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAdapter.alphabet-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAdapter.alphabet-expanded"><a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-427" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-427', 'alphabet', 'link-122');">alphabet</a></tt> </tt>
</div></div><a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"> </tt>
<a name="AlignedSequenceAdapter"></a><div id="AlignedSequenceAdapter-def"><a name="L737"></a><tt class="py-lineno"> 737</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter-toggle" onclick="return toggle('AlignedSequenceAdapter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html">AlignedSequenceAdapter</a><tt class="py-op">(</tt><tt class="py-base-class">SequenceAdapter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AlignedSequenceAdapter-expanded"><a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring">    Adapter, which wraps a gapped L{AbstractSequence} object and makes it</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring">    compatible with the MSA row/entry interface, expected by L{AbstractAlignment}.</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring">    The C{adapter.residues} property operates with an L{UngappedSequenceIndexer},</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">    which provides a gap-free view of the underlying sequence.</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring">    The C{adapter.columns} property operates with a standard L{ColumnIndexer},</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring">    the same indexer which is used to provide the column view in multiple </tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"><tt class="py-docstring">    alignments. Adapted sequences therefore act as alignment rows and allow for</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line"><tt class="py-docstring">    MSA-column-oriented indexing.</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"><tt class="py-docstring">    @param sequence: adaptee</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line"><tt class="py-docstring">    @type sequence: L{AbstractSequence}    </tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"> </tt>
<a name="AlignedSequenceAdapter.__init__"></a><div id="AlignedSequenceAdapter.__init__-def"><a name="L754"></a><tt class="py-lineno"> 754</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.__init__-toggle" onclick="return toggle('AlignedSequenceAdapter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.__init__-expanded"><a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-428" class="py-name" targets="Class csb.bio.sequence.AlignedSequenceAdapter=csb.bio.sequence.AlignedSequenceAdapter-class.html"><a title="csb.bio.sequence.AlignedSequenceAdapter" class="py-name" href="#" onclick="return doclink('link-428', 'AlignedSequenceAdapter', 'link-428');">AlignedSequenceAdapter</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-429" class="py-name"><a title="csb.Version.__init__
csb.apps.AppExit.__init__
csb.apps.AppRunner.__init__
csb.apps.Application.__init__
csb.apps.ArgHandler.__init__
csb.apps.buildhmm.NoOutputError.__init__
csb.apps.buildhmm.ProfileBuilder.__init__
csb.apps.csfrag.CSfrag.__init__
csb.apps.csfrag.MatchInfo.__init__
csb.apps.csfrag.MatchTable.__init__
csb.apps.csfrag.ScoringHelper.__init__
csb.apps.embd.Deconvolution.__init__
csb.apps.embd.OutputPathBuilder.__init__
csb.apps.hhfrag.HHfrag.__init__
csb.apps.hhfrag.PredictionBuilder.__init__
csb.apps.hhfrag.SliceContext.__init__
csb.apps.hhsearch.Context.__init__
csb.apps.hhsearch.HHTask.__init__
csb.apps.hhsearch.HHsearch.__init__
csb.apps.precision.GlobalInfo.__init__
csb.apps.precision.LibrarySuperimposer.__init__
csb.bio.fragments.Assignment.__init__
csb.bio.fragments.BenchmarkAdapter.Connection.__init__
csb.bio.fragments.BenchmarkAdapter.__init__
csb.bio.fragments.ChemShiftAssignment.__init__
csb.bio.fragments.ChemShiftTarget.__init__
csb.bio.fragments.ClusterNode.__init__
csb.bio.fragments.ClusterRep.__init__
csb.bio.fragments.FragmentCluster.__init__
csb.bio.fragments.FragmentMatch.__init__
csb.bio.fragments.ResidueAssignmentInfo.__init__
csb.bio.fragments.ResidueEventInfo.__init__
csb.bio.fragments.RosettaFragsetFactory.__init__
csb.bio.fragments.Target.__init__
csb.bio.fragments.TargetResidue.__init__
csb.bio.fragments.TorsionAnglePredictor.__init__
csb.bio.fragments.TorsionPredictionInfo.__init__
csb.bio.fragments.rosetta.OutputBuilder.__init__
csb.bio.fragments.rosetta.ResidueInfo.__init__
csb.bio.fragments.rosetta.RosettaFragment.__init__
csb.bio.fragments.rosetta.RosettaFragmentMap.__init__
csb.bio.hmm.EVDParameters.__init__
csb.bio.hmm.EmissionTable.__init__
csb.bio.hmm.HHpredHit.__init__
csb.bio.hmm.HHpredHitAlignment.__init__
csb.bio.hmm.HHpredHitList.__init__
csb.bio.hmm.HMMLayer.__init__
csb.bio.hmm.HMMLayersCollection.__init__
csb.bio.hmm.ProfileHMM.__init__
csb.bio.hmm.ProfileHMMRegion.__init__
csb.bio.hmm.ProfileHMMSegment.__init__
csb.bio.hmm.ProfileLength.__init__
csb.bio.hmm.State.__init__
csb.bio.hmm.StateFactory.__init__
csb.bio.hmm.Transition.__init__
csb.bio.hmm.TransitionTable.__init__
csb.bio.hmm.TransitionType.__init__
csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__
csb.bio.io.clans.Clans.__init__
csb.bio.io.clans.ClansEntry.__init__
csb.bio.io.clans.ClansEntryCollection.__init__
csb.bio.io.clans.ClansFileBuilder.__init__
csb.bio.io.clans.ClansFileWriter.__init__
csb.bio.io.clans.ClansParams.__init__
csb.bio.io.clans.ClansParser.__init__
csb.bio.io.clans.ClansSeqgroup.__init__
csb.bio.io.clans.ClansSeqgroupCollection.__init__
csb.bio.io.clans.Color.__init__
csb.bio.io.cs.ChemShiftReader.__init__
csb.bio.io.dssp.ResidueAssignmentInfo.__init__
csb.bio.io.fasta.A3MSequenceIterator.__init__
csb.bio.io.fasta.BaseSequenceParser.__init__
csb.bio.io.fasta.OutputBuilder.__init__
csb.bio.io.fasta.SequenceAlignmentReader.__init__
csb.bio.io.fasta.StructureAlignmentFactory.__init__
csb.bio.io.hhpred.HHMFileBuilder.__init__
csb.bio.io.hhpred.HHOutputParser.__init__
csb.bio.io.hhpred.HHProfileParser.__init__
csb.bio.io.mrc.DensityInfo.__init__
csb.bio.io.mrc.DensityMapReader.__init__
csb.bio.io.mrc.HeaderInfo.__init__
csb.bio.io.noe.SparkyPeakListReader.__init__
csb.bio.io.noe.XeasyFileBuilder.__init__
csb.bio.io.noe.XeasyPeakListReader.__init__
csb.bio.io.procheck.ProcheckParser.__init__
csb.bio.io.psipred.PSIPredResidueInfo.__init__
csb.bio.io.svg.SSCartoonBuilder.__init__
csb.bio.io.vasco.ChemicalShiftContainer.__init__
csb.bio.io.vasco.ShiftInfo.__init__
csb.bio.io.vasco.VascoStructureParser.__init__
csb.bio.io.whatif.WhatCheckParser.__init__
csb.bio.io.wwpdb.AbstractStructureParser.__init__
csb.bio.io.wwpdb.AsyncParseResult.__init__
csb.bio.io.wwpdb.AsyncStructureParser.__init__
csb.bio.io.wwpdb.CustomStructureProvider.__init__
csb.bio.io.wwpdb.EntryID.__init__
csb.bio.io.wwpdb.FastResidueMapper.__init__
csb.bio.io.wwpdb.FileBuilder.__init__
csb.bio.io.wwpdb.FileSystemStructureProvider.__init__
csb.bio.io.wwpdb.RemoteStructureProvider.__init__
csb.bio.io.wwpdb.ResidueInfo.__init__
csb.bio.io.wwpdb.RobustResidueMapper.__init__
csb.bio.nmr.AtomConnectivity.__init__
csb.bio.nmr.ChemShiftInfo.__init__
csb.bio.nmr.ChemShiftScoringModel.__init__
csb.bio.nmr.ChemicalShiftNetwork.__init__
csb.bio.nmr.ContactMap.__init__
csb.bio.nmr.ContactMapComparisonInfo.__init__
csb.bio.nmr.Label.__init__
csb.bio.nmr.NOEPeak.__init__
csb.bio.nmr.NOESpectrum.__init__
csb.bio.nmr.RandomCoil.__init__
csb.bio.sequence.A3MAlignment.__init__
csb.bio.sequence.AbstractAlignment.__init__
csb.bio.sequence.AbstractSequence.__init__
csb.bio.sequence.AlignedSequenceAdapter.__init__
csb.bio.sequence.AlignmentRowsTable.__init__
csb.bio.sequence.ColumnInfo.__init__
csb.bio.sequence.PositionError.__init__
csb.bio.sequence.ResidueInfo.__init__
csb.bio.sequence.Sequence.__init__
csb.bio.sequence.SequenceAdapter.__init__
csb.bio.sequence.SequenceCollection.__init__
csb.bio.sequence.SequenceIndexer.__init__
csb.bio.sequence.SliceHelper.__init__
csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.AlignmentResult.__init__
csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.IdentityMatrix.__init__
csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.SimilarityMatrix.__init__
csb.bio.structure.Atom.__init__
csb.bio.structure.Chain.__init__
csb.bio.structure.ChainResiduesCollection.__init__
csb.bio.structure.CompositeEntityIterator.__init__
csb.bio.structure.ConfinedEntityIterator.__init__
csb.bio.structure.DisorderedAtom.__init__
csb.bio.structure.Ensemble.__init__
csb.bio.structure.EnsembleModelsCollection.__init__
csb.bio.structure.NucleicResidue.__init__
csb.bio.structure.ProteinResidue.__init__
csb.bio.structure.Residue.__init__
csb.bio.structure.ResidueAtomsTable.__init__
csb.bio.structure.SecondaryStructure.__init__
csb.bio.structure.SecondaryStructureElement.__init__
csb.bio.structure.Structure.__init__
csb.bio.structure.StructureChainsTable.__init__
csb.bio.structure.SuperimposeInfo.__init__
csb.bio.structure.TorsionAngles.__init__
csb.bio.structure.TorsionAnglesCollection.__init__
csb.bio.structure.UnknownResidue.__init__
csb.build.Console.__init__
csb.build.GitHandler.__init__
csb.build.MercurialHandler.__init__
csb.build.RevisionError.__init__
csb.build.RevisionHandler.__init__
csb.build.RevisionInfo.__init__
csb.build.SubversionHandler.__init__
csb.core.BaseCollectionContainer.__init__
csb.core.BaseDictionaryContainer.__init__
csb.core.CollectionContainer.__init__
csb.core.Container.__init__
csb.core.DictionaryContainer.__init__
csb.core.EnumItem.__init__
csb.core.InterruptibleThread.__init__
csb.core.Proxy.__init__
csb.core.REMatchProxy.__init__
csb.core.enum.__init__
csb.core.typedproperty.__init__
csb.core.validatedproperty.__init__
csb.io.AutoFlushStream.__init__
csb.io.EntryReader.__init__
csb.io.EntryWriter.__init__
csb.io.InvalidCommandError.__init__
csb.io.ProcessError.__init__
csb.io.ShellInfo.__init__
csb.io.TempFile.__init__
csb.io.TempFolder.__init__
csb.io.TimeoutError.__init__
csb.io.plots.Backend.__init__
csb.io.plots.Chart.__init__
csb.io.plots.PlotsCollection.__init__
csb.io.plots.WxBackendImpl.__init__
csb.io.tsv.Between.__init__
csb.io.tsv.ColumnInfo.__init__
csb.io.tsv.DataRow.__init__
csb.io.tsv.Expression.__init__
csb.io.tsv.Operator.__init__
csb.io.tsv.Predicate.__init__
csb.io.tsv.RepositoryImp.__init__
csb.io.tsv.SQLiteRepository.ChunkedReader.__init__
csb.io.tsv.SQLiteRepository.__init__
csb.io.tsv.Step.__init__
csb.io.tsv.Table.__init__
csb.io.tsv.Where.__init__
csb.numeric.InvertibleMatrix.__init__
csb.numeric.integrators.AbstractIntegrator.__init__
csb.statistics.Cumulative.__init__
csb.statistics.ars.ARS.__init__
csb.statistics.ars.Envelope.__init__
csb.statistics.ars.Gauss.__init__
csb.statistics.maxent.MaxentModel.__init__
csb.statistics.maxent.MaxentPosterior.__init__
csb.statistics.mixtures.AbstractStructureMixture.__init__
csb.statistics.mixtures.GaussianMixture.__init__
csb.statistics.pdf.AbstractDensity.__init__
csb.statistics.pdf.Dirichlet.__init__
csb.statistics.pdf.DirichletEstimator.__init__
csb.statistics.pdf.Gamma.__init__
csb.statistics.pdf.GammaMLEstimator.__init__
csb.statistics.pdf.GenNormalBruteForceEstimator.__init__
csb.statistics.pdf.GeneralizedInverseGaussian.__init__
csb.statistics.pdf.GeneralizedNormal.__init__
csb.statistics.pdf.GumbelMaximum.__init__
csb.statistics.pdf.GumbelMinimum.__init__
csb.statistics.pdf.InverseGamma.__init__
csb.statistics.pdf.InverseGaussian.__init__
csb.statistics.pdf.Laplace.__init__
csb.statistics.pdf.MultivariateGaussian.__init__
csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__
csb.statistics.pdf.Normal.__init__
csb.statistics.pdf.ParameterValueError.__init__
csb.statistics.pdf.parameterized.AbstractParameter.__init__
csb.statistics.pdf.parameterized.Parameter.__init__
csb.statistics.samplers.EnsembleState.__init__
csb.statistics.samplers.State.__init__
csb.statistics.samplers.mc.AbstractMC.__init__
csb.statistics.samplers.mc.MCCollection.__init__
csb.statistics.samplers.mc.PropagationResult.__init__
csb.statistics.samplers.mc.Trajectory.__init__
csb.statistics.samplers.mc.TrajectoryBuilder.__init__
csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__
csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__
csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__
csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__
csb.statistics.samplers.mc.multichain.Gradient.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__
csb.statistics.samplers.mc.multichain.MDRENS.__init__
csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__
csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__
csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__
csb.statistics.samplers.mc.multichain.SwapStatistics.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__
csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.Step.__init__
csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__
csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__
csb.statistics.samplers.mc.propagators.HMCPropagator.__init__
csb.statistics.samplers.mc.propagators.Looper.__init__
csb.statistics.samplers.mc.propagators.MDPropagator.__init__
csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__
csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__
csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__
csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__
csb.statistics.samplers.mc.singlechain.HMCSampler.__init__
csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__
csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__
csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__
csb.statistics.scalemixture.ARSPosteriorAlpha.__init__
csb.statistics.scalemixture.GammaPosteriorMAP.__init__
csb.statistics.scalemixture.GammaPosteriorSampler.__init__
csb.statistics.scalemixture.GammaPrior.__init__
csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__
csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__
csb.statistics.scalemixture.InvGammaPrior.__init__
csb.statistics.scalemixture.ScaleMixture.__init__
csb.statistics.scalemixture.ScaleMixturePrior.__init__
csb.statmech.ensembles.BoltzmannEnsemble.__init__
csb.statmech.ensembles.CompositeEnsemble.__init__
csb.statmech.ensembles.FermiEnsemble.__init__
csb.statmech.ensembles.TsallisEnsemble.__init__
csb.statmech.wham.AbstractWHAM.__init__
csb.statmech.wham.WHAM.__init__
csb.test.Console.__init__" class="py-name" href="#" onclick="return doclink('link-429', '__init__', 'link-132');">__init__</a></tt><tt class="py-op">(</tt><tt id="link-430" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-430', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">         </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rmap</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt id="link-431" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-431', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">         </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-432" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-432', 'column', 'link-146');">column</a></tt><tt class="py-op">,</tt> <tt id="link-433" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-433', 'residue', 'link-152');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-434" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-434', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-435" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-435', 'start', 'link-126');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">             </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-436" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-436', 'residue', 'link-152');">residue</a></tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-437', 'type', 'link-121');">type</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-438', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-439', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-440', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-441', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">                <tt id="link-442" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-442', 'rank', 'link-136');">rank</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt><tt class="py-op">[</tt><tt id="link-443" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-443', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-444" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-444', 'rank', 'link-136');">rank</a></tt>                 </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rmap</tt><tt class="py-op">[</tt><tt id="link-445" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-445', 'rank', 'link-136');">rank</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-446" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-446', 'column', 'link-146');">column</a></tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt><tt class="py-op">[</tt><tt id="link-447" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-447', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"> </tt>
<a name="AlignedSequenceAdapter.__getitem__"></a><div id="AlignedSequenceAdapter.__getitem__-def"><a name="L771"></a><tt class="py-lineno"> 771</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.__getitem__-toggle" onclick="return toggle('AlignedSequenceAdapter.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.__getitem__-expanded"><a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">0</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">index</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-448', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-name">index</tt><tt class="py-op">)</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get_column</tt><tt class="py-op">(</tt><tt class="py-name">index</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">     </tt>
<a name="AlignedSequenceAdapter.__iter__"></a><div id="AlignedSequenceAdapter.__iter__-def"><a name="L776"></a><tt class="py-lineno"> 776</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.__iter__-toggle" onclick="return toggle('AlignedSequenceAdapter.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.__iter__-expanded"><a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get_column</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> </tt>
</div><a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                     </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AlignedSequenceAdapter.columns"></a><div id="AlignedSequenceAdapter.columns-def"><a name="L781"></a><tt class="py-lineno"> 781</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.columns-toggle" onclick="return toggle('AlignedSequenceAdapter.columns');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#columns">columns</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.columns-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.columns-expanded"><a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line"><tt class="py-docstring">        Provides 1-based access to the respective columns in the MSA.</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ColumnIndexer}</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-449" class="py-name" targets="Class csb.bio.sequence.ColumnIndexer=csb.bio.sequence.ColumnIndexer-class.html"><a title="csb.bio.sequence.ColumnIndexer" class="py-name" href="#" onclick="return doclink('link-449', 'ColumnIndexer', 'link-449');">ColumnIndexer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AlignedSequenceAdapter.residues"></a><div id="AlignedSequenceAdapter.residues-def"><a name="L789"></a><tt class="py-lineno"> 789</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.residues-toggle" onclick="return toggle('AlignedSequenceAdapter.residues');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#residues">residues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.residues-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.residues-expanded"><a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring">        Provides 1-based access to the residues of the unaligned (ungapped)</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">        sequence.</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{UngappedSequenceIndexer} </tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-450" class="py-name" targets="Class csb.bio.sequence.UngappedSequenceIndexer=csb.bio.sequence.UngappedSequenceIndexer-class.html"><a title="csb.bio.sequence.UngappedSequenceIndexer" class="py-name" href="#" onclick="return doclink('link-450', 'UngappedSequenceIndexer', 'link-450');">UngappedSequenceIndexer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"> </tt>
<a name="AlignedSequenceAdapter._get_column"></a><div id="AlignedSequenceAdapter._get_column-def"><a name="L797"></a><tt class="py-lineno"> 797</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter._get_column-toggle" onclick="return toggle('AlignedSequenceAdapter._get_column');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#_get_column">_get_column</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter._get_column-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter._get_column-expanded"><a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-451" class="py-name" targets="Class csb.bio.sequence.ColumnInfo=csb.bio.sequence.ColumnInfo-class.html,Class csb.io.tsv.ColumnInfo=csb.io.tsv.ColumnInfo-class.html"><a title="csb.bio.sequence.ColumnInfo
csb.io.tsv.ColumnInfo" class="py-name" href="#" onclick="return doclink('link-451', 'ColumnInfo', 'link-451');">ColumnInfo</a></tt><tt class="py-op">(</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">                <tt id="link-452" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-452', 'column', 'link-146');">column</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-453', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt><tt class="py-op">[</tt><tt id="link-454" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-454', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-455', 'residues', 'link-209');">residues</a></tt><tt class="py-op">[</tt><tt id="link-456" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-456', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">             </tt>
<a name="AlignedSequenceAdapter._get_ungapped"></a><div id="AlignedSequenceAdapter._get_ungapped-def"><a name="L801"></a><tt class="py-lineno"> 801</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter._get_ungapped-toggle" onclick="return toggle('AlignedSequenceAdapter._get_ungapped');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#_get_ungapped">_get_ungapped</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter._get_ungapped-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter._get_ungapped-expanded"><a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_subject</tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-457', 'residues', 'link-209');">residues</a></tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rmap</tt><tt class="py-op">[</tt><tt id="link-458" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-458', 'rank', 'link-136');">rank</a></tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
</div><a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">     </tt>
<a name="AlignedSequenceAdapter.map_residue"></a><div id="AlignedSequenceAdapter.map_residue-def"><a name="L804"></a><tt class="py-lineno"> 804</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.map_residue-toggle" onclick="return toggle('AlignedSequenceAdapter.map_residue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#map_residue">map_residue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.map_residue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.map_residue-expanded"><a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring">        Return the MSA column number corresponding to the specified ungapped</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">        sequence C{rank}.</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"><tt class="py-docstring">        @param rank: 1-based residue rank</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"><tt class="py-docstring">        @type rank: int</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_rmap</tt><tt class="py-op">[</tt><tt id="link-459" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-459', 'rank', 'link-136');">rank</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">     </tt>
<a name="AlignedSequenceAdapter.map_column"></a><div id="AlignedSequenceAdapter.map_column-def"><a name="L815"></a><tt class="py-lineno"> 815</tt> <a class="py-toggle" href="#" id="AlignedSequenceAdapter.map_column-toggle" onclick="return toggle('AlignedSequenceAdapter.map_column');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignedSequenceAdapter-class.html#map_column">map_column</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignedSequenceAdapter.map_column-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignedSequenceAdapter.map_column-expanded"><a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"><tt class="py-docstring">        Return the ungapped sequence rank corresponding to the specified MSA</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring">        C{column} number.</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line"><tt class="py-docstring">        @param column: 1-based alignment column number</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"><tt class="py-docstring">        @type column: int</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fmap</tt><tt class="py-op">[</tt><tt id="link-460" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-460', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt>     </tt>
</div></div><a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">     </tt>
<a name="SliceHelper"></a><div id="SliceHelper-def"><a name="L826"></a><tt class="py-lineno"> 826</tt> <a class="py-toggle" href="#" id="SliceHelper-toggle" onclick="return toggle('SliceHelper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SliceHelper-class.html">SliceHelper</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SliceHelper-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SliceHelper-expanded"><a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">     </tt>
<a name="SliceHelper.__init__"></a><div id="SliceHelper.__init__-def"><a name="L828"></a><tt class="py-lineno"> 828</tt> <a class="py-toggle" href="#" id="SliceHelper.__init__-toggle" onclick="return toggle('SliceHelper.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SliceHelper-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">slice</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">stop</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SliceHelper.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SliceHelper.__init__-expanded"><a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">         </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt><tt class="py-op">,</tt> <tt id="link-461" class="py-name" targets="Variable csb.statistics.mixtures.AbstractStructureMixture.t=csb.statistics.mixtures.AbstractStructureMixture-class.html#t"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-461', 't', 'link-461');">t</a></tt> <tt class="py-op">=</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-462', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt id="link-463" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-463', 'stop', 'link-215');">stop</a></tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt class="py-name">step</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">         </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">s</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-464" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-464', 'start', 'link-126');">start</a></tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">e</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">            <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt id="link-465" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-465', 'stop', 'link-215');">stop</a></tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-466" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-466', 't', 'link-461');">t</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">            <tt id="link-467" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-467', 't', 'link-461');">t</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">             </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-468" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-468', 'value', 'link-265');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt><tt class="py-op">,</tt> <tt id="link-469" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-469', 't', 'link-461');">t</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-470" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-470', 'value', 'link-265');">value</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt id="link-471" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-471', 'value', 'link-265');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">             </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-472', 'start', 'link-126');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-473" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-473', 'stop', 'link-215');">stop</a></tt> <tt class="py-op">=</tt> <tt class="py-name">e</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">step</tt> <tt class="py-op">=</tt> <tt id="link-474" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-474', 't', 'link-461');">t</a></tt>             </tt>
</div></div><a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"> </tt>
<a name="AlignmentRowsTable"></a><div id="AlignmentRowsTable-def"><a name="L847"></a><tt class="py-lineno"> 847</tt> <a class="py-toggle" href="#" id="AlignmentRowsTable-toggle" onclick="return toggle('AlignmentRowsTable');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentRowsTable-class.html">AlignmentRowsTable</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">core</tt><tt class="py-op">.</tt><tt class="py-base-class">BaseDictionaryContainer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentRowsTable-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AlignmentRowsTable-expanded"><a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">     </tt>
<a name="AlignmentRowsTable.__init__"></a><div id="AlignmentRowsTable.__init__-def"><a name="L849"></a><tt class="py-lineno"> 849</tt> <a class="py-toggle" href="#" id="AlignmentRowsTable.__init__-toggle" onclick="return toggle('AlignmentRowsTable.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentRowsTable-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">container</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentRowsTable.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignmentRowsTable.__init__-expanded"><a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">         </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-475" class="py-name" targets="Class csb.bio.sequence.AlignmentRowsTable=csb.bio.sequence.AlignmentRowsTable-class.html"><a title="csb.bio.sequence.AlignmentRowsTable" class="py-name" href="#" onclick="return doclink('link-475', 'AlignmentRowsTable', 'link-475');">AlignmentRowsTable</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="csb.Version.__init__
csb.apps.AppExit.__init__
csb.apps.AppRunner.__init__
csb.apps.Application.__init__
csb.apps.ArgHandler.__init__
csb.apps.buildhmm.NoOutputError.__init__
csb.apps.buildhmm.ProfileBuilder.__init__
csb.apps.csfrag.CSfrag.__init__
csb.apps.csfrag.MatchInfo.__init__
csb.apps.csfrag.MatchTable.__init__
csb.apps.csfrag.ScoringHelper.__init__
csb.apps.embd.Deconvolution.__init__
csb.apps.embd.OutputPathBuilder.__init__
csb.apps.hhfrag.HHfrag.__init__
csb.apps.hhfrag.PredictionBuilder.__init__
csb.apps.hhfrag.SliceContext.__init__
csb.apps.hhsearch.Context.__init__
csb.apps.hhsearch.HHTask.__init__
csb.apps.hhsearch.HHsearch.__init__
csb.apps.precision.GlobalInfo.__init__
csb.apps.precision.LibrarySuperimposer.__init__
csb.bio.fragments.Assignment.__init__
csb.bio.fragments.BenchmarkAdapter.Connection.__init__
csb.bio.fragments.BenchmarkAdapter.__init__
csb.bio.fragments.ChemShiftAssignment.__init__
csb.bio.fragments.ChemShiftTarget.__init__
csb.bio.fragments.ClusterNode.__init__
csb.bio.fragments.ClusterRep.__init__
csb.bio.fragments.FragmentCluster.__init__
csb.bio.fragments.FragmentMatch.__init__
csb.bio.fragments.ResidueAssignmentInfo.__init__
csb.bio.fragments.ResidueEventInfo.__init__
csb.bio.fragments.RosettaFragsetFactory.__init__
csb.bio.fragments.Target.__init__
csb.bio.fragments.TargetResidue.__init__
csb.bio.fragments.TorsionAnglePredictor.__init__
csb.bio.fragments.TorsionPredictionInfo.__init__
csb.bio.fragments.rosetta.OutputBuilder.__init__
csb.bio.fragments.rosetta.ResidueInfo.__init__
csb.bio.fragments.rosetta.RosettaFragment.__init__
csb.bio.fragments.rosetta.RosettaFragmentMap.__init__
csb.bio.hmm.EVDParameters.__init__
csb.bio.hmm.EmissionTable.__init__
csb.bio.hmm.HHpredHit.__init__
csb.bio.hmm.HHpredHitAlignment.__init__
csb.bio.hmm.HHpredHitList.__init__
csb.bio.hmm.HMMLayer.__init__
csb.bio.hmm.HMMLayersCollection.__init__
csb.bio.hmm.ProfileHMM.__init__
csb.bio.hmm.ProfileHMMRegion.__init__
csb.bio.hmm.ProfileHMMSegment.__init__
csb.bio.hmm.ProfileLength.__init__
csb.bio.hmm.State.__init__
csb.bio.hmm.StateFactory.__init__
csb.bio.hmm.Transition.__init__
csb.bio.hmm.TransitionTable.__init__
csb.bio.hmm.TransitionType.__init__
csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__
csb.bio.io.clans.Clans.__init__
csb.bio.io.clans.ClansEntry.__init__
csb.bio.io.clans.ClansEntryCollection.__init__
csb.bio.io.clans.ClansFileBuilder.__init__
csb.bio.io.clans.ClansFileWriter.__init__
csb.bio.io.clans.ClansParams.__init__
csb.bio.io.clans.ClansParser.__init__
csb.bio.io.clans.ClansSeqgroup.__init__
csb.bio.io.clans.ClansSeqgroupCollection.__init__
csb.bio.io.clans.Color.__init__
csb.bio.io.cs.ChemShiftReader.__init__
csb.bio.io.dssp.ResidueAssignmentInfo.__init__
csb.bio.io.fasta.A3MSequenceIterator.__init__
csb.bio.io.fasta.BaseSequenceParser.__init__
csb.bio.io.fasta.OutputBuilder.__init__
csb.bio.io.fasta.SequenceAlignmentReader.__init__
csb.bio.io.fasta.StructureAlignmentFactory.__init__
csb.bio.io.hhpred.HHMFileBuilder.__init__
csb.bio.io.hhpred.HHOutputParser.__init__
csb.bio.io.hhpred.HHProfileParser.__init__
csb.bio.io.mrc.DensityInfo.__init__
csb.bio.io.mrc.DensityMapReader.__init__
csb.bio.io.mrc.HeaderInfo.__init__
csb.bio.io.noe.SparkyPeakListReader.__init__
csb.bio.io.noe.XeasyFileBuilder.__init__
csb.bio.io.noe.XeasyPeakListReader.__init__
csb.bio.io.procheck.ProcheckParser.__init__
csb.bio.io.psipred.PSIPredResidueInfo.__init__
csb.bio.io.svg.SSCartoonBuilder.__init__
csb.bio.io.vasco.ChemicalShiftContainer.__init__
csb.bio.io.vasco.ShiftInfo.__init__
csb.bio.io.vasco.VascoStructureParser.__init__
csb.bio.io.whatif.WhatCheckParser.__init__
csb.bio.io.wwpdb.AbstractStructureParser.__init__
csb.bio.io.wwpdb.AsyncParseResult.__init__
csb.bio.io.wwpdb.AsyncStructureParser.__init__
csb.bio.io.wwpdb.CustomStructureProvider.__init__
csb.bio.io.wwpdb.EntryID.__init__
csb.bio.io.wwpdb.FastResidueMapper.__init__
csb.bio.io.wwpdb.FileBuilder.__init__
csb.bio.io.wwpdb.FileSystemStructureProvider.__init__
csb.bio.io.wwpdb.RemoteStructureProvider.__init__
csb.bio.io.wwpdb.ResidueInfo.__init__
csb.bio.io.wwpdb.RobustResidueMapper.__init__
csb.bio.nmr.AtomConnectivity.__init__
csb.bio.nmr.ChemShiftInfo.__init__
csb.bio.nmr.ChemShiftScoringModel.__init__
csb.bio.nmr.ChemicalShiftNetwork.__init__
csb.bio.nmr.ContactMap.__init__
csb.bio.nmr.ContactMapComparisonInfo.__init__
csb.bio.nmr.Label.__init__
csb.bio.nmr.NOEPeak.__init__
csb.bio.nmr.NOESpectrum.__init__
csb.bio.nmr.RandomCoil.__init__
csb.bio.sequence.A3MAlignment.__init__
csb.bio.sequence.AbstractAlignment.__init__
csb.bio.sequence.AbstractSequence.__init__
csb.bio.sequence.AlignedSequenceAdapter.__init__
csb.bio.sequence.AlignmentRowsTable.__init__
csb.bio.sequence.ColumnInfo.__init__
csb.bio.sequence.PositionError.__init__
csb.bio.sequence.ResidueInfo.__init__
csb.bio.sequence.Sequence.__init__
csb.bio.sequence.SequenceAdapter.__init__
csb.bio.sequence.SequenceCollection.__init__
csb.bio.sequence.SequenceIndexer.__init__
csb.bio.sequence.SliceHelper.__init__
csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.AlignmentResult.__init__
csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.IdentityMatrix.__init__
csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.SimilarityMatrix.__init__
csb.bio.structure.Atom.__init__
csb.bio.structure.Chain.__init__
csb.bio.structure.ChainResiduesCollection.__init__
csb.bio.structure.CompositeEntityIterator.__init__
csb.bio.structure.ConfinedEntityIterator.__init__
csb.bio.structure.DisorderedAtom.__init__
csb.bio.structure.Ensemble.__init__
csb.bio.structure.EnsembleModelsCollection.__init__
csb.bio.structure.NucleicResidue.__init__
csb.bio.structure.ProteinResidue.__init__
csb.bio.structure.Residue.__init__
csb.bio.structure.ResidueAtomsTable.__init__
csb.bio.structure.SecondaryStructure.__init__
csb.bio.structure.SecondaryStructureElement.__init__
csb.bio.structure.Structure.__init__
csb.bio.structure.StructureChainsTable.__init__
csb.bio.structure.SuperimposeInfo.__init__
csb.bio.structure.TorsionAngles.__init__
csb.bio.structure.TorsionAnglesCollection.__init__
csb.bio.structure.UnknownResidue.__init__
csb.build.Console.__init__
csb.build.GitHandler.__init__
csb.build.MercurialHandler.__init__
csb.build.RevisionError.__init__
csb.build.RevisionHandler.__init__
csb.build.RevisionInfo.__init__
csb.build.SubversionHandler.__init__
csb.core.BaseCollectionContainer.__init__
csb.core.BaseDictionaryContainer.__init__
csb.core.CollectionContainer.__init__
csb.core.Container.__init__
csb.core.DictionaryContainer.__init__
csb.core.EnumItem.__init__
csb.core.InterruptibleThread.__init__
csb.core.Proxy.__init__
csb.core.REMatchProxy.__init__
csb.core.enum.__init__
csb.core.typedproperty.__init__
csb.core.validatedproperty.__init__
csb.io.AutoFlushStream.__init__
csb.io.EntryReader.__init__
csb.io.EntryWriter.__init__
csb.io.InvalidCommandError.__init__
csb.io.ProcessError.__init__
csb.io.ShellInfo.__init__
csb.io.TempFile.__init__
csb.io.TempFolder.__init__
csb.io.TimeoutError.__init__
csb.io.plots.Backend.__init__
csb.io.plots.Chart.__init__
csb.io.plots.PlotsCollection.__init__
csb.io.plots.WxBackendImpl.__init__
csb.io.tsv.Between.__init__
csb.io.tsv.ColumnInfo.__init__
csb.io.tsv.DataRow.__init__
csb.io.tsv.Expression.__init__
csb.io.tsv.Operator.__init__
csb.io.tsv.Predicate.__init__
csb.io.tsv.RepositoryImp.__init__
csb.io.tsv.SQLiteRepository.ChunkedReader.__init__
csb.io.tsv.SQLiteRepository.__init__
csb.io.tsv.Step.__init__
csb.io.tsv.Table.__init__
csb.io.tsv.Where.__init__
csb.numeric.InvertibleMatrix.__init__
csb.numeric.integrators.AbstractIntegrator.__init__
csb.statistics.Cumulative.__init__
csb.statistics.ars.ARS.__init__
csb.statistics.ars.Envelope.__init__
csb.statistics.ars.Gauss.__init__
csb.statistics.maxent.MaxentModel.__init__
csb.statistics.maxent.MaxentPosterior.__init__
csb.statistics.mixtures.AbstractStructureMixture.__init__
csb.statistics.mixtures.GaussianMixture.__init__
csb.statistics.pdf.AbstractDensity.__init__
csb.statistics.pdf.Dirichlet.__init__
csb.statistics.pdf.DirichletEstimator.__init__
csb.statistics.pdf.Gamma.__init__
csb.statistics.pdf.GammaMLEstimator.__init__
csb.statistics.pdf.GenNormalBruteForceEstimator.__init__
csb.statistics.pdf.GeneralizedInverseGaussian.__init__
csb.statistics.pdf.GeneralizedNormal.__init__
csb.statistics.pdf.GumbelMaximum.__init__
csb.statistics.pdf.GumbelMinimum.__init__
csb.statistics.pdf.InverseGamma.__init__
csb.statistics.pdf.InverseGaussian.__init__
csb.statistics.pdf.Laplace.__init__
csb.statistics.pdf.MultivariateGaussian.__init__
csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__
csb.statistics.pdf.Normal.__init__
csb.statistics.pdf.ParameterValueError.__init__
csb.statistics.pdf.parameterized.AbstractParameter.__init__
csb.statistics.pdf.parameterized.Parameter.__init__
csb.statistics.samplers.EnsembleState.__init__
csb.statistics.samplers.State.__init__
csb.statistics.samplers.mc.AbstractMC.__init__
csb.statistics.samplers.mc.MCCollection.__init__
csb.statistics.samplers.mc.PropagationResult.__init__
csb.statistics.samplers.mc.Trajectory.__init__
csb.statistics.samplers.mc.TrajectoryBuilder.__init__
csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__
csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__
csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__
csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__
csb.statistics.samplers.mc.multichain.Gradient.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__
csb.statistics.samplers.mc.multichain.MDRENS.__init__
csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__
csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__
csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__
csb.statistics.samplers.mc.multichain.SwapStatistics.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__
csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.Step.__init__
csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__
csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__
csb.statistics.samplers.mc.propagators.HMCPropagator.__init__
csb.statistics.samplers.mc.propagators.Looper.__init__
csb.statistics.samplers.mc.propagators.MDPropagator.__init__
csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__
csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__
csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__
csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__
csb.statistics.samplers.mc.singlechain.HMCSampler.__init__
csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__
csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__
csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__
csb.statistics.scalemixture.ARSPosteriorAlpha.__init__
csb.statistics.scalemixture.GammaPosteriorMAP.__init__
csb.statistics.scalemixture.GammaPosteriorSampler.__init__
csb.statistics.scalemixture.GammaPrior.__init__
csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__
csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__
csb.statistics.scalemixture.InvGammaPrior.__init__
csb.statistics.scalemixture.ScaleMixture.__init__
csb.statistics.scalemixture.ScaleMixturePrior.__init__
csb.statmech.ensembles.BoltzmannEnsemble.__init__
csb.statmech.ensembles.CompositeEnsemble.__init__
csb.statmech.ensembles.FermiEnsemble.__init__
csb.statmech.ensembles.TsallisEnsemble.__init__
csb.statmech.wham.AbstractWHAM.__init__
csb.statmech.wham.WHAM.__init__
csb.test.Console.__init__" class="py-name" href="#" onclick="return doclink('link-476', '__init__', 'link-132');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">         </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_container</tt> <tt class="py-op">=</tt> <tt class="py-name">container</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
</div><a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">         </tt>
<a name="AlignmentRowsTable.__getitem__"></a><div id="AlignmentRowsTable.__getitem__-def"><a name="L856"></a><tt class="py-lineno"> 856</tt> <a class="py-toggle" href="#" id="AlignmentRowsTable.__getitem__-toggle" onclick="return toggle('AlignmentRowsTable.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentRowsTable-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentRowsTable.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignmentRowsTable.__getitem__-expanded"><a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">         </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">item</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">                <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">                <tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">item</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">                 </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-477" class="py-name"><a title="csb.bio.sequence.AlignmentRowsTable" class="py-name" href="#" onclick="return doclink('link-477', 'AlignmentRowsTable', 'link-475');">AlignmentRowsTable</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-478" class="py-name" targets="Method csb.apps.csfrag.MatchTable.__getitem__()=csb.apps.csfrag.MatchTable-class.html#__getitem__,Method csb.bio.hmm.HHpredHitList.__getitem__()=csb.bio.hmm.HHpredHitList-class.html#__getitem__,Method csb.bio.io.clans.Clans.__getitem__()=csb.bio.io.clans.Clans-class.html#__getitem__,Method csb.bio.io.mrc.HeaderInfo.__getitem__()=csb.bio.io.mrc.HeaderInfo-class.html#__getitem__,Method csb.bio.io.wwpdb.StructureProvider.__getitem__()=csb.bio.io.wwpdb.StructureProvider-class.html#__getitem__,Method csb.bio.nmr.NOEPeak.__getitem__()=csb.bio.nmr.NOEPeak-class.html#__getitem__,Method csb.bio.nmr.NOESpectrum.__getitem__()=csb.bio.nmr.NOESpectrum-class.html#__getitem__,Method csb.bio.sequence.AbstractAlignment.__getitem__()=csb.bio.sequence.AbstractAlignment-class.html#__getitem__,Method csb.bio.sequence.AbstractSequence.__getitem__()=csb.bio.sequence.AbstractSequence-class.html#__getitem__,Method csb.bio.sequence.AlignedSequenceAdapter.__getitem__()=csb.bio.sequence.AlignedSequenceAdapter-class.html#__getitem__,Method csb.bio.sequence.AlignmentRowsTable.__getitem__()=csb.bio.sequence.AlignmentRowsTable-class.html#__getitem__,Method csb.bio.sequence.ColumnIndexer.__getitem__()=csb.bio.sequence.ColumnIndexer-class.html#__getitem__,Method csb.bio.sequence.SequenceAdapter.__getitem__()=csb.bio.sequence.SequenceAdapter-class.html#__getitem__,Method csb.bio.sequence.SequenceIndexer.__getitem__()=csb.bio.sequence.SequenceIndexer-class.html#__getitem__,Method csb.bio.sequence.UngappedSequenceIndexer.__getitem__()=csb.bio.sequence.UngappedSequenceIndexer-class.html#__getitem__,Method csb.core.AbstractContainer.__getitem__()=csb.core.AbstractContainer-class.html#__getitem__,Method csb.core.AbstractIndexer.__getitem__()=csb.core.AbstractIndexer-class.html#__getitem__,Method csb.core.AbstractNIContainer.__getitem__()=csb.core.AbstractNIContainer-class.html#__getitem__,Method csb.core.BaseCollectionContainer.__getitem__()=csb.core.BaseCollectionContainer-class.html#__getitem__,Method csb.core.BaseDictionaryContainer.__getitem__()=csb.core.BaseDictionaryContainer-class.html#__getitem__,Method csb.io.plots.Chart.__getitem__()=csb.io.plots.Chart-class.html#__getitem__,Method csb.io.plots.PlotsCollection.__getitem__()=csb.io.plots.PlotsCollection-class.html#__getitem__,Method csb.io.tsv.DataRow.__getitem__()=csb.io.tsv.DataRow-class.html#__getitem__,Method csb.io.tsv.Table.__getitem__()=csb.io.tsv.Table-class.html#__getitem__,Method csb.numeric.InvertibleMatrix.__getitem__()=csb.numeric.InvertibleMatrix-class.html#__getitem__,Method csb.statistics.pdf.AbstractDensity.__getitem__()=csb.statistics.pdf.AbstractDensity-class.html#__getitem__"><a title="csb.apps.csfrag.MatchTable.__getitem__
csb.bio.hmm.HHpredHitList.__getitem__
csb.bio.io.clans.Clans.__getitem__
csb.bio.io.mrc.HeaderInfo.__getitem__
csb.bio.io.wwpdb.StructureProvider.__getitem__
csb.bio.nmr.NOEPeak.__getitem__
csb.bio.nmr.NOESpectrum.__getitem__
csb.bio.sequence.AbstractAlignment.__getitem__
csb.bio.sequence.AbstractSequence.__getitem__
csb.bio.sequence.AlignedSequenceAdapter.__getitem__
csb.bio.sequence.AlignmentRowsTable.__getitem__
csb.bio.sequence.ColumnIndexer.__getitem__
csb.bio.sequence.SequenceAdapter.__getitem__
csb.bio.sequence.SequenceIndexer.__getitem__
csb.bio.sequence.UngappedSequenceIndexer.__getitem__
csb.core.AbstractContainer.__getitem__
csb.core.AbstractIndexer.__getitem__
csb.core.AbstractNIContainer.__getitem__
csb.core.BaseCollectionContainer.__getitem__
csb.core.BaseDictionaryContainer.__getitem__
csb.io.plots.Chart.__getitem__
csb.io.plots.PlotsCollection.__getitem__
csb.io.tsv.DataRow.__getitem__
csb.io.tsv.Table.__getitem__
csb.numeric.InvertibleMatrix.__getitem__
csb.statistics.pdf.AbstractDensity.__getitem__" class="py-name" href="#" onclick="return doclink('link-478', '__getitem__', 'link-478');">__getitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">         </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-479" class="py-name" targets="Class csb.bio.sequence.SequenceNotFoundError=csb.bio.sequence.SequenceNotFoundError-class.html"><a title="csb.bio.sequence.SequenceNotFoundError" class="py-name" href="#" onclick="return doclink('link-479', 'SequenceNotFoundError', 'link-479');">SequenceNotFoundError</a></tt><tt class="py-op">(</tt><tt class="py-name">item</tt><tt class="py-op">)</tt> </tt>
</div><a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"> </tt>
<a name="AlignmentRowsTable._append"></a><div id="AlignmentRowsTable._append-def"><a name="L869"></a><tt class="py-lineno"> 869</tt> <a class="py-toggle" href="#" id="AlignmentRowsTable._append-toggle" onclick="return toggle('AlignmentRowsTable._append');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentRowsTable-class.html#_append">_append</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentRowsTable._append-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignmentRowsTable._append-expanded"><a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">        <tt id="link-480" class="py-name" targets="Variable csb.statistics.maxent.MaxentModel.n=csb.statistics.maxent.MaxentModel-class.html#n"><a title="csb.statistics.maxent.MaxentModel.n" class="py-name" href="#" onclick="return doclink('link-480', 'n', 'link-480');">n</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">        <tt class="py-name">sequence_id</tt> <tt class="py-op">=</tt> <tt id="link-481" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-481', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-482', 'id', 'link-148');">id</a></tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">         </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">sequence_id</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">            <tt id="link-483" class="py-name"><a title="csb.statistics.maxent.MaxentModel.n" class="py-name" href="#" onclick="return doclink('link-483', 'n', 'link-480');">n</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">            <tt class="py-name">sequence_id</tt> <tt class="py-op">=</tt> <tt class="py-string">'{0}:A{1}'</tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-484', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt id="link-485" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-485', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-486" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-486', 'id', 'link-148');">id</a></tt><tt class="py-op">,</tt> <tt id="link-487" class="py-name"><a title="csb.statistics.maxent.MaxentModel.n" class="py-name" href="#" onclick="return doclink('link-487', 'n', 'link-480');">n</a></tt><tt class="py-op">)</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-488" class="py-name"><a title="csb.bio.sequence.AlignmentRowsTable" class="py-name" href="#" onclick="return doclink('link-488', 'AlignmentRowsTable', 'link-475');">AlignmentRowsTable</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">_append_item</tt><tt class="py-op">(</tt><tt class="py-name">sequence_id</tt><tt class="py-op">,</tt> <tt id="link-489" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-489', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-490', 'length', 'link-156');">length</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">sequence_id</tt> </tt>
</div><a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line">     </tt>
<a name="AlignmentRowsTable.__iter__"></a><div id="AlignmentRowsTable.__iter__-def"><a name="L881"></a><tt class="py-lineno"> 881</tt> <a class="py-toggle" href="#" id="AlignmentRowsTable.__iter__-toggle" onclick="return toggle('AlignmentRowsTable.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AlignmentRowsTable-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AlignmentRowsTable.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AlignmentRowsTable.__iter__-expanded"><a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-491" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-491', 'id', 'link-148');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-492" class="py-name"><a title="csb.bio.sequence.AlignmentRowsTable" class="py-name" href="#" onclick="return doclink('link-492', 'AlignmentRowsTable', 'link-475');">AlignmentRowsTable</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-493" class="py-name" targets="Method csb.apps.csfrag.MatchTable.__iter__()=csb.apps.csfrag.MatchTable-class.html#__iter__,Method csb.bio.fragments.rosetta.RosettaFragment.__iter__()=csb.bio.fragments.rosetta.RosettaFragment-class.html#__iter__,Method csb.bio.hmm.HHpredHitList.__iter__()=csb.bio.hmm.HHpredHitList-class.html#__iter__,Method csb.bio.io.fasta.A3MSequenceIterator.__iter__()=csb.bio.io.fasta.A3MSequenceIterator-class.html#__iter__,Method csb.bio.io.mrc.HeaderInfo.__iter__()=csb.bio.io.mrc.HeaderInfo-class.html#__iter__,Method csb.bio.nmr.ChemicalShiftNetwork.__iter__()=csb.bio.nmr.ChemicalShiftNetwork-class.html#__iter__,Method csb.bio.nmr.ContactMap.__iter__()=csb.bio.nmr.ContactMap-class.html#__iter__,Method csb.bio.nmr.NOEPeak.__iter__()=csb.bio.nmr.NOEPeak-class.html#__iter__,Method csb.bio.nmr.NOESpectrum.__iter__()=csb.bio.nmr.NOESpectrum-class.html#__iter__,Method csb.bio.sequence.AbstractAlignment.__iter__()=csb.bio.sequence.AbstractAlignment-class.html#__iter__,Method csb.bio.sequence.AbstractSequence.__iter__()=csb.bio.sequence.AbstractSequence-class.html#__iter__,Method csb.bio.sequence.AlignedSequenceAdapter.__iter__()=csb.bio.sequence.AlignedSequenceAdapter-class.html#__iter__,Method csb.bio.sequence.AlignmentRowsTable.__iter__()=csb.bio.sequence.AlignmentRowsTable-class.html#__iter__,Method csb.bio.sequence.SequenceAdapter.__iter__()=csb.bio.sequence.SequenceAdapter-class.html#__iter__,Method csb.bio.sequence.SequenceIndexer.__iter__()=csb.bio.sequence.SequenceIndexer-class.html#__iter__,Method csb.bio.sequence.UngappedSequenceIndexer.__iter__()=csb.bio.sequence.UngappedSequenceIndexer-class.html#__iter__,Method csb.bio.structure.CompositeEntityIterator.__iter__()=csb.bio.structure.CompositeEntityIterator-class.html#__iter__,Method csb.core.AbstractContainer.__iter__()=csb.core.AbstractContainer-class.html#__iter__,Method csb.core.BaseCollectionContainer.__iter__()=csb.core.BaseCollectionContainer-class.html#__iter__,Method csb.core.BaseDictionaryContainer.__iter__()=csb.core.BaseDictionaryContainer-class.html#__iter__,Method csb.core.EnumMeta.__iter__()=csb.core.EnumMeta-class.html#__iter__,Method csb.io.plots.PlotsCollection.__iter__()=csb.io.plots.PlotsCollection-class.html#__iter__,Method csb.io.tsv.DataRow.__iter__()=csb.io.tsv.DataRow-class.html#__iter__,Method csb.io.tsv.SQLiteRepository.ChunkedReader.__iter__()=csb.io.tsv.SQLiteRepository.ChunkedReader-class.html#__iter__,Method csb.io.tsv.Table.__iter__()=csb.io.tsv.Table-class.html#__iter__,Method csb.statistics.samplers.mc.PropagationResult.__iter__()=csb.statistics.samplers.mc.PropagationResult-class.html#__iter__,Method csb.statistics.samplers.mc.propagators.Looper.__iter__()=csb.statistics.samplers.mc.propagators.Looper-class.html#__iter__"><a title="csb.apps.csfrag.MatchTable.__iter__
csb.bio.fragments.rosetta.RosettaFragment.__iter__
csb.bio.hmm.HHpredHitList.__iter__
csb.bio.io.fasta.A3MSequenceIterator.__iter__
csb.bio.io.mrc.HeaderInfo.__iter__
csb.bio.nmr.ChemicalShiftNetwork.__iter__
csb.bio.nmr.ContactMap.__iter__
csb.bio.nmr.NOEPeak.__iter__
csb.bio.nmr.NOESpectrum.__iter__
csb.bio.sequence.AbstractAlignment.__iter__
csb.bio.sequence.AbstractSequence.__iter__
csb.bio.sequence.AlignedSequenceAdapter.__iter__
csb.bio.sequence.AlignmentRowsTable.__iter__
csb.bio.sequence.SequenceAdapter.__iter__
csb.bio.sequence.SequenceIndexer.__iter__
csb.bio.sequence.UngappedSequenceIndexer.__iter__
csb.bio.structure.CompositeEntityIterator.__iter__
csb.core.AbstractContainer.__iter__
csb.core.BaseCollectionContainer.__iter__
csb.core.BaseDictionaryContainer.__iter__
csb.core.EnumMeta.__iter__
csb.io.plots.PlotsCollection.__iter__
csb.io.tsv.DataRow.__iter__
csb.io.tsv.SQLiteRepository.ChunkedReader.__iter__
csb.io.tsv.Table.__iter__
csb.statistics.samplers.mc.PropagationResult.__iter__
csb.statistics.samplers.mc.propagators.Looper.__iter__" class="py-name" href="#" onclick="return doclink('link-493', '__iter__', 'link-493');">__iter__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt id="link-494" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-494', 'id', 'link-148');">id</a></tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">         </tt>
<a name="AbstractAlignment"></a><div id="AbstractAlignment-def"><a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">     </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt> <a class="py-toggle" href="#" id="AbstractAlignment-toggle" onclick="return toggle('AbstractAlignment');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html">AbstractAlignment</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AbstractAlignment-expanded"><a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">    Base class for all alignment objects.</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"><tt class="py-docstring">    Provides 1-based access to the alignment.rows and alignment.columns.</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line"><tt class="py-docstring">    Alignment rows can also be accessed by sequence ID. In addition, all</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"><tt class="py-docstring">    alignments support 0-based slicing:</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; alignment[rows, columns]</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line"><tt class="py-docstring">        AbstractAlignment (sub-alignment)</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line"><tt class="py-docstring">    where</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line"><tt class="py-docstring">        - C{rows} can be a slice, tuple of row indexes or tuple of sequence IDs</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line"><tt class="py-docstring">        - columns can be a slice or tuple of column indexes</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line"><tt class="py-docstring">    For example:</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; alignment[:, 2:]</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"><tt class="py-docstring">        AbstractAlignment     # all rows, columns [3, alignment.length]</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; alignment[(0, 'seqx'), (3, 5)]</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring">        AbstractAlignment     # rows #1 and 'seq3', columns #4 and #5</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring">    @param sequences: alignment entries (must have equal length)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"><tt class="py-docstring">    @type sequences: list of L{AbstractSequence}s</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line"><tt class="py-docstring">    @param strict: if True, raise {DuplicateSequenceError} when a duplicate ID</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line"><tt class="py-docstring">                   is found (default=True)</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line"><tt class="py-docstring">    @type strict: bool</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"><tt class="py-docstring">    @note: if C{strict} is False and there are C{sequences} with redundant identifiers,</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"><tt class="py-docstring">           those sequences will be added to the C{rows} collection with :An suffix,</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"><tt class="py-docstring">           where n is a serial number. Therefore, rows['ID'] will return only one sequence,</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"><tt class="py-docstring">           the first sequence with id=ID. All remaining sequences can be retrieved</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line"><tt class="py-docstring">           with C{rows['ID:A1']}, {rows['ID:A2']}, etc. However, the sequence objects will</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"><tt class="py-docstring">           remain intact, e.g. {rows['ID:A1'].id} still returns 'ID' and not 'ID:A1'. </tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">     </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-495" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-495', 'ABCMeta', 'link-4');">ABCMeta</a></tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment.__init__"></a><div id="AbstractAlignment.__init__-def"><a name="L924"></a><tt class="py-lineno"> 924</tt> <a class="py-toggle" href="#" id="AbstractAlignment.__init__-toggle" onclick="return toggle('AbstractAlignment.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.__init__-expanded"><a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">         </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt> <tt class="py-op">=</tt> <tt id="link-496" class="py-name"><a title="csb.bio.sequence.AlignmentRowsTable" class="py-name" href="#" onclick="return doclink('link-496', 'AlignmentRowsTable', 'link-475');">AlignmentRowsTable</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colview</tt> <tt class="py-op">=</tt> <tt id="link-497" class="py-name"><a title="csb.bio.sequence.ColumnIndexer" class="py-name" href="#" onclick="return doclink('link-497', 'ColumnIndexer', 'link-449');">ColumnIndexer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> <tt class="py-op">=</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt id="link-498" class="py-name" targets="Variable csb.bio.io.fasta.SequenceAlignmentReader.strict=csb.bio.io.fasta.SequenceAlignmentReader-class.html#strict,Variable csb.bio.io.fasta.StructureAlignmentFactory.strict=csb.bio.io.fasta.StructureAlignmentFactory-class.html#strict"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-498', 'strict', 'link-498');">strict</a></tt><tt class="py-op">)</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">         </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_construct</tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt> </tt>
</div><a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">             </tt>
<a name="AbstractAlignment.__getitem__"></a><div id="AbstractAlignment.__getitem__-def"><a name="L934"></a><tt class="py-lineno"> 934</tt> <a class="py-toggle" href="#" id="AbstractAlignment.__getitem__-toggle" onclick="return toggle('AbstractAlignment.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">spec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.__getitem__-expanded"><a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">         </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">        <tt class="py-comment"># The following code can hardly get more readable than that, sorry.</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">        <tt class="py-comment"># Don't even think of modifying this before there is a 100% unit test coverage </tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">         </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-comment"># 0. expand the input tuple: (rows/, columns/) =&gt; (rows, columns)</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">'Invalid alignment slice expression'</tt><tt class="py-op">)</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">         </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">spec</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">            <tt class="py-name">rowspec</tt><tt class="py-op">,</tt> <tt class="py-name">colspec</tt> <tt class="py-op">=</tt> <tt class="py-name">spec</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">            <tt class="py-name">rowspec</tt><tt class="py-op">,</tt> <tt class="py-name">colspec</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">spec</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">        <tt class="py-comment"># 1. interpret the row slice: int, iter(int), iter(str) or slice(int) =&gt; list(int, 1-based)</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">.</tt><tt id="link-499" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-499', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-500" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-500', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-501', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-502', 'stop', 'link-215');">stop</a></tt><tt class="py-op">,</tt> <tt id="link-503" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-503', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-504', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid row slice: only indexes are supported"</tt><tt class="py-op">)</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">            <tt class="py-name">rowspec</tt> <tt class="py-op">=</tt> <tt id="link-505" class="py-name"><a title="csb.bio.sequence.SliceHelper" class="py-name" href="#" onclick="return doclink('link-505', 'SliceHelper', 'link-211');">SliceHelper</a></tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-506" class="py-name" targets="Variable csb.bio.fragments.rosetta.RosettaFragmentMap.size=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#size,Variable csb.bio.io.clans.ClansSeqgroup.size=csb.bio.io.clans.ClansSeqgroup-class.html#size,Variable csb.bio.sequence.AbstractAlignment.size=csb.bio.sequence.AbstractAlignment-class.html#size"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.size
csb.bio.io.clans.ClansSeqgroup.size
csb.bio.sequence.AbstractAlignment.size" class="py-name" href="#" onclick="return doclink('link-506', 'size', 'link-506');">size</a></tt><tt class="py-op">)</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">            <tt id="link-507" class="py-name" targets="Variable csb.bio.sequence.AbstractAlignment.rows=csb.bio.sequence.AbstractAlignment-class.html#rows,Variable csb.io.plots.Chart.rows=csb.io.plots.Chart-class.html#rows"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-507', 'rows', 'link-507');">rows</a></tt> <tt class="py-op">=</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-508', 'start', 'link-126');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">rowspec</tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-509', 'stop', 'link-215');">stop</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">            <tt id="link-510" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-510', 'rows', 'link-507');">rows</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">rowspec</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt>      </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-511" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-511', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-512" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-512', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-513" class="py-name" targets="Function csb.core.iterable()=csb.core-module.html#iterable"><a title="csb.core.iterable" class="py-name" href="#" onclick="return doclink('link-513', 'iterable', 'link-513');">iterable</a></tt><tt class="py-op">(</tt><tt class="py-name">rowspec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">                <tt id="link-514" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-514', 'rows', 'link-507');">rows</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-515" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-515', 'r', 'link-220');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">rowspec</tt><tt class="py-op">:</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-516" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-516', 'r', 'link-220');">r</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">                        <tt id="link-517" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-517', 'rows', 'link-507');">rows</a></tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-518', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt id="link-519" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-519', 'r', 'link-220');">r</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">                        <tt id="link-520" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-520', 'rows', 'link-507');">rows</a></tt><tt class="py-op">.</tt><tt id="link-521" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-521', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt><tt class="py-op">[</tt><tt id="link-522" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-522', 'r', 'link-220');">r</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ke</tt><tt class="py-op">:</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">'No such Sequence ID: {0!s}'</tt><tt class="py-op">.</tt><tt id="link-523" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-523', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">ke</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">'Unsupported row expression'</tt><tt class="py-op">)</tt>             </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">        <tt class="py-comment"># 2. interpret the column slice: int, iter(int) or slice(int) =&gt; list(int, 1-based)            </tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">colspec</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">            <tt class="py-name">colspec</tt> <tt class="py-op">=</tt> <tt id="link-524" class="py-name"><a title="csb.bio.sequence.SliceHelper" class="py-name" href="#" onclick="return doclink('link-524', 'SliceHelper', 'link-211');">SliceHelper</a></tt><tt class="py-op">(</tt><tt class="py-name">colspec</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">            <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">colspec</tt><tt class="py-op">.</tt><tt id="link-525" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-525', 'start', 'link-126');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">colspec</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name"><a title="csb.io.plots.Backend.stop" class="py-name" href="#" onclick="return doclink('link-526', 'stop', 'link-215');">stop</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">colspec</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">            <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">colspec</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt>  </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-527" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-527', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-528" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-528', 'core', 'link-1');">core</a></tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="csb.core.iterable" class="py-name" href="#" onclick="return doclink('link-529', 'iterable', 'link-513');">iterable</a></tt><tt class="py-op">(</tt><tt class="py-name">colspec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">                <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-name">c</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">colspec</tt> <tt class="py-op">]</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt><tt class="py-op">:</tt>             </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">'Unsupported column expression'</tt><tt class="py-op">)</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">'Unsupported column expression'</tt><tt class="py-op">)</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">         </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">        <tt class="py-comment"># 3. some more checks</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-530" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-530', 'rows', 'link-507');">rows</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"The expression returns zero rows"</tt><tt class="py-op">)</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"The expression returns zero columns"</tt><tt class="py-op">)</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">                 </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt class="py-comment"># 4. we are done</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_extract</tt><tt class="py-op">(</tt><tt id="link-531" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-531', 'rows', 'link-507');">rows</a></tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">)</tt> </tt>
</div><a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment._range"></a><div id="AbstractAlignment._range-def"><a name="L992"></a><tt class="py-lineno"> 992</tt> <a class="py-toggle" href="#" id="AbstractAlignment._range-toggle" onclick="return toggle('AbstractAlignment._range');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_range">_range</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">slice</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._range-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._range-expanded"><a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">         </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">        <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt><tt class="py-op">,</tt> <tt id="link-532" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-532', 't', 'link-461');">t</a></tt> <tt class="py-op">=</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt id="link-533" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-533', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt id="link-534" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-534', 'end', 'link-125');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">.</tt><tt class="py-name">step</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">         </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">s</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-535" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-535', 'start', 'link-126');">start</a></tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">e</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">            <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt id="link-536" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-536', 'end', 'link-125');">end</a></tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-537" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-537', 't', 'link-461');">t</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">            <tt id="link-538" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-538', 't', 'link-461');">t</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">             </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt><tt class="py-op">,</tt> <tt id="link-539" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-539', 't', 'link-461');">t</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment.__iter__"></a><div id="AbstractAlignment.__iter__-def"><a name="L1005"></a><tt class="py-lineno">1005</tt> <a class="py-toggle" href="#" id="AbstractAlignment.__iter__-toggle" onclick="return toggle('AbstractAlignment.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.__iter__-expanded"><a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">cn</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-540" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-540', 'length', 'link-156');">length</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_get_column</tt><tt class="py-op">(</tt><tt class="py-name">cn</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">             </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="AbstractAlignment._construct"></a><div id="AbstractAlignment._construct-def"><a name="L1010"></a><tt class="py-lineno">1010</tt> <a class="py-toggle" href="#" id="AbstractAlignment._construct-toggle" onclick="return toggle('AbstractAlignment._construct');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_construct">_construct</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._construct-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._construct-expanded"><a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring">        Hook method, called internally upon object construction. Subclasses</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">        define how the source alignment sequences are handled during alignment</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-docstring">        construction.</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring">        @param sequences: alignment entries</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">        @type sequences: list of L{AbstractSequence}s</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment._initialize"></a><div id="AbstractAlignment._initialize-def"><a name="L1021"></a><tt class="py-lineno">1021</tt> <a class="py-toggle" href="#" id="AbstractAlignment._initialize-toggle" onclick="return toggle('AbstractAlignment._initialize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_initialize">_initialize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rep_sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._initialize-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._initialize-expanded"><a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring">        Hook method, which is used to initialize various alignment properties</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"><tt class="py-docstring">        (such as length) from the first alignned sequence.</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">rep_sequence</tt><tt class="py-op">.</tt><tt id="link-541" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-541', 'length', 'link-156');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-542" class="py-name" targets="Class csb.bio.sequence.SequenceError=csb.bio.sequence.SequenceError-class.html"><a title="csb.bio.sequence.SequenceError" class="py-name" href="#" onclick="return doclink('link-542', 'SequenceError', 'link-542');">SequenceError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Sequence '{0}' is empty"</tt><tt class="py-op">.</tt><tt id="link-543" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-543', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">rep_sequence</tt><tt class="py-op">.</tt><tt id="link-544" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-544', 'id', 'link-148');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">                 </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-op">=</tt> <tt class="py-name">rep_sequence</tt><tt class="py-op">.</tt><tt id="link-545" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-545', 'length', 'link-156');">length</a></tt>  </tt>
</div><a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">         </tt>
<a name="AbstractAlignment._factory"></a><div id="AbstractAlignment._factory-def"><a name="L1032"></a><tt class="py-lineno">1032</tt> <a class="py-toggle" href="#" id="AbstractAlignment._factory-toggle" onclick="return toggle('AbstractAlignment._factory');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_factory">_factory</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">a</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">k</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._factory-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._factory-expanded"><a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line"><tt class="py-docstring">        Return a new sequence of the current L{AbstractAlignment} sub-class.</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>  </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt id="link-546" class="py-name"><a title="csb.bio.io.clans.Color.a
csb.statistics.pdf.GeneralizedInverseGaussian.a" class="py-name" href="#" onclick="return doclink('link-546', 'a', 'link-218');">a</a></tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">k</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">       </tt>
<a name="AbstractAlignment.add"></a><div id="AbstractAlignment.add-def"><a name="L1038"></a><tt class="py-lineno">1038</tt> <a class="py-toggle" href="#" id="AbstractAlignment.add-toggle" onclick="return toggle('AbstractAlignment.add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#add">add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.add-expanded"><a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line"><tt class="py-docstring">        Append a new sequence to the alignment.</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line"><tt class="py-docstring">        @type sequence: L{AbstractSequence}</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"><tt class="py-docstring">        @raise SequenceError: if the new sequence is too short/long</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line"><tt class="py-docstring">        @raise DuplicateSequenceError: if a sequence with same ID already exists  </tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">         </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">.</tt><tt id="link-547" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-547', 'length', 'link-156');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_initialize</tt><tt class="py-op">(</tt><tt id="link-548" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-548', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line"> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-549" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-549', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-550" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-550', 'length', 'link-156');">length</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt><tt class="py-op">:</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-551" class="py-name"><a title="csb.bio.sequence.SequenceError" class="py-name" href="#" onclick="return doclink('link-551', 'SequenceError', 'link-542');">SequenceError</a></tt><tt class="py-op">(</tt><tt class="py-string">'{0!r} is not of the expected length'</tt><tt class="py-op">.</tt><tt id="link-552" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-552', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt id="link-553" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-553', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">         </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt> <tt class="py-keyword">and</tt> <tt id="link-554" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-554', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-555" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-555', 'id', 'link-148');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">:</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-556" class="py-name" targets="Class csb.bio.sequence.DuplicateSequenceError=csb.bio.sequence.DuplicateSequenceError-class.html"><a title="csb.bio.sequence.DuplicateSequenceError" class="py-name" href="#" onclick="return doclink('link-556', 'DuplicateSequenceError', 'link-556');">DuplicateSequenceError</a></tt><tt class="py-op">(</tt><tt id="link-557" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-557', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-558" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-558', 'id', 'link-148');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">         </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">.</tt><tt class="py-name">_append</tt><tt class="py-op">(</tt><tt id="link-559" class="py-name"><a title="csb.bio.sequence.AlignedSequenceAdapter" class="py-name" href="#" onclick="return doclink('link-559', 'AlignedSequenceAdapter', 'link-428');">AlignedSequenceAdapter</a></tt><tt class="py-op">(</tt><tt id="link-560" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-560', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map</tt><tt class="py-op">[</tt><tt id="link-561" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-561', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-562" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-562', 'id', 'link-148');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-563', 'length', 'link-156');">length</a></tt> </tt>
</div><a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractAlignment.length"></a><div id="AbstractAlignment.length-def"><a name="L1060"></a><tt class="py-lineno">1060</tt> <a class="py-toggle" href="#" id="AbstractAlignment.length-toggle" onclick="return toggle('AbstractAlignment.length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#length">length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.length-expanded"><a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line"><tt class="py-docstring">        Number of columns in the alignment</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_length</tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line"> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractAlignment.size"></a><div id="AbstractAlignment.size-def"><a name="L1068"></a><tt class="py-lineno">1068</tt> <a class="py-toggle" href="#" id="AbstractAlignment.size-toggle" onclick="return toggle('AbstractAlignment.size');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#size">size</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.size-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.size-expanded"><a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"><tt class="py-docstring">        Number of rows (sequences) in the alignment</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">.</tt><tt id="link-564" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-564', 'length', 'link-156');">length</a></tt>     </tt>
</div><a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">             </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractAlignment.rows"></a><div id="AbstractAlignment.rows-def"><a name="L1076"></a><tt class="py-lineno">1076</tt> <a class="py-toggle" href="#" id="AbstractAlignment.rows-toggle" onclick="return toggle('AbstractAlignment.rows');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#rows">rows</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.rows-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.rows-expanded"><a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"><tt class="py-docstring">        1-based access to the alignment entries (sequences)</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AlignmentRowsTable}</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt>  </tt>
</div><a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">         </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractAlignment.columns"></a><div id="AbstractAlignment.columns-def"><a name="L1084"></a><tt class="py-lineno">1084</tt> <a class="py-toggle" href="#" id="AbstractAlignment.columns-toggle" onclick="return toggle('AbstractAlignment.columns');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#columns">columns</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.columns-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.columns-expanded"><a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"><tt class="py-docstring">        1-based access to the alignment columns</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ColumnIndexer}</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colview</tt> </tt>
</div><a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment.gap_at"></a><div id="AbstractAlignment.gap_at-def"><a name="L1091"></a><tt class="py-lineno">1091</tt> <a class="py-toggle" href="#" id="AbstractAlignment.gap_at-toggle" onclick="return toggle('AbstractAlignment.gap_at');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#gap_at">gap_at</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.gap_at-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.gap_at-expanded"><a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-docstring">        Return True of C{column} contains at least one gap.</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-docstring">        @param column: column number, 1-based</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"><tt class="py-docstring">        @type column: int</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">         </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">:</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-565" class="py-name" targets="Variable csb.bio.sequence.AbstractAlignment.columns=csb.bio.sequence.AbstractAlignment-class.html#columns,Variable csb.bio.sequence.AlignedSequenceAdapter.columns=csb.bio.sequence.AlignedSequenceAdapter-class.html#columns,Variable csb.io.plots.Chart.columns=csb.io.plots.Chart-class.html#columns,Variable csb.io.tsv.DataRow.columns=csb.io.tsv.DataRow-class.html#columns,Variable csb.io.tsv.Table.columns=csb.io.tsv.Table-class.html#columns"><a title="csb.bio.sequence.AbstractAlignment.columns
csb.bio.sequence.AlignedSequenceAdapter.columns
csb.io.plots.Chart.columns
csb.io.tsv.DataRow.columns
csb.io.tsv.Table.columns" class="py-name" href="#" onclick="return doclink('link-565', 'columns', 'link-565');">columns</a></tt><tt class="py-op">[</tt><tt id="link-566" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-566', 'column', 'link-146');">column</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-567', 'residue', 'link-152');">residue</a></tt><tt class="py-op">.</tt><tt id="link-568" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-568', 'type', 'link-121');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-569', 'alphabet', 'link-122');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-570" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-570', 'GAP', 'link-32');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">             </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt>         </tt>
</div><a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment._get_column"></a><div id="AbstractAlignment._get_column-def"><a name="L1106"></a><tt class="py-lineno">1106</tt> <a class="py-toggle" href="#" id="AbstractAlignment._get_column-toggle" onclick="return toggle('AbstractAlignment._get_column');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_get_column">_get_column</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._get_column-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._get_column-expanded"><a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt class="py-name">_get_column</tt><tt class="py-op">(</tt><tt id="link-571" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-571', 'column', 'link-146');">column</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-572" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-572', 'rows', 'link-507');">rows</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment._extract"></a><div id="AbstractAlignment._extract-def"><a name="L1109"></a><tt class="py-lineno">1109</tt> <a class="py-toggle" href="#" id="AbstractAlignment._extract-toggle" onclick="return toggle('AbstractAlignment._extract');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#_extract">_extract</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rows</tt><tt class="py-op">,</tt> <tt class="py-param">cols</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment._extract-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment._extract-expanded"><a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">         </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt id="link-573" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-573', 'rows', 'link-507');">rows</a></tt> <tt class="py-op">=</tt> <tt id="link-574" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-574', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt id="link-575" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-575', 'rows', 'link-507');">rows</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">        <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt id="link-576" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-576', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">                 </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">min</tt><tt class="py-op">(</tt><tt id="link-577" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-577', 'rows', 'link-507');">rows</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">max</tt><tt class="py-op">(</tt><tt id="link-578" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-578', 'rows', 'link-507');">rows</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.size
csb.bio.io.clans.ClansSeqgroup.size
csb.bio.sequence.AbstractAlignment.size" class="py-name" href="#" onclick="return doclink('link-579', 'size', 'link-506');">size</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">'Row specification out of range'</tt><tt class="py-op">)</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">                 </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">min</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">max</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-580" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-580', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">'Column specification out of range'</tt><tt class="py-op">)</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">         </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">         </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rn</tt><tt class="py-op">,</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-581" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-581', 'rows', 'link-507');">rows</a></tt><tt class="py-op">,</tt> <tt id="link-582" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-582', 'start', 'link-126');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">rn</tt> <tt class="py-keyword">in</tt> <tt id="link-583" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-583', 'rows', 'link-507');">rows</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">                <tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-584" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-584', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-585" class="py-name"><a title="csb.bio.sequence.AbstractSequence.extract
csb.bio.sequence.Sequence.extract
csb.bio.sequence.SequenceAdapter.extract" class="py-name" href="#" onclick="return doclink('link-585', 'extract', 'link-234');">extract</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">                 </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-586" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-586', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment.subregion"></a><div id="AbstractAlignment.subregion-def"><a name="L1128"></a><tt class="py-lineno">1128</tt> <a class="py-toggle" href="#" id="AbstractAlignment.subregion-toggle" onclick="return toggle('AbstractAlignment.subregion');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#subregion">subregion</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.subregion-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.subregion-expanded"><a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line"><tt class="py-docstring">        Extract a sub-alignment, ranging from C{start} to C{end} columns.</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line"><tt class="py-docstring">        @param start: starting column, 1-based</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line"><tt class="py-docstring">        @type start: int</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"><tt class="py-docstring">        @param end: ending column, 1-based</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line"><tt class="py-docstring">        @type end: int</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new alignment of the current type</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractAlignment}</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line"><tt class="py-docstring">        @raise ColumnPositionError: if start/end is out of range </tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt id="link-587" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-587', 'start', 'link-126');">start</a></tt> <tt class="py-op">&lt;=</tt> <tt id="link-588" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-588', 'end', 'link-125');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-589" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-589', 'length', 'link-156');">length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-590" class="py-name"><a title="csb.bio.sequence.ColumnPositionError" class="py-name" href="#" onclick="return doclink('link-590', 'ColumnPositionError', 'link-174');">ColumnPositionError</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-591', 'length', 'link-156');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">         </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">         </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">row</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-592" class="py-name"><a title="csb.bio.sequence.AbstractAlignment.rows
csb.io.plots.Chart.rows" class="py-name" href="#" onclick="return doclink('link-592', 'rows', 'link-507');">rows</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">            <tt class="py-name">sequences</tt><tt class="py-op">.</tt><tt id="link-593" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-593', 'append', 'link-243');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-594" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.subregion
csb.bio.hmm.ProfileHMM.subregion
csb.bio.sequence.AbstractAlignment.subregion
csb.bio.sequence.AbstractSequence.subregion
csb.bio.sequence.Sequence.subregion
csb.bio.sequence.SequenceAdapter.subregion
csb.bio.structure.Chain.subregion
csb.bio.structure.SecondaryStructure.subregion" class="py-name" href="#" onclick="return doclink('link-594', 'subregion', 'link-213');">subregion</a></tt><tt class="py-op">(</tt><tt id="link-595" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-595', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-596" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-596', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">                 </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_factory</tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-597" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-597', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_strict</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">     </tt>
<a name="AbstractAlignment.format"></a><div id="AbstractAlignment.format-def"><a name="L1152"></a><tt class="py-lineno">1152</tt> <a class="py-toggle" href="#" id="AbstractAlignment.format-toggle" onclick="return toggle('AbstractAlignment.format');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.AbstractAlignment-class.html#format">format</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">=</tt><tt id="link-598" class="py-name" targets="Class csb.bio.sequence.AlignmentFormats=csb.bio.sequence.AlignmentFormats-class.html"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-598', 'AlignmentFormats', 'link-598');">AlignmentFormats</a></tt><tt class="py-op">.</tt><tt id="link-599" class="py-name"><a title="csb.bio.sequence.AlignmentFormats.FASTA" class="py-name" href="#" onclick="return doclink('link-599', 'FASTA', 'link-6');">FASTA</a></tt><tt class="py-op">,</tt> <tt class="py-param">headers</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractAlignment.format-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractAlignment.format-expanded"><a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line"><tt class="py-docstring">        Format the alignment as a string.</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line"><tt class="py-docstring">        @param format: alignment format type, member of L{AlignmentFormats}</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line"><tt class="py-docstring">        @type format: L{EnumItem}</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line"><tt class="py-docstring">        @param headers: if False, omit headers</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line"><tt class="py-docstring">        @type headers: bool</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str </tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-600" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-600', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-601" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-601', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-602" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-602', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-603" class="py-name"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-603', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-604" class="py-name" targets="Class csb.bio.fragments.rosetta.OutputBuilder=csb.bio.fragments.rosetta.OutputBuilder-class.html,Class csb.bio.io.fasta.OutputBuilder=csb.bio.io.fasta.OutputBuilder-class.html"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-604', 'OutputBuilder', 'link-604');">OutputBuilder</a></tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line"> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt id="link-605" class="py-name" targets="Variable csb.test.Config.temp=csb.test.Config-class.html#temp"><a title="csb.test.Config.temp" class="py-name" href="#" onclick="return doclink('link-605', 'temp', 'link-605');">temp</a></tt> <tt class="py-op">=</tt> <tt id="link-606" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-606', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-607', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-608" class="py-name" targets="Class csb.io.MemoryStream=csb.io.MemoryStream-class.html"><a title="csb.io.MemoryStream" class="py-name" href="#" onclick="return doclink('link-608', 'MemoryStream', 'link-608');">MemoryStream</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">                 </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt>             </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">            <tt id="link-609" class="py-name"><a title="csb.test.Console.builder" class="py-name" href="#" onclick="return doclink('link-609', 'builder', 'link-192');">builder</a></tt> <tt class="py-op">=</tt> <tt id="link-610" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder
csb.bio.io.fasta.OutputBuilder" class="py-name" href="#" onclick="return doclink('link-610', 'OutputBuilder', 'link-604');">OutputBuilder</a></tt><tt class="py-op">.</tt><tt id="link-611" class="py-name" targets="Static Method csb.apps.buildhmm.ProfileBuilder.create()=csb.apps.buildhmm.ProfileBuilder-class.html#create,Static Method csb.apps.hhfrag.PredictionBuilder.create()=csb.apps.hhfrag.PredictionBuilder-class.html#create,Static Method csb.bio.fragments.ClusterNode.create()=csb.bio.fragments.ClusterNode-class.html#create,Static Method csb.bio.io.cs.ChemShiftReader.create()=csb.bio.io.cs.ChemShiftReader-class.html#create,Static Method csb.bio.io.fasta.OutputBuilder.create()=csb.bio.io.fasta.OutputBuilder-class.html#create,Static Method csb.bio.io.wwpdb.EntryID.create()=csb.bio.io.wwpdb.EntryID-class.html#create,Static Method csb.bio.io.wwpdb.SparseChainSequence.create()=csb.bio.io.wwpdb.SparseChainSequence-class.html#create,Static Method csb.bio.sequence.ChainSequence.create()=csb.bio.sequence.ChainSequence-class.html#create,Static Method csb.bio.sequence.RichSequence.create()=csb.bio.sequence.RichSequence-class.html#create,Static Method csb.bio.structure.CompositeEntityIterator.create()=csb.bio.structure.CompositeEntityIterator-class.html#create,Static Method csb.bio.structure.Residue.create()=csb.bio.structure.Residue-class.html#create,Static Method csb.core.Enum.create()=csb.core.Enum-class.html#create,Method csb.io.tsv.RepositoryImp.create()=csb.io.tsv.RepositoryImp-class.html#create,Method csb.io.tsv.SQLiteRepository.create()=csb.io.tsv.SQLiteRepository-class.html#create,Static Method csb.statistics.samplers.mc.TrajectoryBuilder.create()=csb.statistics.samplers.mc.TrajectoryBuilder-class.html#create"><a title="csb.apps.buildhmm.ProfileBuilder.create
csb.apps.hhfrag.PredictionBuilder.create
csb.bio.fragments.ClusterNode.create
csb.bio.io.cs.ChemShiftReader.create
csb.bio.io.fasta.OutputBuilder.create
csb.bio.io.wwpdb.EntryID.create
csb.bio.io.wwpdb.SparseChainSequence.create
csb.bio.sequence.ChainSequence.create
csb.bio.sequence.RichSequence.create
csb.bio.structure.CompositeEntityIterator.create
csb.bio.structure.Residue.create
csb.core.Enum.create
csb.io.tsv.RepositoryImp.create
csb.io.tsv.SQLiteRepository.create
csb.statistics.samplers.mc.TrajectoryBuilder.create" class="py-name" href="#" onclick="return doclink('link-611', 'create', 'link-611');">create</a></tt><tt class="py-op">(</tt><tt id="link-612" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-612', 'format', 'link-135');">format</a></tt><tt class="py-op">,</tt> <tt id="link-613" class="py-name"><a title="csb.test.Config.temp" class="py-name" href="#" onclick="return doclink('link-613', 'temp', 'link-605');">temp</a></tt><tt class="py-op">,</tt> <tt id="link-614" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-614', 'headers', 'link-195');">headers</a></tt><tt class="py-op">=</tt><tt id="link-615" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-615', 'headers', 'link-195');">headers</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">            <tt id="link-616" class="py-name"><a title="csb.test.Console.builder" class="py-name" href="#" onclick="return doclink('link-616', 'builder', 'link-192');">builder</a></tt><tt class="py-op">.</tt><tt id="link-617" class="py-name" targets="Method csb.bio.hmm.HHpredHit.add_alignment()=csb.bio.hmm.HHpredHit-class.html#add_alignment,Method csb.bio.io.fasta.A3MOutputBuilder.add_alignment()=csb.bio.io.fasta.A3MOutputBuilder-class.html#add_alignment,Method csb.bio.io.fasta.FASTAOutputBuilder.add_alignment()=csb.bio.io.fasta.FASTAOutputBuilder-class.html#add_alignment,Method csb.bio.io.fasta.OutputBuilder.add_alignment()=csb.bio.io.fasta.OutputBuilder-class.html#add_alignment,Method csb.bio.io.fasta.PIROutputBuilder.add_alignment()=csb.bio.io.fasta.PIROutputBuilder-class.html#add_alignment"><a title="csb.bio.hmm.HHpredHit.add_alignment
csb.bio.io.fasta.A3MOutputBuilder.add_alignment
csb.bio.io.fasta.FASTAOutputBuilder.add_alignment
csb.bio.io.fasta.OutputBuilder.add_alignment
csb.bio.io.fasta.PIROutputBuilder.add_alignment" class="py-name" href="#" onclick="return doclink('link-617', 'add_alignment', 'link-617');">add_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">             </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-618" class="py-name"><a title="csb.test.Config.temp" class="py-name" href="#" onclick="return doclink('link-618', 'temp', 'link-605');">temp</a></tt><tt class="py-op">.</tt><tt class="py-name">getvalue</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">         </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">        <tt class="py-keyword">finally</tt><tt class="py-op">:</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">            <tt id="link-619" class="py-name"><a title="csb.test.Config.temp" class="py-name" href="#" onclick="return doclink('link-619', 'temp', 'link-605');">temp</a></tt><tt class="py-op">.</tt><tt id="link-620" class="py-name"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-620', 'close', 'link-191');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>           </tt>
</div></div><a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line"> </tt>
<a name="SequenceAlignment"></a><div id="SequenceAlignment-def"><a name="L1176"></a><tt class="py-lineno">1176</tt> <a class="py-toggle" href="#" id="SequenceAlignment-toggle" onclick="return toggle('SequenceAlignment');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlignment-class.html">SequenceAlignment</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractAlignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignment-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SequenceAlignment-expanded"><a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"><tt class="py-docstring">    Multiple sequence alignment. See L{AbstractAlignment} for details.</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">         </tt>
<a name="SequenceAlignment._construct"></a><div id="SequenceAlignment._construct-def"><a name="L1181"></a><tt class="py-lineno">1181</tt> <a class="py-toggle" href="#" id="SequenceAlignment._construct-toggle" onclick="return toggle('SequenceAlignment._construct');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlignment-class.html#_construct">_construct</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignment._construct-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAlignment._construct-expanded"><a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">         </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-621" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-621', 'sequence', 'link-387');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-622" class="py-name" targets="Method csb.apps.csfrag.MatchTable.add()=csb.apps.csfrag.MatchTable-class.html#add,Method csb.apps.hhfrag.PredictionBuilder.add()=csb.apps.hhfrag.PredictionBuilder-class.html#add,Method csb.bio.io.clans.ClansSeqgroup.add()=csb.bio.io.clans.ClansSeqgroup-class.html#add,Method csb.bio.io.wwpdb.CustomStructureProvider.add()=csb.bio.io.wwpdb.CustomStructureProvider-class.html#add,Method csb.bio.io.wwpdb.FileSystemStructureProvider.add()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#add,Method csb.bio.nmr.NOESpectrum.add()=csb.bio.nmr.NOESpectrum-class.html#add,Method csb.bio.sequence.AbstractAlignment.add()=csb.bio.sequence.AbstractAlignment-class.html#add,Method csb.io.plots.Backend.add()=csb.io.plots.Backend-class.html#add"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-622', 'add', 'link-622');">add</a></tt><tt class="py-op">(</tt><tt id="link-623" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-623', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">             </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="SequenceAlignment.parse"></a><div id="SequenceAlignment.parse-def"><a name="L1187"></a><tt class="py-lineno">1187</tt> <a class="py-toggle" href="#" id="SequenceAlignment.parse-toggle" onclick="return toggle('SequenceAlignment.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.SequenceAlignment-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">string</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceAlignment.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SequenceAlignment.parse-expanded"><a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{SequenceAlignment} from an mFASTA string.</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: MSA-formatted string</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line"><tt class="py-docstring">        @param strict: see L{AbstractAlignment}</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line"><tt class="py-docstring">        @type strict: bool        </tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SequenceAlignment}</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-624" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-624', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-625" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-625', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-626" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-626', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-627" class="py-name"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-627', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-628" class="py-name" targets="Class csb.bio.io.fasta.SequenceAlignmentReader=csb.bio.io.fasta.SequenceAlignmentReader-class.html"><a title="csb.bio.io.fasta.SequenceAlignmentReader" class="py-name" href="#" onclick="return doclink('link-628', 'SequenceAlignmentReader', 'link-628');">SequenceAlignmentReader</a></tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-629" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader" class="py-name" href="#" onclick="return doclink('link-629', 'SequenceAlignmentReader', 'link-628');">SequenceAlignmentReader</a></tt><tt class="py-op">(</tt><tt id="link-630" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-630', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt id="link-631" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-631', 'strict', 'link-498');">strict</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-632" class="py-name" targets="Method csb.bio.io.fasta.SequenceAlignmentReader.read_fasta()=csb.bio.io.fasta.SequenceAlignmentReader-class.html#read_fasta"><a title="csb.bio.io.fasta.SequenceAlignmentReader.read_fasta" class="py-name" href="#" onclick="return doclink('link-632', 'read_fasta', 'link-632');">read_fasta</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">         </tt>
<a name="StructureAlignment"></a><div id="StructureAlignment-def"><a name="L1201"></a><tt class="py-lineno">1201</tt> <a class="py-toggle" href="#" id="StructureAlignment-toggle" onclick="return toggle('StructureAlignment');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.StructureAlignment-class.html">StructureAlignment</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractAlignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignment-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StructureAlignment-expanded"><a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line"><tt class="py-docstring">    Multiple structure alignment. Similar to a L{SequenceAlignment}, but</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line"><tt class="py-docstring">    the alignment holds the actual L{csb.bio.structure.ProteinResidue} objects,</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"><tt class="py-docstring">    taken from the corresponding source L{csb.bio.structure.Chain}s.</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line"><tt class="py-docstring">    See L{AbstractAlignment} for details.</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">         </tt>
<a name="StructureAlignment._construct"></a><div id="StructureAlignment._construct-def"><a name="L1210"></a><tt class="py-lineno">1210</tt> <a class="py-toggle" href="#" id="StructureAlignment._construct-toggle" onclick="return toggle('StructureAlignment._construct');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.StructureAlignment-class.html#_construct">_construct</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignment._construct-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StructureAlignment._construct-expanded"><a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">         </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-633" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-633', 'sequence', 'link-387');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-634" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-634', 'add', 'link-622');">add</a></tt><tt class="py-op">(</tt><tt id="link-635" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-635', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">             </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="StructureAlignment.parse"></a><div id="StructureAlignment.parse-def"><a name="L1216"></a><tt class="py-lineno">1216</tt> <a class="py-toggle" href="#" id="StructureAlignment.parse-toggle" onclick="return toggle('StructureAlignment.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.StructureAlignment-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">string</tt><tt class="py-op">,</tt> <tt class="py-param">provider</tt><tt class="py-op">,</tt> <tt class="py-param">id_factory</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureAlignment.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StructureAlignment.parse-expanded"><a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{StructureAlignment} from an mFASTA string. See </tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line"><tt class="py-docstring">        L{csb.bio.io.fasta.StructureAlignmentFactory} for details. </tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: MSA-formatted string</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line"><tt class="py-docstring">        @param provider: data source for all structures found in the alignment</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line"><tt class="py-docstring">        @type provider: L{csb.bio.io.wwpdb.StructureProvider}</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line"><tt class="py-docstring">        @param strict: see L{AbstractAlignment}</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line"><tt class="py-docstring">        @type strict: bool</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line"><tt class="py-docstring">        @param id_factory: callable factory, which transforms a sequence ID into</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line"><tt class="py-docstring">                           a L{csb.bio.io.wwpdb.EntryID} object. By default</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line"><tt class="py-docstring">                           this is L{csb.bio.io.wwpdb.EntryID.create}. </tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line"><tt class="py-docstring">        @type id_factory: callable        </tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{StructureAlignment}</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-636" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-636', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-637" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-637', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-638" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-638', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-639" class="py-name"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-639', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-640" class="py-name" targets="Class csb.bio.io.fasta.StructureAlignmentFactory=csb.bio.io.fasta.StructureAlignmentFactory-class.html"><a title="csb.bio.io.fasta.StructureAlignmentFactory" class="py-name" href="#" onclick="return doclink('link-640', 'StructureAlignmentFactory', 'link-640');">StructureAlignmentFactory</a></tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">         </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">        <tt class="py-name">factory</tt> <tt class="py-op">=</tt> <tt id="link-641" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory" class="py-name" href="#" onclick="return doclink('link-641', 'StructureAlignmentFactory', 'link-640');">StructureAlignmentFactory</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">                        <tt id="link-642" class="py-name" targets="Variable csb.bio.io.fasta.StructureAlignmentFactory.provider=csb.bio.io.fasta.StructureAlignmentFactory-class.html#provider"><a title="csb.bio.io.fasta.StructureAlignmentFactory.provider" class="py-name" href="#" onclick="return doclink('link-642', 'provider', 'link-642');">provider</a></tt><tt class="py-op">,</tt> <tt id="link-643" class="py-name" targets="Variable csb.bio.io.fasta.StructureAlignmentFactory.id_factory=csb.bio.io.fasta.StructureAlignmentFactory-class.html#id_factory"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-643', 'id_factory', 'link-643');">id_factory</a></tt><tt class="py-op">=</tt><tt id="link-644" class="py-name"><a title="csb.bio.io.fasta.StructureAlignmentFactory.id_factory" class="py-name" href="#" onclick="return doclink('link-644', 'id_factory', 'link-643');">id_factory</a></tt><tt class="py-op">,</tt> <tt id="link-645" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-645', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt id="link-646" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-646', 'strict', 'link-498');">strict</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">factory</tt><tt class="py-op">.</tt><tt id="link-647" class="py-name" targets="Method csb.bio.io.fasta.StructureAlignmentFactory.make_alignment()=csb.bio.io.fasta.StructureAlignmentFactory-class.html#make_alignment"><a title="csb.bio.io.fasta.StructureAlignmentFactory.make_alignment" class="py-name" href="#" onclick="return doclink('link-647', 'make_alignment', 'link-647');">make_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">     </tt>
<a name="A3MAlignment"></a><div id="A3MAlignment-def"><a name="L1239"></a><tt class="py-lineno">1239</tt> <a class="py-toggle" href="#" id="A3MAlignment-toggle" onclick="return toggle('A3MAlignment');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html">A3MAlignment</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractAlignment</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="A3MAlignment-expanded"><a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line"><tt class="py-docstring">    A specific type of multiple alignment, which provides some operations</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line"><tt class="py-docstring">    relative to a master sequence (the first entry in the alignment). </tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">     </tt>
<a name="A3MAlignment.__init__"></a><div id="A3MAlignment.__init__-def"><a name="L1245"></a><tt class="py-lineno">1245</tt> <a class="py-toggle" href="#" id="A3MAlignment.__init__-toggle" onclick="return toggle('A3MAlignment.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.__init__-expanded"><a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line"> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_matches</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_insertions</tt> <tt class="py-op">=</tt> <tt id="link-648" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-648', 'set', 'link-108');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">                 </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-649" class="py-name" targets="Class csb.bio.sequence.A3MAlignment=csb.bio.sequence.A3MAlignment-class.html"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-649', 'A3MAlignment', 'link-649');">A3MAlignment</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-650" class="py-name"><a title="csb.Version.__init__
csb.apps.AppExit.__init__
csb.apps.AppRunner.__init__
csb.apps.Application.__init__
csb.apps.ArgHandler.__init__
csb.apps.buildhmm.NoOutputError.__init__
csb.apps.buildhmm.ProfileBuilder.__init__
csb.apps.csfrag.CSfrag.__init__
csb.apps.csfrag.MatchInfo.__init__
csb.apps.csfrag.MatchTable.__init__
csb.apps.csfrag.ScoringHelper.__init__
csb.apps.embd.Deconvolution.__init__
csb.apps.embd.OutputPathBuilder.__init__
csb.apps.hhfrag.HHfrag.__init__
csb.apps.hhfrag.PredictionBuilder.__init__
csb.apps.hhfrag.SliceContext.__init__
csb.apps.hhsearch.Context.__init__
csb.apps.hhsearch.HHTask.__init__
csb.apps.hhsearch.HHsearch.__init__
csb.apps.precision.GlobalInfo.__init__
csb.apps.precision.LibrarySuperimposer.__init__
csb.bio.fragments.Assignment.__init__
csb.bio.fragments.BenchmarkAdapter.Connection.__init__
csb.bio.fragments.BenchmarkAdapter.__init__
csb.bio.fragments.ChemShiftAssignment.__init__
csb.bio.fragments.ChemShiftTarget.__init__
csb.bio.fragments.ClusterNode.__init__
csb.bio.fragments.ClusterRep.__init__
csb.bio.fragments.FragmentCluster.__init__
csb.bio.fragments.FragmentMatch.__init__
csb.bio.fragments.ResidueAssignmentInfo.__init__
csb.bio.fragments.ResidueEventInfo.__init__
csb.bio.fragments.RosettaFragsetFactory.__init__
csb.bio.fragments.Target.__init__
csb.bio.fragments.TargetResidue.__init__
csb.bio.fragments.TorsionAnglePredictor.__init__
csb.bio.fragments.TorsionPredictionInfo.__init__
csb.bio.fragments.rosetta.OutputBuilder.__init__
csb.bio.fragments.rosetta.ResidueInfo.__init__
csb.bio.fragments.rosetta.RosettaFragment.__init__
csb.bio.fragments.rosetta.RosettaFragmentMap.__init__
csb.bio.hmm.EVDParameters.__init__
csb.bio.hmm.EmissionTable.__init__
csb.bio.hmm.HHpredHit.__init__
csb.bio.hmm.HHpredHitAlignment.__init__
csb.bio.hmm.HHpredHitList.__init__
csb.bio.hmm.HMMLayer.__init__
csb.bio.hmm.HMMLayersCollection.__init__
csb.bio.hmm.ProfileHMM.__init__
csb.bio.hmm.ProfileHMMRegion.__init__
csb.bio.hmm.ProfileHMMSegment.__init__
csb.bio.hmm.ProfileLength.__init__
csb.bio.hmm.State.__init__
csb.bio.hmm.StateFactory.__init__
csb.bio.hmm.Transition.__init__
csb.bio.hmm.TransitionTable.__init__
csb.bio.hmm.TransitionType.__init__
csb.bio.hmm.pseudocounts.PseudocountBuilder.__init__
csb.bio.io.clans.Clans.__init__
csb.bio.io.clans.ClansEntry.__init__
csb.bio.io.clans.ClansEntryCollection.__init__
csb.bio.io.clans.ClansFileBuilder.__init__
csb.bio.io.clans.ClansFileWriter.__init__
csb.bio.io.clans.ClansParams.__init__
csb.bio.io.clans.ClansParser.__init__
csb.bio.io.clans.ClansSeqgroup.__init__
csb.bio.io.clans.ClansSeqgroupCollection.__init__
csb.bio.io.clans.Color.__init__
csb.bio.io.cs.ChemShiftReader.__init__
csb.bio.io.dssp.ResidueAssignmentInfo.__init__
csb.bio.io.fasta.A3MSequenceIterator.__init__
csb.bio.io.fasta.BaseSequenceParser.__init__
csb.bio.io.fasta.OutputBuilder.__init__
csb.bio.io.fasta.SequenceAlignmentReader.__init__
csb.bio.io.fasta.StructureAlignmentFactory.__init__
csb.bio.io.hhpred.HHMFileBuilder.__init__
csb.bio.io.hhpred.HHOutputParser.__init__
csb.bio.io.hhpred.HHProfileParser.__init__
csb.bio.io.mrc.DensityInfo.__init__
csb.bio.io.mrc.DensityMapReader.__init__
csb.bio.io.mrc.HeaderInfo.__init__
csb.bio.io.noe.SparkyPeakListReader.__init__
csb.bio.io.noe.XeasyFileBuilder.__init__
csb.bio.io.noe.XeasyPeakListReader.__init__
csb.bio.io.procheck.ProcheckParser.__init__
csb.bio.io.psipred.PSIPredResidueInfo.__init__
csb.bio.io.svg.SSCartoonBuilder.__init__
csb.bio.io.vasco.ChemicalShiftContainer.__init__
csb.bio.io.vasco.ShiftInfo.__init__
csb.bio.io.vasco.VascoStructureParser.__init__
csb.bio.io.whatif.WhatCheckParser.__init__
csb.bio.io.wwpdb.AbstractStructureParser.__init__
csb.bio.io.wwpdb.AsyncParseResult.__init__
csb.bio.io.wwpdb.AsyncStructureParser.__init__
csb.bio.io.wwpdb.CustomStructureProvider.__init__
csb.bio.io.wwpdb.EntryID.__init__
csb.bio.io.wwpdb.FastResidueMapper.__init__
csb.bio.io.wwpdb.FileBuilder.__init__
csb.bio.io.wwpdb.FileSystemStructureProvider.__init__
csb.bio.io.wwpdb.RemoteStructureProvider.__init__
csb.bio.io.wwpdb.ResidueInfo.__init__
csb.bio.io.wwpdb.RobustResidueMapper.__init__
csb.bio.nmr.AtomConnectivity.__init__
csb.bio.nmr.ChemShiftInfo.__init__
csb.bio.nmr.ChemShiftScoringModel.__init__
csb.bio.nmr.ChemicalShiftNetwork.__init__
csb.bio.nmr.ContactMap.__init__
csb.bio.nmr.ContactMapComparisonInfo.__init__
csb.bio.nmr.Label.__init__
csb.bio.nmr.NOEPeak.__init__
csb.bio.nmr.NOESpectrum.__init__
csb.bio.nmr.RandomCoil.__init__
csb.bio.sequence.A3MAlignment.__init__
csb.bio.sequence.AbstractAlignment.__init__
csb.bio.sequence.AbstractSequence.__init__
csb.bio.sequence.AlignedSequenceAdapter.__init__
csb.bio.sequence.AlignmentRowsTable.__init__
csb.bio.sequence.ColumnInfo.__init__
csb.bio.sequence.PositionError.__init__
csb.bio.sequence.ResidueInfo.__init__
csb.bio.sequence.Sequence.__init__
csb.bio.sequence.SequenceAdapter.__init__
csb.bio.sequence.SequenceCollection.__init__
csb.bio.sequence.SequenceIndexer.__init__
csb.bio.sequence.SliceHelper.__init__
csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.AlignmentResult.__init__
csb.bio.sequence.alignment.GlobalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.IdentityMatrix.__init__
csb.bio.sequence.alignment.LocalAlignmentAlgorithm.__init__
csb.bio.sequence.alignment.SimilarityMatrix.__init__
csb.bio.structure.Atom.__init__
csb.bio.structure.Chain.__init__
csb.bio.structure.ChainResiduesCollection.__init__
csb.bio.structure.CompositeEntityIterator.__init__
csb.bio.structure.ConfinedEntityIterator.__init__
csb.bio.structure.DisorderedAtom.__init__
csb.bio.structure.Ensemble.__init__
csb.bio.structure.EnsembleModelsCollection.__init__
csb.bio.structure.NucleicResidue.__init__
csb.bio.structure.ProteinResidue.__init__
csb.bio.structure.Residue.__init__
csb.bio.structure.ResidueAtomsTable.__init__
csb.bio.structure.SecondaryStructure.__init__
csb.bio.structure.SecondaryStructureElement.__init__
csb.bio.structure.Structure.__init__
csb.bio.structure.StructureChainsTable.__init__
csb.bio.structure.SuperimposeInfo.__init__
csb.bio.structure.TorsionAngles.__init__
csb.bio.structure.TorsionAnglesCollection.__init__
csb.bio.structure.UnknownResidue.__init__
csb.build.Console.__init__
csb.build.GitHandler.__init__
csb.build.MercurialHandler.__init__
csb.build.RevisionError.__init__
csb.build.RevisionHandler.__init__
csb.build.RevisionInfo.__init__
csb.build.SubversionHandler.__init__
csb.core.BaseCollectionContainer.__init__
csb.core.BaseDictionaryContainer.__init__
csb.core.CollectionContainer.__init__
csb.core.Container.__init__
csb.core.DictionaryContainer.__init__
csb.core.EnumItem.__init__
csb.core.InterruptibleThread.__init__
csb.core.Proxy.__init__
csb.core.REMatchProxy.__init__
csb.core.enum.__init__
csb.core.typedproperty.__init__
csb.core.validatedproperty.__init__
csb.io.AutoFlushStream.__init__
csb.io.EntryReader.__init__
csb.io.EntryWriter.__init__
csb.io.InvalidCommandError.__init__
csb.io.ProcessError.__init__
csb.io.ShellInfo.__init__
csb.io.TempFile.__init__
csb.io.TempFolder.__init__
csb.io.TimeoutError.__init__
csb.io.plots.Backend.__init__
csb.io.plots.Chart.__init__
csb.io.plots.PlotsCollection.__init__
csb.io.plots.WxBackendImpl.__init__
csb.io.tsv.Between.__init__
csb.io.tsv.ColumnInfo.__init__
csb.io.tsv.DataRow.__init__
csb.io.tsv.Expression.__init__
csb.io.tsv.Operator.__init__
csb.io.tsv.Predicate.__init__
csb.io.tsv.RepositoryImp.__init__
csb.io.tsv.SQLiteRepository.ChunkedReader.__init__
csb.io.tsv.SQLiteRepository.__init__
csb.io.tsv.Step.__init__
csb.io.tsv.Table.__init__
csb.io.tsv.Where.__init__
csb.numeric.InvertibleMatrix.__init__
csb.numeric.integrators.AbstractIntegrator.__init__
csb.statistics.Cumulative.__init__
csb.statistics.ars.ARS.__init__
csb.statistics.ars.Envelope.__init__
csb.statistics.ars.Gauss.__init__
csb.statistics.maxent.MaxentModel.__init__
csb.statistics.maxent.MaxentPosterior.__init__
csb.statistics.mixtures.AbstractStructureMixture.__init__
csb.statistics.mixtures.GaussianMixture.__init__
csb.statistics.pdf.AbstractDensity.__init__
csb.statistics.pdf.Dirichlet.__init__
csb.statistics.pdf.DirichletEstimator.__init__
csb.statistics.pdf.Gamma.__init__
csb.statistics.pdf.GammaMLEstimator.__init__
csb.statistics.pdf.GenNormalBruteForceEstimator.__init__
csb.statistics.pdf.GeneralizedInverseGaussian.__init__
csb.statistics.pdf.GeneralizedNormal.__init__
csb.statistics.pdf.GumbelMaximum.__init__
csb.statistics.pdf.GumbelMinimum.__init__
csb.statistics.pdf.InverseGamma.__init__
csb.statistics.pdf.InverseGaussian.__init__
csb.statistics.pdf.Laplace.__init__
csb.statistics.pdf.MultivariateGaussian.__init__
csb.statistics.pdf.MultivariateGaussianMLEstimator.__init__
csb.statistics.pdf.Normal.__init__
csb.statistics.pdf.ParameterValueError.__init__
csb.statistics.pdf.parameterized.AbstractParameter.__init__
csb.statistics.pdf.parameterized.Parameter.__init__
csb.statistics.samplers.EnsembleState.__init__
csb.statistics.samplers.State.__init__
csb.statistics.samplers.mc.AbstractMC.__init__
csb.statistics.samplers.mc.MCCollection.__init__
csb.statistics.samplers.mc.PropagationResult.__init__
csb.statistics.samplers.mc.Trajectory.__init__
csb.statistics.samplers.mc.TrajectoryBuilder.__init__
csb.statistics.samplers.mc.multichain.AbstractEnsembleMC.__init__
csb.statistics.samplers.mc.multichain.AbstractExchangeMC.__init__
csb.statistics.samplers.mc.multichain.AbstractRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractStepRENS.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapCommunicator.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.AbstractSwapScheme.__init__
csb.statistics.samplers.mc.multichain.AlternatingAdjacentSwapScheme.__init__
csb.statistics.samplers.mc.multichain.Gradient.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENS.__init__
csb.statistics.samplers.mc.multichain.HMCStepRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.InterpolationFactory.__init__
csb.statistics.samplers.mc.multichain.MDRENS.__init__
csb.statistics.samplers.mc.multichain.MDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.multichain.RENSTrajInfo.__init__
csb.statistics.samplers.mc.multichain.ReplicaHistory.__init__
csb.statistics.samplers.mc.multichain.SingleSwapStatistics.__init__
csb.statistics.samplers.mc.multichain.SwapStatistics.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENS.__init__
csb.statistics.samplers.mc.multichain.ThermostattedMDRENSSwapParameterInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractHeatWorkJacobianLogger.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.AbstractPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.HMCPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.HamiltonianSysInfo.__init__
csb.statistics.samplers.mc.neqsteppropagator.MDParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumStepPropagator.__init__
csb.statistics.samplers.mc.neqsteppropagator.NonequilibriumTrajectory.__init__
csb.statistics.samplers.mc.neqsteppropagator.PerturbationResult.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagation.__init__
csb.statistics.samplers.mc.neqsteppropagator.PlainMDPropagationParam.__init__
csb.statistics.samplers.mc.neqsteppropagator.Protocol.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonian.__init__
csb.statistics.samplers.mc.neqsteppropagator.ReducedHamiltonianPerturbation.__init__
csb.statistics.samplers.mc.neqsteppropagator.Step.__init__
csb.statistics.samplers.mc.propagators.AbstractMCPropagator.__init__
csb.statistics.samplers.mc.propagators.AbstractNCMCPropagator.__init__
csb.statistics.samplers.mc.propagators.HMCPropagator.__init__
csb.statistics.samplers.mc.propagators.Looper.__init__
csb.statistics.samplers.mc.propagators.MDPropagator.__init__
csb.statistics.samplers.mc.propagators.RWMCPropagator.__init__
csb.statistics.samplers.mc.propagators.ThermostattedMDPropagator.__init__
csb.statistics.samplers.mc.singlechain.AbstractNCMCSampler.__init__
csb.statistics.samplers.mc.singlechain.AbstractSingleChainMC.__init__
csb.statistics.samplers.mc.singlechain.HMCSampler.__init__
csb.statistics.samplers.mc.singlechain.NCMCProposalCommunicator.__init__
csb.statistics.samplers.mc.singlechain.RWMCSampler.__init__
csb.statistics.samplers.mc.singlechain.SimpleProposalCommunicator.__init__
csb.statistics.scalemixture.ARSPosteriorAlpha.__init__
csb.statistics.scalemixture.GammaPosteriorMAP.__init__
csb.statistics.scalemixture.GammaPosteriorSampler.__init__
csb.statistics.scalemixture.GammaPrior.__init__
csb.statistics.scalemixture.InvGammaPosteriorMAP.__init__
csb.statistics.scalemixture.InvGammaPosteriorSampler.__init__
csb.statistics.scalemixture.InvGammaPrior.__init__
csb.statistics.scalemixture.ScaleMixture.__init__
csb.statistics.scalemixture.ScaleMixturePrior.__init__
csb.statmech.ensembles.BoltzmannEnsemble.__init__
csb.statmech.ensembles.CompositeEnsemble.__init__
csb.statmech.ensembles.FermiEnsemble.__init__
csb.statmech.ensembles.TsallisEnsemble.__init__
csb.statmech.wham.AbstractWHAM.__init__
csb.statmech.wham.WHAM.__init__
csb.test.Console.__init__" class="py-name" href="#" onclick="return doclink('link-650', '__init__', 'link-132');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-651" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-651', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt id="link-652" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-652', 'strict', 'link-498');">strict</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line"> </tt>
<a name="A3MAlignment._initialize"></a><div id="A3MAlignment._initialize-def"><a name="L1253"></a><tt class="py-lineno">1253</tt> <a class="py-toggle" href="#" id="A3MAlignment._initialize-toggle" onclick="return toggle('A3MAlignment._initialize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#_initialize">_initialize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rep_sequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment._initialize-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment._initialize-expanded"><a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">         </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-653" class="py-name"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-653', 'A3MAlignment', 'link-649');">A3MAlignment</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">_initialize</tt><tt class="py-op">(</tt><tt class="py-name">rep_sequence</tt><tt class="py-op">)</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt> <tt class="py-op">=</tt> <tt class="py-name">rep_sequence</tt><tt class="py-op">.</tt><tt id="link-654" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-654', 'alphabet', 'link-122');">alphabet</a></tt>         </tt>
</div><a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">             </tt>
<a name="A3MAlignment._construct"></a><div id="A3MAlignment._construct-def"><a name="L1258"></a><tt class="py-lineno">1258</tt> <a class="py-toggle" href="#" id="A3MAlignment._construct-toggle" onclick="return toggle('A3MAlignment._construct');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#_construct">_construct</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sequences</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment._construct-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment._construct-expanded"><a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">         </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-655" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-655', 'sequence', 'link-387');">sequence</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">sequences</tt><tt class="py-op">:</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">             </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-656', 'add', 'link-622');">add</a></tt><tt class="py-op">(</tt><tt id="link-657" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-657', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">                     </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-658" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-658', 'rank', 'link-136');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-659" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-659', 'residue', 'link-152');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-660" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-660', 'sequence', 'link-387');">sequence</a></tt><tt class="py-op">,</tt> <tt id="link-661" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-661', 'start', 'link-126');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-662" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-662', 'residue', 'link-152');">residue</a></tt><tt class="py-op">.</tt><tt id="link-663" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-663', 'type', 'link-121');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alphabet</tt><tt class="py-op">.</tt><tt id="link-664" class="py-name"><a title="csb.bio.sequence.NucleicAlphabet.INSERTION
csb.bio.sequence.ProteinAlphabet.INSERTION
csb.bio.sequence.UnknownAlphabet.INSERTION" class="py-name" href="#" onclick="return doclink('link-664', 'INSERTION', 'link-33');">INSERTION</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_insertions</tt><tt class="py-op">.</tt><tt id="link-665" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-665', 'add', 'link-622');">add</a></tt><tt class="py-op">(</tt><tt id="link-666" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-666', 'rank', 'link-136');">rank</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line"> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-667" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.size
csb.bio.io.clans.ClansSeqgroup.size
csb.bio.sequence.AbstractAlignment.size" class="py-name" href="#" onclick="return doclink('link-667', 'size', 'link-506');">size</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-668" class="py-name"><a title="csb.bio.sequence.SequenceError" class="py-name" href="#" onclick="return doclink('link-668', 'SequenceError', 'link-542');">SequenceError</a></tt><tt class="py-op">(</tt><tt class="py-string">"At least one sequence is required"</tt><tt class="py-op">)</tt>  </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line">         </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_msa</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_matches</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt><tt class="py-op">.</tt><tt id="link-669" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-669', 'strip', 'link-268');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-670" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-670', 'length', 'link-156');">length</a></tt> </tt>
</div><a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">     </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="A3MAlignment.master"></a><div id="A3MAlignment.master-def"><a name="L1275"></a><tt class="py-lineno">1275</tt> <a class="py-toggle" href="#" id="A3MAlignment.master-toggle" onclick="return toggle('A3MAlignment.master');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#master">master</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.master-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.master-expanded"><a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line"><tt class="py-docstring">        The master sequence</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractSequence}</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt> </tt>
</div><a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">     </tt>
<a name="A3MAlignment.insertion_at"></a><div id="A3MAlignment.insertion_at-def"><a name="L1282"></a><tt class="py-lineno">1282</tt> <a class="py-toggle" href="#" id="A3MAlignment.insertion_at-toggle" onclick="return toggle('A3MAlignment.insertion_at');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#insertion_at">insertion_at</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">column</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.insertion_at-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.insertion_at-expanded"><a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line"><tt class="py-docstring">        Return True of C{column} contains at least one insertion.</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line"><tt class="py-docstring">        @param column: column number, 1-based</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line"><tt class="py-docstring">        @type column: int</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: bool</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-671" class="py-name"><a title="csb.io.tsv.Table.column" class="py-name" href="#" onclick="return doclink('link-671', 'column', 'link-146');">column</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_insertions</tt> </tt>
</div><a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line">     </tt>
<a name="A3MAlignment.hmm_subregion"></a><div id="A3MAlignment.hmm_subregion-def"><a name="L1292"></a><tt class="py-lineno">1292</tt> <a class="py-toggle" href="#" id="A3MAlignment.hmm_subregion-toggle" onclick="return toggle('A3MAlignment.hmm_subregion');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#hmm_subregion">hmm_subregion</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">match_start</tt><tt class="py-op">,</tt> <tt class="py-param">match_end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.hmm_subregion-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.hmm_subregion-expanded"><a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"><tt class="py-docstring">        Same as L{AbstractAlignment.subregion}, but start/end positions are</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line"><tt class="py-docstring">        ranks in the ungapped master sequence.</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line"> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-number">1</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">match_start</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">match_end</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-672" class="py-name" targets="Variable csb.bio.fragments.Target.matches=csb.bio.fragments.Target-class.html#matches,Variable csb.bio.sequence.A3MAlignment.matches=csb.bio.sequence.A3MAlignment-class.html#matches"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-672', 'matches', 'link-672');">matches</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-673" class="py-name"><a title="csb.bio.sequence.ColumnPositionError" class="py-name" href="#" onclick="return doclink('link-673', 'ColumnPositionError', 'link-174');">ColumnPositionError</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-674" class="py-name"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-674', 'matches', 'link-672');">matches</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">                 </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">        <tt id="link-675" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-675', 'start', 'link-126');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt><tt class="py-op">.</tt><tt id="link-676" class="py-name" targets="Method csb.bio.sequence.AlignedSequenceAdapter.map_residue()=csb.bio.sequence.AlignedSequenceAdapter-class.html#map_residue"><a title="csb.bio.sequence.AlignedSequenceAdapter.map_residue" class="py-name" href="#" onclick="return doclink('link-676', 'map_residue', 'link-676');">map_residue</a></tt><tt class="py-op">(</tt><tt class="py-name">match_start</tt><tt class="py-op">)</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">        <tt id="link-677" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-677', 'end', 'link-125');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_master</tt><tt class="py-op">.</tt><tt id="link-678" class="py-name"><a title="csb.bio.sequence.AlignedSequenceAdapter.map_residue" class="py-name" href="#" onclick="return doclink('link-678', 'map_residue', 'link-676');">map_residue</a></tt><tt class="py-op">(</tt><tt class="py-name">match_end</tt><tt class="py-op">)</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">         </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-679" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.subregion
csb.bio.hmm.ProfileHMM.subregion
csb.bio.sequence.AbstractAlignment.subregion
csb.bio.sequence.AbstractSequence.subregion
csb.bio.sequence.Sequence.subregion
csb.bio.sequence.SequenceAdapter.subregion
csb.bio.structure.Chain.subregion
csb.bio.structure.SecondaryStructure.subregion" class="py-name" href="#" onclick="return doclink('link-679', 'subregion', 'link-213');">subregion</a></tt><tt class="py-op">(</tt><tt id="link-680" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-680', 'start', 'link-126');">start</a></tt><tt class="py-op">,</tt> <tt id="link-681" class="py-name"><a title="csb.bio.fragments.Assignment.end
csb.bio.fragments.FragmentMatch.end
csb.bio.fragments.rosetta.RosettaFragment.end
csb.bio.hmm.HHpredHit.end
csb.bio.hmm.ProfileHMM.end
csb.bio.sequence.alignment.AlignmentResult.end
csb.bio.structure.SecondaryStructure.end
csb.bio.structure.SecondaryStructureElement.end
csb.core.REMatchProxy.end" class="py-name" href="#" onclick="return doclink('link-681', 'end', 'link-125');">end</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line"> </tt>
<a name="A3MAlignment.format"></a><div id="A3MAlignment.format-def"><a name="L1306"></a><tt class="py-lineno">1306</tt> <a class="py-toggle" href="#" id="A3MAlignment.format-toggle" onclick="return toggle('A3MAlignment.format');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#format">format</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">=</tt><tt id="link-682" class="py-name"><a title="csb.bio.sequence.AlignmentFormats" class="py-name" href="#" onclick="return doclink('link-682', 'AlignmentFormats', 'link-598');">AlignmentFormats</a></tt><tt class="py-op">.</tt><tt id="link-683" class="py-name"><a title="csb.bio.sequence.AlignmentFormats.A3M" class="py-name" href="#" onclick="return doclink('link-683', 'A3M', 'link-5');">A3M</a></tt><tt class="py-op">,</tt> <tt class="py-param">headers</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.format-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.format-expanded"><a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-684" class="py-name"><a title="csb.bio.sequence.A3MAlignment" class="py-name" href="#" onclick="return doclink('link-684', 'A3MAlignment', 'link-649');">A3MAlignment</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-685" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-685', 'format', 'link-135');">format</a></tt><tt class="py-op">(</tt><tt id="link-686" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-686', 'format', 'link-135');">format</a></tt><tt class="py-op">,</tt> <tt id="link-687" class="py-name"><a title="csb.bio.io.fasta.OutputBuilder.headers" class="py-name" href="#" onclick="return doclink('link-687', 'headers', 'link-195');">headers</a></tt><tt class="py-op">)</tt>  </tt>
</div><a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">     </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="A3MAlignment.matches"></a><div id="A3MAlignment.matches-def"><a name="L1310"></a><tt class="py-lineno">1310</tt> <a class="py-toggle" href="#" id="A3MAlignment.matches-toggle" onclick="return toggle('A3MAlignment.matches');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#matches">matches</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.matches-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.matches-expanded"><a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line"><tt class="py-docstring">        Number of match states (residues in the ungapped master).</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_matches</tt> </tt>
</div><a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">     </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="A3MAlignment.parse"></a><div id="A3MAlignment.parse-def"><a name="L1318"></a><tt class="py-lineno">1318</tt> <a class="py-toggle" href="#" id="A3MAlignment.parse-toggle" onclick="return toggle('A3MAlignment.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.sequence.A3MAlignment-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">string</tt><tt class="py-op">,</tt> <tt class="py-param">strict</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="A3MAlignment.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="A3MAlignment.parse-expanded"><a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{A3MAlignment} from an A3M string.</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line"><tt class="py-docstring">        @param string: MSA-formatted string</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line"><tt class="py-docstring">        @type string: str</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line"><tt class="py-docstring">        @param strict: see L{AbstractAlignment}</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line"><tt class="py-docstring">        @type strict: bool</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{A3MAlignment}</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-688" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-688', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-689" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-689', 'bio', 'link-184');">bio</a></tt><tt class="py-op">.</tt><tt id="link-690" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-690', 'io', 'link-3');">io</a></tt><tt class="py-op">.</tt><tt id="link-691" class="py-name"><a title="csb.bio.io.fasta" class="py-name" href="#" onclick="return doclink('link-691', 'fasta', 'link-186');">fasta</a></tt> <tt class="py-keyword">import</tt> <tt id="link-692" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader" class="py-name" href="#" onclick="return doclink('link-692', 'SequenceAlignmentReader', 'link-628');">SequenceAlignmentReader</a></tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-693" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader" class="py-name" href="#" onclick="return doclink('link-693', 'SequenceAlignmentReader', 'link-628');">SequenceAlignmentReader</a></tt><tt class="py-op">(</tt><tt id="link-694" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-694', 'strict', 'link-498');">strict</a></tt><tt class="py-op">=</tt><tt id="link-695" class="py-name"><a title="csb.bio.io.fasta.SequenceAlignmentReader.strict
csb.bio.io.fasta.StructureAlignmentFactory.strict" class="py-name" href="#" onclick="return doclink('link-695', 'strict', 'link-498');">strict</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-696" class="py-name" targets="Method csb.bio.io.fasta.SequenceAlignmentReader.read_a3m()=csb.bio.io.fasta.SequenceAlignmentReader-class.html#read_a3m"><a title="csb.bio.io.fasta.SequenceAlignmentReader.read_a3m" class="py-name" href="#" onclick="return doclink('link-696', 'read_a3m', 'link-696');">read_a3m</a></tt><tt class="py-op">(</tt><tt class="py-name">string</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul  4 20:19:07 2017
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
