<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>csb.bio.io.wwpdb</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="csb-module.html">Package&nbsp;csb</a> ::
        <a href="csb.bio-module.html">Package&nbsp;bio</a> ::
        <a href="csb.bio.io-module.html">Package&nbsp;io</a> ::
        Module&nbsp;wwpdb
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="csb.bio.io.wwpdb-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="csb.bio.io.wwpdb-module.html" onclick="show_private();">Module csb.bio.io.wwpdb</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-docstring">PDB structure parsers, format builders and database providers.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-docstring">The most basic usage is:</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; parser = StructureParser('structure.pdb')</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; parser.parse_structure()</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;Structure&gt;     # a Structure object (model)</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">or if this is an NMR ensemble:</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; parser.parse_models()</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;Ensemble&gt;      # an Ensemble object (collection of alternative Structure-s)</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">This module introduces a family of PDB file parsers. The common interface of all</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">parsers is defined in L{AbstractStructureParser}. This class has several</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">implementations:</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">    - L{RegularStructureParser} - handles normal PDB files with SEQRES fields</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring">    - L{LegacyStructureParser} - reads structures from legacy or malformed PDB files,</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">      which are lacking SEQRES records (initializes all residues from the ATOMs instead)</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">    - L{PDBHeaderParser} - reads only the headers of the PDB files and produces structures</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">      without coordinates. Useful for reading metadata (e.g. accession numbers or just</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">      plain SEQRES sequences) with minimum overhead</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">      </tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">Unless you have a special reason, you should use the L{StructureParser} factory,</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">which returns a proper L{AbstractStructureParser} implementation, depending on the</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">input PDB file. If the input file looks like a regular PDB file, the factory</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">returns a L{RegularStructureParser}, otherwise it instantiates L{LegacyStructureParser}.</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">L{StructureParser} is in fact an alias for L{AbstractStructureParser.create_parser}.</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">Writing your own, customized PDB parser is easy. Suppose that you are trying to</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">parse a PDB-like file which misuses the charge column to store custom info. This</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">will certainly crash L{RegularStructureParser} (for good), but you can create your</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">own parser as a workaround. All you need to to is to override the virtual</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">C{_read_charge} hook method::</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">    class CustomParser(RegularStructureParser):</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">        def _read_charge(self, line):</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring">            try:</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">                return super(CustomParser, self)._read_charge(line)</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">            except StructureFormatError:</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">                return None</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">Another important abstraction in this module is L{StructureProvider}. It has several</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">implementations which can be used to retrieve PDB L{Structure}s from various sources:</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">file system directories, remote URLs, etc. You can easily create your own provider as</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">well. See L{StructureProvider} for details.</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">Finally, this module gives you some L{FileBuilder}s, used for text serialization </tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">of L{Structure}s and L{Ensemble}s:</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; builder = PDBFileBuilder(stream)</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; builder.add_header(structure)</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; builder.add_structure(structure)</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">where stream is any Python stream, e.g. an open file or sys.stdout.</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">See L{Ensemble} and L{Structure} from L{csb.bio.structure} for details on these</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">objects.</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">numpy</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">datetime</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">multiprocessing</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Package csb=csb-module.html"><a title="csb" class="py-name" href="#" onclick="return doclink('link-0', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package csb.bio=csb.bio-module.html"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-1', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Method csb.bio.fragments.BenchmarkAdapter.structure()=csb.bio.fragments.BenchmarkAdapter-class.html#structure,Method csb.bio.hmm.ProfileHMM.structure()=csb.bio.hmm.ProfileHMM-class.html#structure,Package csb.bio.structure=csb.bio.structure-module.html"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-2', 'structure', 'link-2');">structure</a></tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-3" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-3', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-4', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Variable csb.bio.fragments.Assignment.sequence=csb.bio.fragments.Assignment-class.html#sequence,Variable csb.bio.fragments.Target.sequence=csb.bio.fragments.Target-class.html#sequence,Package csb.bio.sequence=csb.bio.sequence-module.html,Variable csb.bio.sequence.AbstractSequence.sequence=csb.bio.sequence.AbstractSequence-class.html#sequence,Variable csb.bio.sequence.Sequence.sequence=csb.bio.sequence.Sequence-class.html#sequence,Variable csb.bio.sequence.SequenceAdapter.sequence=csb.bio.sequence.SequenceAdapter-class.html#sequence,Variable csb.bio.structure.Chain.sequence=csb.bio.structure.Chain-class.html#sequence"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-5', 'sequence', 'link-5');">sequence</a></tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-6" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-6', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-7', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-8', 'sequence', 'link-5');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Variable csb.bio.hmm.HHpredHit.alignment=csb.bio.hmm.HHpredHit-class.html#alignment,Variable csb.bio.hmm.ProfileHMM.alignment=csb.bio.hmm.ProfileHMM-class.html#alignment,Module csb.bio.sequence.alignment=csb.bio.sequence.alignment-module.html,Method csb.bio.sequence.alignment.AlignmentResult.alignment()=csb.bio.sequence.alignment.AlignmentResult-class.html#alignment"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-9', 'alignment', 'link-9');">alignment</a></tt> <tt class="py-keyword">as</tt> <tt id="link-10" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-10', 'alignment', 'link-9');">alignment</a></tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-11" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-11', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package csb.core=csb.core-module.html"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-12', 'core', 'link-12');">core</a></tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-13" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-13', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Package csb.bio.io=csb.bio.io-module.html,Package csb.io=csb.io-module.html"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-14', 'io', 'link-14');">io</a></tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">abc</tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Class abc.ABCMeta=abc.ABCMeta-class.html"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-15', 'ABCMeta', 'link-15');">ABCMeta</a></tt><tt class="py-op">,</tt> <tt class="py-name">abstractmethod</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-16" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-16', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-17', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="csb.bio.fragments.Assignment.sequence
csb.bio.fragments.Target.sequence
csb.bio.sequence
csb.bio.sequence.AbstractSequence.sequence
csb.bio.sequence.Sequence.sequence
csb.bio.sequence.SequenceAdapter.sequence
csb.bio.structure.Chain.sequence" class="py-name" href="#" onclick="return doclink('link-18', 'sequence', 'link-5');">sequence</a></tt> <tt class="py-keyword">import</tt> <tt id="link-19" class="py-name" targets="Class csb.bio.sequence.SequenceTypes=csb.bio.sequence.SequenceTypes-class.html"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-19', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name" targets="Class csb.bio.sequence.SequenceAlphabets=csb.bio.sequence.SequenceAlphabets-class.html"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-20', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-21" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-21', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-22', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-23', 'structure', 'link-2');">structure</a></tt> <tt class="py-keyword">import</tt> <tt id="link-24" class="py-name" targets="Class csb.bio.structure.ChemElements=csb.bio.structure.ChemElements-class.html"><a title="csb.bio.structure.ChemElements" class="py-name" href="#" onclick="return doclink('link-24', 'ChemElements', 'link-24');">ChemElements</a></tt><tt class="py-op">,</tt> <tt id="link-25" class="py-name" targets="Class csb.bio.structure.SecStructures=csb.bio.structure.SecStructures-class.html"><a title="csb.bio.structure.SecStructures" class="py-name" href="#" onclick="return doclink('link-25', 'SecStructures', 'link-25');">SecStructures</a></tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt id="link-26" class="py-name" targets="Variable csb.bio.io.wwpdb.PDB_AMINOACIDS=csb.bio.io.wwpdb-module.html#PDB_AMINOACIDS"><a title="csb.bio.io.wwpdb.PDB_AMINOACIDS" class="py-name" href="#" onclick="return doclink('link-26', 'PDB_AMINOACIDS', 'link-26');">PDB_AMINOACIDS</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">    <tt class="py-string">'PAQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'AGM'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'ILE'</tt><tt class="py-op">:</tt> <tt class="py-string">'ILE'</tt><tt class="py-op">,</tt> <tt class="py-string">'PR3'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLN'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLN'</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line">    <tt class="py-string">'DVA'</tt><tt class="py-op">:</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">,</tt> <tt class="py-string">'CCS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'ACL'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLX'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLX'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line">    <tt class="py-string">'GLZ'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> <tt class="py-string">'DTH'</tt><tt class="py-op">:</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'OAS'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'C6C'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'NEM'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line">    <tt class="py-string">'DLY'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'MIS'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'SMC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'NEP'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line">    <tt class="py-string">'BCS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'SCY'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'SER'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line">    <tt class="py-string">'LYS'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'SAC'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'PRO'</tt><tt class="py-op">:</tt> <tt class="py-string">'PRO'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASX'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASX'</tt><tt class="py-op">,</tt> <tt class="py-string">'DGN'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLN'</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">    <tt class="py-string">'DGL'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'MHS'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASB'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'NLE'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">    <tt class="py-string">'DCY'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASK'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'GGL'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'STY'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'SEL'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">    <tt class="py-string">'CGU'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASN'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASN'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASL'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'LTR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'DAR'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">    <tt class="py-string">'VAL'</tt><tt class="py-op">:</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">,</tt> <tt class="py-string">'CHG'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'TPO'</tt><tt class="py-op">:</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'CLE'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'GMA'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line">    <tt class="py-string">'HAC'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'AYA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'THR'</tt><tt class="py-op">:</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'TIH'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'SVA'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line">    <tt class="py-string">'MVA'</tt><tt class="py-op">:</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">,</tt> <tt class="py-string">'SAR'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> <tt class="py-string">'LYZ'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'BNN'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'5HP'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">    <tt class="py-string">'IIL'</tt><tt class="py-op">:</tt> <tt class="py-string">'ILE'</tt><tt class="py-op">,</tt> <tt class="py-string">'SHR'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HAR'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'FME'</tt><tt class="py-op">:</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> <tt class="py-string">'ALO'</tt><tt class="py-op">:</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">    <tt class="py-string">'PHI'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'ALM'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'PHL'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'MEN'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASN'</tt><tt class="py-op">,</tt> <tt class="py-string">'TPQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line">    <tt class="py-string">'GSC'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'MAA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'MET'</tt><tt class="py-op">:</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line">    <tt class="py-string">'UNK'</tt><tt class="py-op">:</tt> <tt class="py-string">'UNK'</tt><tt class="py-op">,</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'ALY'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'SET'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'GL3'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">    <tt class="py-string">'TRG'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CXM'</tt><tt class="py-op">:</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'SCS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'DIL'</tt><tt class="py-op">:</tt> <tt class="py-string">'ILE'</tt><tt class="py-op">,</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">    <tt class="py-string">'TYQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'3AH'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'DPR'</tt><tt class="py-op">:</tt> <tt class="py-string">'PRO'</tt><tt class="py-op">,</tt> <tt class="py-string">'PRR'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'CME'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line">    <tt class="py-string">'IYR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'CY1'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYY'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'HYP'</tt><tt class="py-op">:</tt> <tt class="py-string">'PRO'</tt><tt class="py-op">,</tt> <tt class="py-string">'DTY'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line">    <tt class="py-string">'2AS'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'DTR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'FLA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'DPN'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'DIV'</tt><tt class="py-op">:</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">    <tt class="py-string">'PCA'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'MSE'</tt><tt class="py-op">:</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> <tt class="py-string">'MSA'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> <tt class="py-string">'AIB'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">    <tt class="py-string">'NLP'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'DLE'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'CEA'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line">    <tt class="py-string">'DAL'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'LLP'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'DAH'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HMR'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'TRO'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">    <tt class="py-string">'HIC'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYG'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'BMT'</tt><tt class="py-op">:</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'DAS'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYB'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">    <tt class="py-string">'BUC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'PEC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'BUG'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYM'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'NLN'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">    <tt class="py-string">'CY3'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSO'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'TPL'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'LYM'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">    <tt class="py-string">'DHI'</tt><tt class="py-op">:</tt> <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'MLE'</tt><tt class="py-op">:</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSD'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'HPQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'MPQ'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line">    <tt class="py-string">'LLY'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'DHA'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'DSN'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'SOC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSX'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line">    <tt class="py-string">'OMT'</tt><tt class="py-op">:</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> <tt class="py-string">'DSP'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'PTR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSW'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line">    <tt class="py-string">'EFC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSP'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CSS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'SCH'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'OCS'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">    <tt class="py-string">'NMC'</tt><tt class="py-op">:</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> <tt class="py-string">'SEP'</tt><tt class="py-op">:</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'BHD'</tt><tt class="py-op">:</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'KCX'</tt><tt class="py-op">:</tt> <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'SHC'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">    <tt class="py-string">'C5C'</tt><tt class="py-op">:</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HTR'</tt><tt class="py-op">:</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYS'</tt><tt class="py-op">:</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'ARM'</tt><tt class="py-op">:</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line">    <tt class="py-string">'DNP'</tt><tt class="py-op">:</tt> <tt class="py-string">'ALA'</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">    <tt class="py-op">}</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-string">Dictionary of non-standard amino acids, which could be found in PDB.</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt id="link-27" class="py-name" targets="Variable csb.bio.io.wwpdb.PDB_NUCLEOTIDES=csb.bio.io.wwpdb-module.html#PDB_NUCLEOTIDES"><a title="csb.bio.io.wwpdb.PDB_NUCLEOTIDES" class="py-name" href="#" onclick="return doclink('link-27', 'PDB_NUCLEOTIDES', 'link-27');">PDB_NUCLEOTIDES</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">    <tt class="py-string">'DA'</tt><tt class="py-op">:</tt> <tt class="py-string">'Adenine'</tt><tt class="py-op">,</tt> <tt class="py-string">'DG'</tt><tt class="py-op">:</tt> <tt class="py-string">'Guanine'</tt><tt class="py-op">,</tt> <tt class="py-string">'DC'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cytosine'</tt><tt class="py-op">,</tt> <tt class="py-string">'DT'</tt><tt class="py-op">:</tt> <tt class="py-string">'Thymine'</tt><tt class="py-op">,</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">     <tt class="py-string">'A'</tt><tt class="py-op">:</tt> <tt class="py-string">'Adenine'</tt><tt class="py-op">,</tt> <tt class="py-string">'G'</tt><tt class="py-op">:</tt> <tt class="py-string">'Guanine'</tt><tt class="py-op">,</tt> <tt class="py-string">'C'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cytosine'</tt><tt class="py-op">,</tt> <tt class="py-string">'T'</tt><tt class="py-op">:</tt> <tt class="py-string">'Thymine'</tt><tt class="py-op">,</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">     <tt class="py-string">'U'</tt><tt class="py-op">:</tt> <tt class="py-string">'Uracil'</tt><tt class="py-op">,</tt> <tt class="py-string">'DOC'</tt><tt class="py-op">:</tt> <tt class="py-string">'Cytosine'</tt><tt class="py-op">,</tt> <tt class="py-string">'R'</tt><tt class="py-op">:</tt> <tt class="py-string">'Purine'</tt><tt class="py-op">,</tt> <tt class="py-string">'Y'</tt><tt class="py-op">:</tt> <tt class="py-string">'Pyrimidine'</tt><tt class="py-op">,</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">     <tt class="py-string">'K'</tt><tt class="py-op">:</tt> <tt class="py-string">'Ketone'</tt><tt class="py-op">,</tt> <tt class="py-string">'  M'</tt><tt class="py-op">:</tt> <tt class="py-string">'Amino'</tt><tt class="py-op">,</tt> <tt class="py-string">'S'</tt><tt class="py-op">:</tt> <tt class="py-string">'Strong'</tt><tt class="py-op">,</tt> <tt class="py-string">'W'</tt><tt class="py-op">:</tt> <tt class="py-string">'Weak'</tt><tt class="py-op">,</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">     <tt class="py-string">'B'</tt><tt class="py-op">:</tt> <tt class="py-string">'NotA'</tt><tt class="py-op">,</tt> <tt class="py-string">'D'</tt>  <tt class="py-op">:</tt> <tt class="py-string">'NotC'</tt><tt class="py-op">,</tt> <tt class="py-string">'H'</tt><tt class="py-op">:</tt> <tt class="py-string">'NotG'</tt><tt class="py-op">,</tt> <tt class="py-string">'V'</tt><tt class="py-op">:</tt> <tt class="py-string">'NotT'</tt><tt class="py-op">,</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">     <tt class="py-string">'N'</tt><tt class="py-op">:</tt> <tt class="py-string">'Any'</tt><tt class="py-op">,</tt> <tt class="py-string">'X'</tt>  <tt class="py-op">:</tt> <tt class="py-string">'Masked'</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">    <tt class="py-op">}</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"><tt class="py-string">Dictionary of non-standard nucleotides, which could be found in PDB.</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="PDBParseError"></a><div id="PDBParseError-def"><a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt> <a class="py-toggle" href="#" id="PDBParseError-toggle" onclick="return toggle('PDBParseError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBParseError-class.html">PDBParseError</a><tt class="py-op">(</tt><tt class="py-base-class">ValueError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBParseError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PDBParseError-expanded"><a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"> </tt>
<a name="HeaderFormatError"></a><div id="HeaderFormatError-def"><a name="L136"></a><tt class="py-lineno"> 136</tt> <a class="py-toggle" href="#" id="HeaderFormatError-toggle" onclick="return toggle('HeaderFormatError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.HeaderFormatError-class.html">HeaderFormatError</a><tt class="py-op">(</tt><tt class="py-base-class">PDBParseError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HeaderFormatError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="HeaderFormatError-expanded"><a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"> </tt>
<a name="SecStructureFormatError"></a><div id="SecStructureFormatError-def"><a name="L139"></a><tt class="py-lineno"> 139</tt> <a class="py-toggle" href="#" id="SecStructureFormatError-toggle" onclick="return toggle('SecStructureFormatError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SecStructureFormatError-class.html">SecStructureFormatError</a><tt class="py-op">(</tt><tt class="py-base-class">PDBParseError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SecStructureFormatError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SecStructureFormatError-expanded"><a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"> </tt>
<a name="StructureFormatError"></a><div id="StructureFormatError-def"><a name="L142"></a><tt class="py-lineno"> 142</tt> <a class="py-toggle" href="#" id="StructureFormatError-toggle" onclick="return toggle('StructureFormatError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureFormatError-class.html">StructureFormatError</a><tt class="py-op">(</tt><tt class="py-base-class">PDBParseError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureFormatError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StructureFormatError-expanded"><a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"> </tt>
<a name="UnknownPDBResidueError"></a><div id="UnknownPDBResidueError-def"><a name="L145"></a><tt class="py-lineno"> 145</tt> <a class="py-toggle" href="#" id="UnknownPDBResidueError-toggle" onclick="return toggle('UnknownPDBResidueError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.UnknownPDBResidueError-class.html">UnknownPDBResidueError</a><tt class="py-op">(</tt><tt class="py-base-class">PDBParseError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="UnknownPDBResidueError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="UnknownPDBResidueError-expanded"><a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"> </tt>
<a name="StructureNotFoundError"></a><div id="StructureNotFoundError-def"><a name="L148"></a><tt class="py-lineno"> 148</tt> <a class="py-toggle" href="#" id="StructureNotFoundError-toggle" onclick="return toggle('StructureNotFoundError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureNotFoundError-class.html">StructureNotFoundError</a><tt class="py-op">(</tt><tt class="py-base-class">KeyError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureNotFoundError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StructureNotFoundError-expanded"><a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"> </tt>
<a name="InvalidEntryIDError"></a><div id="InvalidEntryIDError-def"><a name="L151"></a><tt class="py-lineno"> 151</tt> <a class="py-toggle" href="#" id="InvalidEntryIDError-toggle" onclick="return toggle('InvalidEntryIDError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.InvalidEntryIDError-class.html">InvalidEntryIDError</a><tt class="py-op">(</tt><tt class="py-base-class">StructureFormatError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="InvalidEntryIDError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="InvalidEntryIDError-expanded"><a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"> </tt>
<a name="ResidueMappingError"></a><div id="ResidueMappingError-def"><a name="L154"></a><tt class="py-lineno"> 154</tt> <a class="py-toggle" href="#" id="ResidueMappingError-toggle" onclick="return toggle('ResidueMappingError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.ResidueMappingError-class.html">ResidueMappingError</a><tt class="py-op">(</tt><tt class="py-base-class">StructureFormatError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueMappingError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ResidueMappingError-expanded"><a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"> </tt>
<a name="EntryID"></a><div id="EntryID-def"><a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt> <a class="py-toggle" href="#" id="EntryID-toggle" onclick="return toggle('EntryID');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html">EntryID</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="EntryID-expanded"><a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">    Represents a PDB Chain identifier. Implementing classes must define</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">    how the original ID is split into accession number and chain ID.</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">    @param id: identifier</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">    @type id: str</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-28" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-28', 'ABCMeta', 'link-15');">ABCMeta</a></tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"> </tt>
<a name="EntryID.__init__"></a><div id="EntryID.__init__-def"><a name="L168"></a><tt class="py-lineno"> 168</tt> <a class="py-toggle" href="#" id="EntryID.__init__-toggle" onclick="return toggle('EntryID.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.__init__-expanded"><a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">         </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_accession</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_chain</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">         </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">        <tt id="link-29" class="py-name" targets="Variable csb.bio.fragments.FragmentMatch.id=csb.bio.fragments.FragmentMatch-class.html#id,Variable csb.bio.fragments.Target.id=csb.bio.fragments.Target-class.html#id,Variable csb.bio.fragments.rosetta.RosettaFragment.id=csb.bio.fragments.rosetta.RosettaFragment-class.html#id,Variable csb.bio.hmm.HHpredHit.id=csb.bio.hmm.HHpredHit-class.html#id,Variable csb.bio.hmm.ProfileHMM.id=csb.bio.hmm.ProfileHMM-class.html#id,Variable csb.bio.io.wwpdb.ResidueInfo.id=csb.bio.io.wwpdb.ResidueInfo-class.html#id,Variable csb.bio.sequence.AbstractSequence.id=csb.bio.sequence.AbstractSequence-class.html#id,Variable csb.bio.sequence.SequenceAdapter.id=csb.bio.sequence.SequenceAdapter-class.html#id,Variable csb.bio.structure.Chain.id=csb.bio.structure.Chain-class.html#id,Variable csb.bio.structure.Residue.id=csb.bio.structure.Residue-class.html#id"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-29', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-30" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-30', 'id', 'link-29');">id</a></tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method csb.bio.sequence.AbstractSequence.strip()=csb.bio.sequence.AbstractSequence-class.html#strip,Method csb.bio.sequence.Sequence.strip()=csb.bio.sequence.Sequence-class.html#strip,Method csb.bio.sequence.SequenceAdapter.strip()=csb.bio.sequence.SequenceAdapter-class.html#strip"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-31', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>         </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_accession</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_chain</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method csb.apps.ArgHandler.parse()=csb.apps.ArgHandler-class.html#parse,Method csb.bio.io.dssp.DSSPParser.parse()=csb.bio.io.dssp.DSSPParser-class.html#parse,Method csb.bio.io.dssp.StrideParser.parse()=csb.bio.io.dssp.StrideParser-class.html#parse,Method csb.bio.io.hhpred.HHProfileParser.parse()=csb.bio.io.hhpred.HHProfileParser-class.html#parse,Method csb.bio.io.procheck.ProcheckParser.parse()=csb.bio.io.procheck.ProcheckParser-class.html#parse,Method csb.bio.io.psipred.PSIPredParser.parse()=csb.bio.io.psipred.PSIPredParser-class.html#parse,Method csb.bio.io.vasco.VascoStructureParser.parse()=csb.bio.io.vasco.VascoStructureParser-class.html#parse,Method csb.bio.io.whatif.WhatCheckParser.parse()=csb.bio.io.whatif.WhatCheckParser-class.html#parse,Method csb.bio.io.wwpdb.AbstractStructureParser.parse()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse,Method csb.bio.io.wwpdb.DegenerateID.parse()=csb.bio.io.wwpdb.DegenerateID-class.html#parse,Method csb.bio.io.wwpdb.EntryID.parse()=csb.bio.io.wwpdb.EntryID-class.html#parse,Method csb.bio.io.wwpdb.SeqResID.parse()=csb.bio.io.wwpdb.SeqResID-class.html#parse,Method csb.bio.io.wwpdb.StandardID.parse()=csb.bio.io.wwpdb.StandardID-class.html#parse,Static Method csb.bio.nmr.Label.parse()=csb.bio.nmr.Label-class.html#parse,Static Method csb.bio.sequence.A3MAlignment.parse()=csb.bio.sequence.A3MAlignment-class.html#parse,Static Method csb.bio.sequence.SequenceAlignment.parse()=csb.bio.sequence.SequenceAlignment-class.html#parse,Static Method csb.bio.sequence.StructureAlignment.parse()=csb.bio.sequence.StructureAlignment-class.html#parse,Static Method csb.bio.sequence.alignment.SimilarityMatrix.parse()=csb.bio.sequence.alignment.SimilarityMatrix-class.html#parse,Static Method csb.bio.structure.SecondaryStructure.parse()=csb.bio.structure.SecondaryStructure-class.html#parse,Static Method csb.core.Enum.parse()=csb.core.Enum-class.html#parse"><a title="csb.apps.ArgHandler.parse
csb.bio.io.dssp.DSSPParser.parse
csb.bio.io.dssp.StrideParser.parse
csb.bio.io.hhpred.HHProfileParser.parse
csb.bio.io.procheck.ProcheckParser.parse
csb.bio.io.psipred.PSIPredParser.parse
csb.bio.io.vasco.VascoStructureParser.parse
csb.bio.io.whatif.WhatCheckParser.parse
csb.bio.io.wwpdb.AbstractStructureParser.parse
csb.bio.io.wwpdb.DegenerateID.parse
csb.bio.io.wwpdb.EntryID.parse
csb.bio.io.wwpdb.SeqResID.parse
csb.bio.io.wwpdb.StandardID.parse
csb.bio.nmr.Label.parse
csb.bio.sequence.A3MAlignment.parse
csb.bio.sequence.SequenceAlignment.parse
csb.bio.sequence.StructureAlignment.parse
csb.bio.sequence.alignment.SimilarityMatrix.parse
csb.bio.structure.SecondaryStructure.parse
csb.core.Enum.parse" class="py-name" href="#" onclick="return doclink('link-32', 'parse', 'link-32');">parse</a></tt><tt class="py-op">(</tt><tt id="link-33" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-33', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">     </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="EntryID.create"></a><div id="EntryID.create-def"><a name="L177"></a><tt class="py-lineno"> 177</tt> <a class="py-toggle" href="#" id="EntryID.create-toggle" onclick="return toggle('EntryID.create');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#create">create</a><tt class="py-op">(</tt><tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.create-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.create-expanded"><a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"><tt class="py-docstring">        Guess the format of C{id} and parse it. </tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new PDB ID of the appropriate type</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{EntryID}</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">         </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-34', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-number">5</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-35" class="py-name" targets="Class csb.bio.io.wwpdb.StandardID=csb.bio.io.wwpdb.StandardID-class.html"><a title="csb.bio.io.wwpdb.StandardID" class="py-name" href="#" onclick="return doclink('link-35', 'StandardID', 'link-35');">StandardID</a></tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-36', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-37', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">6</tt> <tt class="py-keyword">and</tt> <tt id="link-38" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-38', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'_'</tt><tt class="py-op">:</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-39" class="py-name" targets="Class csb.bio.io.wwpdb.SeqResID=csb.bio.io.wwpdb.SeqResID-class.html"><a title="csb.bio.io.wwpdb.SeqResID" class="py-name" href="#" onclick="return doclink('link-39', 'SeqResID', 'link-39');">SeqResID</a></tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-40', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-41" class="py-name" targets="Class csb.bio.io.wwpdb.DegenerateID=csb.bio.io.wwpdb.DegenerateID-class.html"><a title="csb.bio.io.wwpdb.DegenerateID" class="py-name" href="#" onclick="return doclink('link-41', 'DegenerateID', 'link-41');">DegenerateID</a></tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-42', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">     </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="EntryID.parse"></a><div id="EntryID.parse-def"><a name="L193"></a><tt class="py-lineno"> 193</tt> <a class="py-toggle" href="#" id="EntryID.parse-toggle" onclick="return toggle('EntryID.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.parse-expanded"><a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring">        Split C{id} into accession number and chain ID.</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring">        @param id: PDB identifier</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">        @type id: str</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">        @return: (accession, chain)</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: tuple of str</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">        @raise InvalidEntryIDError: when C{id} is in an inappropriate format</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">     </tt>
<a name="EntryID.format"></a><div id="EntryID.format-def"><a name="L206"></a><tt class="py-lineno"> 206</tt> <a class="py-toggle" href="#" id="EntryID.format-toggle" onclick="return toggle('EntryID.format');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#format">format</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.format-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.format-expanded"><a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">        @return: the identifier in its original format</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Variable csb.bio.io.wwpdb.EntryID.entry_id=csb.bio.io.wwpdb.EntryID-class.html#entry_id,Variable csb.bio.structure.Chain.entry_id=csb.bio.structure.Chain-class.html#entry_id"><a title="csb.bio.io.wwpdb.EntryID.entry_id
csb.bio.structure.Chain.entry_id" class="py-name" href="#" onclick="return doclink('link-43', 'entry_id', 'link-43');">entry_id</a></tt> </tt>
</div><a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">         </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="EntryID.accession"></a><div id="EntryID.accession-def"><a name="L214"></a><tt class="py-lineno"> 214</tt> <a class="py-toggle" href="#" id="EntryID.accession-toggle" onclick="return toggle('EntryID.accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#accession">accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.accession-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.accession-expanded"><a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring">        Accession number part of the Entry ID</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_accession</tt> </tt>
</div><a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="EntryID.chain"></a><div id="EntryID.chain-def"><a name="L222"></a><tt class="py-lineno"> 222</tt> <a class="py-toggle" href="#" id="EntryID.chain-toggle" onclick="return toggle('EntryID.chain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#chain">chain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.chain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.chain-expanded"><a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring">        Chain ID part of the Entry ID</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str      </tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_chain</tt> </tt>
</div><a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">     </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="EntryID.entry_id"></a><div id="EntryID.entry_id-def"><a name="L230"></a><tt class="py-lineno"> 230</tt> <a class="py-toggle" href="#" id="EntryID.entry_id-toggle" onclick="return toggle('EntryID.entry_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#entry_id">entry_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.entry_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.entry_id-expanded"><a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">        Accession number + Chain ID</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"{0.accession}{0.chain}"</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method csb.bio.io.wwpdb.EntryID.format()=csb.bio.io.wwpdb.EntryID-class.html#format,Method csb.bio.io.wwpdb.SeqResID.format()=csb.bio.io.wwpdb.SeqResID-class.html#format,Method csb.bio.sequence.A3MAlignment.format()=csb.bio.sequence.A3MAlignment-class.html#format,Method csb.bio.sequence.AbstractAlignment.format()=csb.bio.sequence.AbstractAlignment-class.html#format"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-44', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">     </tt>
<a name="EntryID.__str__"></a><div id="EntryID.__str__-def"><a name="L237"></a><tt class="py-lineno"> 237</tt> <a class="py-toggle" href="#" id="EntryID.__str__-toggle" onclick="return toggle('EntryID.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.EntryID-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="EntryID.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="EntryID.__str__-expanded"><a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.entry_id
csb.bio.structure.Chain.entry_id" class="py-name" href="#" onclick="return doclink('link-45', 'entry_id', 'link-43');">entry_id</a></tt> </tt>
</div></div><a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">     </tt>
<a name="StandardID"></a><div id="StandardID-def"><a name="L240"></a><tt class="py-lineno"> 240</tt> <a class="py-toggle" href="#" id="StandardID-toggle" onclick="return toggle('StandardID');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StandardID-class.html">StandardID</a><tt class="py-op">(</tt><tt class="py-base-class">EntryID</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StandardID-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StandardID-expanded"><a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-docstring">    Standard PDB ID in the following form: xxxxY, where xxxx is the accession</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-docstring">    number (lower case) and Y is an optional chain identifier.</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>     </tt>
<a name="StandardID.parse"></a><div id="StandardID.parse-def"><a name="L245"></a><tt class="py-lineno"> 245</tt> <a class="py-toggle" href="#" id="StandardID.parse-toggle" onclick="return toggle('StandardID.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StandardID-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StandardID.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StandardID.parse-expanded"><a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">         </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-46" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-46', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-number">5</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-47" class="py-name" targets="Class csb.bio.io.wwpdb.InvalidEntryIDError=csb.bio.io.wwpdb.InvalidEntryIDError-class.html"><a title="csb.bio.io.wwpdb.InvalidEntryIDError" class="py-name" href="#" onclick="return doclink('link-47', 'InvalidEntryIDError', 'link-47');">InvalidEntryIDError</a></tt><tt class="py-op">(</tt><tt id="link-48" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-48', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">         </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-49" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-49', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method csb.io.tsv.Where.lower()=csb.io.tsv.Where-class.html#lower"><a title="csb.io.tsv.Where.lower" class="py-name" href="#" onclick="return doclink('link-50', 'lower', 'link-50');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-51" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-51', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">         </tt>
<a name="DegenerateID"></a><div id="DegenerateID-def"><a name="L252"></a><tt class="py-lineno"> 252</tt> <a class="py-toggle" href="#" id="DegenerateID-toggle" onclick="return toggle('DegenerateID');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.DegenerateID-class.html">DegenerateID</a><tt class="py-op">(</tt><tt class="py-base-class">EntryID</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DegenerateID-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DegenerateID-expanded"><a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring">    Looks like a L{StandardID}, except that the accession number may have</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring">    arbitrary length.</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>     </tt>
<a name="DegenerateID.parse"></a><div id="DegenerateID.parse-def"><a name="L257"></a><tt class="py-lineno"> 257</tt> <a class="py-toggle" href="#" id="DegenerateID.parse-toggle" onclick="return toggle('DegenerateID.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.DegenerateID-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DegenerateID.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DegenerateID.parse-expanded"><a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-52" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-52', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-53" class="py-name"><a title="csb.bio.io.wwpdb.InvalidEntryIDError" class="py-name" href="#" onclick="return doclink('link-53', 'InvalidEntryIDError', 'link-47');">InvalidEntryIDError</a></tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-54', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">         </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-55" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-55', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="csb.io.tsv.Where.lower" class="py-name" href="#" onclick="return doclink('link-56', 'lower', 'link-50');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-57" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-57', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">         </tt>
<a name="SeqResID"></a><div id="SeqResID-def"><a name="L264"></a><tt class="py-lineno"> 264</tt> <a class="py-toggle" href="#" id="SeqResID-toggle" onclick="return toggle('SeqResID');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SeqResID-class.html">SeqResID</a><tt class="py-op">(</tt><tt class="py-base-class">EntryID</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SeqResID-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SeqResID-expanded"><a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"><tt class="py-docstring">    Same as a L{StandardID}, but contains an additional underscore between</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"><tt class="py-docstring">    te accession number and the chain identifier.</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt>       </tt>
<a name="SeqResID.parse"></a><div id="SeqResID.parse-def"><a name="L269"></a><tt class="py-lineno"> 269</tt> <a class="py-toggle" href="#" id="SeqResID.parse-toggle" onclick="return toggle('SeqResID.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SeqResID-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SeqResID.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SeqResID.parse-expanded"><a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">         </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-58" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-58', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">6</tt> <tt class="py-keyword">and</tt> <tt id="link-59" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-59', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'_'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-60" class="py-name"><a title="csb.bio.io.wwpdb.InvalidEntryIDError" class="py-name" href="#" onclick="return doclink('link-60', 'InvalidEntryIDError', 'link-47');">InvalidEntryIDError</a></tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-61', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">         </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-62', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="csb.io.tsv.Where.lower" class="py-name" href="#" onclick="return doclink('link-63', 'lower', 'link-50');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-64" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-64', 'id', 'link-29');">id</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">     </tt>
<a name="SeqResID.format"></a><div id="SeqResID.format-def"><a name="L276"></a><tt class="py-lineno"> 276</tt> <a class="py-toggle" href="#" id="SeqResID.format-toggle" onclick="return toggle('SeqResID.format');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SeqResID-class.html#format">format</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SeqResID.format-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SeqResID.format-expanded"><a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"{0.accession}_{0.chain}"</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-65', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser"></a><div id="AbstractStructureParser-def"><a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line"> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt> <a class="py-toggle" href="#" id="AbstractStructureParser-toggle" onclick="return toggle('AbstractStructureParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html">AbstractStructureParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AbstractStructureParser-expanded"><a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line"><tt class="py-docstring">    A base PDB structure format-aware parser. Subclasses must implement the</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"><tt class="py-docstring">    internal abstract method C{_parse_header} in order to complete the </tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line"><tt class="py-docstring">    implementation.</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"><tt class="py-docstring">    @param structure_file: the input PD file to parse</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line"><tt class="py-docstring">    @type structure_file: str</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line"><tt class="py-docstring">    @param check_ss: if True, secondary structure errors in the file will cause </tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring">                     L{SecStructureFormatError} exceptions</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">    @type check_ss: bool</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">    @param mapper: residue mapper, used to align ATOM records to SEQRES.</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">                   If None, use the default (L{CombinedResidueMapper})</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring">    @type mapper: L{AbstractResidueMapper}    </tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">    @raise IOError: when the input file cannot be found</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-66', 'ABCMeta', 'link-15');">ABCMeta</a></tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="AbstractStructureParser.create_parser"></a><div id="AbstractStructureParser.create_parser-def"><a name="L301"></a><tt class="py-lineno"> 301</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.create_parser-toggle" onclick="return toggle('AbstractStructureParser.create_parser');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#create_parser">create_parser</a><tt class="py-op">(</tt><tt class="py-param">structure_file</tt><tt class="py-op">,</tt> <tt class="py-param">check_ss</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">mapper</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.create_parser-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.create_parser-expanded"><a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">        A StructureParser factory, which instantiates and returns the proper parser</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring">        object based on the contents of the PDB file.</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">        If the file contains a SEQRES section, L{RegularStructureParser} is returned,</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">        otherwise L{LegacyStructureParser} is instantiated. In the latter case</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">        LegacyStructureParser will read the sequence data directly from the ATOMs.        </tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure_file: the PDB file to parse</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure_file: str</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">        @param check_ss: if True, secondary structure errors in the file will cause </tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">                         L{SecStructureFormatError} exceptions</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">        @type check_ss: bool</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">        @param mapper: residue mapper, used to align ATOM records to SEQRES.</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">                       If None, use the default (L{CombinedResidueMapper})</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">        @type mapper: L{AbstractResidueMapper}          </tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractStructureParser}</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">        <tt class="py-name">has_seqres</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">         </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">open</tt><tt class="py-op">(</tt><tt class="py-name">structure_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'SEQRES'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                <tt class="py-name">has_seqres</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">         </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">has_seqres</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-67" class="py-name" targets="Class csb.bio.io.wwpdb.RegularStructureParser=csb.bio.io.wwpdb.RegularStructureParser-class.html"><a title="csb.bio.io.wwpdb.RegularStructureParser" class="py-name" href="#" onclick="return doclink('link-67', 'RegularStructureParser', 'link-67');">RegularStructureParser</a></tt><tt class="py-op">(</tt><tt class="py-name">structure_file</tt><tt class="py-op">,</tt> <tt class="py-name">check_ss</tt><tt class="py-op">,</tt> <tt id="link-68" class="py-name" targets="Variable csb.bio.io.wwpdb.AbstractStructureParser.mapper=csb.bio.io.wwpdb.AbstractStructureParser-class.html#mapper"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-68', 'mapper', 'link-68');">mapper</a></tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-69" class="py-name" targets="Class csb.bio.io.wwpdb.LegacyStructureParser=csb.bio.io.wwpdb.LegacyStructureParser-class.html"><a title="csb.bio.io.wwpdb.LegacyStructureParser" class="py-name" href="#" onclick="return doclink('link-69', 'LegacyStructureParser', 'link-69');">LegacyStructureParser</a></tt><tt class="py-op">(</tt><tt class="py-name">structure_file</tt><tt class="py-op">,</tt> <tt class="py-name">check_ss</tt><tt class="py-op">,</tt> <tt id="link-70" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-70', 'mapper', 'link-68');">mapper</a></tt><tt class="py-op">)</tt>         </tt>
</div><a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.__init__"></a><div id="AbstractStructureParser.__init__-def"><a name="L333"></a><tt class="py-lineno"> 333</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.__init__-toggle" onclick="return toggle('AbstractStructureParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure_file</tt><tt class="py-op">,</tt> <tt class="py-param">check_ss</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">mapper</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.__init__-expanded"><a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_file</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mapper</tt> <tt class="py-op">=</tt> <tt id="link-71" class="py-name" targets="Class csb.bio.io.wwpdb.CombinedResidueMapper=csb.bio.io.wwpdb.CombinedResidueMapper-class.html"><a title="csb.bio.io.wwpdb.CombinedResidueMapper" class="py-name" href="#" onclick="return doclink('link-71', 'CombinedResidueMapper', 'link-71');">CombinedResidueMapper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt> <tt class="py-op">=</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt class="py-name">check_ss</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Variable csb.apps.embd.OutputPathBuilder.filename=csb.apps.embd.OutputPathBuilder-class.html#filename,Variable csb.bio.io.clans.Clans.filename=csb.bio.io.clans.Clans-class.html#filename,Variable csb.bio.io.mrc.DensityMapReader.filename=csb.bio.io.mrc.DensityMapReader-class.html#filename,Variable csb.bio.io.wwpdb.AbstractStructureParser.filename=csb.bio.io.wwpdb.AbstractStructureParser-class.html#filename"><a title="csb.apps.embd.OutputPathBuilder.filename
csb.bio.io.clans.Clans.filename
csb.bio.io.mrc.DensityMapReader.filename
csb.bio.io.wwpdb.AbstractStructureParser.filename" class="py-name" href="#" onclick="return doclink('link-72', 'filename', 'link-72');">filename</a></tt> <tt class="py-op">=</tt> <tt class="py-name">structure_file</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-73" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-73', 'mapper', 'link-68');">mapper</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-74', 'mapper', 'link-68');">mapper</a></tt> <tt class="py-op">=</tt> <tt id="link-75" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-75', 'mapper', 'link-68');">mapper</a></tt> </tt>
</div><a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.__del__"></a><div id="AbstractStructureParser.__del__-def"><a name="L344"></a><tt class="py-lineno"> 344</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.__del__-toggle" onclick="return toggle('AbstractStructureParser.__del__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#__del__">__del__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.__del__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.__del__-expanded"><a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method csb.io.EntryWriter.close()=csb.io.EntryWriter-class.html#close,Method csb.io.TempFile.close()=csb.io.TempFile-class.html#close,Method csb.io.TempFolder.close()=csb.io.TempFolder-class.html#close,Method csb.io.tsv.RepositoryImp.close()=csb.io.tsv.RepositoryImp-class.html#close,Method csb.io.tsv.SQLiteRepository.close()=csb.io.tsv.SQLiteRepository-class.html#close"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-76', 'close', 'link-76');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
</div><a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">         </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractStructureParser.mapper"></a><div id="AbstractStructureParser.mapper-def"><a name="L351"></a><tt class="py-lineno"> 351</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.mapper-toggle" onclick="return toggle('AbstractStructureParser.mapper');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#mapper">mapper</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.mapper-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.mapper-expanded"><a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-docstring">        Current residue mapping strategy</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AbstractResidueMapper}</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mapper</tt> </tt>
</div><a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">mapper</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="AbstractStructureParser.mapper"></a><div id="AbstractStructureParser.mapper-def"><a name="L358"></a><tt class="py-lineno"> 358</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.mapper-toggle" onclick="return toggle('AbstractStructureParser.mapper');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#mapper">mapper</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.mapper-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.mapper-expanded"><a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-77" class="py-name" targets="Variable csb.core.EnumItem.value=csb.core.EnumItem-class.html#value,Variable csb.statistics.pdf.parameterized.AbstractParameter.value=csb.statistics.pdf.parameterized.AbstractParameter-class.html#value"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-77', 'value', 'link-77');">value</a></tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name" targets="Class csb.bio.io.wwpdb.AbstractResidueMapper=csb.bio.io.wwpdb.AbstractResidueMapper-class.html"><a title="csb.bio.io.wwpdb.AbstractResidueMapper" class="py-name" href="#" onclick="return doclink('link-78', 'AbstractResidueMapper', 'link-78');">AbstractResidueMapper</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-79', 'value', 'link-77');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_mapper</tt> <tt class="py-op">=</tt> <tt id="link-80" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-80', 'value', 'link-77');">value</a></tt> </tt>
</div><a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="AbstractStructureParser.filename"></a><div id="AbstractStructureParser.filename-def"><a name="L364"></a><tt class="py-lineno"> 364</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.filename-toggle" onclick="return toggle('AbstractStructureParser.filename');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#filename">filename</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.filename-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.filename-expanded"><a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"><tt class="py-docstring">        Current input PDB file name</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_file</tt> </tt>
</div><a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">filename</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="AbstractStructureParser.filename"></a><div id="AbstractStructureParser.filename-def"><a name="L371"></a><tt class="py-lineno"> 371</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.filename-toggle" onclick="return toggle('AbstractStructureParser.filename');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#filename">filename</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.filename-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.filename-expanded"><a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">            <tt id="link-81" class="py-name" targets="Variable csb.io.EntryWriter.stream=csb.io.EntryWriter-class.html#stream"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-81', 'stream', 'link-81');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">open</tt><tt class="py-op">(</tt><tt id="link-82" class="py-name" targets="Variable csb.bio.hmm.ProfileHMM.name=csb.bio.hmm.ProfileHMM-class.html#name,Variable csb.bio.io.clans.ClansEntry.name=csb.bio.io.clans.ClansEntry-class.html#name,Variable csb.bio.io.clans.ClansSeqgroup.name=csb.bio.io.clans.ClansSeqgroup-class.html#name,Variable csb.bio.structure.Atom.name=csb.bio.structure.Atom-class.html#name,Variable csb.bio.structure.Chain.name=csb.bio.structure.Chain-class.html#name,Variable csb.core.EnumItem.name=csb.core.EnumItem-class.html#name,Variable csb.io.TempFile.name=csb.io.TempFile-class.html#name,Variable csb.io.TempFolder.name=csb.io.TempFolder-class.html#name,Variable csb.io.tsv.ColumnInfo.name=csb.io.tsv.ColumnInfo-class.html#name,Variable csb.io.tsv.Table.name=csb.io.tsv.Table-class.html#name,Variable csb.statistics.pdf.parameterized.AbstractParameter.name=csb.statistics.pdf.parameterized.AbstractParameter-class.html#name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-82', 'name', 'link-82');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IOError</tt><tt class="py-op">:</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IOError</tt><tt class="py-op">(</tt><tt class="py-string">'File not found: {0}'</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-83', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-84', 'name', 'link-82');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">         </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">:</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-85', 'close', 'link-76');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt> <tt class="py-op">=</tt> <tt id="link-86" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-86', 'stream', 'link-81');">stream</a></tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_file</tt> <tt class="py-op">=</tt> <tt id="link-87" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-87', 'name', 'link-82');">name</a></tt> </tt>
</div><a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.models"></a><div id="AbstractStructureParser.models-def"><a name="L385"></a><tt class="py-lineno"> 385</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.models-toggle" onclick="return toggle('AbstractStructureParser.models');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#models">models</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.models-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.models-expanded"><a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-docstring">        Find all available model identifiers in the structure.</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a list of model IDs</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: list</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">        <tt id="link-88" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.models()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#models,Variable csb.bio.structure.Ensemble.models=csb.bio.structure.Ensemble-class.html#models"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-88', 'models', 'link-88');">models</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">        <tt class="py-name">check</tt> <tt class="py-op">=</tt> <tt id="link-89" class="py-name" targets="Method csb.bio.hmm.EmissionTable.set()=csb.bio.hmm.EmissionTable-class.html#set,Method csb.bio.hmm.TransitionTable.set()=csb.bio.hmm.TransitionTable-class.html#set,Method csb.statistics.maxent.MaxentModel.set()=csb.statistics.maxent.MaxentModel-class.html#set,Method csb.statistics.pdf.parameterized.AbstractParameter.set()=csb.statistics.pdf.parameterized.AbstractParameter-class.html#set"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-89', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line"> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt class="py-name">open</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_file</tt><tt class="py-op">,</tt> <tt class="py-string">'r'</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">f</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">f</tt><tt class="py-op">:</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">                    <tt id="link-90" class="py-name" targets="Variable csb.bio.structure.Structure.model_id=csb.bio.structure.Structure-class.html#model_id"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-90', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-91" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-91', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">check</tt><tt class="py-op">:</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-92" class="py-name" targets="Class csb.bio.io.hhpred.StructureFormatError=csb.bio.io.hhpred.StructureFormatError-class.html,Class csb.bio.io.wwpdb.StructureFormatError=csb.bio.io.wwpdb.StructureFormatError-class.html"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-92', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Duplicate model identifier: {0}'</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-93', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-94" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-94', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">                    <tt id="link-95" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-95', 'models', 'link-88');">models</a></tt><tt class="py-op">.</tt><tt id="link-96" class="py-name" targets="Method csb.bio.fragments.rosetta.RosettaFragmentMap.append()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#append,Method csb.bio.hmm.EmissionTable.append()=csb.bio.hmm.EmissionTable-class.html#append,Method csb.bio.hmm.HMMLayer.append()=csb.bio.hmm.HMMLayer-class.html#append,Method csb.bio.hmm.TransitionTable.append()=csb.bio.hmm.TransitionTable-class.html#append,Method csb.bio.structure.ChainResiduesCollection.append()=csb.bio.structure.ChainResiduesCollection-class.html#append,Method csb.bio.structure.DisorderedAtom.append()=csb.bio.structure.DisorderedAtom-class.html#append,Method csb.bio.structure.EnsembleModelsCollection.append()=csb.bio.structure.EnsembleModelsCollection-class.html#append,Method csb.bio.structure.ResidueAtomsTable.append()=csb.bio.structure.ResidueAtomsTable-class.html#append,Method csb.bio.structure.SecondaryStructure.append()=csb.bio.structure.SecondaryStructure-class.html#append,Method csb.bio.structure.StructureChainsTable.append()=csb.bio.structure.StructureChainsTable-class.html#append,Method csb.core.CollectionContainer.append()=csb.core.CollectionContainer-class.html#append,Method csb.core.DictionaryContainer.append()=csb.core.DictionaryContainer-class.html#append"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-96', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-97" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-97', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">                    <tt class="py-name">check</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name" targets="Method csb.apps.csfrag.MatchTable.add()=csb.apps.csfrag.MatchTable-class.html#add,Method csb.apps.hhfrag.PredictionBuilder.add()=csb.apps.hhfrag.PredictionBuilder-class.html#add,Method csb.bio.io.clans.ClansSeqgroup.add()=csb.bio.io.clans.ClansSeqgroup-class.html#add,Method csb.bio.io.wwpdb.CustomStructureProvider.add()=csb.bio.io.wwpdb.CustomStructureProvider-class.html#add,Method csb.bio.io.wwpdb.FileSystemStructureProvider.add()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#add,Method csb.bio.nmr.NOESpectrum.add()=csb.bio.nmr.NOESpectrum-class.html#add,Method csb.bio.sequence.AbstractAlignment.add()=csb.bio.sequence.AbstractAlignment-class.html#add,Method csb.io.plots.Backend.add()=csb.io.plots.Backend-class.html#add"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-98', 'add', 'link-98');">add</a></tt><tt class="py-op">(</tt><tt id="link-99" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-99', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line"> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-100" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-100', 'models', 'link-88');">models</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt><tt class="py-op">(</tt><tt class="py-name">min</tt><tt class="py-op">(</tt><tt class="py-name">check</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">and</tt> <tt class="py-name">max</tt><tt class="py-op">(</tt><tt class="py-name">check</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-101', 'models', 'link-88');">models</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-102" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-102', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Non-consecutive model identifier(s) encountered'</tt><tt class="py-op">)</tt>                 </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-103" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-103', 'models', 'link-88');">models</a></tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser.guess_chain_type"></a><div id="AbstractStructureParser.guess_chain_type-def"><a name="L411"></a><tt class="py-lineno"> 411</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.guess_chain_type-toggle" onclick="return toggle('AbstractStructureParser.guess_chain_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#guess_chain_type">guess_chain_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_labels</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.guess_chain_type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.guess_chain_type-expanded"><a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-docstring">        Try to guess what is the sequence type of a chunk of PDB</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-docstring">        C{residue_label}s. The list of labels is probed starting from the middle</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-docstring">        first, because PDB chains often contain modified / unknown residues at</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">        the termini. If none of the probed residues can be used to determine</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line"><tt class="py-docstring">        chain's type, just give up and return L{SequenceTypes.Unknown}.</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line"><tt class="py-docstring">        @param residue_labels: an iterable of PDB residue labels</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"><tt class="py-docstring">        @type residue_labels: iterable </tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a L{SequenceTypes} enum member</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.core.EnumItem}              </tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">         </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">        <tt id="link-104" class="py-name" targets="Variable csb.test.AbstractTestBuilder.labels=csb.test.AbstractTestBuilder-class.html#labels,Variable csb.test.AnyTestBuilder.labels=csb.test.AnyTestBuilder-class.html#labels,Variable csb.test.CustomTestBuilder.labels=csb.test.CustomTestBuilder-class.html#labels,Variable csb.test.FunctionalTestBuilder.labels=csb.test.FunctionalTestBuilder-class.html#labels,Variable csb.test.RegressionTestBuilder.labels=csb.test.RegressionTestBuilder-class.html#labels,Variable csb.test.UnitTestBuilder.labels=csb.test.UnitTestBuilder-class.html#labels"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-104', 'labels', 'link-104');">labels</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">residue_labels</tt><tt class="py-op">)</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-name">middle</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-105" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-105', 'labels', 'link-104');">labels</a></tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">         </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-name">reordered</tt> <tt class="py-op">=</tt> <tt id="link-106" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-106', 'labels', 'link-104');">labels</a></tt><tt class="py-op">[</tt><tt class="py-name">middle</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">reversed</tt><tt class="py-op">(</tt><tt id="link-107" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-107', 'labels', 'link-104');">labels</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">middle</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">         </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-108" class="py-name" targets="Variable csb.bio.io.wwpdb.ResidueInfo.label=csb.bio.io.wwpdb.ResidueInfo-class.html#label,Variable csb.bio.nmr.ChemShiftInfo.label=csb.bio.nmr.ChemShiftInfo-class.html#label,Variable csb.bio.structure.Residue.label=csb.bio.structure.Residue-class.html#label"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-108', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">reordered</tt><tt class="py-op">:</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">                <tt id="link-109" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.type=csb.bio.fragments.ResidueEventInfo-class.html#type,Variable csb.bio.fragments.TargetResidue.type=csb.bio.fragments.TargetResidue-class.html#type,Variable csb.bio.hmm.State.type=csb.bio.hmm.State-class.html#type,Variable csb.bio.hmm.Transition.type=csb.bio.hmm.Transition-class.html#type,Variable csb.bio.io.clans.ClansSeqgroup.type=csb.bio.io.clans.ClansSeqgroup-class.html#type,Variable csb.bio.io.wwpdb.ResidueInfo.type=csb.bio.io.wwpdb.ResidueInfo-class.html#type,Variable csb.bio.sequence.AbstractSequence.type=csb.bio.sequence.AbstractSequence-class.html#type,Variable csb.bio.sequence.ResidueInfo.type=csb.bio.sequence.ResidueInfo-class.html#type,Variable csb.bio.sequence.SequenceAdapter.type=csb.bio.sequence.SequenceAdapter-class.html#type,Variable csb.bio.structure.Chain.type=csb.bio.structure.Chain-class.html#type,Variable csb.bio.structure.Residue.type=csb.bio.structure.Residue-class.html#type,Variable csb.bio.structure.SecondaryStructureElement.type=csb.bio.structure.SecondaryStructureElement-class.html#type,Variable csb.io.tsv.ColumnInfo.type=csb.io.tsv.ColumnInfo-class.html#type"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-109', 'type', 'link-109');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.guess_sequence_type()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#guess_sequence_type"><a title="csb.bio.io.wwpdb.AbstractStructureParser.guess_sequence_type" class="py-name" href="#" onclick="return doclink('link-110', 'guess_sequence_type', 'link-110');">guess_sequence_type</a></tt><tt class="py-op">(</tt><tt id="link-111" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-111', 'label', 'link-108');">label</a></tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-112" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-112', 'type', 'link-109');">type</a></tt> <tt class="py-op">!=</tt> <tt id="link-113" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-113', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Unknown=csb.bio.sequence.SequenceTypes-class.html#Unknown"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-114', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt id="link-115" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-115', 'type', 'link-109');">type</a></tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">                 </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt id="link-116" class="py-name" targets="Class csb.bio.io.wwpdb.UnknownPDBResidueError=csb.bio.io.wwpdb.UnknownPDBResidueError-class.html"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-116', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">             </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-117" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-117', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-118', 'Unknown', 'link-114');">Unknown</a></tt> </tt>
</div><a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.guess_sequence_type"></a><div id="AbstractStructureParser.guess_sequence_type-def"><a name="L442"></a><tt class="py-lineno"> 442</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.guess_sequence_type-toggle" onclick="return toggle('AbstractStructureParser.guess_sequence_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#guess_sequence_type">guess_sequence_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_label</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.guess_sequence_type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.guess_sequence_type-expanded"><a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"><tt class="py-docstring">        Try to guess what is the sequence type of a PDB C{residue_label}.</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"><tt class="py-docstring">        @param residue_label: a PDB-conforming name of a residue</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"><tt class="py-docstring">        @type residue_label: str</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a L{SequenceTypes} enum member</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.core.EnumItem}</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-docstring">        @raise UnknownPDBResidueError: when there is no such PDB residue name</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"><tt class="py-docstring">                                       in the catalog tables</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">residue_label</tt> <tt class="py-keyword">in</tt> <tt id="link-119" class="py-name"><a title="csb.bio.io.wwpdb.PDB_AMINOACIDS" class="py-name" href="#" onclick="return doclink('link-119', 'PDB_AMINOACIDS', 'link-26');">PDB_AMINOACIDS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-120" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-120', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.Protein=csb.bio.sequence.SequenceTypes-class.html#Protein"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-121', 'Protein', 'link-121');">Protein</a></tt>                                 </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">residue_label</tt> <tt class="py-keyword">in</tt> <tt id="link-122" class="py-name"><a title="csb.bio.io.wwpdb.PDB_NUCLEOTIDES" class="py-name" href="#" onclick="return doclink('link-122', 'PDB_NUCLEOTIDES', 'link-27');">PDB_NUCLEOTIDES</a></tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-123" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-123', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name" targets="Variable csb.bio.sequence.SequenceTypes.NucleicAcid=csb.bio.sequence.SequenceTypes-class.html#NucleicAcid"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-124', 'NucleicAcid', 'link-124');">NucleicAcid</a></tt>                             </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-125" class="py-name"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-125', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_label</tt><tt class="py-op">)</tt> </tt>
</div><a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.parse_residue"></a><div id="AbstractStructureParser.parse_residue-def"><a name="L462"></a><tt class="py-lineno"> 462</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse_residue-toggle" onclick="return toggle('AbstractStructureParser.parse_residue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_residue">parse_residue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_label</tt><tt class="py-op">,</tt> <tt class="py-param">as_type</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse_residue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse_residue-expanded"><a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line"><tt class="py-docstring">        Try to parse a PDB C{residue_label} and return its closest 'normal'</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line"><tt class="py-docstring">        string representation. If a sequence type (C{as_type}) is defined,</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line"><tt class="py-docstring">        guess the alphabet based on that information, otherwise try first to</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"><tt class="py-docstring">        parse it as a protein residue.</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line"><tt class="py-docstring">        @param residue_label: a PDB-conforming name of a residue</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"><tt class="py-docstring">        @type residue_label: str</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"><tt class="py-docstring">        @param as_type: suggest a sequence type (L{SequenceTypes} member)</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"><tt class="py-docstring">        @type L{scb.core.EnumItem}</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a normalized residue name</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"><tt class="py-docstring">        @raise UnknownPDBResidueError: when there is no such PDB residue name</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"><tt class="py-docstring">                                       in the catalog table(s)</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">as_type</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">            <tt class="py-name">as_type</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.guess_sequence_type" class="py-name" href="#" onclick="return doclink('link-126', 'guess_sequence_type', 'link-110');">guess_sequence_type</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_label</tt><tt class="py-op">)</tt>          </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line"> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">as_type</tt> <tt class="py-op">==</tt> <tt id="link-127" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-127', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-128', 'Protein', 'link-121');">Protein</a></tt><tt class="py-op">:</tt>                             </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-129" class="py-name"><a title="csb.bio.io.wwpdb.PDB_AMINOACIDS" class="py-name" href="#" onclick="return doclink('link-129', 'PDB_AMINOACIDS', 'link-26');">PDB_AMINOACIDS</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_label</tt><tt class="py-op">]</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">as_type</tt> <tt class="py-op">==</tt> <tt id="link-130" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-130', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-131', 'NucleicAcid', 'link-124');">NucleicAcid</a></tt><tt class="py-op">:</tt>                       </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-132" class="py-name"><a title="csb.bio.io.wwpdb.PDB_NUCLEOTIDES" class="py-name" href="#" onclick="return doclink('link-132', 'PDB_NUCLEOTIDES', 'link-27');">PDB_NUCLEOTIDES</a></tt><tt class="py-op">[</tt><tt class="py-name">residue_label</tt><tt class="py-op">]</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-133" class="py-name"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-133', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_label</tt><tt class="py-op">)</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-134" class="py-name"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-134', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_label</tt><tt class="py-op">)</tt> </tt>
</div><a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.parse_residue_safe"></a><div id="AbstractStructureParser.parse_residue_safe-def"><a name="L493"></a><tt class="py-lineno"> 493</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse_residue_safe-toggle" onclick="return toggle('AbstractStructureParser.parse_residue_safe');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_residue_safe">parse_residue_safe</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_label</tt><tt class="py-op">,</tt> <tt class="py-param">as_type</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse_residue_safe-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse_residue_safe-expanded"><a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring">        Same as C{parse_residue}, but returns UNK/Any instead of raising</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring">        UnknownPDBResidueError.</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">        @param residue_label: a PDB-conforming name of a residue</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"><tt class="py-docstring">        @type residue_label: str</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line"><tt class="py-docstring">        @param as_type: suggest a sequence type (L{SequenceTypes} member)</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"><tt class="py-docstring">        @type L{scb.core.EnumItem}</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a normalized residue name</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.parse_residue()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_residue"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_residue" class="py-name" href="#" onclick="return doclink('link-135', 'parse_residue', 'link-135');">parse_residue</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_label</tt><tt class="py-op">,</tt> <tt class="py-name">as_type</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-136" class="py-name"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-136', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">as_type</tt> <tt class="py-op">==</tt> <tt id="link-137" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-137', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-138', 'Protein', 'link-121');">Protein</a></tt><tt class="py-op">:</tt>                         </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-139', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-140', 'Protein', 'link-121');">Protein</a></tt><tt class="py-op">.</tt><tt id="link-141" class="py-name" targets="Variable csb.bio.sequence.ProteinAlphabet.UNK=csb.bio.sequence.ProteinAlphabet-class.html#UNK,Variable csb.bio.sequence.UnknownAlphabet.UNK=csb.bio.sequence.UnknownAlphabet-class.html#UNK"><a title="csb.bio.sequence.ProteinAlphabet.UNK
csb.bio.sequence.UnknownAlphabet.UNK" class="py-name" href="#" onclick="return doclink('link-141', 'UNK', 'link-141');">UNK</a></tt><tt class="py-op">)</tt>               </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">as_type</tt> <tt class="py-op">==</tt> <tt id="link-142" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-142', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-143', 'NucleicAcid', 'link-124');">NucleicAcid</a></tt><tt class="py-op">:</tt>                   </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-144" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-144', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt class="py-name">Nucleic</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name" targets="Variable csb.bio.sequence.NucleicAlphabet.Any=csb.bio.sequence.NucleicAlphabet-class.html#Any"><a title="csb.bio.sequence.NucleicAlphabet.Any" class="py-name" href="#" onclick="return doclink('link-145', 'Any', 'link-145');">Any</a></tt><tt class="py-op">)</tt>               </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt id="link-146" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-146', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-147', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="csb.bio.sequence.ProteinAlphabet.UNK
csb.bio.sequence.UnknownAlphabet.UNK" class="py-name" href="#" onclick="return doclink('link-148', 'UNK', 'link-141');">UNK</a></tt><tt class="py-op">)</tt>               </tt>
</div><a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.parse"></a><div id="AbstractStructureParser.parse-def"><a name="L516"></a><tt class="py-lineno"> 516</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse-toggle" onclick="return toggle('AbstractStructureParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse-expanded"><a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-149" class="py-name"><a title="csb.apps.embd.OutputPathBuilder.filename
csb.bio.io.clans.Clans.filename
csb.bio.io.mrc.DensityMapReader.filename
csb.bio.io.wwpdb.AbstractStructureParser.filename" class="py-name" href="#" onclick="return doclink('link-149', 'filename', 'link-72');">filename</a></tt><tt class="py-op">:</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="csb.apps.embd.OutputPathBuilder.filename
csb.bio.io.clans.Clans.filename
csb.bio.io.mrc.DensityMapReader.filename
csb.bio.io.wwpdb.AbstractStructureParser.filename" class="py-name" href="#" onclick="return doclink('link-150', 'filename', 'link-72');">filename</a></tt> <tt class="py-op">=</tt> <tt id="link-151" class="py-name"><a title="csb.apps.embd.OutputPathBuilder.filename
csb.bio.io.clans.Clans.filename
csb.bio.io.mrc.DensityMapReader.filename
csb.bio.io.wwpdb.AbstractStructureParser.filename" class="py-name" href="#" onclick="return doclink('link-151', 'filename', 'link-72');">filename</a></tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.parse_structure()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_structure,Method csb.bio.io.wwpdb.AsyncStructureParser.parse_structure()=csb.bio.io.wwpdb.AsyncStructureParser-class.html#parse_structure"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-152', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt id="link-153" class="py-name" targets="Variable csb.statistics.maxent.MaxentPosterior.model=csb.statistics.maxent.MaxentPosterior-class.html#model"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-153', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser.parse_structure"></a><div id="AbstractStructureParser.parse_structure-def"><a name="L522"></a><tt class="py-lineno"> 522</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse_structure-toggle" onclick="return toggle('AbstractStructureParser.parse_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_structure">parse_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse_structure-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse_structure-expanded"><a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line"><tt class="py-docstring">        Parse and return the L{Structure} with the specified model identifier.</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring">        If no explicit model is specified, parse the first model in the</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line"><tt class="py-docstring">        structure.</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: parse exactly the model with this ID</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: str</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"><tt class="py-docstring">        @return: object representation of the selected model</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{Structure}</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line"><tt class="py-docstring">        @raise ValueError: When an invalid model ID is specified</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"><tt class="py-docstring">        @raise PDBParseError: When the input PDB file suffers from unrecoverable</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line"><tt class="py-docstring">                              corruption. More specialized exceptions will be</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"><tt class="py-docstring">                              raised depending on the context (see L{PDBParseError}'s</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line"><tt class="py-docstring">                              subclasses). </tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-154" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-154', 'model', 'link-153');">model</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">            <tt id="link-155" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-155', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-156', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">         </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt>             </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">            <tt id="link-157" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-157', 'structure', 'link-2');">structure</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_header</tt><tt class="py-op">(</tt><tt id="link-158" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-158', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-159" class="py-name" targets="Class csb.bio.io.wwpdb.PDBParseError=csb.bio.io.wwpdb.PDBParseError-class.html"><a title="csb.bio.io.wwpdb.PDBParseError" class="py-name" href="#" onclick="return doclink('link-159', 'PDBParseError', 'link-159');">PDBParseError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ex</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-160" class="py-name" targets="Class csb.bio.io.wwpdb.HeaderFormatError=csb.bio.io.wwpdb.HeaderFormatError-class.html"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-160', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Malformed header: {0}"</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-161', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">ex</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">         </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_atoms</tt><tt class="py-op">(</tt><tt id="link-162" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-162', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt id="link-163" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-163', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_ss</tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-164', 'structure', 'link-2');">structure</a></tt><tt class="py-op">)</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-165" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-165', 'structure', 'link-2');">structure</a></tt>     </tt>
</div><a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser.parse_models"></a><div id="AbstractStructureParser.parse_models-def"><a name="L555"></a><tt class="py-lineno"> 555</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse_models-toggle" onclick="return toggle('AbstractStructureParser.parse_models');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_models">parse_models</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">models</tt><tt class="py-op">=</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse_models-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse_models-expanded"><a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"><tt class="py-docstring">        Parse the specified models in the file and build an L{Ensemble}.</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line"><tt class="py-docstring">        @param models: an iterable object providing model identifiers.</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line"><tt class="py-docstring">                       If not specified, all models will be parsed.</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"><tt class="py-docstring">        @type models: tuple</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"><tt class="py-docstring">        @return: an ensemble with all parsed models</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{Ensemble}</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">         </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-166" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-166', 'models', 'link-88');">models</a></tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">            <tt id="link-167" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-167', 'models', 'link-88');">models</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-168', 'models', 'link-88');">models</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">            <tt id="link-169" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-169', 'models', 'link-88');">models</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-170" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractResidueMapper.map()=csb.bio.io.wwpdb.AbstractResidueMapper-class.html#map,Method csb.bio.io.wwpdb.CombinedResidueMapper.map()=csb.bio.io.wwpdb.CombinedResidueMapper-class.html#map,Method csb.bio.io.wwpdb.FastResidueMapper.map()=csb.bio.io.wwpdb.FastResidueMapper-class.html#map,Method csb.bio.io.wwpdb.RobustResidueMapper.map()=csb.bio.io.wwpdb.RobustResidueMapper-class.html#map"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-170', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">,</tt> <tt id="link-171" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-171', 'models', 'link-88');">models</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">         </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-name">ensemble</tt> <tt class="py-op">=</tt> <tt id="link-172" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-172', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-173', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-174', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Class csb.bio.structure.Ensemble=csb.bio.structure.Ensemble-class.html"><a title="csb.bio.structure.Ensemble" class="py-name" href="#" onclick="return doclink('link-175', 'Ensemble', 'link-175');">Ensemble</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">         </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-176" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-176', 'models', 'link-88');">models</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-177" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-177', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-keyword">in</tt> <tt id="link-178" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-178', 'models', 'link-88');">models</a></tt><tt class="py-op">:</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">                <tt id="link-179" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-179', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-180', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-181', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">                <tt class="py-name">ensemble</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-182', 'models', 'link-88');">models</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-183', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-184" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-184', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">            <tt id="link-185" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-185', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-186', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">            <tt id="link-187" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-187', 'model', 'link-153');">model</a></tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-188', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">            <tt class="py-name">ensemble</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.models
csb.bio.structure.Ensemble.models" class="py-name" href="#" onclick="return doclink('link-189', 'models', 'link-88');">models</a></tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-190', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-191" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-191', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">         </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">ensemble</tt> </tt>
</div><a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser.parse_biomolecule"></a><div id="AbstractStructureParser.parse_biomolecule-def"><a name="L585"></a><tt class="py-lineno"> 585</tt> <a class="py-toggle" href="#" id="AbstractStructureParser.parse_biomolecule-toggle" onclick="return toggle('AbstractStructureParser.parse_biomolecule');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_biomolecule">parse_biomolecule</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">single</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser.parse_biomolecule-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser.parse_biomolecule-expanded"><a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"><tt class="py-docstring">        Parse and return the L{Structure} of the biological unit (quaternary</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line"><tt class="py-docstring">        structure) as annotated by the REMARK 350 BIOMOLECULE record.</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: biomolecule number</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: int</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"><tt class="py-docstring">        @param single: if True, assign new single-letter chain</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring">                       identifiers. If False, assign multi-letter chain identifiers whith a</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring">                       number appended to the original identifier, like "A1", "A2", ...</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring">        @type single: bool</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">        @return: structure of biological unit</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{Structure}</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt class="py-name">remarks</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_remarks</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-number">350</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">remarks</tt><tt class="py-op">:</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-192" class="py-name"><a title="csb.bio.io.wwpdb.PDBParseError" class="py-name" href="#" onclick="return doclink('link-192', 'PDBParseError', 'link-159');">PDBParseError</a></tt><tt class="py-op">(</tt><tt class="py-string">'There is no REMARK 350'</tt><tt class="py-op">)</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-name">current</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">        <tt class="py-name">biomt</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-name">current</tt><tt class="py-op">:</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">}</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">        <tt id="link-193" class="py-name" targets="Variable csb.bio.structure.Structure.chains=csb.bio.structure.Structure-class.html#chains"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-193', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">split</tt><tt class="py-op">(</tt><tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-194', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-195', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">''</tt><tt class="py-op">]</tt> </tt>
</div><a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">remarks</tt><tt class="py-op">[</tt><tt class="py-number">350</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'BIOMOLECULE:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">                <tt class="py-name">current</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">                <tt class="py-name">biomt</tt><tt class="py-op">[</tt><tt class="py-name">current</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'APPLY THE FOLLOWING TO CHAINS:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">                <tt id="link-196" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-196', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">30</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'                   AND CHAINS:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">                <tt id="link-197" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-197', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">30</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'  BIOMT'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">                <tt class="py-name">num</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">:</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">                <tt class="py-name">vec</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">                <tt class="py-name">vec</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-198', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">,</tt> <tt class="py-name">vec</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">                <tt class="py-name">biomt</tt><tt class="py-op">[</tt><tt class="py-name">current</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">setdefault</tt><tt class="py-op">(</tt><tt id="link-199" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-199', 'chains', 'link-193');">chains</a></tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">setdefault</tt><tt class="py-op">(</tt><tt class="py-name">num</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name" targets="Variable csb.bio.fragments.TorsionAnglePredictor.extend=csb.bio.fragments.TorsionAnglePredictor-class.html#extend"><a title="csb.bio.fragments.TorsionAnglePredictor.extend" class="py-name" href="#" onclick="return doclink('link-200', 'extend', 'link-200');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">vec</tt><tt class="py-op">)</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line"> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-201" class="py-name" targets="Variable csb.io.plots.Chart.number=csb.io.plots.Chart-class.html#number,Variable csb.io.tsv.DataRow.number=csb.io.tsv.DataRow-class.html#number"><a title="csb.io.plots.Chart.number
csb.io.tsv.DataRow.number" class="py-name" href="#" onclick="return doclink('link-201', 'number', 'link-201');">number</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">biomt</tt> <tt class="py-keyword">or</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">biomt</tt><tt class="py-op">[</tt><tt id="link-202" class="py-name"><a title="csb.io.plots.Chart.number
csb.io.tsv.DataRow.number" class="py-name" href="#" onclick="return doclink('link-202', 'number', 'link-201');">number</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">'no BIOMOLECULE number {0}'</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-203', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-204" class="py-name"><a title="csb.io.plots.Chart.number
csb.io.tsv.DataRow.number" class="py-name" href="#" onclick="return doclink('link-204', 'number', 'link-201');">number</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line"> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-name">asu</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-205', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">        <tt id="link-206" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-206', 'structure', 'link-2');">structure</a></tt> <tt class="py-op">=</tt> <tt id="link-207" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-207', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-208', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-209', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name" targets="Class csb.bio.structure.Structure=csb.bio.structure.Structure-class.html"><a title="csb.bio.structure.Structure" class="py-name" href="#" onclick="return doclink('link-210', 'Structure', 'link-210');">Structure</a></tt><tt class="py-op">(</tt><tt class="py-string">'{0}_{1}'</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-211', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">asu</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name" targets="Variable csb.bio.fragments.Target.accession=csb.bio.fragments.Target-class.html#accession,Variable csb.bio.fragments.rosetta.RosettaFragment.accession=csb.bio.fragments.rosetta.RosettaFragment-class.html#accession,Variable csb.bio.io.wwpdb.EntryID.accession=csb.bio.io.wwpdb.EntryID-class.html#accession,Variable csb.bio.structure.Chain.accession=csb.bio.structure.Chain-class.html#accession,Variable csb.bio.structure.Structure.accession=csb.bio.structure.Structure-class.html#accession"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-212', 'accession', 'link-212');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-213" class="py-name"><a title="csb.io.plots.Chart.number
csb.io.tsv.DataRow.number" class="py-name" href="#" onclick="return doclink('link-213', 'number', 'link-201');">number</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-name">newchainiditer</tt> <tt class="py-op">=</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-string">'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'</tt><tt class="py-op">)</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-214" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-214', 'chains', 'link-193');">chains</a></tt><tt class="py-op">,</tt> <tt class="py-name">matrices</tt> <tt class="py-keyword">in</tt> <tt class="py-name">biomt</tt><tt class="py-op">[</tt><tt id="link-215" class="py-name"><a title="csb.io.plots.Chart.number
csb.io.tsv.DataRow.number" class="py-name" href="#" onclick="return doclink('link-215', 'number', 'link-201');">number</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name" targets="Variable csb.bio.fragments.FragmentCluster.items=csb.bio.fragments.FragmentCluster-class.html#items,Variable csb.bio.structure.AbstractEntity.items=csb.bio.structure.AbstractEntity-class.html#items,Variable csb.bio.structure.Atom.items=csb.bio.structure.Atom-class.html#items,Variable csb.bio.structure.Chain.items=csb.bio.structure.Chain-class.html#items,Variable csb.bio.structure.Ensemble.items=csb.bio.structure.Ensemble-class.html#items,Variable csb.bio.structure.Residue.items=csb.bio.structure.Residue-class.html#items,Variable csb.bio.structure.Structure.items=csb.bio.structure.Structure-class.html#items"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-216', 'items', 'link-216');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">num</tt> <tt class="py-keyword">in</tt> <tt class="py-name">matrices</tt><tt class="py-op">:</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">                <tt class="py-name">mat</tt> <tt class="py-op">=</tt> <tt class="py-name">numpy</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name" targets="Static Method csb.io.tsv.Expression.array()=csb.io.tsv.Expression-class.html#array"><a title="csb.io.tsv.Expression.array" class="py-name" href="#" onclick="return doclink('link-217', 'array', 'link-217');">array</a></tt><tt class="py-op">(</tt><tt class="py-name">matrices</tt><tt class="py-op">[</tt><tt class="py-name">num</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">reshape</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt><tt class="py-number">4</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">                <tt id="link-218" class="py-name" targets="Variable csb.statistics.mixtures.AbstractStructureMixture.R=csb.statistics.mixtures.AbstractStructureMixture-class.html#R"><a title="csb.statistics.mixtures.AbstractStructureMixture.R" class="py-name" href="#" onclick="return doclink('link-218', 'R', 'link-218');">R</a></tt><tt class="py-op">,</tt> <tt id="link-219" class="py-name" targets="Variable csb.statistics.mixtures.AbstractStructureMixture.t=csb.statistics.mixtures.AbstractStructureMixture-class.html#t"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-219', 't', 'link-219');">t</a></tt> <tt class="py-op">=</tt> <tt class="py-name">mat</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">,</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">mat</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">,</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line"> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-220" class="py-name" targets="Variable csb.bio.fragments.rosetta.RosettaFragment.chain=csb.bio.fragments.rosetta.RosettaFragment-class.html#chain,Method csb.bio.hmm.ProfileHMM.chain()=csb.bio.hmm.ProfileHMM-class.html#chain,Variable csb.bio.io.wwpdb.EntryID.chain=csb.bio.io.wwpdb.EntryID-class.html#chain,Variable csb.bio.io.wwpdb.ResidueInfo.chain=csb.bio.io.wwpdb.ResidueInfo-class.html#chain,Variable csb.bio.nmr.ContactMap.chain=csb.bio.nmr.ContactMap-class.html#chain"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-220', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-221" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-221', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-222" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-222', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">asu</tt><tt class="py-op">:</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-223" class="py-name"><a title="csb.bio.io.wwpdb.PDBParseError" class="py-name" href="#" onclick="return doclink('link-223', 'PDBParseError', 'link-159');">PDBParseError</a></tt><tt class="py-op">(</tt><tt class="py-string">'chain {0} missing'</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-224', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-225" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-225', 'chain', 'link-220');">chain</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">                    <tt id="link-226" class="py-name" targets="Method csb.bio.fragments.rosetta.ResidueInfo.copy()=csb.bio.fragments.rosetta.ResidueInfo-class.html#copy,Method csb.bio.structure.TorsionAngles.copy()=csb.bio.structure.TorsionAngles-class.html#copy,Method csb.io.tsv.ColumnInfo.copy()=csb.io.tsv.ColumnInfo-class.html#copy"><a title="csb.bio.fragments.rosetta.ResidueInfo.copy
csb.bio.structure.TorsionAngles.copy
csb.io.tsv.ColumnInfo.copy" class="py-name" href="#" onclick="return doclink('link-226', 'copy', 'link-226');">copy</a></tt> <tt class="py-op">=</tt> <tt class="py-name">asu</tt><tt class="py-op">[</tt><tt id="link-227" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-227', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method csb.bio.fragments.ChemShiftTarget.clone()=csb.bio.fragments.ChemShiftTarget-class.html#clone,Method csb.bio.fragments.Target.clone()=csb.bio.fragments.Target-class.html#clone,Method csb.bio.nmr.ChemShiftInfo.clone()=csb.bio.nmr.ChemShiftInfo-class.html#clone,Method csb.bio.structure.Atom.clone()=csb.bio.structure.Atom-class.html#clone,Method csb.bio.structure.Chain.clone()=csb.bio.structure.Chain-class.html#clone,Method csb.bio.structure.Residue.clone()=csb.bio.structure.Residue-class.html#clone,Method csb.bio.structure.SecondaryStructure.clone()=csb.bio.structure.SecondaryStructure-class.html#clone,Method csb.statistics.samplers.State.clone()=csb.statistics.samplers.State-class.html#clone"><a title="csb.bio.fragments.ChemShiftTarget.clone
csb.bio.fragments.Target.clone
csb.bio.nmr.ChemShiftInfo.clone
csb.bio.structure.Atom.clone
csb.bio.structure.Chain.clone
csb.bio.structure.Residue.clone
csb.bio.structure.SecondaryStructure.clone
csb.statistics.samplers.State.clone" class="py-name" href="#" onclick="return doclink('link-228', 'clone', 'link-228');">clone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">                    <tt id="link-229" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo.copy
csb.bio.structure.TorsionAngles.copy
csb.io.tsv.ColumnInfo.copy" class="py-name" href="#" onclick="return doclink('link-229', 'copy', 'link-226');">copy</a></tt><tt class="py-op">.</tt><tt id="link-230" class="py-name" targets="Method csb.bio.fragments.Assignment.transform()=csb.bio.fragments.Assignment-class.html#transform,Method csb.bio.structure.AbstractEntity.transform()=csb.bio.structure.AbstractEntity-class.html#transform,Method csb.bio.structure.Atom.transform()=csb.bio.structure.Atom-class.html#transform,Method csb.bio.structure.DisorderedAtom.transform()=csb.bio.structure.DisorderedAtom-class.html#transform,Function csb.bio.utils.transform()=csb.bio.utils-module.html#transform"><a title="csb.bio.fragments.Assignment.transform
csb.bio.structure.AbstractEntity.transform
csb.bio.structure.Atom.transform
csb.bio.structure.DisorderedAtom.transform
csb.bio.utils.transform" class="py-name" href="#" onclick="return doclink('link-230', 'transform', 'link-230');">transform</a></tt><tt class="py-op">(</tt><tt id="link-231" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.R" class="py-name" href="#" onclick="return doclink('link-231', 'R', 'link-218');">R</a></tt><tt class="py-op">,</tt> <tt id="link-232" class="py-name"><a title="csb.statistics.mixtures.AbstractStructureMixture.t" class="py-name" href="#" onclick="return doclink('link-232', 't', 'link-219');">t</a></tt><tt class="py-op">)</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">single</tt><tt class="py-op">:</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-233" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-233', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-234', 'chains', 'link-193');">chains</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">62</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'too many chains for single=True'</tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">                        <tt id="link-235" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo.copy
csb.bio.structure.TorsionAngles.copy
csb.io.tsv.ColumnInfo.copy" class="py-name" href="#" onclick="return doclink('link-235', 'copy', 'link-226');">copy</a></tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-236', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-237" class="py-name" targets="Method csb.bio.io.fasta.A3MSequenceIterator.next()=csb.bio.io.fasta.A3MSequenceIterator-class.html#next,Method csb.bio.structure.CompositeEntityIterator.next()=csb.bio.structure.CompositeEntityIterator-class.html#next,Method csb.statistics.samplers.mc.propagators.Looper.next()=csb.statistics.samplers.mc.propagators.Looper-class.html#next"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-237', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">newchainiditer</tt><tt class="py-op">)</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">                        <tt id="link-238" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo.copy
csb.bio.structure.TorsionAngles.copy
csb.io.tsv.ColumnInfo.copy" class="py-name" href="#" onclick="return doclink('link-238', 'copy', 'link-226');">copy</a></tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-239', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'{0}{1}'</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-240', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-241" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-241', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt class="py-name">num</tt><tt class="py-op">)</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">                    <tt id="link-242" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-242', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-243', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-244', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-245" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo.copy
csb.bio.structure.TorsionAngles.copy
csb.io.tsv.ColumnInfo.copy" class="py-name" href="#" onclick="return doclink('link-245', 'copy', 'link-226');">copy</a></tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line"> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-246" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-246', 'structure', 'link-2');">structure</a></tt> </tt>
</div><a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line"> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="AbstractStructureParser._parse_header"></a><div id="AbstractStructureParser._parse_header-def"><a name="L655"></a><tt class="py-lineno"> 655</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_header-toggle" onclick="return toggle('AbstractStructureParser._parse_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_header">_parse_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_header-expanded"><a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        An abstract method, which subclasses must use to customize the way the</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-docstring">        PDB header (or the absence of a such) is handled. The implementation</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-docstring">        must rely on reading character data from the current internal </tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-docstring">        self._stream and must return a new L{csb.bio.structure.Structure} </tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-docstring">        instance with properly initialized header data: accession, model, </tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"><tt class="py-docstring">        molecule identifiers, chains and residues. This structure object is</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line"><tt class="py-docstring">        then internally passed to the C{_parse_atoms} hook, responsible for</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line"><tt class="py-docstring">        attachment of the atoms to the residues in the structure.</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: model ID to parse</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: str</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{Structure}</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._scroll_model"></a><div id="AbstractStructureParser._scroll_model-def"><a name="L673"></a><tt class="py-lineno"> 673</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._scroll_model-toggle" onclick="return toggle('AbstractStructureParser._scroll_model');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_scroll_model">_scroll_model</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">,</tt> <tt class="py-param">stream</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._scroll_model-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._scroll_model-expanded"><a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line"><tt class="py-docstring">        Scroll the C{stream} to the specified C{model}.</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">         </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-247" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-247', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt id="link-248" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-248', 'stream', 'link-81');">stream</a></tt><tt class="py-op">)</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'No such model {0} in the structure.'</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-249', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-250" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-250', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">             </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">                <tt id="link-251" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-251', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_model</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-252" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-252', 'model', 'link-153');">model</a></tt> <tt class="py-op">==</tt> <tt id="link-253" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-253', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt id="link-254" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-254', 'model_id', 'link-90');">model_id</a></tt>       </tt>
</div><a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._parse_atoms"></a><div id="AbstractStructureParser._parse_atoms-def"><a name="L689"></a><tt class="py-lineno"> 689</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_atoms-toggle" onclick="return toggle('AbstractStructureParser._parse_atoms');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_atoms">_parse_atoms</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_atoms-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_atoms-expanded"><a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"><tt class="py-docstring">        Parse the ATOMs from the specified C{model} and attach them to the</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line"><tt class="py-docstring">        C{structure}.</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure: L{Structure} being constructed</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure:L{Structure}</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: model ID to parse</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: str        </tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">         </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">        <tt id="link-255" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-255', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-256', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line"> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">        <tt id="link-257" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-257', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt id="link-258" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-258', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">        <tt class="py-name">total_chains</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">c</tt> <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt id="link-259" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-259', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-260', 'items', 'link-216');">items</a></tt> <tt class="py-keyword">if</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Variable csb.apps.hhfrag.SliceContext.length=csb.apps.hhfrag.SliceContext-class.html#length,Variable csb.bio.fragments.FragmentMatch.length=csb.bio.fragments.FragmentMatch-class.html#length,Variable csb.bio.fragments.Target.length=csb.bio.fragments.Target-class.html#length,Variable csb.bio.fragments.rosetta.RosettaFragment.length=csb.bio.fragments.rosetta.RosettaFragment-class.html#length,Variable csb.bio.hmm.HHpredHit.length=csb.bio.hmm.HHpredHit-class.html#length,Variable csb.bio.hmm.ProfileHMM.length=csb.bio.hmm.ProfileHMM-class.html#length,Variable csb.bio.sequence.AbstractAlignment.length=csb.bio.sequence.AbstractAlignment-class.html#length,Variable csb.bio.sequence.AbstractSequence.length=csb.bio.sequence.AbstractSequence-class.html#length,Variable csb.bio.sequence.SequenceAdapter.length=csb.bio.sequence.SequenceAdapter-class.html#length,Variable csb.bio.sequence.alignment.AlignmentResult.length=csb.bio.sequence.alignment.AlignmentResult-class.html#length,Variable csb.bio.structure.Chain.length=csb.bio.structure.Chain-class.html#length,Variable csb.bio.structure.SecondaryStructureElement.length=csb.bio.structure.SecondaryStructureElement-class.html#length,Variable csb.core.BaseCollectionContainer.length=csb.core.BaseCollectionContainer-class.html#length,Variable csb.core.BaseDictionaryContainer.length=csb.core.BaseDictionaryContainer-class.html#length"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-261', 'length', 'link-261');">length</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">         </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt id="link-262" class="py-name" targets="Variable csb.bio.fragments.Target.residues=csb.bio.fragments.Target-class.html#residues,Variable csb.bio.fragments.rosetta.RosettaFragment.residues=csb.bio.fragments.rosetta.RosettaFragment-class.html#residues,Variable csb.bio.hmm.ProfileHMM.residues=csb.bio.hmm.ProfileHMM-class.html#residues,Variable csb.bio.sequence.AbstractSequence.residues=csb.bio.sequence.AbstractSequence-class.html#residues,Variable csb.bio.sequence.AlignedSequenceAdapter.residues=csb.bio.sequence.AlignedSequenceAdapter-class.html#residues,Variable csb.bio.structure.Chain.residues=csb.bio.structure.Chain-class.html#residues"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-262', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt> <tt class="py-op">(</tt><tt id="link-263" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-263', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-264" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-264', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-265" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-265', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-266', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">)</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">        <tt class="py-name">seen_residues</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt> <tt class="py-op">(</tt><tt id="link-267" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-267', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-268" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-268', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-269" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-269', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-270', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">)</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">         </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">        <tt class="py-name">in_ligands</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt>                 </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">        <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">        <tt class="py-name">read_model</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line"> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line"> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-271" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-271', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">             </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">read_model</tt><tt class="py-op">:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_model_line</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-272" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-272', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt id="link-273" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-273', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">                    <tt id="link-274" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-274', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt id="link-275" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-275', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-276', 'model_id', 'link-90');">model_id</a></tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">                    <tt class="py-name">read_model</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ATOM'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'HETATM'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">in_ligands</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">                <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">                 </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">                <tt class="py-name">info</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_atom_line</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-277" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-277', 'structure', 'link-2');">structure</a></tt><tt class="py-op">)</tt>                 </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">                <tt id="link-278" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-278', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-279', 'add', 'link-98');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-280', 'chain', 'link-220');">chain</a></tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">                 </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-281', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seen_residues</tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-282', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">                    <tt id="link-283" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-283', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-284', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-285', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">info</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">                    <tt class="py-name">seen_residues</tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-286', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-287', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">info</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">                    <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name" targets="Variable csb.bio.io.wwpdb.ResidueInfo.atoms=csb.bio.io.wwpdb.ResidueInfo-class.html#atoms,Variable csb.bio.nmr.ContactMap.atoms=csb.bio.nmr.ContactMap-class.html#atoms,Variable csb.bio.structure.Residue.atoms=csb.bio.structure.Residue-class.html#atoms"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-288', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">                    <tt class="py-name">seen_residues</tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-289', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-290" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-290', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-291', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-292', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">                     </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">in_atom</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'TER'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line">                <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-293" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-293', 'chains', 'link-193');">chains</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">total_chains</tt><tt class="py-op">:</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">                    <tt class="py-name">in_ligands</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ENDMDL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'END'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line"> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-294" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-294', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-295', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">!=</tt> <tt id="link-296" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-296', 'model', 'link-153');">model</a></tt><tt class="py-op">:</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'No such model {0} in the structure.'</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-297', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-298" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-298', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">                 </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_map_residues</tt><tt class="py-op">(</tt><tt id="link-299" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-299', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt id="link-300" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-300', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._parse_model_line"></a><div id="AbstractStructureParser._parse_model_line-def"><a name="L758"></a><tt class="py-lineno"> 758</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_model_line-toggle" onclick="return toggle('AbstractStructureParser._parse_model_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_model_line">_parse_model_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_model_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_model_line-expanded"><a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line"><tt class="py-docstring">        Handle a new MODEL line. The default implementation will read the model</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"><tt class="py-docstring">        ID and attach it to the C{structure}.</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line"><tt class="py-docstring">        @param line: raw string line to parse</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line"><tt class="py-docstring">        @type line: str</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure: L{Structure} being constructed</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure:L{Structure}</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line"><tt class="py-docstring">        @note: this method may have side effects: scrolls the current stream</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"><tt class="py-docstring">        @return: read model ID</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-301" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-301', 'model', 'link-153');">model</a></tt> <tt class="py-keyword">and</tt> <tt id="link-302" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-302', 'model', 'link-153');">model</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_model</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scroll_model</tt><tt class="py-op">(</tt><tt id="link-303" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-303', 'model', 'link-153');">model</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">            <tt id="link-304" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-304', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-305" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-305', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt id="link-306" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-306', 'model', 'link-153');">model</a></tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">            <tt id="link-307" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-307', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_model</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">            <tt id="link-308" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-308', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-309', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt id="link-310" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-310', 'model', 'link-153');">model</a></tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                             </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-311" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-311', 'model', 'link-153');">model</a></tt> </tt>
</div><a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._parse_atom_line"></a><div id="AbstractStructureParser._parse_atom_line-def"><a name="L782"></a><tt class="py-lineno"> 782</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_atom_line-toggle" onclick="return toggle('AbstractStructureParser._parse_atom_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_atom_line">_parse_atom_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_atom_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_atom_line-expanded"><a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"><tt class="py-docstring">        Handle a new ATOM or HETATM line. The default implementation will read</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"><tt class="py-docstring">        all data fields and create a new L{Atom}.</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"><tt class="py-docstring">        @param line: raw string line to parse</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"><tt class="py-docstring">        @type line: str</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure: L{Structure} being constructed</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure:L{Structure}</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">        @return: newly constructed atom</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ResidueInfo}</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">        <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_atom</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">         </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt id="link-312" class="py-name" targets="Variable csb.bio.fragments.ResidueEventInfo.rank=csb.bio.fragments.ResidueEventInfo-class.html#rank,Variable csb.bio.hmm.HHpredHit.rank=csb.bio.hmm.HHpredHit-class.html#rank,Variable csb.bio.hmm.HMMLayer.rank=csb.bio.hmm.HMMLayer-class.html#rank,Variable csb.bio.hmm.State.rank=csb.bio.hmm.State-class.html#rank,Variable csb.bio.io.wwpdb.ResidueInfo.rank=csb.bio.io.wwpdb.ResidueInfo-class.html#rank,Variable csb.bio.nmr.Label.rank=csb.bio.nmr.Label-class.html#rank,Variable csb.bio.sequence.ResidueInfo.rank=csb.bio.sequence.ResidueInfo-class.html#rank,Variable csb.bio.structure.Residue.rank=csb.bio.structure.Residue-class.html#rank"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-312', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_sequence_number</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt>             </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt id="link-313" class="py-name" targets="Variable csb.bio.io.wwpdb.ResidueInfo.sequence_number=csb.bio.io.wwpdb.ResidueInfo-class.html#sequence_number,Variable csb.bio.structure.Residue.sequence_number=csb.bio.structure.Residue-class.html#sequence_number"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-313', 'sequence_number', 'link-313');">sequence_number</a></tt> <tt class="py-op">=</tt> <tt id="link-314" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-314', 'rank', 'link-312');">rank</a></tt>             </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt id="link-315" class="py-name" targets="Variable csb.bio.io.wwpdb.ResidueInfo.insertion_code=csb.bio.io.wwpdb.ResidueInfo-class.html#insertion_code,Variable csb.bio.structure.Residue.insertion_code=csb.bio.structure.Residue-class.html#insertion_code"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-315', 'insertion_code', 'link-315');">insertion_code</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_insertion_code</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt>         </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">         </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">        <tt id="link-316" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-316', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-317', 'sequence_number', 'link-313');">sequence_number</a></tt><tt class="py-op">)</tt>         </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-318" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-318', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">:</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">            <tt id="link-319" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-319', 'id', 'link-29');">id</a></tt> <tt class="py-op">+=</tt> <tt id="link-320" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-320', 'insertion_code', 'link-315');">insertion_code</a></tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line">        <tt id="link-321" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-321', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_chain_id</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-322" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-322', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-323" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-323', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-324', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-325" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-325', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Chain {0} is undefined"</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-326', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-327" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-327', 'chain', 'link-220');">chain</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">        <tt id="link-328" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-328', 'type', 'link-109');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_residue</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-329" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-329', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-330" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-330', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-331" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-331', 'chain', 'link-220');">chain</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">        <tt id="link-332" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-332', 'label', 'link-108');">label</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_residue_raw</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">         </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name" targets="Variable csb.bio.structure.Atom.alternate=csb.bio.structure.Atom-class.html#alternate"><a title="csb.bio.structure.Atom.alternate" class="py-name" href="#" onclick="return doclink('link-333', 'alternate', 'link-333');">alternate</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_alternate</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-334" class="py-name" targets="Variable csb.bio.structure.Atom.occupancy=csb.bio.structure.Atom-class.html#occupancy"><a title="csb.bio.structure.Atom.occupancy" class="py-name" href="#" onclick="return doclink('link-334', 'occupancy', 'link-334');">occupancy</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_occupancy</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name" targets="Variable csb.bio.structure.Atom.bfactor=csb.bio.structure.Atom-class.html#bfactor"><a title="csb.bio.structure.Atom.bfactor" class="py-name" href="#" onclick="return doclink('link-335', 'bfactor', 'link-335');">bfactor</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_bfactor</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name" targets="Variable csb.bio.structure.Atom.charge=csb.bio.structure.Atom-class.html#charge"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-336', 'charge', 'link-336');">charge</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_charge</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">        <tt class="py-name">info</tt> <tt class="py-op">=</tt> <tt id="link-337" class="py-name" targets="Class csb.bio.fragments.rosetta.ResidueInfo=csb.bio.fragments.rosetta.ResidueInfo-class.html,Class csb.bio.io.wwpdb.ResidueInfo=csb.bio.io.wwpdb.ResidueInfo-class.html,Class csb.bio.sequence.ResidueInfo=csb.bio.sequence.ResidueInfo-class.html"><a title="csb.bio.fragments.rosetta.ResidueInfo
csb.bio.io.wwpdb.ResidueInfo
csb.bio.sequence.ResidueInfo" class="py-name" href="#" onclick="return doclink('link-337', 'ResidueInfo', 'link-337');">ResidueInfo</a></tt><tt class="py-op">(</tt><tt id="link-338" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-338', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt id="link-339" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-339', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-340" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-340', 'id', 'link-29');">id</a></tt><tt class="py-op">,</tt> <tt id="link-341" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-341', 'sequence_number', 'link-313');">sequence_number</a></tt><tt class="py-op">,</tt> <tt id="link-342" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-342', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">,</tt> <tt id="link-343" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-343', 'type', 'link-109');">type</a></tt><tt class="py-op">,</tt> <tt id="link-344" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-344', 'label', 'link-108');">label</a></tt><tt class="py-op">)</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">        <tt class="py-name">info</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-345', 'atoms', 'link-288');">atoms</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">]</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">         </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">info</tt> </tt>
</div><a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser._read_model"></a><div id="AbstractStructureParser._read_model-def"><a name="L823"></a><tt class="py-lineno"> 823</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_model-toggle" onclick="return toggle('AbstractStructureParser._read_model');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_model">_read_model</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_model-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_model-expanded"><a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line"><tt class="py-docstring">        @return: model identifier</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>           </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-346" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-346', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Invalid model ID: {0}"</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-347', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>   </tt>
</div><a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser._read_atom"></a><div id="AbstractStructureParser._read_atom-def"><a name="L833"></a><tt class="py-lineno"> 833</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_atom-toggle" onclick="return toggle('AbstractStructureParser._read_atom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_atom">_read_atom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_atom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_atom-expanded"><a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a new atom (serial_number, name, element, vector)</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{Atom}</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">            <tt id="link-348" class="py-name" targets="Variable csb.bio.structure.Atom.serial_number=csb.bio.structure.Atom-class.html#serial_number"><a title="csb.bio.structure.Atom.serial_number" class="py-name" href="#" onclick="return doclink('link-348', 'serial_number', 'link-348');">serial_number</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">            <tt id="link-349" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-349', 'name', 'link-82');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">:</tt><tt class="py-number">16</tt><tt class="py-op">]</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">            <tt id="link-350" class="py-name" targets="Variable csb.bio.io.mrc.HeaderInfo.x=csb.bio.io.mrc.HeaderInfo-class.html#x,Variable csb.bio.structure.ChemElements.x=csb.bio.structure.ChemElements-class.html#x"><a title="csb.bio.io.mrc.HeaderInfo.x
csb.bio.structure.ChemElements.x" class="py-name" href="#" onclick="return doclink('link-350', 'x', 'link-350');">x</a></tt><tt class="py-op">,</tt> <tt id="link-351" class="py-name" targets="Variable csb.bio.io.mrc.HeaderInfo.y=csb.bio.io.mrc.HeaderInfo-class.html#y"><a title="csb.bio.io.mrc.HeaderInfo.y" class="py-name" href="#" onclick="return doclink('link-351', 'y', 'link-351');">y</a></tt><tt class="py-op">,</tt> <tt id="link-352" class="py-name" targets="Variable csb.bio.io.mrc.HeaderInfo.z=csb.bio.io.mrc.HeaderInfo-class.html#z,Method csb.statistics.ars.Envelope.z()=csb.statistics.ars.Envelope-class.html#z,Method csb.statistics.maxent.MaxentModel.z()=csb.statistics.maxent.MaxentModel-class.html#z"><a title="csb.bio.io.mrc.HeaderInfo.z
csb.statistics.ars.Envelope.z
csb.statistics.maxent.MaxentModel.z" class="py-name" href="#" onclick="return doclink('link-352', 'z', 'link-352');">z</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">30</tt><tt class="py-op">:</tt><tt class="py-number">38</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">38</tt><tt class="py-op">:</tt><tt class="py-number">46</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">46</tt><tt class="py-op">:</tt><tt class="py-number">54</tt><tt class="py-op">]</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">            <tt id="link-353" class="py-name" targets="Variable csb.bio.structure.Atom.vector=csb.bio.structure.Atom-class.html#vector"><a title="csb.bio.structure.Atom.vector" class="py-name" href="#" onclick="return doclink('link-353', 'vector', 'link-353');">vector</a></tt> <tt class="py-op">=</tt> <tt class="py-name">numpy</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="csb.io.tsv.Expression.array" class="py-name" href="#" onclick="return doclink('link-354', 'array', 'link-217');">array</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-355" class="py-name"><a title="csb.bio.io.mrc.HeaderInfo.x
csb.bio.structure.ChemElements.x" class="py-name" href="#" onclick="return doclink('link-355', 'x', 'link-350');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-356" class="py-name"><a title="csb.bio.io.mrc.HeaderInfo.y" class="py-name" href="#" onclick="return doclink('link-356', 'y', 'link-351');">y</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-357" class="py-name"><a title="csb.bio.io.mrc.HeaderInfo.z
csb.statistics.ars.Envelope.z
csb.statistics.maxent.MaxentModel.z" class="py-name" href="#" onclick="return doclink('link-357', 'z', 'link-352');">z</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ve</tt><tt class="py-op">:</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-358" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-358', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Invalid ATOM line: {0}"</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-359', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">ve</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">         </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">        <tt id="link-360" class="py-name" targets="Method csb.bio.nmr.NOEPeak.element()=csb.bio.nmr.NOEPeak-class.html#element,Method csb.bio.nmr.NOESpectrum.element()=csb.bio.nmr.NOESpectrum-class.html#element,Variable csb.bio.structure.Atom.element=csb.bio.structure.Atom-class.html#element"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-360', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_element</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-361" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-361', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-362', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-363', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-364" class="py-name" targets="Class csb.bio.structure.Atom=csb.bio.structure.Atom-class.html"><a title="csb.bio.structure.Atom" class="py-name" href="#" onclick="return doclink('link-364', 'Atom', 'link-364');">Atom</a></tt><tt class="py-op">(</tt><tt id="link-365" class="py-name"><a title="csb.bio.structure.Atom.serial_number" class="py-name" href="#" onclick="return doclink('link-365', 'serial_number', 'link-348');">serial_number</a></tt><tt class="py-op">,</tt> <tt id="link-366" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-366', 'name', 'link-82');">name</a></tt><tt class="py-op">,</tt> <tt id="link-367" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-367', 'element', 'link-360');">element</a></tt><tt class="py-op">,</tt> <tt id="link-368" class="py-name"><a title="csb.bio.structure.Atom.vector" class="py-name" href="#" onclick="return doclink('link-368', 'vector', 'link-353');">vector</a></tt><tt class="py-op">)</tt>         </tt>
</div><a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._read_sequence_number"></a><div id="AbstractStructureParser._read_sequence_number-def"><a name="L849"></a><tt class="py-lineno"> 849</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_sequence_number-toggle" onclick="return toggle('AbstractStructureParser._read_sequence_number');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_sequence_number">_read_sequence_number</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_sequence_number-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_sequence_number-expanded"><a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line"><tt class="py-docstring">        @return: PDB sequence number</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">22</tt><tt class="py-op">:</tt><tt class="py-number">26</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-369" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-369', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Invalid sequence number"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._read_insertion_code"></a><div id="AbstractStructureParser._read_insertion_code-def"><a name="L859"></a><tt class="py-lineno"> 859</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_insertion_code-toggle" onclick="return toggle('AbstractStructureParser._read_insertion_code');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_insertion_code">_read_insertion_code</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_insertion_code-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_insertion_code-expanded"><a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring">        @return: PDB insertion code</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str or None</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>        </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">        <tt class="py-name">code</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">26</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-370', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">         </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">code</tt><tt class="py-op">:</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">code</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._read_chain_id"></a><div id="AbstractStructureParser._read_chain_id-def"><a name="L871"></a><tt class="py-lineno"> 871</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_chain_id-toggle" onclick="return toggle('AbstractStructureParser._read_chain_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_chain_id">_read_chain_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_chain_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_chain_id-expanded"><a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring">        @return: chain identifier</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>           </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-371', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._read_residue"></a><div id="AbstractStructureParser._read_residue-def"><a name="L878"></a><tt class="py-lineno"> 878</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_residue-toggle" onclick="return toggle('AbstractStructureParser._read_residue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_residue">_read_residue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_residue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_residue-expanded"><a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">        @param chain: owning L{Chain} object </tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain: L{Chain}</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a member of any alphabet (e.g. L{SequenceAlphabets.Protein})</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{EnumItem}</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line">        <tt class="py-name">raw</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_residue_raw</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">        <tt id="link-372" class="py-name" targets="Method csb.bio.fragments.AssignmentFactory.residue()=csb.bio.fragments.AssignmentFactory-class.html#residue,Method csb.bio.fragments.ChemShiftAssignmentFactory.residue()=csb.bio.fragments.ChemShiftAssignmentFactory-class.html#residue,Variable csb.bio.hmm.HMMLayer.residue=csb.bio.hmm.HMMLayer-class.html#residue,Variable csb.bio.nmr.Label.residue=csb.bio.nmr.Label-class.html#residue,Variable csb.bio.structure.Atom.residue=csb.bio.structure.Atom-class.html#residue"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-372', 'residue', 'link-372');">residue</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.parse_residue_safe()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#parse_residue_safe"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_residue_safe" class="py-name" href="#" onclick="return doclink('link-373', 'parse_residue_safe', 'link-373');">parse_residue_safe</a></tt><tt class="py-op">(</tt><tt class="py-name">raw</tt><tt class="py-op">,</tt> <tt class="py-name">as_type</tt><tt class="py-op">=</tt><tt id="link-374" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-374', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-375', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line">         </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-376" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-376', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-377', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt id="link-378" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-378', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="csb.bio.sequence.SequenceTypes.NucleicAcid" class="py-name" href="#" onclick="return doclink('link-379', 'NucleicAcid', 'link-124');">NucleicAcid</a></tt><tt class="py-op">:</tt>                                </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-380" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-380', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-381', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-382" class="py-name" targets="Class csb.core.Enum=csb.core.Enum-class.html"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-382', 'Enum', 'link-382');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-383" class="py-name" targets="Static Method csb.core.Enum.parsename()=csb.core.Enum-class.html#parsename"><a title="csb.core.Enum.parsename" class="py-name" href="#" onclick="return doclink('link-383', 'parsename', 'link-383');">parsename</a></tt><tt class="py-op">(</tt><tt id="link-384" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-384', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt class="py-name">Nucleic</tt><tt class="py-op">,</tt> <tt id="link-385" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-385', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-386" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-386', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-387" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-387', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-388', 'Enum', 'link-382');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-389" class="py-name"><a title="csb.core.Enum.parsename" class="py-name" href="#" onclick="return doclink('link-389', 'parsename', 'link-383');">parsename</a></tt><tt class="py-op">(</tt><tt id="link-390" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-390', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-391" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-391', 'Protein', 'link-121');">Protein</a></tt><tt class="py-op">,</tt> <tt id="link-392" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-392', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-393" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-393', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-394', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-395" class="py-name" targets="Class csb.core.EnumMemberError=csb.core.EnumMemberError-class.html"><a title="csb.core.EnumMemberError" class="py-name" href="#" onclick="return doclink('link-395', 'EnumMemberError', 'link-395');">EnumMemberError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-396" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-396', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"{0} is not a valid {1} residue"</tt><tt class="py-op">.</tt><tt id="link-397" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-397', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">raw</tt><tt class="py-op">,</tt> <tt id="link-398" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-398', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-399" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-399', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt>   </tt>
</div><a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._read_residue_raw"></a><div id="AbstractStructureParser._read_residue_raw-def"><a name="L897"></a><tt class="py-lineno"> 897</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_residue_raw-toggle" onclick="return toggle('AbstractStructureParser._read_residue_raw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_residue_raw">_read_residue_raw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_residue_raw-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_residue_raw-expanded"><a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">17</tt><tt class="py-op">:</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-400', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>               </tt>
</div><a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">             </tt>
<a name="AbstractStructureParser._read_element"></a><div id="AbstractStructureParser._read_element-def"><a name="L903"></a><tt class="py-lineno"> 903</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_element-toggle" onclick="return toggle('AbstractStructureParser._read_element');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_element">_read_element</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_element-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_element-expanded"><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a member of L{ChemElements}</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{EnumItem} or None</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">        <tt id="link-401" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-401', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">76</tt><tt class="py-op">:</tt><tt class="py-number">78</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-402" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-402', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-403" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-403', 'element', 'link-360');">element</a></tt><tt class="py-op">:</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">                <tt id="link-404" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-404', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt id="link-405" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-405', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-406" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-406', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="csb.core.Enum" class="py-name" href="#" onclick="return doclink('link-407', 'Enum', 'link-382');">Enum</a></tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="csb.core.Enum.parsename" class="py-name" href="#" onclick="return doclink('link-408', 'parsename', 'link-383');">parsename</a></tt><tt class="py-op">(</tt><tt id="link-409" class="py-name"><a title="csb.bio.structure.ChemElements" class="py-name" href="#" onclick="return doclink('link-409', 'ChemElements', 'link-24');">ChemElements</a></tt><tt class="py-op">,</tt> <tt id="link-410" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-410', 'element', 'link-360');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt id="link-411" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-411', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-412', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="csb.core.EnumMemberError" class="py-name" href="#" onclick="return doclink('link-413', 'EnumMemberError', 'link-395');">EnumMemberError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-414" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-414', 'element', 'link-360');">element</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'D'</tt><tt class="py-op">,</tt> <tt class="py-string">'X'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">                    <tt id="link-415" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-415', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt id="link-416" class="py-name"><a title="csb.bio.structure.ChemElements" class="py-name" href="#" onclick="return doclink('link-416', 'ChemElements', 'link-24');">ChemElements</a></tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="csb.bio.io.mrc.HeaderInfo.x
csb.bio.structure.ChemElements.x" class="py-name" href="#" onclick="return doclink('link-417', 'x', 'link-350');">x</a></tt>                             </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt id="link-418" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-418', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Unknown chemical element: {0}'</tt><tt class="py-op">.</tt><tt id="link-419" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-419', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-420" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-420', 'element', 'link-360');">element</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">            <tt id="link-421" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-421', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">             </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-422" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-422', 'element', 'link-360');">element</a></tt> </tt>
</div><a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser._read_alternate"></a><div id="AbstractStructureParser._read_alternate-def"><a name="L922"></a><tt class="py-lineno"> 922</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_alternate-toggle" onclick="return toggle('AbstractStructureParser._read_alternate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_alternate">_read_alternate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_alternate-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_alternate-expanded"><a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line"><tt class="py-docstring">        @return: alt identifier</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str or None</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt id="link-423" class="py-name"><a title="csb.bio.structure.Atom.alternate" class="py-name" href="#" onclick="return doclink('link-423', 'alternate', 'link-333');">alternate</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">16</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-424', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">         </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-425" class="py-name"><a title="csb.bio.structure.Atom.alternate" class="py-name" href="#" onclick="return doclink('link-425', 'alternate', 'link-333');">alternate</a></tt><tt class="py-op">:</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-426" class="py-name"><a title="csb.bio.structure.Atom.alternate" class="py-name" href="#" onclick="return doclink('link-426', 'alternate', 'link-333');">alternate</a></tt> </tt>
</div><a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">     </tt>
<a name="AbstractStructureParser._read_occupancy"></a><div id="AbstractStructureParser._read_occupancy-def"><a name="L934"></a><tt class="py-lineno"> 934</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_occupancy-toggle" onclick="return toggle('AbstractStructureParser._read_occupancy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_occupancy">_read_occupancy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_occupancy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_occupancy-expanded"><a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring">        @return: occupancy</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: float or None</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">54</tt><tt class="py-op">:</tt><tt class="py-number">60</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-427" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-427', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-428" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-428', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Malformed occupancy field"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._read_bfactor"></a><div id="AbstractStructureParser._read_bfactor-def"><a name="L944"></a><tt class="py-lineno"> 944</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_bfactor-toggle" onclick="return toggle('AbstractStructureParser._read_bfactor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_bfactor">_read_bfactor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_bfactor-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_bfactor-expanded"><a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"><tt class="py-docstring">        @return: b-factor</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: float or None</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">60</tt><tt class="py-op">:</tt><tt class="py-number">66</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-429" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-429', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-430" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-430', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Malformed bfactor field"</tt><tt class="py-op">)</tt>         </tt>
</div><a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">             </tt>
<a name="AbstractStructureParser._read_charge"></a><div id="AbstractStructureParser._read_charge-def"><a name="L954"></a><tt class="py-lineno"> 954</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._read_charge-toggle" onclick="return toggle('AbstractStructureParser._read_charge');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_read_charge">_read_charge</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._read_charge-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._read_charge-expanded"><a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line"><tt class="py-docstring">        @return: charge</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: int or None</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">        <tt id="link-431" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-431', 'charge', 'link-336');">charge</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">78</tt><tt class="py-op">:</tt><tt class="py-number">80</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-432', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">         </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-433" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-433', 'charge', 'link-336');">charge</a></tt><tt class="py-op">:</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-434" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-434', 'charge', 'link-336');">charge</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'+'</tt><tt class="py-op">,</tt> <tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">                <tt id="link-435" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-435', 'charge', 'link-336');">charge</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">'1'</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-436" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-436', 'charge', 'link-336');">charge</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'+'</tt><tt class="py-op">,</tt> <tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">                <tt id="link-437" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-437', 'charge', 'link-336');">charge</a></tt> <tt class="py-op">=</tt> <tt id="link-438" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-438', 'charge', 'link-336');">charge</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-439" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-439', 'charge', 'link-336');">charge</a></tt><tt class="py-op">)</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-440" class="py-name"><a title="csb.bio.io.hhpred.StructureFormatError
csb.bio.io.wwpdb.StructureFormatError" class="py-name" href="#" onclick="return doclink('link-440', 'StructureFormatError', 'link-92');">StructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Malformed charge field"</tt><tt class="py-op">)</tt>  </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._map_residues"></a><div id="AbstractStructureParser._map_residues-def"><a name="L973"></a><tt class="py-lineno"> 973</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._map_residues-toggle" onclick="return toggle('AbstractStructureParser._map_residues');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_map_residues">_map_residues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._map_residues-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._map_residues-expanded"><a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line"><tt class="py-docstring">        Attach each L{Atom} to its corresponding L{Residue}.</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"><tt class="py-docstring">        So far we have constructed a sparse (fragmented) chain given the information</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line"><tt class="py-docstring">        we have read from the ATOM/HETATM records. That includes PDB sequence</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"><tt class="py-docstring">        identifiers and insertion codes, which cover only residues with XYZ coordinates</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line"><tt class="py-docstring">        and often do not correspond to our L{Residue} ranks.</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"><tt class="py-docstring">        Our job is to find the right correspondence by matching this unreal sequence</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"><tt class="py-docstring">        to what we have got from the SEQRES fields (the complete, gap-free protein</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"><tt class="py-docstring">        sequence). Here we delegate this task to the current L{AbstractResidueMapper}</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line"><tt class="py-docstring">        strategy, which does all the magic.</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure: L{Structure} being constructed</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure:L{Structure}</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"><tt class="py-docstring">        @param residues: all L{Atom}s which have been constructed so far.</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line"><tt class="py-docstring">                         This must be a map of the form:</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring">                         &lt;chainID: [L{ResidueInfo}1, L{ResidueInfo}2...]&gt;</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring">        @type residues: dict of L{ResidueInfo}        </tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line"> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-441" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-441', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt id="link-442" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-442', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-443', 'chains', 'link-193');">chains</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt id="link-444" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-444', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt id="link-445" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-445', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-446" class="py-name" targets="Method csb.core.BaseDictionaryContainer.keys()=csb.core.BaseDictionaryContainer-class.html#keys"><a title="csb.core.BaseDictionaryContainer.keys" class="py-name" href="#" onclick="return doclink('link-446', 'keys', 'link-446');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-447" class="py-name"><a title="csb.bio.io.wwpdb.PDBParseError" class="py-name" href="#" onclick="return doclink('link-447', 'PDBParseError', 'link-159');">PDBParseError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Corrupt PDB file"</tt><tt class="py-op">)</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line"> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-448" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-448', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-449" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-449', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-450', 'items', 'link-216');">items</a></tt><tt class="py-op">:</tt>             </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-451" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-451', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-452', 'length', 'link-261');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-453" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-453', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt id="link-454" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-454', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-455', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">             </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">            <tt class="py-name">reference</tt> <tt class="py-op">=</tt> <tt id="link-456" class="py-name" targets="Class csb.bio.io.wwpdb.SparseChainSequence=csb.bio.io.wwpdb.SparseChainSequence-class.html"><a title="csb.bio.io.wwpdb.SparseChainSequence" class="py-name" href="#" onclick="return doclink('link-456', 'SparseChainSequence', 'link-456');">SparseChainSequence</a></tt><tt class="py-op">.</tt><tt id="link-457" class="py-name" targets="Static Method csb.apps.buildhmm.ProfileBuilder.create()=csb.apps.buildhmm.ProfileBuilder-class.html#create,Static Method csb.apps.hhfrag.PredictionBuilder.create()=csb.apps.hhfrag.PredictionBuilder-class.html#create,Static Method csb.bio.fragments.ClusterNode.create()=csb.bio.fragments.ClusterNode-class.html#create,Static Method csb.bio.io.cs.ChemShiftReader.create()=csb.bio.io.cs.ChemShiftReader-class.html#create,Static Method csb.bio.io.fasta.OutputBuilder.create()=csb.bio.io.fasta.OutputBuilder-class.html#create,Static Method csb.bio.io.wwpdb.EntryID.create()=csb.bio.io.wwpdb.EntryID-class.html#create,Static Method csb.bio.io.wwpdb.SparseChainSequence.create()=csb.bio.io.wwpdb.SparseChainSequence-class.html#create,Static Method csb.bio.sequence.ChainSequence.create()=csb.bio.sequence.ChainSequence-class.html#create,Static Method csb.bio.sequence.RichSequence.create()=csb.bio.sequence.RichSequence-class.html#create,Static Method csb.bio.structure.CompositeEntityIterator.create()=csb.bio.structure.CompositeEntityIterator-class.html#create,Static Method csb.bio.structure.Residue.create()=csb.bio.structure.Residue-class.html#create,Static Method csb.core.Enum.create()=csb.core.Enum-class.html#create,Method csb.io.tsv.RepositoryImp.create()=csb.io.tsv.RepositoryImp-class.html#create,Method csb.io.tsv.SQLiteRepository.create()=csb.io.tsv.SQLiteRepository-class.html#create,Static Method csb.statistics.samplers.mc.TrajectoryBuilder.create()=csb.statistics.samplers.mc.TrajectoryBuilder-class.html#create"><a title="csb.apps.buildhmm.ProfileBuilder.create
csb.apps.hhfrag.PredictionBuilder.create
csb.bio.fragments.ClusterNode.create
csb.bio.io.cs.ChemShiftReader.create
csb.bio.io.fasta.OutputBuilder.create
csb.bio.io.wwpdb.EntryID.create
csb.bio.io.wwpdb.SparseChainSequence.create
csb.bio.sequence.ChainSequence.create
csb.bio.sequence.RichSequence.create
csb.bio.structure.CompositeEntityIterator.create
csb.bio.structure.Residue.create
csb.core.Enum.create
csb.io.tsv.RepositoryImp.create
csb.io.tsv.SQLiteRepository.create
csb.statistics.samplers.mc.TrajectoryBuilder.create" class="py-name" href="#" onclick="return doclink('link-457', 'create', 'link-457');">create</a></tt><tt class="py-op">(</tt><tt id="link-458" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-458', 'chain', 'link-220');">chain</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">            <tt class="py-name">sparse</tt> <tt class="py-op">=</tt> <tt id="link-459" class="py-name"><a title="csb.bio.io.wwpdb.SparseChainSequence" class="py-name" href="#" onclick="return doclink('link-459', 'SparseChainSequence', 'link-456');">SparseChainSequence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">                            <tt class="py-string">"atoms"</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">,</tt> <tt id="link-460" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-460', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt id="link-461" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-461', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-462', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-463" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-463', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-464" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-464', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">             </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">            <tt class="py-name">aligned</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.mapper" class="py-name" href="#" onclick="return doclink('link-465', 'mapper', 'link-68');">mapper</a></tt><tt class="py-op">.</tt><tt id="link-466" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-466', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">reference</tt><tt class="py-op">)</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-467', 'length', 'link-261');">length</a></tt> <tt class="py-op">==</tt> <tt id="link-468" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-468', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-469" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-469', 'length', 'link-261');">length</a></tt>  </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">             </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-470" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-470', 'residue', 'link-372');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-name">mapped</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt id="link-471" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-471', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-472', 'residues', 'link-262');">residues</a></tt><tt class="py-op">,</tt> <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-473" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-473', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">mapped</tt><tt class="py-op">.</tt><tt id="link-474" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-474', 'type', 'link-109');">type</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.alphabet=csb.bio.sequence.AbstractSequence-class.html#alphabet,Variable csb.bio.sequence.SequenceAdapter.alphabet=csb.bio.sequence.SequenceAdapter-class.html#alphabet,Variable csb.bio.structure.Chain.alphabet=csb.bio.structure.Chain-class.html#alphabet"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-475', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-476" class="py-name" targets="Variable csb.bio.hmm.HHpredHitAlignment.GAP=csb.bio.hmm.HHpredHitAlignment-class.html#GAP,Variable csb.bio.sequence.NucleicAlphabet.GAP=csb.bio.sequence.NucleicAlphabet-class.html#GAP,Variable csb.bio.sequence.ProteinAlphabet.GAP=csb.bio.sequence.ProteinAlphabet-class.html#GAP,Variable csb.bio.sequence.UnknownAlphabet.GAP=csb.bio.sequence.UnknownAlphabet-class.html#GAP"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-476', 'GAP', 'link-476');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">                    <tt id="link-477" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-477', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-478" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-478', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">mapped</tt><tt class="py-op">.</tt><tt id="link-479" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-479', 'sequence_number', 'link-313');">sequence_number</a></tt><tt class="py-op">,</tt> <tt class="py-name">mapped</tt><tt class="py-op">.</tt><tt id="link-480" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-480', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">                     </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">mapped</tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-481', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">                        <tt id="link-482" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-482', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-483" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-483', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-484', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"> </tt>
<a name="AbstractStructureParser._parse_ss"></a><div id="AbstractStructureParser._parse_ss-def"><a name="L1015"></a><tt class="py-lineno">1015</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_ss-toggle" onclick="return toggle('AbstractStructureParser._parse_ss');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_ss">_parse_ss</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_ss-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_ss-expanded"><a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">        Parse and attach secondary structure data.</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"><tt class="py-docstring">        @bug: Currently the PDB helix types are ignored. Each HELIX line is treated</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line"><tt class="py-docstring">              as a regular SecStructures.Helix. This is due to incompatibility </tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line"><tt class="py-docstring">              between DSSP and PDB helix types.</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"><tt class="py-docstring">        @todo: Implement a proper workaround for the previous bug (e.g. skip all</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring">               helices types not included in the DSSP enum)</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring">        @warning: In this implementation only the start/end positions of the SS </tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line"><tt class="py-docstring">                  elements are parsed. Additional data like H-bonding is ignored.</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line"><tt class="py-docstring">                  </tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line"><tt class="py-docstring">        @bug: Currently structure.to_pdb() is not writing any SS data. </tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">        <tt class="py-name">elements</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">         </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-485" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-485', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line"> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'HELIX'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">                 </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">                <tt id="link-486" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-486', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-487" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-487', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-488', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">19</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-489" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-489', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-490" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-490', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-491" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-491', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">elements</tt><tt class="py-op">:</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">                    <tt class="py-name">elements</tt><tt class="py-op">[</tt><tt id="link-492" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-492', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-493" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-493', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-494" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-494', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-495" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-495', 'id', 'link-29');">id</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">31</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-496" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-496', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-497" class="py-name" targets="Class csb.bio.io.wwpdb.SecStructureFormatError=csb.bio.io.wwpdb.SecStructureFormatError-class.html"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-497', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Helix {0} spans multiple chains'</tt><tt class="py-op">.</tt><tt id="link-498" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-498', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt>                 </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">                    <tt class="py-name">startres</tt> <tt class="py-op">=</tt> <tt id="link-499" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-499', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-500" class="py-name" targets="Method csb.bio.io.wwpdb.CustomStructureProvider.find()=csb.bio.io.wwpdb.CustomStructureProvider-class.html#find,Method csb.bio.io.wwpdb.FileSystemStructureProvider.find()=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#find,Method csb.bio.io.wwpdb.RemoteStructureProvider.find()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#find,Method csb.bio.io.wwpdb.StructureProvider.find()=csb.bio.io.wwpdb.StructureProvider-class.html#find,Function csb.bio.io.wwpdb.find()=csb.bio.io.wwpdb-module.html#find,Method csb.bio.structure.Chain.find()=csb.bio.structure.Chain-class.html#find,Method csb.bio.structure.DisorderedAtom.find()=csb.bio.structure.DisorderedAtom-class.html#find"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-500', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">:</tt><tt class="py-number">25</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-501', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">25</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-502', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>                 </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">                    <tt class="py-name">endres</tt> <tt class="py-op">=</tt> <tt id="link-503" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-503', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-504', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">33</tt><tt class="py-op">:</tt><tt class="py-number">37</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-505" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-505', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">37</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-506" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-506', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt id="link-507" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-507', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-508', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-509" class="py-name" targets="Class csb.core.ItemNotFoundError=csb.core.ItemNotFoundError-class.html"><a title="csb.core.ItemNotFoundError" class="py-name" href="#" onclick="return doclink('link-509', 'ItemNotFoundError', 'link-509');">ItemNotFoundError</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">ex</tt><tt class="py-op">:</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-510" class="py-name"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-510', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">                                <tt class="py-string">'Helix {0} refers to an undefined residue ID: {1}'</tt><tt class="py-op">.</tt><tt id="link-511" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-511', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">ex</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">startres</tt><tt class="py-op">.</tt><tt id="link-512" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-512', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endres</tt><tt class="py-op">.</tt><tt id="link-513" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-513', 'rank', 'link-312');">rank</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-514" class="py-name"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-514', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Helix {0} is out of range'</tt><tt class="py-op">.</tt><tt id="link-515" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-515', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt>                     </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">                <tt class="py-name">helix</tt> <tt class="py-op">=</tt> <tt id="link-516" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-516', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-517" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-517', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-518', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-519" class="py-name" targets="Class csb.bio.structure.SecondaryStructureElement=csb.bio.structure.SecondaryStructureElement-class.html"><a title="csb.bio.structure.SecondaryStructureElement" class="py-name" href="#" onclick="return doclink('link-519', 'SecondaryStructureElement', 'link-519');">SecondaryStructureElement</a></tt><tt class="py-op">(</tt><tt class="py-name">startres</tt><tt class="py-op">.</tt><tt id="link-520" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-520', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">endres</tt><tt class="py-op">.</tt><tt id="link-521" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-521', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-522" class="py-name"><a title="csb.bio.structure.SecStructures" class="py-name" href="#" onclick="return doclink('link-522', 'SecStructures', 'link-25');">SecStructures</a></tt><tt class="py-op">.</tt><tt id="link-523" class="py-name" targets="Variable csb.bio.structure.SecStructures.Helix=csb.bio.structure.SecStructures-class.html#Helix"><a title="csb.bio.structure.SecStructures.Helix" class="py-name" href="#" onclick="return doclink('link-523', 'Helix', 'link-523');">Helix</a></tt><tt class="py-op">)</tt>         </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">                <tt class="py-name">elements</tt><tt class="py-op">[</tt><tt id="link-524" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-524', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-525" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-525', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-526', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">helix</tt><tt class="py-op">)</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">             </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'SHEET'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">                 </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">                <tt id="link-527" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-527', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-528" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-528', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-529', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-530" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-530', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-531" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-531', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-532" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-532', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">elements</tt><tt class="py-op">:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">                    <tt class="py-name">elements</tt><tt class="py-op">[</tt><tt id="link-533" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-533', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-534" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-534', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>                 </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-535" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-535', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-536" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-536', 'id', 'link-29');">id</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">32</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-537" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-537', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt>                     </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-538" class="py-name"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-538', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sheet {0} spans multiple chains'</tt><tt class="py-op">.</tt><tt id="link-539" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-539', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">                    <tt class="py-name">startres</tt> <tt class="py-op">=</tt> <tt id="link-540" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-540', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-541" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-541', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">22</tt><tt class="py-op">:</tt><tt class="py-number">26</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-542" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-542', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">26</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-543" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-543', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>                 </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">                    <tt class="py-name">endres</tt> <tt class="py-op">=</tt> <tt id="link-544" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-544', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-545" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-545', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">33</tt><tt class="py-op">:</tt><tt class="py-number">37</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-546" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-546', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">37</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-547" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-547', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt id="link-548" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-548', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-549" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-549', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-550" class="py-name"><a title="csb.core.ItemNotFoundError" class="py-name" href="#" onclick="return doclink('link-550', 'ItemNotFoundError', 'link-509');">ItemNotFoundError</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">ex</tt><tt class="py-op">:</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt>                     </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-551" class="py-name"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-551', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">                                <tt class="py-string">'Sheet {0} refers to an undefined residue ID: {1}'</tt><tt class="py-op">.</tt><tt id="link-552" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-552', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">ex</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">startres</tt><tt class="py-op">.</tt><tt id="link-553" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-553', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endres</tt><tt class="py-op">.</tt><tt id="link-554" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-554', 'rank', 'link-312');">rank</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_check_ss</tt><tt class="py-op">:</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-555" class="py-name"><a title="csb.bio.io.wwpdb.SecStructureFormatError" class="py-name" href="#" onclick="return doclink('link-555', 'SecStructureFormatError', 'link-497');">SecStructureFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sheet {0} is out of range'</tt><tt class="py-op">.</tt><tt id="link-556" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-556', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">                <tt class="py-name">strand</tt> <tt class="py-op">=</tt> <tt id="link-557" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-557', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-558" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-558', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-559" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-559', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-560" class="py-name"><a title="csb.bio.structure.SecondaryStructureElement" class="py-name" href="#" onclick="return doclink('link-560', 'SecondaryStructureElement', 'link-519');">SecondaryStructureElement</a></tt><tt class="py-op">(</tt><tt class="py-name">startres</tt><tt class="py-op">.</tt><tt id="link-561" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-561', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">endres</tt><tt class="py-op">.</tt><tt id="link-562" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-562', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-563" class="py-name"><a title="csb.bio.structure.SecStructures" class="py-name" href="#" onclick="return doclink('link-563', 'SecStructures', 'link-25');">SecStructures</a></tt><tt class="py-op">.</tt><tt id="link-564" class="py-name" targets="Variable csb.bio.structure.SecStructures.Strand=csb.bio.structure.SecStructures-class.html#Strand"><a title="csb.bio.structure.SecStructures.Strand" class="py-name" href="#" onclick="return doclink('link-564', 'Strand', 'link-564');">Strand</a></tt><tt class="py-op">)</tt>       </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">                <tt class="py-name">elements</tt><tt class="py-op">[</tt><tt id="link-565" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-565', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-566" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-566', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-567', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">strand</tt><tt class="py-op">)</tt>          </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">             </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ATOM'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt>         </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">             </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-568" class="py-name" targets="Variable csb.bio.fragments.Target.chain_id=csb.bio.fragments.Target-class.html#chain_id"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-568', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">elements</tt><tt class="py-op">:</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">            <tt id="link-569" class="py-name" targets="Variable csb.apps.hhsearch.HHsearch.ss=csb.apps.hhsearch.HHsearch-class.html#ss"><a title="csb.apps.hhsearch.HHsearch.ss" class="py-name" href="#" onclick="return doclink('link-569', 'ss', 'link-569');">ss</a></tt> <tt class="py-op">=</tt> <tt id="link-570" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-570', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-571" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-571', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-572" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-572', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-573" class="py-name" targets="Class csb.bio.structure.SecondaryStructure=csb.bio.structure.SecondaryStructure-class.html"><a title="csb.bio.structure.SecondaryStructure" class="py-name" href="#" onclick="return doclink('link-573', 'SecondaryStructure', 'link-573');">SecondaryStructure</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">e</tt> <tt class="py-keyword">in</tt> <tt class="py-name">elements</tt><tt class="py-op">[</tt><tt id="link-574" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-574', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">                <tt id="link-575" class="py-name"><a title="csb.apps.hhsearch.HHsearch.ss" class="py-name" href="#" onclick="return doclink('link-575', 'ss', 'link-569');">ss</a></tt><tt class="py-op">.</tt><tt id="link-576" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-576', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">            <tt id="link-577" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-577', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-578" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-578', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-579" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-579', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-580" class="py-name" targets="Variable csb.bio.fragments.Assignment.secondary_structure=csb.bio.fragments.Assignment-class.html#secondary_structure,Variable csb.bio.structure.Chain.secondary_structure=csb.bio.structure.Chain-class.html#secondary_structure,Variable csb.bio.structure.Residue.secondary_structure=csb.bio.structure.Residue-class.html#secondary_structure"><a title="csb.bio.fragments.Assignment.secondary_structure
csb.bio.structure.Chain.secondary_structure
csb.bio.structure.Residue.secondary_structure" class="py-name" href="#" onclick="return doclink('link-580', 'secondary_structure', 'link-580');">secondary_structure</a></tt> <tt class="py-op">=</tt> <tt id="link-581" class="py-name"><a title="csb.apps.hhsearch.HHsearch.ss" class="py-name" href="#" onclick="return doclink('link-581', 'ss', 'link-569');">ss</a></tt> </tt>
</div><a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">         </tt>
<a name="AbstractStructureParser._parse_remarks"></a><div id="AbstractStructureParser._parse_remarks-def"><a name="L1102"></a><tt class="py-lineno">1102</tt> <a class="py-toggle" href="#" id="AbstractStructureParser._parse_remarks-toggle" onclick="return toggle('AbstractStructureParser._parse_remarks');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractStructureParser-class.html#_parse_remarks">_parse_remarks</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractStructureParser._parse_remarks-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractStructureParser._parse_remarks-expanded"><a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line"><tt class="py-docstring">        Read REMARK lines from PDB file.</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line"><tt class="py-docstring">        @return: dictionary with remark numbers as keys, and lists of lines as values.</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: dict</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">         </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt class="py-name">remarks</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">         </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">:</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'REMARK'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">                <tt class="py-name">num</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">                <tt class="py-name">lstring</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">                <tt class="py-name">remarks</tt><tt class="py-op">.</tt><tt class="py-name">setdefault</tt><tt class="py-op">(</tt><tt class="py-name">num</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-582" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-582', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">lstring</tt><tt class="py-op">)</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'DBREF'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ATOM'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt>         </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">         </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">remarks</tt> </tt>
</div></div><a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line"> </tt>
<a name="RegularStructureParser"></a><div id="RegularStructureParser-def"><a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt> <a class="py-toggle" href="#" id="RegularStructureParser-toggle" onclick="return toggle('RegularStructureParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RegularStructureParser-class.html">RegularStructureParser</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractStructureParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RegularStructureParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="RegularStructureParser-expanded"><a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line"><tt class="py-docstring">    This is the de facto PDB parser, which is designed to read SEQRES and ATOM</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line"><tt class="py-docstring">    sections separately, and them map them. Intentionally fails to parse</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line"><tt class="py-docstring">    malformed PDB files, e.g. a PDB file without a HEADER section.</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">     </tt>
<a name="RegularStructureParser._parse_header"></a><div id="RegularStructureParser._parse_header-def"><a name="L1131"></a><tt class="py-lineno">1131</tt> <a class="py-toggle" href="#" id="RegularStructureParser._parse_header-toggle" onclick="return toggle('RegularStructureParser._parse_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RegularStructureParser-class.html#_parse_header">_parse_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RegularStructureParser._parse_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RegularStructureParser._parse_header-expanded"><a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line"><tt class="py-docstring">        Parse the HEADER section of a regular PDB file.</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a L{csb.bio.structure.Structure} instance with properly </tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"><tt class="py-docstring">                 initialized residues from the SEQRES.</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.bio.structure.Structure}</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line"><tt class="py-docstring">        @raise PDBParseError: if the stream has no HEADER at byte 0</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">         </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line"> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt id="link-583" class="py-name" targets="Variable csb.bio.sequence.AbstractSequence.header=csb.bio.sequence.AbstractSequence-class.html#header,Variable csb.bio.sequence.SequenceAdapter.header=csb.bio.sequence.SequenceAdapter-class.html#header,Variable csb.bio.structure.Chain.header=csb.bio.structure.Chain-class.html#header"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-583', 'header', 'link-583');">header</a></tt> <tt class="py-op">=</tt> <tt id="link-584" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-584', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-585" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-585', 'header', 'link-583');">header</a></tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'HEADER'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-586" class="py-name"><a title="csb.bio.io.wwpdb.PDBParseError" class="py-name" href="#" onclick="return doclink('link-586', 'PDBParseError', 'link-159');">PDBParseError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Does not look like a regular PDB file.'</tt><tt class="py-op">)</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line"> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt id="link-587" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-587', 'structure', 'link-2');">structure</a></tt> <tt class="py-op">=</tt> <tt id="link-588" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-588', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-589" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-589', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-590" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-590', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="csb.bio.structure.Structure" class="py-name" href="#" onclick="return doclink('link-591', 'Structure', 'link-210');">Structure</a></tt><tt class="py-op">(</tt><tt id="link-592" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-592', 'header', 'link-583');">header</a></tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line"> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line"> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-593" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-593', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line"> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'COMPND'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">lstrip</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MOL_ID:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">                    <tt class="py-name">mol_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">18</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-594" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-594', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">                    <tt class="py-name">chain_name</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">                    <tt id="link-595" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-595', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'COMPND'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">                        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-596" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-596', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MOLECULE:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">                            <tt class="py-name">chain_name</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">20</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-597" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-597', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">                            <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">chain_name</tt><tt class="py-op">.</tt><tt class="py-name">endswith</tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">                                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-598" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-598', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">                                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'COMPND'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">                                        <tt class="py-keyword">break</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">                                <tt class="py-name">chain_name</tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-599" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-599', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">                            <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'CHAIN:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">                                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-600" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-600', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">                                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'COMPND'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">                                    <tt class="py-keyword">raise</tt> <tt id="link-601" class="py-name"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-601', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Missing chain identifier in COMPND section'</tt><tt class="py-op">)</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">                            <tt id="link-602" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-602', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">17</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-603" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-603', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">                            <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt id="link-604" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-604', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt class="py-name">endswith</tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">                                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-605" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-605', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">                                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'COMPND'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">                                    <tt class="py-keyword">break</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">                                <tt id="link-606" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-606', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">', '</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-607', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">                            <tt class="py-keyword">break</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line"> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">                    <tt class="py-name">chain_ids</tt> <tt class="py-op">=</tt> <tt id="link-608" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-608', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-string">''</tt><tt class="py-op">]</tt>  <tt class="py-comment"># the second part deals with an empty chain id</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_chains</tt><tt class="py-op">(</tt><tt id="link-609" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-609', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt class="py-name">chain_name</tt><tt class="py-op">,</tt> <tt class="py-name">mol_id</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">chain_ids</tt><tt class="py-op">)</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'REMARK   2 RESOLUTION'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">                <tt id="link-610" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-610', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-611" class="py-name" targets="Variable csb.bio.structure.Structure.resolution=csb.bio.structure.Structure-class.html#resolution"><a title="csb.bio.structure.Structure.resolution" class="py-name" href="#" onclick="return doclink('link-611', 'resolution', 'link-611');">resolution</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_resolution</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">                                         </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'SEQRES'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">                <tt id="link-612" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-612', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">,</tt> <tt id="link-613" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-613', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_seqres_line</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-614" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-614', 'structure', 'link-2');">structure</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">                <tt id="link-615" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-615', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-616" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-616', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-617" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-617', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-618" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-618', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">                 </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-619" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-619', 'residue', 'link-372');">residue</a></tt> <tt class="py-keyword">in</tt> <tt id="link-620" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-620', 'residues', 'link-262');">residues</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">                    <tt id="link-621" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-621', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-622" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-622', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-623" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-623', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-624" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-624', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-625" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-625', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-626" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-626', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-627" class="py-name" targets="Variable csb.core.BaseCollectionContainer.last_index=csb.core.BaseCollectionContainer-class.html#last_index"><a title="csb.core.BaseCollectionContainer.last_index" class="py-name" href="#" onclick="return doclink('link-627', 'last_index', 'link-627');">last_index</a></tt> <tt class="py-op">!=</tt> <tt id="link-628" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-628', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-629" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-629', 'rank', 'link-312');">rank</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt id="link-630" class="py-name"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-630', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Malformed SEQRES"</tt><tt class="py-op">)</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ATOM'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">                         </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-631" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-631', 'structure', 'link-2');">structure</a></tt>   </tt>
</div><a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">     </tt>
<a name="RegularStructureParser._add_chains"></a><div id="RegularStructureParser._add_chains-def"><a name="L1204"></a><tt class="py-lineno">1204</tt> <a class="py-toggle" href="#" id="RegularStructureParser._add_chains-toggle" onclick="return toggle('RegularStructureParser._add_chains');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RegularStructureParser-class.html#_add_chains">_add_chains</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">mol_id</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">chain_ids</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RegularStructureParser._add_chains-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RegularStructureParser._add_chains-expanded"><a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">        <tt id="link-632" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-632', 'name', 'link-82');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-633" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-633', 'name', 'link-82');">name</a></tt><tt class="py-op">.</tt><tt id="link-634" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-634', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">rstrip</tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">         </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-635" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-635', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">chain_ids</tt><tt class="py-op">:</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">            <tt class="py-name">new_chain</tt> <tt class="py-op">=</tt> <tt id="link-636" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-636', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-637" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-637', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-638" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-638', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-639" class="py-name" targets="Class csb.bio.structure.Chain=csb.bio.structure.Chain-class.html"><a title="csb.bio.structure.Chain" class="py-name" href="#" onclick="return doclink('link-639', 'Chain', 'link-639');">Chain</a></tt><tt class="py-op">(</tt><tt id="link-640" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-640', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt id="link-641" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-641', 'type', 'link-109');">type</a></tt><tt class="py-op">=</tt><tt id="link-642" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-642', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-643" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-643', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">                                                <tt id="link-644" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-644', 'name', 'link-82');">name</a></tt><tt class="py-op">=</tt><tt id="link-645" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-645', 'name', 'link-82');">name</a></tt><tt class="py-op">,</tt> <tt id="link-646" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-646', 'accession', 'link-212');">accession</a></tt><tt class="py-op">=</tt><tt id="link-647" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-647', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-648" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-648', 'accession', 'link-212');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">            <tt class="py-name">new_chain</tt><tt class="py-op">.</tt><tt id="link-649" class="py-name" targets="Variable csb.bio.structure.Chain.molecule_id=csb.bio.structure.Chain-class.html#molecule_id"><a title="csb.bio.structure.Chain.molecule_id" class="py-name" href="#" onclick="return doclink('link-649', 'molecule_id', 'link-649');">molecule_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">mol_id</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">                <tt id="link-650" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-650', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-651" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-651', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt id="link-652" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-652', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">new_chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt id="link-653" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-653', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-654" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-654', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-655" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-655', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-656" class="py-name" targets="Class csb.bio.structure.DuplicateChainIDError=csb.bio.structure.DuplicateChainIDError-class.html"><a title="csb.bio.structure.DuplicateChainIDError" class="py-name" href="#" onclick="return doclink('link-656', 'DuplicateChainIDError', 'link-656');">DuplicateChainIDError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-657" class="py-name"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-657', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Chain {0} is already defined.'</tt><tt class="py-op">.</tt><tt id="link-658" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-658', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">new_chain</tt><tt class="py-op">.</tt><tt id="link-659" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-659', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">     </tt>
<a name="RegularStructureParser._read_resolution"></a><div id="RegularStructureParser._read_resolution-def"><a name="L1217"></a><tt class="py-lineno">1217</tt> <a class="py-toggle" href="#" id="RegularStructureParser._read_resolution-toggle" onclick="return toggle('RegularStructureParser._read_resolution');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RegularStructureParser-class.html#_read_resolution">_read_resolution</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RegularStructureParser._read_resolution-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RegularStructureParser._read_resolution-expanded"><a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line"><tt class="py-docstring">        @return: resolution</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: float or None</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt class="py-name">res</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">search</tt><tt class="py-op">(</tt><tt class="py-string">"(\d+(?:\.\d+)?)\s+ANGSTROM"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">         </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">res</tt> <tt class="py-keyword">and</tt> <tt class="py-name">res</tt><tt class="py-op">.</tt><tt id="link-660" class="py-name" targets="Variable csb.bio.io.clans.ClansEntry.groups=csb.bio.io.clans.ClansEntry-class.html#groups,Method csb.core.REMatchProxy.groups()=csb.core.REMatchProxy-class.html#groups"><a title="csb.bio.io.clans.ClansEntry.groups
csb.core.REMatchProxy.groups" class="py-name" href="#" onclick="return doclink('link-660', 'groups', 'link-660');">groups</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">res</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt>          </tt>
</div><a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line"> </tt>
<a name="RegularStructureParser._parse_seqres_line"></a><div id="RegularStructureParser._parse_seqres_line-def"><a name="L1229"></a><tt class="py-lineno">1229</tt> <a class="py-toggle" href="#" id="RegularStructureParser._parse_seqres_line-toggle" onclick="return toggle('RegularStructureParser._parse_seqres_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RegularStructureParser-class.html#_parse_seqres_line">_parse_seqres_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RegularStructureParser._parse_seqres_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RegularStructureParser._parse_seqres_line-expanded"><a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line"><tt class="py-docstring">        Parse a SEQRES line, build and return newly constructed residues.</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line"><tt class="py-docstring">        If the current sequence type of the chain is unknown, try to guess it</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line"><tt class="py-docstring">        before parsing the residues.</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line"><tt class="py-docstring">        @return: parsed chain_id and L{Residue}s</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: 2-tuple: (str, iterable of L{Residue})</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">        <tt id="link-661" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-661', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">         </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">        <tt class="py-name">rownum</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">        <tt id="link-662" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-662', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-663" class="py-name"><a title="csb.bio.sequence.AbstractSequence.strip
csb.bio.sequence.Sequence.strip
csb.bio.sequence.SequenceAdapter.strip" class="py-name" href="#" onclick="return doclink('link-663', 'strip', 'link-31');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">        <tt id="link-664" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-664', 'labels', 'link-104');">labels</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">18</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">         </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-665" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-665', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-666" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-666', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-667" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-667', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-668" class="py-name"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-668', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">'Chain {0} is undefined'</tt><tt class="py-op">.</tt><tt id="link-669" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-669', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-670" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-670', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">         </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt id="link-671" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-671', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-672" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-672', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-673" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-673', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-674" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-674', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">         </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-675" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-675', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-676" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-676', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt id="link-677" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-677', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-678" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-678', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">            <tt id="link-679" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-679', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-680" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-680', 'type', 'link-109');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-681" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractStructureParser.guess_chain_type()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#guess_chain_type"><a title="csb.bio.io.wwpdb.AbstractStructureParser.guess_chain_type" class="py-name" href="#" onclick="return doclink('link-681', 'guess_chain_type', 'link-681');">guess_chain_type</a></tt><tt class="py-op">(</tt><tt id="link-682" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-682', 'labels', 'link-104');">labels</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line"> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rn</tt><tt class="py-op">,</tt> <tt id="link-683" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-683', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-684" class="py-name"><a title="csb.test.AbstractTestBuilder.labels
csb.test.AnyTestBuilder.labels
csb.test.CustomTestBuilder.labels
csb.test.FunctionalTestBuilder.labels
csb.test.RegressionTestBuilder.labels
csb.test.UnitTestBuilder.labels" class="py-name" href="#" onclick="return doclink('link-684', 'labels', 'link-104');">labels</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt id="link-685" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-685', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rownum</tt> <tt class="py-op">*</tt> <tt class="py-number">13</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-number">13</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">rn</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">            <tt class="py-name">rtype</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-686" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_residue_safe" class="py-name" href="#" onclick="return doclink('link-686', 'parse_residue_safe', 'link-373');">parse_residue_safe</a></tt><tt class="py-op">(</tt><tt id="link-687" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-687', 'label', 'link-108');">label</a></tt><tt class="py-op">,</tt> <tt class="py-name">as_type</tt><tt class="py-op">=</tt><tt id="link-688" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-688', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-689" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-689', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">             </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">            <tt id="link-690" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-690', 'residue', 'link-372');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-691" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-691', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-692" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-692', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-693" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-693', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-694" class="py-name" targets="Class csb.bio.structure.Residue=csb.bio.structure.Residue-class.html"><a title="csb.bio.structure.Residue" class="py-name" href="#" onclick="return doclink('link-694', 'Residue', 'link-694');">Residue</a></tt><tt class="py-op">.</tt><tt id="link-695" class="py-name"><a title="csb.apps.buildhmm.ProfileBuilder.create
csb.apps.hhfrag.PredictionBuilder.create
csb.bio.fragments.ClusterNode.create
csb.bio.io.cs.ChemShiftReader.create
csb.bio.io.fasta.OutputBuilder.create
csb.bio.io.wwpdb.EntryID.create
csb.bio.io.wwpdb.SparseChainSequence.create
csb.bio.sequence.ChainSequence.create
csb.bio.sequence.RichSequence.create
csb.bio.structure.CompositeEntityIterator.create
csb.bio.structure.Residue.create
csb.core.Enum.create
csb.io.tsv.RepositoryImp.create
csb.io.tsv.SQLiteRepository.create
csb.statistics.samplers.mc.TrajectoryBuilder.create" class="py-name" href="#" onclick="return doclink('link-695', 'create', 'link-457');">create</a></tt><tt class="py-op">(</tt><tt id="link-696" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-696', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-697" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-697', 'type', 'link-109');">type</a></tt><tt class="py-op">,</tt> <tt id="link-698" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-698', 'rank', 'link-312');">rank</a></tt><tt class="py-op">=</tt><tt id="link-699" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-699', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-700" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-700', 'type', 'link-109');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">rtype</tt><tt class="py-op">)</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">            <tt id="link-701" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-701', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-702" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-702', 'label', 'link-108');">label</a></tt> <tt class="py-op">=</tt> <tt id="link-703" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-703', 'label', 'link-108');">label</a></tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">            <tt id="link-704" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-704', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-705" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-705', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-706" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-706', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">             </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-707" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-707', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">,</tt> <tt id="link-708" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-708', 'residues', 'link-262');">residues</a></tt> </tt>
</div></div><a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">     </tt>
<a name="PDBHeaderParser"></a><div id="PDBHeaderParser-def"><a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line"> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt> <a class="py-toggle" href="#" id="PDBHeaderParser-toggle" onclick="return toggle('PDBHeaderParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBHeaderParser-class.html">PDBHeaderParser</a><tt class="py-op">(</tt><tt class="py-base-class">RegularStructureParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBHeaderParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PDBHeaderParser-expanded"><a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line"><tt class="py-docstring">    Ultra fast PDB HEADER parser. Does not read any structural data.</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">     </tt>
<a name="PDBHeaderParser._parse_atoms"></a><div id="PDBHeaderParser._parse_atoms-def"><a name="L1268"></a><tt class="py-lineno">1268</tt> <a class="py-toggle" href="#" id="PDBHeaderParser._parse_atoms-toggle" onclick="return toggle('PDBHeaderParser._parse_atoms');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBHeaderParser-class.html#_parse_atoms">_parse_atoms</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBHeaderParser._parse_atoms-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBHeaderParser._parse_atoms-expanded"><a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line">     </tt>
<a name="PDBHeaderParser._parse_ss"></a><div id="PDBHeaderParser._parse_ss-def"><a name="L1271"></a><tt class="py-lineno">1271</tt> <a class="py-toggle" href="#" id="PDBHeaderParser._parse_ss-toggle" onclick="return toggle('PDBHeaderParser._parse_ss');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBHeaderParser-class.html#_parse_ss">_parse_ss</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBHeaderParser._parse_ss-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBHeaderParser._parse_ss-expanded"><a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">     </tt>
<a name="PDBHeaderParser._parse_header"></a><div id="PDBHeaderParser._parse_header-def"><a name="L1274"></a><tt class="py-lineno">1274</tt> <a class="py-toggle" href="#" id="PDBHeaderParser._parse_header-toggle" onclick="return toggle('PDBHeaderParser._parse_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBHeaderParser-class.html#_parse_header">_parse_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBHeaderParser._parse_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBHeaderParser._parse_header-expanded"><a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-709" class="py-name" targets="Class csb.bio.io.wwpdb.PDBHeaderParser=csb.bio.io.wwpdb.PDBHeaderParser-class.html"><a title="csb.bio.io.wwpdb.PDBHeaderParser" class="py-name" href="#" onclick="return doclink('link-709', 'PDBHeaderParser', 'link-709');">PDBHeaderParser</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">_parse_header</tt><tt class="py-op">(</tt><tt id="link-710" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-710', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">         </tt>
<a name="LegacyStructureParser"></a><div id="LegacyStructureParser-def"><a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line"> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt> <a class="py-toggle" href="#" id="LegacyStructureParser-toggle" onclick="return toggle('LegacyStructureParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html">LegacyStructureParser</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractStructureParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="LegacyStructureParser-expanded"><a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line"><tt class="py-docstring">    This is a customized PDB parser, which is designed to read both sequence and</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line"><tt class="py-docstring">    atom data from the ATOM section. This is especially useful when parsing PDB</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line"><tt class="py-docstring">    files without a header.</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">     </tt>
<a name="LegacyStructureParser._parse_header"></a><div id="LegacyStructureParser._parse_header-def"><a name="L1285"></a><tt class="py-lineno">1285</tt> <a class="py-toggle" href="#" id="LegacyStructureParser._parse_header-toggle" onclick="return toggle('LegacyStructureParser._parse_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html#_parse_header">_parse_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser._parse_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="LegacyStructureParser._parse_header-expanded"><a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line"><tt class="py-docstring">        Initialize a structure with residues from the ATOMs section.</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: model identifier (e.g. if multiple models exist)</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: str </tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a L{csb.bio.structure.Structure} instance with properly </tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line"><tt class="py-docstring">                 initialized residues from ATOMs under the specified C{model}.</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.bio.structure.Structure}</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">          </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">        <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">        <tt class="py-name">has_atoms</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">        <tt class="py-name">has_model</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">        <tt id="link-711" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-711', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt id="link-712" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-712', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-713" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-713', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-714" class="py-name" targets="Class csb.core.OrderedDict=csb.core.OrderedDict-class.html"><a title="csb.core.OrderedDict" class="py-name" href="#" onclick="return doclink('link-714', 'OrderedDict', 'link-714');">OrderedDict</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line"> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">        <tt id="link-715" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-715', 'header', 'link-583');">header</a></tt> <tt class="py-op">=</tt> <tt id="link-716" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-716', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-717" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-717', 'header', 'link-583');">header</a></tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'HEADER'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">            <tt id="link-718" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-718', 'structure', 'link-2');">structure</a></tt> <tt class="py-op">=</tt> <tt id="link-719" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-719', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-720" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-720', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-721" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-721', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-722" class="py-name"><a title="csb.bio.structure.Structure" class="py-name" href="#" onclick="return doclink('link-722', 'Structure', 'link-210');">Structure</a></tt><tt class="py-op">(</tt><tt id="link-723" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-723', 'header', 'link-583');">header</a></tt><tt class="py-op">.</tt><tt class="py-name">split</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">.</tt><tt class="py-name">seek</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">            <tt id="link-724" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-724', 'structure', 'link-2');">structure</a></tt> <tt class="py-op">=</tt> <tt id="link-725" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-725', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-726" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-726', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-727" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-727', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-728" class="py-name"><a title="csb.bio.structure.Structure" class="py-name" href="#" onclick="return doclink('link-728', 'Structure', 'link-210');">Structure</a></tt><tt class="py-op">(</tt><tt class="py-string">'NONE'</tt><tt class="py-op">)</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line"> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">        <tt id="link-729" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-729', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-730" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-730', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">         </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line"> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-731" class="py-name"><a title="csb.bio.io.fasta.A3MSequenceIterator.next
csb.bio.structure.CompositeEntityIterator.next
csb.statistics.samplers.mc.propagators.Looper.next" class="py-name" href="#" onclick="return doclink('link-731', 'next', 'link-237');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_stream</tt><tt class="py-op">)</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line"> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'MODEL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">has_model</tt><tt class="py-op">:</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parse_model_line</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-732" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-732', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt id="link-733" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-733', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">                    <tt id="link-734" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-734', 'model', 'link-153');">model</a></tt> <tt class="py-op">=</tt> <tt id="link-735" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-735', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-736" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-736', 'model_id', 'link-90');">model_id</a></tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">                    <tt class="py-name">has_model</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line"> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ATOM'</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">                     <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-name">in_atom</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'HETATM'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">                    <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">                    <tt class="py-name">has_atoms</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">                     </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">                    <tt class="py-name">seq_number</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_sequence_number</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">                    <tt class="py-name">ins_code</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_insertion_code</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">                    <tt class="py-name">residue_id</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">seq_number</tt><tt class="py-op">,</tt> <tt class="py-name">ins_code</tt><tt class="py-op">)</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">                    <tt id="link-737" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-737', 'label', 'link-108');">label</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_residue_raw</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">                    <tt id="link-738" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-738', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_read_chain_id</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">                     </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-739" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-739', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-740" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-740', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">                        <tt id="link-741" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-741', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-742" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-742', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-743" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-743', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-744" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-744', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-745" class="py-name"><a title="csb.core.OrderedDict" class="py-name" href="#" onclick="return doclink('link-745', 'OrderedDict', 'link-714');">OrderedDict</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_add_chain</tt><tt class="py-op">(</tt><tt id="link-746" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-746', 'structure', 'link-2');">structure</a></tt><tt class="py-op">,</tt> <tt id="link-747" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-747', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">                         </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">residue_id</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-748" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-748', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-749" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-749', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">                        <tt id="link-750" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-750', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-751" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-751', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">residue_id</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-752" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-752', 'label', 'link-108');">label</a></tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">                        <tt id="link-753" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-753', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-754" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-754', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-755" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-755', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-756" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-756', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt>  </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt id="link-757" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-757', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-758" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-758', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt id="link-759" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-759', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-760" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-760', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">                            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_fix_chain</tt><tt class="py-op">(</tt><tt id="link-761" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-761', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt id="link-762" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-762', 'label', 'link-108');">label</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line"> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">in_atom</tt> <tt class="py-keyword">and</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'TER'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">                <tt class="py-name">in_atom</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'ENDMDL'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">startswith</tt><tt class="py-op">(</tt><tt class="py-string">'END'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line"> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_atoms</tt><tt class="py-op">:</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-763" class="py-name"><a title="csb.bio.io.wwpdb.HeaderFormatError" class="py-name" href="#" onclick="return doclink('link-763', 'HeaderFormatError', 'link-160');">HeaderFormatError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Can't parse legacy structure: no ATOMs found"</tt><tt class="py-op">)</tt>                                      </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line"> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-764" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-764', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-765" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-765', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-766" class="py-name"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-766', 'items', 'link-216');">items</a></tt><tt class="py-op">:</tt>     </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_build_chain</tt><tt class="py-op">(</tt><tt id="link-767" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-767', 'chain', 'link-220');">chain</a></tt><tt class="py-op">,</tt> <tt id="link-768" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-768', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-769" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-769', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-770" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-770', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">         </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-771" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-771', 'structure', 'link-2');">structure</a></tt> </tt>
</div><a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">     </tt>
<a name="LegacyStructureParser._add_chain"></a><div id="LegacyStructureParser._add_chain-def"><a name="L1363"></a><tt class="py-lineno">1363</tt> <a class="py-toggle" href="#" id="LegacyStructureParser._add_chain-toggle" onclick="return toggle('LegacyStructureParser._add_chain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html#_add_chain">_add_chain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">chain_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser._add_chain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="LegacyStructureParser._add_chain-expanded"><a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line"> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">        <tt class="py-name">new_chain</tt> <tt class="py-op">=</tt> <tt id="link-772" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-772', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-773" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-773', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-774" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-774', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-775" class="py-name"><a title="csb.bio.structure.Chain" class="py-name" href="#" onclick="return doclink('link-775', 'Chain', 'link-639');">Chain</a></tt><tt class="py-op">(</tt><tt id="link-776" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-776', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">                                            <tt id="link-777" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-777', 'type', 'link-109');">type</a></tt><tt class="py-op">=</tt><tt id="link-778" class="py-name"><a title="csb.bio.sequence.SequenceTypes" class="py-name" href="#" onclick="return doclink('link-778', 'SequenceTypes', 'link-19');">SequenceTypes</a></tt><tt class="py-op">.</tt><tt id="link-779" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Unknown" class="py-name" href="#" onclick="return doclink('link-779', 'Unknown', 'link-114');">Unknown</a></tt><tt class="py-op">,</tt>  </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">                                            <tt id="link-780" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-780', 'accession', 'link-212');">accession</a></tt><tt class="py-op">=</tt><tt id="link-781" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-781', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-782" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-782', 'accession', 'link-212');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">        <tt class="py-name">new_chain</tt><tt class="py-op">.</tt><tt id="link-783" class="py-name"><a title="csb.bio.structure.Chain.molecule_id" class="py-name" href="#" onclick="return doclink('link-783', 'molecule_id', 'link-649');">molecule_id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'1'</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt id="link-784" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-784', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-785" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-785', 'chains', 'link-193');">chains</a></tt><tt class="py-op">.</tt><tt id="link-786" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-786', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">new_chain</tt><tt class="py-op">)</tt>       </tt>
</div><a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">                      </tt>
<a name="LegacyStructureParser._build_chain"></a><div id="LegacyStructureParser._build_chain-def"><a name="L1371"></a><tt class="py-lineno">1371</tt> <a class="py-toggle" href="#" id="LegacyStructureParser._build_chain-toggle" onclick="return toggle('LegacyStructureParser._build_chain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html#_build_chain">_build_chain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser._build_chain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="LegacyStructureParser._build_chain-expanded"><a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">             </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">residue_id</tt><tt class="py-op">,</tt> <tt id="link-787" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-787', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">in</tt> <tt id="link-788" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-788', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-789" class="py-name"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-789', 'items', 'link-216');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">            <tt id="link-790" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-790', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt id="link-791" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-791', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-792" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-792', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-793" class="py-name"><a title="csb.core.BaseCollectionContainer.last_index" class="py-name" href="#" onclick="return doclink('link-793', 'last_index', 'link-627');">last_index</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">             </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">            <tt class="py-name">rname</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-794" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_residue_safe" class="py-name" href="#" onclick="return doclink('link-794', 'parse_residue_safe', 'link-373');">parse_residue_safe</a></tt><tt class="py-op">(</tt><tt id="link-795" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-795', 'label', 'link-108');">label</a></tt><tt class="py-op">,</tt> <tt class="py-name">as_type</tt><tt class="py-op">=</tt><tt id="link-796" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-796', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-797" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-797', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">            <tt id="link-798" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-798', 'residue', 'link-372');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-799" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-799', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-800" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-800', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-801" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-801', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-802" class="py-name"><a title="csb.bio.structure.Residue" class="py-name" href="#" onclick="return doclink('link-802', 'Residue', 'link-694');">Residue</a></tt><tt class="py-op">.</tt><tt id="link-803" class="py-name"><a title="csb.apps.buildhmm.ProfileBuilder.create
csb.apps.hhfrag.PredictionBuilder.create
csb.bio.fragments.ClusterNode.create
csb.bio.io.cs.ChemShiftReader.create
csb.bio.io.fasta.OutputBuilder.create
csb.bio.io.wwpdb.EntryID.create
csb.bio.io.wwpdb.SparseChainSequence.create
csb.bio.sequence.ChainSequence.create
csb.bio.sequence.RichSequence.create
csb.bio.structure.CompositeEntityIterator.create
csb.bio.structure.Residue.create
csb.core.Enum.create
csb.io.tsv.RepositoryImp.create
csb.io.tsv.SQLiteRepository.create
csb.statistics.samplers.mc.TrajectoryBuilder.create" class="py-name" href="#" onclick="return doclink('link-803', 'create', 'link-457');">create</a></tt><tt class="py-op">(</tt><tt id="link-804" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-804', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-805" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-805', 'type', 'link-109');">type</a></tt><tt class="py-op">,</tt> <tt id="link-806" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-806', 'rank', 'link-312');">rank</a></tt><tt class="py-op">=</tt><tt id="link-807" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-807', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-808" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-808', 'type', 'link-109');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">rname</tt><tt class="py-op">)</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line">            <tt id="link-809" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-809', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-810" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-810', 'label', 'link-108');">label</a></tt> <tt class="py-op">=</tt> <tt id="link-811" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-811', 'label', 'link-108');">label</a></tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line">            <tt id="link-812" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-812', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-813" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-813', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">residue_id</tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">            <tt id="link-814" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-814', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-815" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-815', 'residues', 'link-262');">residues</a></tt><tt class="py-op">.</tt><tt id="link-816" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-816', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-817" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-817', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line"> </tt>
<a name="LegacyStructureParser._fix_chain"></a><div id="LegacyStructureParser._fix_chain-def"><a name="L1382"></a><tt class="py-lineno">1382</tt> <a class="py-toggle" href="#" id="LegacyStructureParser._fix_chain-toggle" onclick="return toggle('LegacyStructureParser._fix_chain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html#_fix_chain">_fix_chain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain</tt><tt class="py-op">,</tt> <tt class="py-param">probe</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser._fix_chain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="LegacyStructureParser._fix_chain-expanded"><a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line">                 </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line">            <tt id="link-818" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-818', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-819" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-819', 'type', 'link-109');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-820" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.guess_sequence_type" class="py-name" href="#" onclick="return doclink('link-820', 'guess_sequence_type', 'link-110');">guess_sequence_type</a></tt><tt class="py-op">(</tt><tt class="py-name">probe</tt><tt class="py-op">)</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-821" class="py-name"><a title="csb.bio.io.wwpdb.UnknownPDBResidueError" class="py-name" href="#" onclick="return doclink('link-821', 'UnknownPDBResidueError', 'link-116');">UnknownPDBResidueError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt>   </tt>
</div><a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">     </tt>
<a name="LegacyStructureParser._map_residues"></a><div id="LegacyStructureParser._map_residues-def"><a name="L1389"></a><tt class="py-lineno">1389</tt> <a class="py-toggle" href="#" id="LegacyStructureParser._map_residues-toggle" onclick="return toggle('LegacyStructureParser._map_residues');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.LegacyStructureParser-class.html#_map_residues">_map_residues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LegacyStructureParser._map_residues-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="LegacyStructureParser._map_residues-expanded"><a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line"> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-822" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-822', 'chain', 'link-220');">chain</a></tt> <tt class="py-keyword">in</tt> <tt id="link-823" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-823', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-824" class="py-name"><a title="csb.bio.fragments.FragmentCluster.items
csb.bio.structure.AbstractEntity.items
csb.bio.structure.Atom.items
csb.bio.structure.Chain.items
csb.bio.structure.Ensemble.items
csb.bio.structure.Residue.items
csb.bio.structure.Structure.items" class="py-name" href="#" onclick="return doclink('link-824', 'items', 'link-216');">items</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">residue_info</tt> <tt class="py-keyword">in</tt> <tt id="link-825" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-825', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt id="link-826" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-826', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-827" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-827', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line">                    <tt id="link-828" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-828', 'residue', 'link-372');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-829" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-829', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-830" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-830', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_info</tt><tt class="py-op">.</tt><tt id="link-831" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-831', 'sequence_number', 'link-313');">sequence_number</a></tt><tt class="py-op">,</tt> <tt class="py-name">residue_info</tt><tt class="py-op">.</tt><tt id="link-832" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-832', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">residue_info</tt><tt class="py-op">.</tt><tt id="link-833" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-833', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">                        <tt id="link-834" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-834', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-835" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-835', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-836" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-836', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line">                     </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt id="link-837" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-837', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-838" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-838', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-839" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-839', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-840" class="py-name" targets="Class csb.bio.structure.EntityNotFoundError=csb.bio.structure.EntityNotFoundError-class.html"><a title="csb.bio.structure.EntityNotFoundError" class="py-name" href="#" onclick="return doclink('link-840', 'EntityNotFoundError', 'link-840');">EntityNotFoundError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line">                    <tt class="py-keyword">pass</tt>          </tt>
</div></div><a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line">                 </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">     </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line"><tt id="link-841" class="py-name" targets="Function csb.bio.io.StructureParser()=csb.bio.io-module.html#StructureParser"><a title="csb.bio.io.StructureParser" class="py-name" href="#" onclick="return doclink('link-841', 'StructureParser', 'link-841');">StructureParser</a></tt> <tt class="py-op">=</tt> <tt id="link-842" class="py-name" targets="Class csb.bio.io.wwpdb.AbstractStructureParser=csb.bio.io.wwpdb.AbstractStructureParser-class.html"><a title="csb.bio.io.wwpdb.AbstractStructureParser" class="py-name" href="#" onclick="return doclink('link-842', 'AbstractStructureParser', 'link-842');">AbstractStructureParser</a></tt><tt class="py-op">.</tt><tt id="link-843" class="py-name" targets="Static Method csb.bio.io.wwpdb.AbstractStructureParser.create_parser()=csb.bio.io.wwpdb.AbstractStructureParser-class.html#create_parser"><a title="csb.bio.io.wwpdb.AbstractStructureParser.create_parser" class="py-name" href="#" onclick="return doclink('link-843', 'create_parser', 'link-843');">create_parser</a></tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line"><tt class="py-string">Alias for L{AbstractStructureParser.create_parser}.</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line"><tt class="py-string">"""</tt> </tt>
<a name="ResidueInfo"></a><div id="ResidueInfo-def"><a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line"> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line"> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt> <a class="py-toggle" href="#" id="ResidueInfo-toggle" onclick="return toggle('ResidueInfo');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.ResidueInfo-class.html">ResidueInfo</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ResidueInfo-expanded"><a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line"><tt class="py-docstring">    High-performance struct, which functions as a container for unmapped</tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line"><tt class="py-docstring">    L{Atom}s.</tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line"><tt class="py-docstring">    @note: This object must implement the L{csb.bio.sequence.ResidueInfo}</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line"><tt class="py-docstring">           interface. This is not enforced through inheritance solely</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line"><tt class="py-docstring">           to save some CPU (by exposing public fields and no properties).</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line"><tt class="py-docstring">           However, on an abstract level this object is_a ResidueInfo</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line"><tt class="py-docstring">           and is used to build L{AbstractSequence}s.</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">    <tt class="py-name">__slots__</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'chain'</tt><tt class="py-op">,</tt> <tt class="py-string">'rank'</tt><tt class="py-op">,</tt> <tt class="py-string">'id'</tt> <tt class="py-op">,</tt> <tt class="py-string">'sequence_number'</tt><tt class="py-op">,</tt> <tt class="py-string">'insertion_code'</tt><tt class="py-op">,</tt> <tt class="py-string">'type'</tt><tt class="py-op">,</tt> <tt class="py-string">'label'</tt><tt class="py-op">,</tt> <tt class="py-string">'atoms'</tt><tt class="py-op">]</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">     </tt>
<a name="ResidueInfo.__init__"></a><div id="ResidueInfo.__init__-def"><a name="L1421"></a><tt class="py-lineno">1421</tt> <a class="py-toggle" href="#" id="ResidueInfo.__init__-toggle" onclick="return toggle('ResidueInfo.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.ResidueInfo-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">seq_number</tt><tt class="py-op">,</tt> <tt class="py-param">ins_code</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">,</tt> <tt class="py-param">label</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.__init__-expanded"><a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">         </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-844" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-844', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-845" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-845', 'chain', 'link-220');">chain</a></tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-846" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-846', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-847" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-847', 'rank', 'link-312');">rank</a></tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-848" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-848', 'id', 'link-29');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-849" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-849', 'id', 'link-29');">id</a></tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-850" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-850', 'sequence_number', 'link-313');">sequence_number</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seq_number</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-851" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-851', 'insertion_code', 'link-315');">insertion_code</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ins_code</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-852" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-852', 'type', 'link-109');">type</a></tt> <tt class="py-op">=</tt> <tt id="link-853" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-853', 'type', 'link-109');">type</a></tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-854" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-854', 'label', 'link-108');">label</a></tt> <tt class="py-op">=</tt> <tt id="link-855" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-855', 'label', 'link-108');">label</a></tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-856" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-856', 'atoms', 'link-288');">atoms</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">         </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="ResidueInfo.is_modified"></a><div id="ResidueInfo.is_modified-def"><a name="L1433"></a><tt class="py-lineno">1433</tt> <a class="py-toggle" href="#" id="ResidueInfo.is_modified-toggle" onclick="return toggle('ResidueInfo.is_modified');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.ResidueInfo-class.html#is_modified">is_modified</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueInfo.is_modified-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueInfo.is_modified-expanded"><a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">         </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-857" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-857', 'type', 'link-109');">type</a></tt><tt class="py-op">.</tt><tt id="link-858" class="py-name" targets="Variable csb.core.EnumItem.enum=csb.core.EnumItem-class.html#enum,Class csb.core.enum=csb.core.enum-class.html"><a title="csb.core.EnumItem.enum
csb.core.enum" class="py-name" href="#" onclick="return doclink('link-858', 'enum', 'link-858');">enum</a></tt> <tt class="py-keyword">is</tt> <tt id="link-859" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-859', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt class="py-name">Nucleic</tt><tt class="py-op">:</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-860" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-860', 'label', 'link-108');">label</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-861" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-861', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-862" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-862', 'label', 'link-108');">label</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-863" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-863', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line"> </tt>
<a name="SparseChainSequence"></a><div id="SparseChainSequence-def"><a name="L1440"></a><tt class="py-lineno">1440</tt> <a class="py-toggle" href="#" id="SparseChainSequence-toggle" onclick="return toggle('SparseChainSequence');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SparseChainSequence-class.html">SparseChainSequence</a><tt class="py-op">(</tt><tt class="py-base-class">csb</tt><tt class="py-op">.</tt><tt class="py-base-class">bio</tt><tt class="py-op">.</tt><tt class="py-base-class">sequence</tt><tt class="py-op">.</tt><tt class="py-base-class">ChainSequence</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SparseChainSequence-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="SparseChainSequence-expanded"><a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line"><tt class="py-docstring">    Sequence view for reference (SEQRES) or sparse (ATOM) PDB chains.</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line"><tt class="py-docstring">    The residue instances passed to the constructor must be</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line"><tt class="py-docstring">    L{csb.bio.structure.Residue} or L{csb.bio.io.wwpdb.ResidueInfo} objects.</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line"><tt class="py-docstring">    See L{csb.bio.sequence.AbstractSequence} for details.</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">             </tt>
<a name="SparseChainSequence._add"></a><div id="SparseChainSequence._add-def"><a name="L1449"></a><tt class="py-lineno">1449</tt> <a class="py-toggle" href="#" id="SparseChainSequence._add-toggle" onclick="return toggle('SparseChainSequence._add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SparseChainSequence-class.html#_add">_add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SparseChainSequence._add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SparseChainSequence._add-expanded"><a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">         </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-864" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-864', 'residue', 'link-372');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-865" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-865', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-866" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-866', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-867" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-867', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-868" class="py-name"><a title="csb.bio.structure.Residue" class="py-name" href="#" onclick="return doclink('link-868', 'Residue', 'link-694');">Residue</a></tt><tt class="py-op">,</tt> <tt id="link-869" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo
csb.bio.io.wwpdb.ResidueInfo
csb.bio.sequence.ResidueInfo" class="py-name" href="#" onclick="return doclink('link-869', 'ResidueInfo', 'link-337');">ResidueInfo</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-870" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-870', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1454"></a><tt class="py-lineno">1454</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">.</tt><tt id="link-871" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-871', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-872" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-872', 'residue', 'link-372');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">             </tt>
<a name="SparseChainSequence._get"></a><div id="SparseChainSequence._get-def"><a name="L1456"></a><tt class="py-lineno">1456</tt> <a class="py-toggle" href="#" id="SparseChainSequence._get-toggle" onclick="return toggle('SparseChainSequence._get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SparseChainSequence-class.html#_get">_get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SparseChainSequence._get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SparseChainSequence._get-expanded"><a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_residues</tt><tt class="py-op">[</tt><tt id="link-873" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-873', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line">     </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">staticmethod</tt> </tt>
<a name="SparseChainSequence.create"></a><div id="SparseChainSequence.create-def"><a name="L1460"></a><tt class="py-lineno">1460</tt> <a class="py-toggle" href="#" id="SparseChainSequence.create-toggle" onclick="return toggle('SparseChainSequence.create');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.SparseChainSequence-class.html#create">create</a><tt class="py-op">(</tt><tt class="py-param">chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SparseChainSequence.create-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="SparseChainSequence.create-expanded"><a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new L{SparseChainSequence} from existing L{Chain}.</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain: L{csb.bio.structure.Chain}</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SparseChainSequence}</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-874" class="py-name"><a title="csb.bio.io.wwpdb.SparseChainSequence" class="py-name" href="#" onclick="return doclink('link-874', 'SparseChainSequence', 'link-456');">SparseChainSequence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line">                <tt id="link-875" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-875', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-876" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.entry_id
csb.bio.structure.Chain.entry_id" class="py-name" href="#" onclick="return doclink('link-876', 'entry_id', 'link-43');">entry_id</a></tt><tt class="py-op">,</tt> <tt id="link-877" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-877', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-878" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-878', 'header', 'link-583');">header</a></tt><tt class="py-op">,</tt> <tt id="link-879" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-879', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-880" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-880', 'residues', 'link-262');">residues</a></tt><tt class="py-op">,</tt> <tt id="link-881" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-881', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-882" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-882', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt>     </tt>
</div></div><a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line">     </tt>
<a name="AbstractResidueMapper"></a><div id="AbstractResidueMapper-def"><a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line">     </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt> <a class="py-toggle" href="#" id="AbstractResidueMapper-toggle" onclick="return toggle('AbstractResidueMapper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractResidueMapper-class.html">AbstractResidueMapper</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractResidueMapper-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AbstractResidueMapper-expanded"><a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line"><tt class="py-docstring">    Defines the base interface of all residue mappers, used to align PDB ATOM</tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"><tt class="py-docstring">    records to the real (SEQRES) sequence of a chain.</tt> </tt>
<a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line">     </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-883" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-883', 'ABCMeta', 'link-15');">ABCMeta</a></tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line">     </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="AbstractResidueMapper.map"></a><div id="AbstractResidueMapper.map-def"><a name="L1480"></a><tt class="py-lineno">1480</tt> <a class="py-toggle" href="#" id="AbstractResidueMapper.map-toggle" onclick="return toggle('AbstractResidueMapper.map');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractResidueMapper-class.html#map">map</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sparse</tt><tt class="py-op">,</tt> <tt class="py-param">reference</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractResidueMapper.map-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractResidueMapper.map-expanded"><a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line"><tt class="py-docstring">        Map L{sparse}'s residues to L{reference}. Return all C{sparse} residues,</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"><tt class="py-docstring">        aligned over C{reference}, with artificial gap residues inserted at</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"><tt class="py-docstring">        relevant positions. The resulting sequence of sparse residues will</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line"><tt class="py-docstring">        always have the same length as the C{reference} sequence.</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line"><tt class="py-docstring">        @note: C{sparse}'s ranks won't be touched because the C{rank} property</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line"><tt class="py-docstring">        of the underlying L{ResidueInfo} implementation is not necessarily r/w.</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line"><tt class="py-docstring">        @param sparse: sparse sequence (e.g. derived from ATOMS records)</tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line"><tt class="py-docstring">        @type sparse: L{SparseChainSequence}</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line"><tt class="py-docstring">        @param reference: reference, complete sequence</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line"><tt class="py-docstring">                          (e.g. derived from SEQRES records)</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line"><tt class="py-docstring">        @type reference: L{SparseChainSequence}</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line"><tt class="py-docstring">        @return: all C{sparse} residues, optimally aligned over C{reference}</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line"><tt class="py-docstring">                 (with gaps)</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{SparseChainSequence}</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line"><tt class="py-docstring">        @raise ResidueMappingError: if the specified sequences are not alignable</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line">     </tt>
<a name="AbstractResidueMapper.create_gap"></a><div id="AbstractResidueMapper.create_gap-def"><a name="L1504"></a><tt class="py-lineno">1504</tt> <a class="py-toggle" href="#" id="AbstractResidueMapper.create_gap-toggle" onclick="return toggle('AbstractResidueMapper.create_gap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractResidueMapper-class.html#create_gap">create_gap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-884" class="py-name"><a title="csb.bio.sequence.SequenceAlphabets" class="py-name" href="#" onclick="return doclink('link-884', 'SequenceAlphabets', 'link-20');">SequenceAlphabets</a></tt><tt class="py-op">.</tt><tt id="link-885" class="py-name"><a title="csb.bio.sequence.SequenceTypes.Protein" class="py-name" href="#" onclick="return doclink('link-885', 'Protein', 'link-121');">Protein</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractResidueMapper.create_gap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractResidueMapper.create_gap-expanded"><a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line"><tt class="py-docstring">        Create and return a new gap residue.</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"><tt class="py-docstring">        @param alphabet: sequence alphabet; a member of L{SequenceAlphabets}</tt> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line"><tt class="py-docstring">                         which has GAP item</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line"><tt class="py-docstring">        @type alphabet: L{enum}</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{ResidueInfo}</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-886" class="py-name"><a title="csb.bio.fragments.rosetta.ResidueInfo
csb.bio.io.wwpdb.ResidueInfo
csb.bio.sequence.ResidueInfo" class="py-name" href="#" onclick="return doclink('link-886', 'ResidueInfo', 'link-337');">ResidueInfo</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt id="link-887" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-887', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-888" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-888', 'GAP', 'link-476');">GAP</a></tt><tt class="py-op">,</tt> <tt class="py-string">"-"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line">     </tt>
<a name="AbstractResidueMapper._build"></a><div id="AbstractResidueMapper._build-def"><a name="L1516"></a><tt class="py-lineno">1516</tt> <a class="py-toggle" href="#" id="AbstractResidueMapper._build-toggle" onclick="return toggle('AbstractResidueMapper._build');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AbstractResidueMapper-class.html#_build">_build</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sparse</tt><tt class="py-op">,</tt> <tt class="py-param">aligned</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AbstractResidueMapper._build-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AbstractResidueMapper._build-expanded"><a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line">         </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-889" class="py-name"><a title="csb.bio.io.wwpdb.SparseChainSequence" class="py-name" href="#" onclick="return doclink('link-889', 'SparseChainSequence', 'link-456');">SparseChainSequence</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line">                    <tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-890" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-890', 'id', 'link-29');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-891" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-891', 'header', 'link-583');">header</a></tt><tt class="py-op">,</tt> <tt class="py-name">aligned</tt><tt class="py-op">,</tt> <tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-892" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-892', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt>         </tt>
</div></div><a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line"> </tt>
<a name="FastResidueMapper"></a><div id="FastResidueMapper-def"><a name="L1521"></a><tt class="py-lineno">1521</tt> <a class="py-toggle" href="#" id="FastResidueMapper-toggle" onclick="return toggle('FastResidueMapper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html">FastResidueMapper</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractResidueMapper</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastResidueMapper-expanded"><a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line"><tt class="py-docstring">    RegExp-based residue mapper. Fails on heavily malformed input (i.e. it cannot</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line"><tt class="py-docstring">    insert gaps in the C{reference}), but it is very fast (linear) and memory</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line"><tt class="py-docstring">    efficient. </tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line">     </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line">    <tt id="link-893" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.MAX_FRAGMENTS=csb.bio.io.wwpdb.FastResidueMapper-class.html#MAX_FRAGMENTS"><a title="csb.bio.io.wwpdb.FastResidueMapper.MAX_FRAGMENTS" class="py-name" href="#" onclick="return doclink('link-893', 'MAX_FRAGMENTS', 'link-893');">MAX_FRAGMENTS</a></tt> <tt class="py-op">=</tt> <tt class="py-number">20</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line">     </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line">    <tt id="link-894" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.MIN_UNICODE_CHAR=csb.bio.io.wwpdb.FastResidueMapper-class.html#MIN_UNICODE_CHAR"><a title="csb.bio.io.wwpdb.FastResidueMapper.MIN_UNICODE_CHAR" class="py-name" href="#" onclick="return doclink('link-894', 'MIN_UNICODE_CHAR', 'link-894');">MIN_UNICODE_CHAR</a></tt> <tt class="py-op">=</tt> <tt class="py-number">300</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line">    <tt id="link-895" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.FORBIDDEN_CHARS=csb.bio.io.wwpdb.FastResidueMapper-class.html#FORBIDDEN_CHARS"><a title="csb.bio.io.wwpdb.FastResidueMapper.FORBIDDEN_CHARS" class="py-name" href="#" onclick="return doclink('link-895', 'FORBIDDEN_CHARS', 'link-895');">FORBIDDEN_CHARS</a></tt> <tt class="py-op">=</tt> <tt id="link-896" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-896', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt class="py-string">'^.*?()-'</tt><tt class="py-op">)</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line">     </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line">    <tt id="link-897" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.CODEC=csb.bio.io.wwpdb.FastResidueMapper-class.html#CODEC"><a title="csb.bio.io.wwpdb.FastResidueMapper.CODEC" class="py-name" href="#" onclick="return doclink('link-897', 'CODEC', 'link-897');">CODEC</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"utf-8"</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line">    <tt id="link-898" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.DELIMITER=csb.bio.io.wwpdb.FastResidueMapper-class.html#DELIMITER,Variable csb.bio.sequence.AbstractSequence.DELIMITER=csb.bio.sequence.AbstractSequence-class.html#DELIMITER"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-898', 'DELIMITER', 'link-898');">DELIMITER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">").*?("</tt><tt class="py-op">.</tt><tt class="py-name">encode</tt><tt class="py-op">(</tt><tt id="link-899" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.CODEC" class="py-name" href="#" onclick="return doclink('link-899', 'CODEC', 'link-897');">CODEC</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">decode</tt><tt class="py-op">(</tt><tt id="link-900" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.CODEC" class="py-name" href="#" onclick="return doclink('link-900', 'CODEC', 'link-897');">CODEC</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line">    <tt id="link-901" class="py-name" targets="Variable csb.bio.io.wwpdb.FastResidueMapper.PATTERN=csb.bio.io.wwpdb.FastResidueMapper-class.html#PATTERN"><a title="csb.bio.io.wwpdb.FastResidueMapper.PATTERN" class="py-name" href="#" onclick="return doclink('link-901', 'PATTERN', 'link-901');">PATTERN</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"^.*?({0}).*?$"</tt><tt class="py-op">.</tt><tt class="py-name">encode</tt><tt class="py-op">(</tt><tt id="link-902" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.CODEC" class="py-name" href="#" onclick="return doclink('link-902', 'CODEC', 'link-897');">CODEC</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">decode</tt><tt class="py-op">(</tt><tt id="link-903" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.CODEC" class="py-name" href="#" onclick="return doclink('link-903', 'CODEC', 'link-897');">CODEC</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line">     </tt>
<a name="FastResidueMapper.__init__"></a><div id="FastResidueMapper.__init__-def"><a name="L1537"></a><tt class="py-lineno">1537</tt> <a class="py-toggle" href="#" id="FastResidueMapper.__init__-toggle" onclick="return toggle('FastResidueMapper.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper.__init__-expanded"><a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_charcode</tt> <tt class="py-op">=</tt> <tt id="link-904" class="py-name" targets="Class csb.bio.io.wwpdb.FastResidueMapper=csb.bio.io.wwpdb.FastResidueMapper-class.html"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-904', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-905" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.MIN_UNICODE_CHAR" class="py-name" href="#" onclick="return doclink('link-905', 'MIN_UNICODE_CHAR', 'link-894');">MIN_UNICODE_CHAR</a></tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cache</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>  </tt>
</div><a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line">     </tt>
<a name="FastResidueMapper.map"></a><div id="FastResidueMapper.map-def"><a name="L1541"></a><tt class="py-lineno">1541</tt> <a class="py-toggle" href="#" id="FastResidueMapper.map-toggle" onclick="return toggle('FastResidueMapper.map');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#map">map</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sparse</tt><tt class="py-op">,</tt> <tt class="py-param">reference</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper.map-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper.map-expanded"><a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line"> </tt>
<a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line">        <tt class="py-name">aligned</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line">        <tt class="py-name">mapping</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line"> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line">        <tt id="link-906" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-906', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-907" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-907', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line"> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line">        <tt class="py-name">pattern</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_build_pattern</tt><tt class="py-op">(</tt><tt id="link-908" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-908', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line">        <tt class="py-name">seqres</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_encode_sequence</tt><tt class="py-op">(</tt><tt class="py-name">reference</tt><tt class="py-op">)</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line"> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line">        <tt id="link-909" class="py-name" targets="Variable csb.bio.fragments.Target.matches=csb.bio.fragments.Target-class.html#matches,Variable csb.bio.sequence.A3MAlignment.matches=csb.bio.sequence.A3MAlignment-class.html#matches"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-909', 'matches', 'link-909');">matches</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-910" class="py-name" targets="Static Method csb.bio.nmr.Label.match()=csb.bio.nmr.Label-class.html#match,Variable csb.bio.sequence.alignment.IdentityMatrix.match=csb.bio.sequence.alignment.IdentityMatrix-class.html#match"><a title="csb.bio.nmr.Label.match
csb.bio.sequence.alignment.IdentityMatrix.match" class="py-name" href="#" onclick="return doclink('link-910', 'match', 'link-910');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">,</tt> <tt class="py-name">seqres</tt><tt class="py-op">)</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line">                 </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-911" class="py-name"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-911', 'matches', 'link-909');">matches</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line">            <tt class="py-name">unmapped_item</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line">             </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">fn</tt><tt class="py-op">,</tt> <tt id="link-912" class="py-name" targets="Variable csb.bio.fragments.ResidueAssignmentInfo.fragment=csb.bio.fragments.ResidueAssignmentInfo-class.html#fragment"><a title="csb.bio.fragments.ResidueAssignmentInfo.fragment" class="py-name" href="#" onclick="return doclink('link-912', 'fragment', 'link-912');">fragment</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-913" class="py-name"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-913', 'matches', 'link-909');">matches</a></tt><tt class="py-op">.</tt><tt id="link-914" class="py-name"><a title="csb.bio.io.clans.ClansEntry.groups
csb.core.REMatchProxy.groups" class="py-name" href="#" onclick="return doclink('link-914', 'groups', 'link-660');">groups</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-915" class="py-name" targets="Variable csb.bio.fragments.Assignment.start=csb.bio.fragments.Assignment-class.html#start,Variable csb.bio.fragments.FragmentMatch.start=csb.bio.fragments.FragmentMatch-class.html#start,Variable csb.bio.fragments.rosetta.RosettaFragment.start=csb.bio.fragments.rosetta.RosettaFragment-class.html#start,Variable csb.bio.hmm.HHpredHit.start=csb.bio.hmm.HHpredHit-class.html#start,Variable csb.bio.hmm.ProfileHMM.start=csb.bio.hmm.ProfileHMM-class.html#start,Variable csb.bio.sequence.alignment.AlignmentResult.start=csb.bio.sequence.alignment.AlignmentResult-class.html#start,Variable csb.bio.structure.SecondaryStructure.start=csb.bio.structure.SecondaryStructure-class.html#start,Variable csb.bio.structure.SecondaryStructureElement.start=csb.bio.structure.SecondaryStructureElement-class.html#start,Method csb.core.REMatchProxy.start()=csb.core.REMatchProxy-class.html#start,Method csb.io.plots.Backend.start()=csb.io.plots.Backend-class.html#start"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-915', 'start', 'link-915');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt id="link-916" class="py-name"><a title="csb.bio.fragments.ResidueAssignmentInfo.fragment" class="py-name" href="#" onclick="return doclink('link-916', 'fragment', 'link-912');">fragment</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line">                 </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">offset</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-917" class="py-name"><a title="csb.bio.fragments.ResidueAssignmentInfo.fragment" class="py-name" href="#" onclick="return doclink('link-917', 'fragment', 'link-912');">fragment</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line">                    <tt class="py-name">unmapped_item</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line">                    <tt id="link-918" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-918', 'rank', 'link-312');">rank</a></tt> <tt class="py-op">=</tt> <tt id="link-919" class="py-name"><a title="csb.bio.fragments.Target.matches
csb.bio.sequence.A3MAlignment.matches" class="py-name" href="#" onclick="return doclink('link-919', 'matches', 'link-909');">matches</a></tt><tt class="py-op">.</tt><tt id="link-920" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-920', 'start', 'link-915');">start</a></tt><tt class="py-op">(</tt><tt class="py-name">fn</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt>  </tt>
<a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line">                     </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line">                    <tt class="py-name">mapped_residue</tt> <tt class="py-op">=</tt> <tt id="link-921" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-921', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt class="py-name">unmapped_item</tt><tt class="py-op">]</tt>  </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line">                    <tt class="py-name">real_residue</tt> <tt class="py-op">=</tt> <tt class="py-name">reference</tt><tt class="py-op">.</tt><tt id="link-922" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-922', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt id="link-923" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-923', 'rank', 'link-312');">rank</a></tt><tt class="py-op">]</tt>                     </tt>
<a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line">                     </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">real_residue</tt><tt class="py-op">.</tt><tt id="link-924" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-924', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">mapped_residue</tt><tt class="py-op">.</tt><tt id="link-925" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-925', 'type', 'link-109');">type</a></tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line">                    <tt class="py-name">mapping</tt><tt class="py-op">[</tt><tt class="py-name">real_residue</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">mapped_residue</tt>                     </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line">         </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-926" class="py-name" targets="Class csb.bio.io.wwpdb.ResidueMappingError=csb.bio.io.wwpdb.ResidueMappingError-class.html"><a title="csb.bio.io.wwpdb.ResidueMappingError" class="py-name" href="#" onclick="return doclink('link-926', 'ResidueMappingError', 'link-926');">ResidueMappingError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Can't map ATOM records"</tt><tt class="py-op">)</tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line">         </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-927" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-927', 'rank', 'link-312');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-928" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-928', 'residue', 'link-372');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">reference</tt><tt class="py-op">.</tt><tt id="link-929" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-929', 'residues', 'link-262');">residues</a></tt><tt class="py-op">,</tt> <tt id="link-930" class="py-name"><a title="csb.bio.fragments.Assignment.start
csb.bio.fragments.FragmentMatch.start
csb.bio.fragments.rosetta.RosettaFragment.start
csb.bio.hmm.HHpredHit.start
csb.bio.hmm.ProfileHMM.start
csb.bio.sequence.alignment.AlignmentResult.start
csb.bio.structure.SecondaryStructure.start
csb.bio.structure.SecondaryStructureElement.start
csb.core.REMatchProxy.start
csb.io.plots.Backend.start" class="py-name" href="#" onclick="return doclink('link-930', 'start', 'link-915');">start</a></tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-931" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-931', 'residue', 'link-372');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">mapping</tt><tt class="py-op">:</tt> </tt>
<a name="L1574"></a><tt class="py-lineno">1574</tt>  <tt class="py-line">                <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-932" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-932', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">mapping</tt><tt class="py-op">[</tt><tt id="link-933" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-933', 'residue', 'link-372');">residue</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1575"></a><tt class="py-lineno">1575</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1576"></a><tt class="py-lineno">1576</tt>  <tt class="py-line">                <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-934" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-934', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-935" class="py-name" targets="Method csb.bio.io.wwpdb.AbstractResidueMapper.create_gap()=csb.bio.io.wwpdb.AbstractResidueMapper-class.html#create_gap"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.create_gap" class="py-name" href="#" onclick="return doclink('link-935', 'create_gap', 'link-935');">create_gap</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-936" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-936', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1577"></a><tt class="py-lineno">1577</tt>  <tt class="py-line">         </tt>
<a name="L1578"></a><tt class="py-lineno">1578</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">aligned</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">reference</tt><tt class="py-op">.</tt><tt id="link-937" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-937', 'length', 'link-261');">length</a></tt> </tt>
<a name="L1579"></a><tt class="py-lineno">1579</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_build</tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">aligned</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1580"></a><tt class="py-lineno">1580</tt>  <tt class="py-line">     </tt>
<a name="FastResidueMapper._build_pattern"></a><div id="FastResidueMapper._build_pattern-def"><a name="L1581"></a><tt class="py-lineno">1581</tt> <a class="py-toggle" href="#" id="FastResidueMapper._build_pattern-toggle" onclick="return toggle('FastResidueMapper._build_pattern');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#_build_pattern">_build_pattern</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper._build_pattern-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper._build_pattern-expanded"><a name="L1582"></a><tt class="py-lineno">1582</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1583"></a><tt class="py-lineno">1583</tt>  <tt class="py-line"><tt class="py-docstring">        Build and return a sparse regular rexpression for C{residues}.</tt> </tt>
<a name="L1584"></a><tt class="py-lineno">1584</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1585"></a><tt class="py-lineno">1585</tt>  <tt class="py-line"> </tt>
<a name="L1586"></a><tt class="py-lineno">1586</tt>  <tt class="py-line">        <tt id="link-938" class="py-name" targets="Package csb.bio.fragments=csb.bio.fragments-module.html,Variable csb.bio.fragments.FragmentCluster.fragments=csb.bio.fragments.FragmentCluster-class.html#fragments"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-938', 'fragments', 'link-938');">fragments</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1587"></a><tt class="py-lineno">1587</tt>  <tt class="py-line">         </tt>
<a name="L1588"></a><tt class="py-lineno">1588</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">rn</tt><tt class="py-op">,</tt> <tt id="link-939" class="py-name" targets="Variable csb.bio.io.clans.Color.r=csb.bio.io.clans.Color-class.html#r"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-939', 'r', 'link-939');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-940" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-940', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1589"></a><tt class="py-lineno">1589</tt>  <tt class="py-line">            <tt class="py-name">res_name</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_encode</tt><tt class="py-op">(</tt><tt id="link-941" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-941', 'r', 'link-939');">r</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1590"></a><tt class="py-lineno">1590</tt>  <tt class="py-line">             </tt>
<a name="L1591"></a><tt class="py-lineno">1591</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">rn</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1592"></a><tt class="py-lineno">1592</tt>  <tt class="py-line">                <tt class="py-comment"># First residue, start a new fragment:</tt> </tt>
<a name="L1593"></a><tt class="py-lineno">1593</tt>  <tt class="py-line">                <tt id="link-942" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-942', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">.</tt><tt id="link-943" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-943', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">res_name</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1594"></a><tt class="py-lineno">1594</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-944" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-944', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-945" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-945', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">:</tt> <tt class="py-comment"># and not residues[rn - 1].insertion_code:</tt> </tt>
<a name="L1595"></a><tt class="py-lineno">1595</tt>  <tt class="py-line">                <tt class="py-comment"># If residue i has an insertion code, initiate a new fragment:</tt> </tt>
<a name="L1596"></a><tt class="py-lineno">1596</tt>  <tt class="py-line">                <tt id="link-946" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-946', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">.</tt><tt id="link-947" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-947', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">res_name</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1597"></a><tt class="py-lineno">1597</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-948" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-948', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-949" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-949', 'sequence_number', 'link-313');">sequence_number</a></tt> <tt class="py-op">-</tt> <tt id="link-950" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-950', 'residues', 'link-262');">residues</a></tt><tt class="py-op">[</tt><tt class="py-name">rn</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-951" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-951', 'sequence_number', 'link-313');">sequence_number</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1598"></a><tt class="py-lineno">1598</tt>  <tt class="py-line">                <tt class="py-comment"># If the seq numbers of residues [i-1, i] are consecutive, extend the last fragment:</tt> </tt>
<a name="L1599"></a><tt class="py-lineno">1599</tt>  <tt class="py-line">                <tt id="link-952" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-952', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-953" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-953', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">res_name</tt><tt class="py-op">)</tt> </tt>
<a name="L1600"></a><tt class="py-lineno">1600</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1601"></a><tt class="py-lineno">1601</tt>  <tt class="py-line">                <tt class="py-comment"># They are not consecutive, so we better start a new fragment:</tt> </tt>
<a name="L1602"></a><tt class="py-lineno">1602</tt>  <tt class="py-line">                <tt id="link-954" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-954', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">.</tt><tt id="link-955" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-955', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">res_name</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1603"></a><tt class="py-lineno">1603</tt>  <tt class="py-line">         </tt>
<a name="L1604"></a><tt class="py-lineno">1604</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">i</tt><tt class="py-op">,</tt> <tt class="py-name">frag</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-956" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-956', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1605"></a><tt class="py-lineno">1605</tt>  <tt class="py-line">            <tt id="link-957" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-957', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">[</tt><tt class="py-name">i</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-958" class="py-name" targets="Static Method csb.bio.nmr.NOESpectrum.join()=csb.bio.nmr.NOESpectrum-class.html#join"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-958', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt> </tt>
<a name="L1606"></a><tt class="py-lineno">1606</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-959" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-959', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt id="link-960" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-960', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-961" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.MAX_FRAGMENTS" class="py-name" href="#" onclick="return doclink('link-961', 'MAX_FRAGMENTS', 'link-893');">MAX_FRAGMENTS</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1607"></a><tt class="py-lineno">1607</tt>  <tt class="py-line">            <tt class="py-comment"># Wow, that's a lot of fragments. Better use a different mapper</tt> </tt>
<a name="L1608"></a><tt class="py-lineno">1608</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-962" class="py-name"><a title="csb.bio.io.wwpdb.ResidueMappingError" class="py-name" href="#" onclick="return doclink('link-962', 'ResidueMappingError', 'link-926');">ResidueMappingError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Can't map chain with large number of fragments"</tt><tt class="py-op">)</tt> </tt>
<a name="L1609"></a><tt class="py-lineno">1609</tt>  <tt class="py-line">         </tt>
<a name="L1610"></a><tt class="py-lineno">1610</tt>  <tt class="py-line">        <tt class="py-name">blocks</tt> <tt class="py-op">=</tt> <tt id="link-963" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-963', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-964" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.DELIMITER
csb.bio.sequence.AbstractSequence.DELIMITER" class="py-name" href="#" onclick="return doclink('link-964', 'DELIMITER', 'link-898');">DELIMITER</a></tt><tt class="py-op">.</tt><tt id="link-965" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-965', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt id="link-966" class="py-name"><a title="csb.bio.fragments
csb.bio.fragments.FragmentCluster.fragments" class="py-name" href="#" onclick="return doclink('link-966', 'fragments', 'link-938');">fragments</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1611"></a><tt class="py-lineno">1611</tt>  <tt class="py-line">        <tt class="py-name">pattern</tt> <tt class="py-op">=</tt> <tt id="link-967" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-967', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-968" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.PATTERN" class="py-name" href="#" onclick="return doclink('link-968', 'PATTERN', 'link-901');">PATTERN</a></tt><tt class="py-op">.</tt><tt id="link-969" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-969', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">blocks</tt><tt class="py-op">)</tt> </tt>
<a name="L1612"></a><tt class="py-lineno">1612</tt>  <tt class="py-line">         </tt>
<a name="L1613"></a><tt class="py-lineno">1613</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">pattern</tt> </tt>
</div><a name="L1614"></a><tt class="py-lineno">1614</tt>  <tt class="py-line">     </tt>
<a name="FastResidueMapper._encode"></a><div id="FastResidueMapper._encode-def"><a name="L1615"></a><tt class="py-lineno">1615</tt> <a class="py-toggle" href="#" id="FastResidueMapper._encode-toggle" onclick="return toggle('FastResidueMapper._encode');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#_encode">_encode</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">r</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper._encode-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper._encode-expanded"><a name="L1616"></a><tt class="py-lineno">1616</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1617"></a><tt class="py-lineno">1617</tt>  <tt class="py-line"><tt class="py-docstring">        Return a unique single-letter representation of C{r.type}. </tt> </tt>
<a name="L1618"></a><tt class="py-lineno">1618</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1619"></a><tt class="py-lineno">1619</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-970" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-970', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-971" class="py-name" targets="Variable csb.bio.io.wwpdb.ResidueInfo.is_modified=csb.bio.io.wwpdb.ResidueInfo-class.html#is_modified,Variable csb.bio.structure.NucleicResidue.is_modified=csb.bio.structure.NucleicResidue-class.html#is_modified,Variable csb.bio.structure.Residue.is_modified=csb.bio.structure.Residue-class.html#is_modified"><a title="csb.bio.io.wwpdb.ResidueInfo.is_modified
csb.bio.structure.NucleicResidue.is_modified
csb.bio.structure.Residue.is_modified" class="py-name" href="#" onclick="return doclink('link-971', 'is_modified', 'link-971');">is_modified</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1620"></a><tt class="py-lineno">1620</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-972" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-972', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-973" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-973', 'type', 'link-109');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1621"></a><tt class="py-lineno">1621</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>              </tt>
<a name="L1622"></a><tt class="py-lineno">1622</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_register_label</tt><tt class="py-op">(</tt><tt id="link-974" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-974', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-975" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-975', 'label', 'link-108');">label</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1623"></a><tt class="py-lineno">1623</tt>  <tt class="py-line">         </tt>
<a name="FastResidueMapper._encode_sequence"></a><div id="FastResidueMapper._encode_sequence-def"><a name="L1624"></a><tt class="py-lineno">1624</tt> <a class="py-toggle" href="#" id="FastResidueMapper._encode_sequence-toggle" onclick="return toggle('FastResidueMapper._encode_sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#_encode_sequence">_encode_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">s</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper._encode_sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper._encode_sequence-expanded"><a name="L1625"></a><tt class="py-lineno">1625</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt id="link-976" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-976', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt id="link-977" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-977', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_encode</tt><tt class="py-op">,</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-978" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-978', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1626"></a><tt class="py-lineno">1626</tt>  <tt class="py-line">             </tt>
<a name="FastResidueMapper._register_label"></a><div id="FastResidueMapper._register_label-def"><a name="L1627"></a><tt class="py-lineno">1627</tt> <a class="py-toggle" href="#" id="FastResidueMapper._register_label-toggle" onclick="return toggle('FastResidueMapper._register_label');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FastResidueMapper-class.html#_register_label">_register_label</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">label</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastResidueMapper._register_label-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastResidueMapper._register_label-expanded"><a name="L1628"></a><tt class="py-lineno">1628</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1629"></a><tt class="py-lineno">1629</tt>  <tt class="py-line"><tt class="py-docstring">        Assign a new unicode character to C{label} and cache it.</tt> </tt>
<a name="L1630"></a><tt class="py-lineno">1630</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1631"></a><tt class="py-lineno">1631</tt>  <tt class="py-line"><tt class="py-docstring">        @return: cached single-letter representation of label.</tt> </tt>
<a name="L1632"></a><tt class="py-lineno">1632</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: unicode char</tt> </tt>
<a name="L1633"></a><tt class="py-lineno">1633</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1634"></a><tt class="py-lineno">1634</tt>  <tt class="py-line"> </tt>
<a name="L1635"></a><tt class="py-lineno">1635</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-979" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-979', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cache</tt><tt class="py-op">:</tt> </tt>
<a name="L1636"></a><tt class="py-lineno">1636</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-980" class="py-name"><a title="csb.bio.hmm.EmissionTable.set
csb.bio.hmm.TransitionTable.set
csb.statistics.maxent.MaxentModel.set
csb.statistics.pdf.parameterized.AbstractParameter.set" class="py-name" href="#" onclick="return doclink('link-980', 'set', 'link-89');">set</a></tt><tt class="py-op">(</tt><tt id="link-981" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-981', 'label', 'link-108');">label</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">intersection</tt><tt class="py-op">(</tt><tt id="link-982" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-982', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-983" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper.FORBIDDEN_CHARS" class="py-name" href="#" onclick="return doclink('link-983', 'FORBIDDEN_CHARS', 'link-895');">FORBIDDEN_CHARS</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1637"></a><tt class="py-lineno">1637</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-984" class="py-name"><a title="csb.bio.io.wwpdb.ResidueMappingError" class="py-name" href="#" onclick="return doclink('link-984', 'ResidueMappingError', 'link-926');">ResidueMappingError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Invalid residue label"</tt><tt class="py-op">)</tt> </tt>
<a name="L1638"></a><tt class="py-lineno">1638</tt>  <tt class="py-line">             </tt>
<a name="L1639"></a><tt class="py-lineno">1639</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_charcode</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1640"></a><tt class="py-lineno">1640</tt>  <tt class="py-line">            <tt class="py-name">code</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_charcode</tt> </tt>
<a name="L1641"></a><tt class="py-lineno">1641</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cache</tt><tt class="py-op">[</tt><tt id="link-985" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-985', 'label', 'link-108');">label</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-986" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-986', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-987" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-987', 'io', 'link-14');">io</a></tt><tt class="py-op">.</tt><tt class="py-name">unichr</tt><tt class="py-op">(</tt><tt class="py-name">code</tt><tt class="py-op">)</tt> </tt>
<a name="L1642"></a><tt class="py-lineno">1642</tt>  <tt class="py-line">             </tt>
<a name="L1643"></a><tt class="py-lineno">1643</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cache</tt><tt class="py-op">[</tt><tt id="link-988" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-988', 'label', 'link-108');">label</a></tt><tt class="py-op">]</tt>     </tt>
</div></div><a name="L1644"></a><tt class="py-lineno">1644</tt>  <tt class="py-line">                 </tt>
<a name="RobustResidueMapper"></a><div id="RobustResidueMapper-def"><a name="L1645"></a><tt class="py-lineno">1645</tt>  <tt class="py-line">         </tt>
<a name="L1646"></a><tt class="py-lineno">1646</tt> <a class="py-toggle" href="#" id="RobustResidueMapper-toggle" onclick="return toggle('RobustResidueMapper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RobustResidueMapper-class.html">RobustResidueMapper</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractResidueMapper</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RobustResidueMapper-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="RobustResidueMapper-expanded"><a name="L1647"></a><tt class="py-lineno">1647</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1648"></a><tt class="py-lineno">1648</tt>  <tt class="py-line"><tt class="py-docstring">    Exhaustive residue mapper, which uses Needleman-Wunsch global alignment.</tt> </tt>
<a name="L1649"></a><tt class="py-lineno">1649</tt>  <tt class="py-line"><tt class="py-docstring">    Much slower (quadratic), but fail-proof even with incompatible sequences</tt> </tt>
<a name="L1650"></a><tt class="py-lineno">1650</tt>  <tt class="py-line"><tt class="py-docstring">    (can insert gaps in both the C{sparse} and the C{reference} sequence).</tt> </tt>
<a name="L1651"></a><tt class="py-lineno">1651</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1652"></a><tt class="py-lineno">1652</tt>  <tt class="py-line"><tt class="py-docstring">    @param match: score for a match</tt> </tt>
<a name="L1653"></a><tt class="py-lineno">1653</tt>  <tt class="py-line"><tt class="py-docstring">    @type match: float</tt> </tt>
<a name="L1654"></a><tt class="py-lineno">1654</tt>  <tt class="py-line"><tt class="py-docstring">    @param mismatch: score for a mismatch (by default mismatches are heavily</tt> </tt>
<a name="L1655"></a><tt class="py-lineno">1655</tt>  <tt class="py-line"><tt class="py-docstring">                     penalized, while gaps are allowed)</tt> </tt>
<a name="L1656"></a><tt class="py-lineno">1656</tt>  <tt class="py-line"><tt class="py-docstring">    @type mismatch: float</tt> </tt>
<a name="L1657"></a><tt class="py-lineno">1657</tt>  <tt class="py-line"><tt class="py-docstring">    @param gap: gap penalty</tt> </tt>
<a name="L1658"></a><tt class="py-lineno">1658</tt>  <tt class="py-line"><tt class="py-docstring">    @type gap: float  </tt> </tt>
<a name="L1659"></a><tt class="py-lineno">1659</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1660"></a><tt class="py-lineno">1660</tt>  <tt class="py-line">     </tt>
<a name="RobustResidueMapper.GlobalAligner"></a><div id="RobustResidueMapper.GlobalAligner-def"><a name="L1661"></a><tt class="py-lineno">1661</tt> <a class="py-toggle" href="#" id="RobustResidueMapper.GlobalAligner-toggle" onclick="return toggle('RobustResidueMapper.GlobalAligner');">-</a><tt class="py-line">    <tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RobustResidueMapper.GlobalAligner-class.html">GlobalAligner</a><tt class="py-op">(</tt><tt class="py-base-class">alignment</tt><tt class="py-op">.</tt><tt class="py-base-class">GlobalAlignmentAlgorithm</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RobustResidueMapper.GlobalAligner-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RobustResidueMapper.GlobalAligner-expanded"><a name="L1662"></a><tt class="py-lineno">1662</tt>  <tt class="py-line">         </tt>
<a name="RobustResidueMapper.GlobalAligner._sequence"></a><div id="RobustResidueMapper.GlobalAligner._sequence-def"><a name="L1663"></a><tt class="py-lineno">1663</tt> <a class="py-toggle" href="#" id="RobustResidueMapper.GlobalAligner._sequence-toggle" onclick="return toggle('RobustResidueMapper.GlobalAligner._sequence');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RobustResidueMapper.GlobalAligner-class.html#_sequence">_sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">s</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RobustResidueMapper.GlobalAligner._sequence-collapsed" style="display:none;" pad="++++" indent="++++++++++++++++"></div><div id="RobustResidueMapper.GlobalAligner._sequence-expanded"><a name="L1664"></a><tt class="py-lineno">1664</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-989" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-989', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-990" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-990', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">for</tt> <tt id="link-991" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-991', 'r', 'link-939');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-992" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-992', 'residues', 'link-262');">residues</a></tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L1665"></a><tt class="py-lineno">1665</tt>  <tt class="py-line">            </tt>
<a name="L1666"></a><tt class="py-lineno">1666</tt>  <tt class="py-line">     </tt>
<a name="RobustResidueMapper.__init__"></a><div id="RobustResidueMapper.__init__-def"><a name="L1667"></a><tt class="py-lineno">1667</tt> <a class="py-toggle" href="#" id="RobustResidueMapper.__init__-toggle" onclick="return toggle('RobustResidueMapper.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RobustResidueMapper-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">match</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">mismatch</tt><tt class="py-op">=</tt><tt class="py-op">-</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-param">gap</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RobustResidueMapper.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RobustResidueMapper.__init__-expanded"><a name="L1668"></a><tt class="py-lineno">1668</tt>  <tt class="py-line">         </tt>
<a name="L1669"></a><tt class="py-lineno">1669</tt>  <tt class="py-line">        <tt class="py-name">scoring</tt> <tt class="py-op">=</tt> <tt id="link-993" class="py-name"><a title="csb.bio.hmm.HHpredHit.alignment
csb.bio.hmm.ProfileHMM.alignment
csb.bio.sequence.alignment
csb.bio.sequence.alignment.AlignmentResult.alignment" class="py-name" href="#" onclick="return doclink('link-993', 'alignment', 'link-9');">alignment</a></tt><tt class="py-op">.</tt><tt id="link-994" class="py-name" targets="Class csb.bio.sequence.alignment.IdentityMatrix=csb.bio.sequence.alignment.IdentityMatrix-class.html"><a title="csb.bio.sequence.alignment.IdentityMatrix" class="py-name" href="#" onclick="return doclink('link-994', 'IdentityMatrix', 'link-994');">IdentityMatrix</a></tt><tt class="py-op">(</tt><tt id="link-995" class="py-name"><a title="csb.bio.nmr.Label.match
csb.bio.sequence.alignment.IdentityMatrix.match" class="py-name" href="#" onclick="return doclink('link-995', 'match', 'link-910');">match</a></tt><tt class="py-op">=</tt><tt id="link-996" class="py-name"><a title="csb.bio.nmr.Label.match
csb.bio.sequence.alignment.IdentityMatrix.match" class="py-name" href="#" onclick="return doclink('link-996', 'match', 'link-910');">match</a></tt><tt class="py-op">,</tt> <tt id="link-997" class="py-name" targets="Variable csb.bio.sequence.alignment.IdentityMatrix.mismatch=csb.bio.sequence.alignment.IdentityMatrix-class.html#mismatch"><a title="csb.bio.sequence.alignment.IdentityMatrix.mismatch" class="py-name" href="#" onclick="return doclink('link-997', 'mismatch', 'link-997');">mismatch</a></tt><tt class="py-op">=</tt><tt id="link-998" class="py-name"><a title="csb.bio.sequence.alignment.IdentityMatrix.mismatch" class="py-name" href="#" onclick="return doclink('link-998', 'mismatch', 'link-997');">mismatch</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1670"></a><tt class="py-lineno">1670</tt>  <tt class="py-line">        <tt id="link-999" class="py-name" targets="Variable csb.apps.hhfrag.HHfrag.aligner=csb.apps.hhfrag.HHfrag-class.html#aligner"><a title="csb.apps.hhfrag.HHfrag.aligner" class="py-name" href="#" onclick="return doclink('link-999', 'aligner', 'link-999');">aligner</a></tt> <tt class="py-op">=</tt> <tt id="link-1000" class="py-name" targets="Class csb.bio.io.wwpdb.RobustResidueMapper=csb.bio.io.wwpdb.RobustResidueMapper-class.html"><a title="csb.bio.io.wwpdb.RobustResidueMapper" class="py-name" href="#" onclick="return doclink('link-1000', 'RobustResidueMapper', 'link-1000');">RobustResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-1001" class="py-name" targets="Class csb.bio.io.wwpdb.RobustResidueMapper.GlobalAligner=csb.bio.io.wwpdb.RobustResidueMapper.GlobalAligner-class.html"><a title="csb.bio.io.wwpdb.RobustResidueMapper.GlobalAligner" class="py-name" href="#" onclick="return doclink('link-1001', 'GlobalAligner', 'link-1001');">GlobalAligner</a></tt><tt class="py-op">(</tt><tt class="py-name">scoring</tt><tt class="py-op">=</tt><tt class="py-name">scoring</tt><tt class="py-op">,</tt> <tt id="link-1002" class="py-name" targets="Variable csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap=csb.bio.sequence.alignment.AbstractAlignmentAlgorithm-class.html#gap"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-1002', 'gap', 'link-1002');">gap</a></tt><tt class="py-op">=</tt><tt id="link-1003" class="py-name"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.gap" class="py-name" href="#" onclick="return doclink('link-1003', 'gap', 'link-1002');">gap</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1671"></a><tt class="py-lineno">1671</tt>  <tt class="py-line">         </tt>
<a name="L1672"></a><tt class="py-lineno">1672</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_aligner</tt> <tt class="py-op">=</tt> <tt id="link-1004" class="py-name"><a title="csb.apps.hhfrag.HHfrag.aligner" class="py-name" href="#" onclick="return doclink('link-1004', 'aligner', 'link-999');">aligner</a></tt> </tt>
</div><a name="L1673"></a><tt class="py-lineno">1673</tt>  <tt class="py-line">         </tt>
<a name="RobustResidueMapper.map"></a><div id="RobustResidueMapper.map-def"><a name="L1674"></a><tt class="py-lineno">1674</tt> <a class="py-toggle" href="#" id="RobustResidueMapper.map-toggle" onclick="return toggle('RobustResidueMapper.map');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RobustResidueMapper-class.html#map">map</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sparse</tt><tt class="py-op">,</tt> <tt class="py-param">reference</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RobustResidueMapper.map-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RobustResidueMapper.map-expanded"><a name="L1675"></a><tt class="py-lineno">1675</tt>  <tt class="py-line">         </tt>
<a name="L1676"></a><tt class="py-lineno">1676</tt>  <tt class="py-line">        <tt class="py-name">aligned</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1677"></a><tt class="py-lineno">1677</tt>  <tt class="py-line">        <tt class="py-name">ali</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_aligner</tt><tt class="py-op">.</tt><tt id="link-1005" class="py-name" targets="Method csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.align()=csb.bio.sequence.alignment.AbstractAlignmentAlgorithm-class.html#align,Method csb.bio.structure.Chain.align()=csb.bio.structure.Chain-class.html#align"><a title="csb.bio.sequence.alignment.AbstractAlignmentAlgorithm.align
csb.bio.structure.Chain.align" class="py-name" href="#" onclick="return doclink('link-1005', 'align', 'link-1005');">align</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">reference</tt><tt class="py-op">)</tt> </tt>
<a name="L1678"></a><tt class="py-lineno">1678</tt>  <tt class="py-line">         </tt>
<a name="L1679"></a><tt class="py-lineno">1679</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">ali</tt><tt class="py-op">.</tt><tt id="link-1006" class="py-name" targets="Method csb.bio.io.clans.ClansSeqgroup.is_empty()=csb.bio.io.clans.ClansSeqgroup-class.html#is_empty,Variable csb.bio.sequence.alignment.AlignmentResult.is_empty=csb.bio.sequence.alignment.AlignmentResult-class.html#is_empty"><a title="csb.bio.io.clans.ClansSeqgroup.is_empty
csb.bio.sequence.alignment.AlignmentResult.is_empty" class="py-name" href="#" onclick="return doclink('link-1006', 'is_empty', 'link-1006');">is_empty</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1680"></a><tt class="py-lineno">1680</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1007" class="py-name"><a title="csb.bio.io.wwpdb.ResidueMappingError" class="py-name" href="#" onclick="return doclink('link-1007', 'ResidueMappingError', 'link-926');">ResidueMappingError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Global alignment failed"</tt><tt class="py-op">)</tt> </tt>
<a name="L1681"></a><tt class="py-lineno">1681</tt>  <tt class="py-line">         </tt>
<a name="L1682"></a><tt class="py-lineno">1682</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">mapped</tt><tt class="py-op">,</tt> <tt id="link-1008" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1008', 'residue', 'link-372');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">ali</tt><tt class="py-op">.</tt><tt id="link-1009" class="py-name" targets="Variable csb.apps.csfrag.CSfrag.query=csb.apps.csfrag.CSfrag-class.html#query,Variable csb.apps.hhfrag.HHfrag.query=csb.apps.hhfrag.HHfrag-class.html#query,Variable csb.apps.hhsearch.Context.query=csb.apps.hhsearch.Context-class.html#query,Variable csb.bio.hmm.HHpredHitAlignment.query=csb.bio.hmm.HHpredHitAlignment-class.html#query,Variable csb.bio.sequence.alignment.AlignmentResult.query=csb.bio.sequence.alignment.AlignmentResult-class.html#query,Static Method csb.io.plots.Backend.query()=csb.io.plots.Backend-class.html#query,Method csb.io.tsv.RepositoryImp.query()=csb.io.tsv.RepositoryImp-class.html#query,Method csb.io.tsv.SQLiteRepository.query()=csb.io.tsv.SQLiteRepository-class.html#query,Method csb.io.tsv.Table.query()=csb.io.tsv.Table-class.html#query"><a title="csb.apps.csfrag.CSfrag.query
csb.apps.hhfrag.HHfrag.query
csb.apps.hhsearch.Context.query
csb.bio.hmm.HHpredHitAlignment.query
csb.bio.sequence.alignment.AlignmentResult.query
csb.io.plots.Backend.query
csb.io.tsv.RepositoryImp.query
csb.io.tsv.SQLiteRepository.query
csb.io.tsv.Table.query" class="py-name" href="#" onclick="return doclink('link-1009', 'query', 'link-1009');">query</a></tt><tt class="py-op">,</tt> <tt class="py-name">ali</tt><tt class="py-op">.</tt><tt id="link-1010" class="py-name" targets="Variable csb.bio.hmm.HHpredHitAlignment.subject=csb.bio.hmm.HHpredHitAlignment-class.html#subject,Variable csb.bio.sequence.alignment.AlignmentResult.subject=csb.bio.sequence.alignment.AlignmentResult-class.html#subject"><a title="csb.bio.hmm.HHpredHitAlignment.subject
csb.bio.sequence.alignment.AlignmentResult.subject" class="py-name" href="#" onclick="return doclink('link-1010', 'subject', 'link-1010');">subject</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1683"></a><tt class="py-lineno">1683</tt>  <tt class="py-line">             </tt>
<a name="L1684"></a><tt class="py-lineno">1684</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-1011" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1011', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1012" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-1012', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">reference</tt><tt class="py-op">.</tt><tt id="link-1013" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-1013', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-1014" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-1014', 'GAP', 'link-476');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1685"></a><tt class="py-lineno">1685</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L1686"></a><tt class="py-lineno">1686</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">mapped</tt><tt class="py-op">.</tt><tt id="link-1015" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.type
csb.bio.fragments.TargetResidue.type
csb.bio.hmm.State.type
csb.bio.hmm.Transition.type
csb.bio.io.clans.ClansSeqgroup.type
csb.bio.io.wwpdb.ResidueInfo.type
csb.bio.sequence.AbstractSequence.type
csb.bio.sequence.ResidueInfo.type
csb.bio.sequence.SequenceAdapter.type
csb.bio.structure.Chain.type
csb.bio.structure.Residue.type
csb.bio.structure.SecondaryStructureElement.type
csb.io.tsv.ColumnInfo.type" class="py-name" href="#" onclick="return doclink('link-1015', 'type', 'link-109');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-1016" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-1016', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">.</tt><tt id="link-1017" class="py-name"><a title="csb.bio.hmm.HHpredHitAlignment.GAP
csb.bio.sequence.NucleicAlphabet.GAP
csb.bio.sequence.ProteinAlphabet.GAP
csb.bio.sequence.UnknownAlphabet.GAP" class="py-name" href="#" onclick="return doclink('link-1017', 'GAP', 'link-476');">GAP</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1687"></a><tt class="py-lineno">1687</tt>  <tt class="py-line">                <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-1018" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1018', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1019" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.create_gap" class="py-name" href="#" onclick="return doclink('link-1019', 'create_gap', 'link-935');">create_gap</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">.</tt><tt id="link-1020" class="py-name"><a title="csb.bio.sequence.AbstractSequence.alphabet
csb.bio.sequence.SequenceAdapter.alphabet
csb.bio.structure.Chain.alphabet" class="py-name" href="#" onclick="return doclink('link-1020', 'alphabet', 'link-475');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1688"></a><tt class="py-lineno">1688</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1689"></a><tt class="py-lineno">1689</tt>  <tt class="py-line">                <tt class="py-name">aligned</tt><tt class="py-op">.</tt><tt id="link-1021" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1021', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">mapped</tt><tt class="py-op">)</tt> </tt>
<a name="L1690"></a><tt class="py-lineno">1690</tt>  <tt class="py-line">                 </tt>
<a name="L1691"></a><tt class="py-lineno">1691</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_build</tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">aligned</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1692"></a><tt class="py-lineno">1692</tt>  <tt class="py-line"> </tt>
<a name="CombinedResidueMapper"></a><div id="CombinedResidueMapper-def"><a name="L1693"></a><tt class="py-lineno">1693</tt> <a class="py-toggle" href="#" id="CombinedResidueMapper-toggle" onclick="return toggle('CombinedResidueMapper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CombinedResidueMapper-class.html">CombinedResidueMapper</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractResidueMapper</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CombinedResidueMapper-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="CombinedResidueMapper-expanded"><a name="L1694"></a><tt class="py-lineno">1694</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1695"></a><tt class="py-lineno">1695</tt>  <tt class="py-line"><tt class="py-docstring">    The best of both worlds: attempts to map the residues using</tt> </tt>
<a name="L1696"></a><tt class="py-lineno">1696</tt>  <tt class="py-line"><tt class="py-docstring">    L{FastResidueMapper}, but upon failure secures success by switching to</tt> </tt>
<a name="L1697"></a><tt class="py-lineno">1697</tt>  <tt class="py-line"><tt class="py-docstring">    L{RobustResidueMapper}. </tt> </tt>
<a name="L1698"></a><tt class="py-lineno">1698</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1699"></a><tt class="py-lineno">1699</tt>  <tt class="py-line">    <tt id="link-1022" class="py-name" targets="Variable csb.bio.io.wwpdb.CombinedResidueMapper.FAST=csb.bio.io.wwpdb.CombinedResidueMapper-class.html#FAST"><a title="csb.bio.io.wwpdb.CombinedResidueMapper.FAST" class="py-name" href="#" onclick="return doclink('link-1022', 'FAST', 'link-1022');">FAST</a></tt> <tt class="py-op">=</tt> <tt id="link-1023" class="py-name"><a title="csb.bio.io.wwpdb.FastResidueMapper" class="py-name" href="#" onclick="return doclink('link-1023', 'FastResidueMapper', 'link-904');">FastResidueMapper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1700"></a><tt class="py-lineno">1700</tt>  <tt class="py-line">    <tt id="link-1024" class="py-name" targets="Variable csb.bio.io.wwpdb.CombinedResidueMapper.ROBUST=csb.bio.io.wwpdb.CombinedResidueMapper-class.html#ROBUST"><a title="csb.bio.io.wwpdb.CombinedResidueMapper.ROBUST" class="py-name" href="#" onclick="return doclink('link-1024', 'ROBUST', 'link-1024');">ROBUST</a></tt> <tt class="py-op">=</tt> <tt id="link-1025" class="py-name"><a title="csb.bio.io.wwpdb.RobustResidueMapper" class="py-name" href="#" onclick="return doclink('link-1025', 'RobustResidueMapper', 'link-1000');">RobustResidueMapper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1701"></a><tt class="py-lineno">1701</tt>  <tt class="py-line">     </tt>
<a name="CombinedResidueMapper.map"></a><div id="CombinedResidueMapper.map-def"><a name="L1702"></a><tt class="py-lineno">1702</tt> <a class="py-toggle" href="#" id="CombinedResidueMapper.map-toggle" onclick="return toggle('CombinedResidueMapper.map');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CombinedResidueMapper-class.html#map">map</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sparse</tt><tt class="py-op">,</tt> <tt class="py-param">reference</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CombinedResidueMapper.map-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CombinedResidueMapper.map-expanded"><a name="L1703"></a><tt class="py-lineno">1703</tt>  <tt class="py-line">         </tt>
<a name="L1704"></a><tt class="py-lineno">1704</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1705"></a><tt class="py-lineno">1705</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-1026" class="py-name"><a title="csb.bio.io.wwpdb.CombinedResidueMapper" class="py-name" href="#" onclick="return doclink('link-1026', 'CombinedResidueMapper', 'link-71');">CombinedResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-1027" class="py-name"><a title="csb.bio.io.wwpdb.CombinedResidueMapper.FAST" class="py-name" href="#" onclick="return doclink('link-1027', 'FAST', 'link-1022');">FAST</a></tt><tt class="py-op">.</tt><tt id="link-1028" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-1028', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">reference</tt><tt class="py-op">)</tt>         </tt>
<a name="L1706"></a><tt class="py-lineno">1706</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-1029" class="py-name"><a title="csb.bio.io.wwpdb.ResidueMappingError" class="py-name" href="#" onclick="return doclink('link-1029', 'ResidueMappingError', 'link-926');">ResidueMappingError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1707"></a><tt class="py-lineno">1707</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-1030" class="py-name"><a title="csb.bio.io.wwpdb.CombinedResidueMapper" class="py-name" href="#" onclick="return doclink('link-1030', 'CombinedResidueMapper', 'link-71');">CombinedResidueMapper</a></tt><tt class="py-op">.</tt><tt id="link-1031" class="py-name"><a title="csb.bio.io.wwpdb.CombinedResidueMapper.ROBUST" class="py-name" href="#" onclick="return doclink('link-1031', 'ROBUST', 'link-1024');">ROBUST</a></tt><tt class="py-op">.</tt><tt id="link-1032" class="py-name"><a title="csb.bio.io.wwpdb.AbstractResidueMapper.map
csb.bio.io.wwpdb.CombinedResidueMapper.map
csb.bio.io.wwpdb.FastResidueMapper.map
csb.bio.io.wwpdb.RobustResidueMapper.map" class="py-name" href="#" onclick="return doclink('link-1032', 'map', 'link-170');">map</a></tt><tt class="py-op">(</tt><tt class="py-name">sparse</tt><tt class="py-op">,</tt> <tt class="py-name">reference</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1708"></a><tt class="py-lineno">1708</tt>  <tt class="py-line">                 </tt>
<a name="FileBuilder"></a><div id="FileBuilder-def"><a name="L1709"></a><tt class="py-lineno">1709</tt>  <tt class="py-line"> </tt>
<a name="L1710"></a><tt class="py-lineno">1710</tt> <a class="py-toggle" href="#" id="FileBuilder-toggle" onclick="return toggle('FileBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html">FileBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FileBuilder-expanded"><a name="L1711"></a><tt class="py-lineno">1711</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1712"></a><tt class="py-lineno">1712</tt>  <tt class="py-line"><tt class="py-docstring">    Base abstract files for all structure file formatters.</tt> </tt>
<a name="L1713"></a><tt class="py-lineno">1713</tt>  <tt class="py-line"><tt class="py-docstring">    Defines a common step-wise interface according to the Builder pattern.</tt> </tt>
<a name="L1714"></a><tt class="py-lineno">1714</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1715"></a><tt class="py-lineno">1715</tt>  <tt class="py-line"><tt class="py-docstring">    @param output: output stream (this is where the product is constructed)</tt> </tt>
<a name="L1716"></a><tt class="py-lineno">1716</tt>  <tt class="py-line"><tt class="py-docstring">    @type output: stream</tt> </tt>
<a name="L1717"></a><tt class="py-lineno">1717</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1718"></a><tt class="py-lineno">1718</tt>  <tt class="py-line">     </tt>
<a name="L1719"></a><tt class="py-lineno">1719</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-1033" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-1033', 'ABCMeta', 'link-15');">ABCMeta</a></tt> </tt>
<a name="L1720"></a><tt class="py-lineno">1720</tt>  <tt class="py-line"> </tt>
<a name="FileBuilder.__init__"></a><div id="FileBuilder.__init__-def"><a name="L1721"></a><tt class="py-lineno">1721</tt> <a class="py-toggle" href="#" id="FileBuilder.__init__-toggle" onclick="return toggle('FileBuilder.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.__init__-expanded"><a name="L1722"></a><tt class="py-lineno">1722</tt>  <tt class="py-line"> </tt>
<a name="L1723"></a><tt class="py-lineno">1723</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-1034" class="py-name" targets="Variable csb.bio.fragments.rosetta.OutputBuilder.output=csb.bio.fragments.rosetta.OutputBuilder-class.html#output,Variable csb.bio.io.clans.ClansFileBuilder.output=csb.bio.io.clans.ClansFileBuilder-class.html#output,Variable csb.bio.io.fasta.OutputBuilder.output=csb.bio.io.fasta.OutputBuilder-class.html#output,Variable csb.bio.io.hhpred.HHMFileBuilder.output=csb.bio.io.hhpred.HHMFileBuilder-class.html#output,Variable csb.bio.io.wwpdb.FileBuilder.output=csb.bio.io.wwpdb.FileBuilder-class.html#output,Variable csb.build.Console.output=csb.build.Console-class.html#output"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-1034', 'output', 'link-1034');">output</a></tt><tt class="py-op">,</tt> <tt class="py-string">'write'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1724"></a><tt class="py-lineno">1724</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt id="link-1035" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-1035', 'output', 'link-1034');">output</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1725"></a><tt class="py-lineno">1725</tt>  <tt class="py-line">         </tt>
<a name="L1726"></a><tt class="py-lineno">1726</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">isnull</tt><tt class="py-op">(</tt><tt class="py-param">this</tt><tt class="py-op">,</tt> <tt class="py-param">that</tt><tt class="py-op">,</tt> <tt class="py-param">null</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1727"></a><tt class="py-lineno">1727</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">this</tt> <tt class="py-keyword">is</tt> <tt class="py-name">null</tt><tt class="py-op">:</tt> </tt>
<a name="L1728"></a><tt class="py-lineno">1728</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">that</tt> </tt>
<a name="L1729"></a><tt class="py-lineno">1729</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1730"></a><tt class="py-lineno">1730</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">this</tt>         </tt>
</div><a name="L1731"></a><tt class="py-lineno">1731</tt>  <tt class="py-line"> </tt>
<a name="L1732"></a><tt class="py-lineno">1732</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt> <tt class="py-op">=</tt> <tt id="link-1036" class="py-name"><a title="csb.bio.fragments.rosetta.OutputBuilder.output
csb.bio.io.clans.ClansFileBuilder.output
csb.bio.io.fasta.OutputBuilder.output
csb.bio.io.hhpred.HHMFileBuilder.output
csb.bio.io.wwpdb.FileBuilder.output
csb.build.Console.output" class="py-name" href="#" onclick="return doclink('link-1036', 'output', 'link-1034');">output</a></tt> </tt>
<a name="L1733"></a><tt class="py-lineno">1733</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_isnull</tt> <tt class="py-op">=</tt> <tt id="link-1037" class="py-name" targets="Variable csb.bio.io.wwpdb.FileBuilder.isnull=csb.bio.io.wwpdb.FileBuilder-class.html#isnull"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1037', 'isnull', 'link-1037');">isnull</a></tt> </tt>
</div><a name="L1734"></a><tt class="py-lineno">1734</tt>  <tt class="py-line">         </tt>
<a name="L1735"></a><tt class="py-lineno">1735</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="FileBuilder.output"></a><div id="FileBuilder.output-def"><a name="L1736"></a><tt class="py-lineno">1736</tt> <a class="py-toggle" href="#" id="FileBuilder.output-toggle" onclick="return toggle('FileBuilder.output');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#output">output</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.output-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.output-expanded"><a name="L1737"></a><tt class="py-lineno">1737</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1738"></a><tt class="py-lineno">1738</tt>  <tt class="py-line"><tt class="py-docstring">        Destination stream</tt> </tt>
<a name="L1739"></a><tt class="py-lineno">1739</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: stream</tt> </tt>
<a name="L1740"></a><tt class="py-lineno">1740</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1741"></a><tt class="py-lineno">1741</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt> </tt>
</div><a name="L1742"></a><tt class="py-lineno">1742</tt>  <tt class="py-line">     </tt>
<a name="L1743"></a><tt class="py-lineno">1743</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="FileBuilder.isnull"></a><div id="FileBuilder.isnull-def"><a name="L1744"></a><tt class="py-lineno">1744</tt> <a class="py-toggle" href="#" id="FileBuilder.isnull-toggle" onclick="return toggle('FileBuilder.isnull');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#isnull">isnull</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.isnull-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.isnull-expanded"><a name="L1745"></a><tt class="py-lineno">1745</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1746"></a><tt class="py-lineno">1746</tt>  <tt class="py-line"><tt class="py-docstring">        ISNULL(X, Y) function</tt> </tt>
<a name="L1747"></a><tt class="py-lineno">1747</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: callable</tt> </tt>
<a name="L1748"></a><tt class="py-lineno">1748</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1749"></a><tt class="py-lineno">1749</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_isnull</tt> </tt>
</div><a name="L1750"></a><tt class="py-lineno">1750</tt>  <tt class="py-line">     </tt>
<a name="FileBuilder.write"></a><div id="FileBuilder.write-def"><a name="L1751"></a><tt class="py-lineno">1751</tt> <a class="py-toggle" href="#" id="FileBuilder.write-toggle" onclick="return toggle('FileBuilder.write');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.write-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.write-expanded"><a name="L1752"></a><tt class="py-lineno">1752</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1753"></a><tt class="py-lineno">1753</tt>  <tt class="py-line"><tt class="py-docstring">        Write a chunk of text</tt> </tt>
<a name="L1754"></a><tt class="py-lineno">1754</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1755"></a><tt class="py-lineno">1755</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-1038" class="py-name" targets="Method csb.bio.io.clans.Clans.write()=csb.bio.io.clans.Clans-class.html#write,Method csb.bio.io.clans.ClansFileBuilder.write()=csb.bio.io.clans.ClansFileBuilder-class.html#write,Method csb.bio.io.fasta.OutputBuilder.write()=csb.bio.io.fasta.OutputBuilder-class.html#write,Method csb.bio.io.hhpred.HHMFileBuilder.write()=csb.bio.io.hhpred.HHMFileBuilder-class.html#write,Method csb.bio.io.mrc.DensityMapWriter.write()=csb.bio.io.mrc.DensityMapWriter-class.html#write,Method csb.bio.io.wwpdb.FileBuilder.write()=csb.bio.io.wwpdb.FileBuilder-class.html#write,Method csb.build.RevisionHandler.write()=csb.build.RevisionHandler-class.html#write,Method csb.io.AutoFlushStream.write()=csb.io.AutoFlushStream-class.html#write,Method csb.io.EntryWriter.write()=csb.io.EntryWriter-class.html#write"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1038', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt>    </tt>
</div><a name="L1756"></a><tt class="py-lineno">1756</tt>  <tt class="py-line"> </tt>
<a name="FileBuilder.writeline"></a><div id="FileBuilder.writeline-def"><a name="L1757"></a><tt class="py-lineno">1757</tt> <a class="py-toggle" href="#" id="FileBuilder.writeline-toggle" onclick="return toggle('FileBuilder.writeline');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#writeline">writeline</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.writeline-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.writeline-expanded"><a name="L1758"></a><tt class="py-lineno">1758</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1759"></a><tt class="py-lineno">1759</tt>  <tt class="py-line"><tt class="py-docstring">        Write a chunk of text and append a new line terminator</tt> </tt>
<a name="L1760"></a><tt class="py-lineno">1760</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L1761"></a><tt class="py-lineno">1761</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-1039" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1039', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L1762"></a><tt class="py-lineno">1762</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt id="link-1040" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1040', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1763"></a><tt class="py-lineno">1763</tt>  <tt class="py-line">                     </tt>
<a name="L1764"></a><tt class="py-lineno">1764</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="FileBuilder.add_header"></a><div id="FileBuilder.add_header-def"><a name="L1765"></a><tt class="py-lineno">1765</tt> <a class="py-toggle" href="#" id="FileBuilder.add_header-toggle" onclick="return toggle('FileBuilder.add_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#add_header">add_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">master_structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.add_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.add_header-expanded"><a name="L1766"></a><tt class="py-lineno">1766</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1767"></a><tt class="py-lineno">1767</tt>  <tt class="py-line">     </tt>
<a name="L1768"></a><tt class="py-lineno">1768</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="FileBuilder.add_structure"></a><div id="FileBuilder.add_structure-def"><a name="L1769"></a><tt class="py-lineno">1769</tt> <a class="py-toggle" href="#" id="FileBuilder.add_structure-toggle" onclick="return toggle('FileBuilder.add_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#add_structure">add_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.add_structure-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.add_structure-expanded"><a name="L1770"></a><tt class="py-lineno">1770</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1771"></a><tt class="py-lineno">1771</tt>  <tt class="py-line">     </tt>
<a name="FileBuilder.finalize"></a><div id="FileBuilder.finalize-def"><a name="L1772"></a><tt class="py-lineno">1772</tt> <a class="py-toggle" href="#" id="FileBuilder.finalize-toggle" onclick="return toggle('FileBuilder.finalize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileBuilder-class.html#finalize">finalize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileBuilder.finalize-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileBuilder.finalize-expanded"><a name="L1773"></a><tt class="py-lineno">1773</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L1774"></a><tt class="py-lineno">1774</tt>  <tt class="py-line"> </tt>
<a name="PDBFileBuilder"></a><div id="PDBFileBuilder-def"><a name="L1775"></a><tt class="py-lineno">1775</tt> <a class="py-toggle" href="#" id="PDBFileBuilder-toggle" onclick="return toggle('PDBFileBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBFileBuilder-class.html">PDBFileBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">FileBuilder</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFileBuilder-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PDBFileBuilder-expanded"><a name="L1776"></a><tt class="py-lineno">1776</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1777"></a><tt class="py-lineno">1777</tt>  <tt class="py-line"><tt class="py-docstring">    PDB file format builder.</tt> </tt>
<a name="L1778"></a><tt class="py-lineno">1778</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1779"></a><tt class="py-lineno">1779</tt>  <tt class="py-line">         </tt>
<a name="PDBFileBuilder.writeline"></a><div id="PDBFileBuilder.writeline-def"><a name="L1780"></a><tt class="py-lineno">1780</tt> <a class="py-toggle" href="#" id="PDBFileBuilder.writeline-toggle" onclick="return toggle('PDBFileBuilder.writeline');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBFileBuilder-class.html#writeline">writeline</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFileBuilder.writeline-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFileBuilder.writeline-expanded"><a name="L1781"></a><tt class="py-lineno">1781</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1041" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1041', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'{0:80}\n'</tt><tt class="py-op">.</tt><tt id="link-1042" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1042', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1782"></a><tt class="py-lineno">1782</tt>  <tt class="py-line">         </tt>
<a name="PDBFileBuilder.add_header"></a><div id="PDBFileBuilder.add_header-def"><a name="L1783"></a><tt class="py-lineno">1783</tt> <a class="py-toggle" href="#" id="PDBFileBuilder.add_header-toggle" onclick="return toggle('PDBFileBuilder.add_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBFileBuilder-class.html#add_header">add_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">master</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFileBuilder.add_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFileBuilder.add_header-expanded"><a name="L1784"></a><tt class="py-lineno">1784</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1785"></a><tt class="py-lineno">1785</tt>  <tt class="py-line"><tt class="py-docstring">        Write the HEADER of the file using C{master}</tt> </tt>
<a name="L1786"></a><tt class="py-lineno">1786</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1787"></a><tt class="py-lineno">1787</tt>  <tt class="py-line"><tt class="py-docstring">        @type master: L{Structure}</tt> </tt>
<a name="L1788"></a><tt class="py-lineno">1788</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1789"></a><tt class="py-lineno">1789</tt>  <tt class="py-line"> </tt>
<a name="L1790"></a><tt class="py-lineno">1790</tt>  <tt class="py-line">        <tt id="link-1043" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1043', 'isnull', 'link-1037');">isnull</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1044" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1044', 'isnull', 'link-1037');">isnull</a></tt> </tt>
<a name="L1791"></a><tt class="py-lineno">1791</tt>  <tt class="py-line">         </tt>
<a name="L1792"></a><tt class="py-lineno">1792</tt>  <tt class="py-line">        <tt id="link-1045" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-1045', 'header', 'link-583');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'HEADER    {0:40}{1:%d-%b-%y}   {2:4}'</tt> </tt>
<a name="L1793"></a><tt class="py-lineno">1793</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1046" class="py-name" targets="Method csb.bio.io.clans.ClansFileBuilder.writeline()=csb.bio.io.clans.ClansFileBuilder-class.html#writeline,Method csb.bio.io.fasta.OutputBuilder.writeline()=csb.bio.io.fasta.OutputBuilder-class.html#writeline,Method csb.bio.io.hhpred.HHMFileBuilder.writeline()=csb.bio.io.hhpred.HHMFileBuilder-class.html#writeline,Method csb.bio.io.wwpdb.FileBuilder.writeline()=csb.bio.io.wwpdb.FileBuilder-class.html#writeline,Method csb.bio.io.wwpdb.PDBFileBuilder.writeline()=csb.bio.io.wwpdb.PDBFileBuilder-class.html#writeline,Method csb.io.EntryWriter.writeline()=csb.io.EntryWriter-class.html#writeline"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1046', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt id="link-1047" class="py-name"><a title="csb.bio.sequence.AbstractSequence.header
csb.bio.sequence.SequenceAdapter.header
csb.bio.structure.Chain.header" class="py-name" href="#" onclick="return doclink('link-1047', 'header', 'link-583');">header</a></tt><tt class="py-op">.</tt><tt id="link-1048" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1048', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">,</tt> <tt class="py-name">datetime</tt><tt class="py-op">.</tt><tt class="py-name">datetime</tt><tt class="py-op">.</tt><tt class="py-name">now</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-1049" class="py-name" targets="Variable csb.bio.sequence.A3MAlignment.master=csb.bio.sequence.A3MAlignment-class.html#master"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1049', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1050" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-1050', 'accession', 'link-212');">accession</a></tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1794"></a><tt class="py-lineno">1794</tt>  <tt class="py-line">         </tt>
<a name="L1795"></a><tt class="py-lineno">1795</tt>  <tt class="py-line">        <tt class="py-name">molecules</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> <tt class="py-op">}</tt> </tt>
<a name="L1796"></a><tt class="py-lineno">1796</tt>  <tt class="py-line">         </tt>
<a name="L1797"></a><tt class="py-lineno">1797</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1051" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1051', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1052" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1052', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1053" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1053', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1798"></a><tt class="py-lineno">1798</tt>  <tt class="py-line">            <tt id="link-1054" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1054', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-1055" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1055', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1056" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1056', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-1057" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1057', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1799"></a><tt class="py-lineno">1799</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-1058" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1058', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1059" class="py-name"><a title="csb.bio.structure.Chain.molecule_id" class="py-name" href="#" onclick="return doclink('link-1059', 'molecule_id', 'link-649');">molecule_id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">molecules</tt><tt class="py-op">:</tt> </tt>
<a name="L1800"></a><tt class="py-lineno">1800</tt>  <tt class="py-line">                <tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt id="link-1060" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1060', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1061" class="py-name"><a title="csb.bio.structure.Chain.molecule_id" class="py-name" href="#" onclick="return doclink('link-1061', 'molecule_id', 'link-649');">molecule_id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-op">]</tt> </tt>
<a name="L1801"></a><tt class="py-lineno">1801</tt>  <tt class="py-line">            <tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt id="link-1062" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1062', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1063" class="py-name"><a title="csb.bio.structure.Chain.molecule_id" class="py-name" href="#" onclick="return doclink('link-1063', 'molecule_id', 'link-649');">molecule_id</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-1064" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1064', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-1065" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1065', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1802"></a><tt class="py-lineno">1802</tt>  <tt class="py-line">         </tt>
<a name="L1803"></a><tt class="py-lineno">1803</tt>  <tt class="py-line">        <tt class="py-name">k</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1804"></a><tt class="py-lineno">1804</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">mol_id</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">molecules</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1805"></a><tt class="py-lineno">1805</tt>  <tt class="py-line">             </tt>
<a name="L1806"></a><tt class="py-lineno">1806</tt>  <tt class="py-line">            <tt id="link-1066" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1066', 'chains', 'link-193');">chains</a></tt> <tt class="py-op">=</tt> <tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt class="py-name">mol_id</tt><tt class="py-op">]</tt> </tt>
<a name="L1807"></a><tt class="py-lineno">1807</tt>  <tt class="py-line">            <tt id="link-1067" class="py-name" targets="Variable csb.bio.structure.Structure.first_chain=csb.bio.structure.Structure-class.html#first_chain"><a title="csb.bio.structure.Structure.first_chain" class="py-name" href="#" onclick="return doclink('link-1067', 'first_chain', 'link-1067');">first_chain</a></tt> <tt class="py-op">=</tt> <tt id="link-1068" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1068', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1069" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1069', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt> <tt id="link-1070" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1070', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">]</tt>             </tt>
<a name="L1808"></a><tt class="py-lineno">1808</tt>  <tt class="py-line">             </tt>
<a name="L1809"></a><tt class="py-lineno">1809</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1071" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1071', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'COMPND {0:3} MOL_ID: {1};'</tt><tt class="py-op">.</tt><tt id="link-1072" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1072', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt id="link-1073" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1073', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt class="py-name">mol_id</tt><tt class="py-op">,</tt> <tt class="py-string">'0'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1810"></a><tt class="py-lineno">1810</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1074" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1074', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'COMPND {0:3} MOLECULE: {1};'</tt><tt class="py-op">.</tt><tt id="link-1075" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1075', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt> <tt class="py-op">+</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt id="link-1076" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1076', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt id="link-1077" class="py-name"><a title="csb.bio.structure.Structure.first_chain" class="py-name" href="#" onclick="return doclink('link-1077', 'first_chain', 'link-1067');">first_chain</a></tt><tt class="py-op">.</tt><tt id="link-1078" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-1078', 'name', 'link-82');">name</a></tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1811"></a><tt class="py-lineno">1811</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1079" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1079', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'COMPND {0:3} CHAIN: {1};'</tt><tt class="py-op">.</tt><tt id="link-1080" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1080', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-string">', '</tt><tt class="py-op">.</tt><tt id="link-1081" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-1081', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt id="link-1082" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1082', 'chains', 'link-193');">chains</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1812"></a><tt class="py-lineno">1812</tt>  <tt class="py-line">            <tt class="py-name">k</tt> <tt class="py-op">+=</tt> <tt class="py-number">3</tt> </tt>
<a name="L1813"></a><tt class="py-lineno">1813</tt>  <tt class="py-line">             </tt>
<a name="L1814"></a><tt class="py-lineno">1814</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1083" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1083', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1084" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1084', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1085" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1085', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1815"></a><tt class="py-lineno">1815</tt>  <tt class="py-line">             </tt>
<a name="L1816"></a><tt class="py-lineno">1816</tt>  <tt class="py-line">            <tt id="link-1086" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1086', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-1087" class="py-name"><a title="csb.bio.sequence.A3MAlignment.master" class="py-name" href="#" onclick="return doclink('link-1087', 'master', 'link-1049');">master</a></tt><tt class="py-op">.</tt><tt id="link-1088" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1088', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-1089" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1089', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1817"></a><tt class="py-lineno">1817</tt>  <tt class="py-line">            <tt class="py-name">res</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt id="link-1090" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1090', 'r', 'link-939');">r</a></tt><tt class="py-op">.</tt><tt id="link-1091" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-1091', 'label', 'link-108');">label</a></tt> <tt class="py-keyword">for</tt> <tt id="link-1092" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1092', 'r', 'link-939');">r</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1093" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1093', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1094" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-1094', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">]</tt> </tt>
<a name="L1818"></a><tt class="py-lineno">1818</tt>  <tt class="py-line"> </tt>
<a name="L1819"></a><tt class="py-lineno">1819</tt>  <tt class="py-line">            <tt class="py-name">rn</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1820"></a><tt class="py-lineno">1820</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">j</tt> <tt class="py-keyword">in</tt> <tt class="py-name">range</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-1095" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1095', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1096" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-1096', 'length', 'link-261');">length</a></tt><tt class="py-op">,</tt> <tt class="py-number">13</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1821"></a><tt class="py-lineno">1821</tt>  <tt class="py-line">                <tt class="py-name">rn</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1822"></a><tt class="py-lineno">1822</tt>  <tt class="py-line">                <tt id="link-1097" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-1097', 'residues', 'link-262');">residues</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-string">'{0:&gt;3}'</tt><tt class="py-op">.</tt><tt id="link-1098" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1098', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-1099" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1099', 'r', 'link-939');">r</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-1100" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1100', 'r', 'link-939');">r</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">res</tt><tt class="py-op">[</tt><tt class="py-name">j</tt> <tt class="py-op">:</tt> <tt class="py-name">j</tt> <tt class="py-op">+</tt> <tt class="py-number">13</tt><tt class="py-op">]</tt> <tt class="py-op">]</tt> </tt>
<a name="L1823"></a><tt class="py-lineno">1823</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1101" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1101', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'SEQRES {0:&gt;3} {1} {2:&gt;4}  {3}'</tt><tt class="py-op">.</tt><tt id="link-1102" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1102', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1824"></a><tt class="py-lineno">1824</tt>  <tt class="py-line">                                            <tt class="py-name">rn</tt><tt class="py-op">,</tt> <tt id="link-1103" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1103', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1104" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1104', 'id', 'link-29');">id</a></tt><tt class="py-op">,</tt> <tt id="link-1105" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1105', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1106" class="py-name"><a title="csb.apps.hhfrag.SliceContext.length
csb.bio.fragments.FragmentMatch.length
csb.bio.fragments.Target.length
csb.bio.fragments.rosetta.RosettaFragment.length
csb.bio.hmm.HHpredHit.length
csb.bio.hmm.ProfileHMM.length
csb.bio.sequence.AbstractAlignment.length
csb.bio.sequence.AbstractSequence.length
csb.bio.sequence.SequenceAdapter.length
csb.bio.sequence.alignment.AlignmentResult.length
csb.bio.structure.Chain.length
csb.bio.structure.SecondaryStructureElement.length
csb.core.BaseCollectionContainer.length
csb.core.BaseDictionaryContainer.length" class="py-name" href="#" onclick="return doclink('link-1106', 'length', 'link-261');">length</a></tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt id="link-1107" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-1107', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt id="link-1108" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-1108', 'residues', 'link-262');">residues</a></tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1825"></a><tt class="py-lineno">1825</tt>  <tt class="py-line">                 </tt>
<a name="PDBFileBuilder.add_structure"></a><div id="PDBFileBuilder.add_structure-def"><a name="L1826"></a><tt class="py-lineno">1826</tt> <a class="py-toggle" href="#" id="PDBFileBuilder.add_structure-toggle" onclick="return toggle('PDBFileBuilder.add_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBFileBuilder-class.html#add_structure">add_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFileBuilder.add_structure-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFileBuilder.add_structure-expanded"><a name="L1827"></a><tt class="py-lineno">1827</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1828"></a><tt class="py-lineno">1828</tt>  <tt class="py-line"><tt class="py-docstring">        Append a new model to the file</tt> </tt>
<a name="L1829"></a><tt class="py-lineno">1829</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1830"></a><tt class="py-lineno">1830</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure: L{Structure}</tt> </tt>
<a name="L1831"></a><tt class="py-lineno">1831</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1832"></a><tt class="py-lineno">1832</tt>  <tt class="py-line"> </tt>
<a name="L1833"></a><tt class="py-lineno">1833</tt>  <tt class="py-line">        <tt id="link-1109" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1109', 'isnull', 'link-1037');">isnull</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1110" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1110', 'isnull', 'link-1037');">isnull</a></tt> </tt>
<a name="L1834"></a><tt class="py-lineno">1834</tt>  <tt class="py-line">          </tt>
<a name="L1835"></a><tt class="py-lineno">1835</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1111" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1111', 'chain_id', 'link-568');">chain_id</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1112" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-1112', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-1113" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1113', 'chains', 'link-193');">chains</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1836"></a><tt class="py-lineno">1836</tt>  <tt class="py-line">         </tt>
<a name="L1837"></a><tt class="py-lineno">1837</tt>  <tt class="py-line">            <tt id="link-1114" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1114', 'chain', 'link-220');">chain</a></tt> <tt class="py-op">=</tt> <tt id="link-1115" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-1115', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-1116" class="py-name"><a title="csb.bio.structure.Structure.chains" class="py-name" href="#" onclick="return doclink('link-1116', 'chains', 'link-193');">chains</a></tt><tt class="py-op">[</tt><tt id="link-1117" class="py-name"><a title="csb.bio.fragments.Target.chain_id" class="py-name" href="#" onclick="return doclink('link-1117', 'chain_id', 'link-568');">chain_id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1838"></a><tt class="py-lineno">1838</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-1118" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1118', 'residue', 'link-372');">residue</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1119" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1119', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1120" class="py-name"><a title="csb.bio.fragments.Target.residues
csb.bio.fragments.rosetta.RosettaFragment.residues
csb.bio.hmm.ProfileHMM.residues
csb.bio.sequence.AbstractSequence.residues
csb.bio.sequence.AlignedSequenceAdapter.residues
csb.bio.structure.Chain.residues" class="py-name" href="#" onclick="return doclink('link-1120', 'residues', 'link-262');">residues</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1839"></a><tt class="py-lineno">1839</tt>  <tt class="py-line">         </tt>
<a name="L1840"></a><tt class="py-lineno">1840</tt>  <tt class="py-line">                <tt id="link-1121" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1121', 'atoms', 'link-288');">atoms</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-op">]</tt> </tt>
<a name="L1841"></a><tt class="py-lineno">1841</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">an</tt> <tt class="py-keyword">in</tt> <tt id="link-1122" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1122', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1123" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1123', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1842"></a><tt class="py-lineno">1842</tt>  <tt class="py-line">                    <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt id="link-1124" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1124', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1125" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1125', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">[</tt><tt class="py-name">an</tt><tt class="py-op">]</tt> </tt>
<a name="L1843"></a><tt class="py-lineno">1843</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">,</tt> <tt id="link-1126" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1126', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1127" class="py-name"><a title="csb.bio" class="py-name" href="#" onclick="return doclink('link-1127', 'bio', 'link-1');">bio</a></tt><tt class="py-op">.</tt><tt id="link-1128" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-1128', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-1129" class="py-name" targets="Class csb.bio.structure.DisorderedAtom=csb.bio.structure.DisorderedAtom-class.html"><a title="csb.bio.structure.DisorderedAtom" class="py-name" href="#" onclick="return doclink('link-1129', 'DisorderedAtom', 'link-1129');">DisorderedAtom</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1844"></a><tt class="py-lineno">1844</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">dis_atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atom</tt><tt class="py-op">:</tt> <tt id="link-1130" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1130', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-1131" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1131', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dis_atom</tt><tt class="py-op">)</tt> </tt>
<a name="L1845"></a><tt class="py-lineno">1845</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1846"></a><tt class="py-lineno">1846</tt>  <tt class="py-line">                        <tt id="link-1132" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1132', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-1133" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1133', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L1847"></a><tt class="py-lineno">1847</tt>  <tt class="py-line">                <tt id="link-1134" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1134', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">.</tt><tt id="link-1135" class="py-name" targets="Method csb.bio.fragments.rosetta.RosettaFragmentMap.sort()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#sort,Method csb.bio.hmm.HHpredHitList.sort()=csb.bio.hmm.HHpredHitList-class.html#sort,Method csb.bio.io.clans.Clans.sort()=csb.bio.io.clans.Clans-class.html#sort"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.sort
csb.bio.hmm.HHpredHitList.sort
csb.bio.io.clans.Clans.sort" class="py-name" href="#" onclick="return doclink('link-1135', 'sort', 'link-1135');">sort</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1848"></a><tt class="py-lineno">1848</tt>  <tt class="py-line">                 </tt>
<a name="L1849"></a><tt class="py-lineno">1849</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt id="link-1136" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.atoms
csb.bio.nmr.ContactMap.atoms
csb.bio.structure.Residue.atoms" class="py-name" href="#" onclick="return doclink('link-1136', 'atoms', 'link-288');">atoms</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1850"></a><tt class="py-lineno">1850</tt>  <tt class="py-line"> </tt>
<a name="L1851"></a><tt class="py-lineno">1851</tt>  <tt class="py-line">                    <tt class="py-name">alt</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1137" class="py-name"><a title="csb.bio.structure.Atom.alternate" class="py-name" href="#" onclick="return doclink('link-1137', 'alternate', 'link-333');">alternate</a></tt> </tt>
<a name="L1852"></a><tt class="py-lineno">1852</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">alt</tt> <tt class="py-keyword">is</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1853"></a><tt class="py-lineno">1853</tt>  <tt class="py-line">                        <tt class="py-name">alt</tt> <tt class="py-op">=</tt> <tt class="py-string">'A'</tt> </tt>
<a name="L1854"></a><tt class="py-lineno">1854</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">alt</tt> <tt class="py-keyword">is</tt> <tt class="py-name">False</tt><tt class="py-op">:</tt> </tt>
<a name="L1855"></a><tt class="py-lineno">1855</tt>  <tt class="py-line">                        <tt class="py-name">alt</tt> <tt class="py-op">=</tt> <tt class="py-string">' '</tt> </tt>
<a name="L1856"></a><tt class="py-lineno">1856</tt>  <tt class="py-line">                     </tt>
<a name="L1857"></a><tt class="py-lineno">1857</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1138" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-1138', 'element', 'link-360');">element</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1858"></a><tt class="py-lineno">1858</tt>  <tt class="py-line">                        <tt id="link-1139" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-1139', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1140" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-1140', 'element', 'link-360');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1859"></a><tt class="py-lineno">1859</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1860"></a><tt class="py-lineno">1860</tt>  <tt class="py-line">                        <tt id="link-1141" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-1141', 'element', 'link-360');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-string">' '</tt> </tt>
<a name="L1861"></a><tt class="py-lineno">1861</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1142" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1142', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'ATOM  {0:&gt;5} {1:&gt;4}{2}{3:&gt;3} {4}{5:&gt;4}{6}   {7:&gt;8.3f}{8:&gt;8.3f}{9:&gt;8.3f}{10:&gt;6.2f}{11:&gt;6.2f}{12:&gt;12}{13:2}'</tt><tt class="py-op">.</tt><tt id="link-1143" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1143', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1862"></a><tt class="py-lineno">1862</tt>  <tt class="py-line">                                        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1144" class="py-name"><a title="csb.bio.structure.Atom.serial_number" class="py-name" href="#" onclick="return doclink('link-1144', 'serial_number', 'link-348');">serial_number</a></tt><tt class="py-op">,</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">_full_name</tt><tt class="py-op">,</tt> <tt id="link-1145" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1145', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt class="py-name">alt</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L1863"></a><tt class="py-lineno">1863</tt>  <tt class="py-line">                                        <tt id="link-1146" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1146', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1147" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.label
csb.bio.nmr.ChemShiftInfo.label
csb.bio.structure.Residue.label" class="py-name" href="#" onclick="return doclink('link-1147', 'label', 'link-108');">label</a></tt><tt class="py-op">,</tt> <tt id="link-1148" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragment.chain
csb.bio.hmm.ProfileHMM.chain
csb.bio.io.wwpdb.EntryID.chain
csb.bio.io.wwpdb.ResidueInfo.chain
csb.bio.nmr.ContactMap.chain" class="py-name" href="#" onclick="return doclink('link-1148', 'chain', 'link-220');">chain</a></tt><tt class="py-op">.</tt><tt id="link-1149" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1149', 'id', 'link-29');">id</a></tt><tt class="py-op">,</tt>  </tt>
<a name="L1864"></a><tt class="py-lineno">1864</tt>  <tt class="py-line">                                        <tt id="link-1150" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1150', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt id="link-1151" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1151', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1152" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.sequence_number
csb.bio.structure.Residue.sequence_number" class="py-name" href="#" onclick="return doclink('link-1152', 'sequence_number', 'link-313');">sequence_number</a></tt><tt class="py-op">,</tt> <tt id="link-1153" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1153', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1154" class="py-name"><a title="csb.bio.fragments.ResidueEventInfo.rank
csb.bio.hmm.HHpredHit.rank
csb.bio.hmm.HMMLayer.rank
csb.bio.hmm.State.rank
csb.bio.io.wwpdb.ResidueInfo.rank
csb.bio.nmr.Label.rank
csb.bio.sequence.ResidueInfo.rank
csb.bio.structure.Residue.rank" class="py-name" href="#" onclick="return doclink('link-1154', 'rank', 'link-312');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-1155" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1155', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt id="link-1156" class="py-name"><a title="csb.bio.fragments.AssignmentFactory.residue
csb.bio.fragments.ChemShiftAssignmentFactory.residue
csb.bio.hmm.HMMLayer.residue
csb.bio.nmr.Label.residue
csb.bio.structure.Atom.residue" class="py-name" href="#" onclick="return doclink('link-1156', 'residue', 'link-372');">residue</a></tt><tt class="py-op">.</tt><tt id="link-1157" class="py-name"><a title="csb.bio.io.wwpdb.ResidueInfo.insertion_code
csb.bio.structure.Residue.insertion_code" class="py-name" href="#" onclick="return doclink('link-1157', 'insertion_code', 'link-315');">insertion_code</a></tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L1865"></a><tt class="py-lineno">1865</tt>  <tt class="py-line">                                        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1158" class="py-name"><a title="csb.bio.structure.Atom.vector" class="py-name" href="#" onclick="return doclink('link-1158', 'vector', 'link-353');">vector</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1159" class="py-name"><a title="csb.bio.structure.Atom.vector" class="py-name" href="#" onclick="return doclink('link-1159', 'vector', 'link-353');">vector</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1160" class="py-name"><a title="csb.bio.structure.Atom.vector" class="py-name" href="#" onclick="return doclink('link-1160', 'vector', 'link-353');">vector</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-1161" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1161', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1162" class="py-name"><a title="csb.bio.structure.Atom.occupancy" class="py-name" href="#" onclick="return doclink('link-1162', 'occupancy', 'link-334');">occupancy</a></tt><tt class="py-op">,</tt> <tt class="py-number">0.0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-1163" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1163', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1164" class="py-name"><a title="csb.bio.structure.Atom.bfactor" class="py-name" href="#" onclick="return doclink('link-1164', 'bfactor', 'link-335');">bfactor</a></tt><tt class="py-op">,</tt> <tt class="py-number">0.0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt>  </tt>
<a name="L1866"></a><tt class="py-lineno">1866</tt>  <tt class="py-line">                                        <tt id="link-1165" class="py-name"><a title="csb.bio.nmr.NOEPeak.element
csb.bio.nmr.NOESpectrum.element
csb.bio.structure.Atom.element" class="py-name" href="#" onclick="return doclink('link-1165', 'element', 'link-360');">element</a></tt><tt class="py-op">,</tt> <tt id="link-1166" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1166', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-1167" class="py-name"><a title="csb.bio.structure.Atom.charge" class="py-name" href="#" onclick="return doclink('link-1167', 'charge', 'link-336');">charge</a></tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> <tt class="py-op">)</tt><tt class="py-op">)</tt>         </tt>
<a name="L1867"></a><tt class="py-lineno">1867</tt>  <tt class="py-line"> </tt>
<a name="L1868"></a><tt class="py-lineno">1868</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1168" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1168', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'TER'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1869"></a><tt class="py-lineno">1869</tt>  <tt class="py-line">         </tt>
<a name="PDBFileBuilder.finalize"></a><div id="PDBFileBuilder.finalize-def"><a name="L1870"></a><tt class="py-lineno">1870</tt> <a class="py-toggle" href="#" id="PDBFileBuilder.finalize-toggle" onclick="return toggle('PDBFileBuilder.finalize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBFileBuilder-class.html#finalize">finalize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFileBuilder.finalize-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFileBuilder.finalize-expanded"><a name="L1871"></a><tt class="py-lineno">1871</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1872"></a><tt class="py-lineno">1872</tt>  <tt class="py-line"><tt class="py-docstring">        Add the END marker</tt> </tt>
<a name="L1873"></a><tt class="py-lineno">1873</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1874"></a><tt class="py-lineno">1874</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1169" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1169', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'END'</tt><tt class="py-op">)</tt> </tt>
<a name="L1875"></a><tt class="py-lineno">1875</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_out</tt><tt class="py-op">.</tt><tt class="py-name">flush</tt><tt class="py-op">(</tt><tt class="py-op">)</tt>      </tt>
</div></div><a name="L1876"></a><tt class="py-lineno">1876</tt>  <tt class="py-line"> </tt>
<a name="PDBEnsembleFileBuilder"></a><div id="PDBEnsembleFileBuilder-def"><a name="L1877"></a><tt class="py-lineno">1877</tt> <a class="py-toggle" href="#" id="PDBEnsembleFileBuilder-toggle" onclick="return toggle('PDBEnsembleFileBuilder');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBEnsembleFileBuilder-class.html">PDBEnsembleFileBuilder</a><tt class="py-op">(</tt><tt class="py-base-class">PDBFileBuilder</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBEnsembleFileBuilder-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PDBEnsembleFileBuilder-expanded"><a name="L1878"></a><tt class="py-lineno">1878</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1879"></a><tt class="py-lineno">1879</tt>  <tt class="py-line"><tt class="py-docstring">    Supports serialization of NMR ensembles.</tt> </tt>
<a name="L1880"></a><tt class="py-lineno">1880</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1881"></a><tt class="py-lineno">1881</tt>  <tt class="py-line"><tt class="py-docstring">    Functions as a simple decorator, which wraps C{add_structure} with</tt> </tt>
<a name="L1882"></a><tt class="py-lineno">1882</tt>  <tt class="py-line"><tt class="py-docstring">    MODEL/ENDMDL records.</tt> </tt>
<a name="L1883"></a><tt class="py-lineno">1883</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1884"></a><tt class="py-lineno">1884</tt>  <tt class="py-line"> </tt>
<a name="PDBEnsembleFileBuilder.add_structure"></a><div id="PDBEnsembleFileBuilder.add_structure-def"><a name="L1885"></a><tt class="py-lineno">1885</tt> <a class="py-toggle" href="#" id="PDBEnsembleFileBuilder.add_structure-toggle" onclick="return toggle('PDBEnsembleFileBuilder.add_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.PDBEnsembleFileBuilder-class.html#add_structure">add_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBEnsembleFileBuilder.add_structure-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBEnsembleFileBuilder.add_structure-expanded"><a name="L1886"></a><tt class="py-lineno">1886</tt>  <tt class="py-line">         </tt>
<a name="L1887"></a><tt class="py-lineno">1887</tt>  <tt class="py-line">        <tt id="link-1170" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-1170', 'model_id', 'link-90');">model_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1171" class="py-name"><a title="csb.bio.io.wwpdb.FileBuilder.isnull" class="py-name" href="#" onclick="return doclink('link-1171', 'isnull', 'link-1037');">isnull</a></tt><tt class="py-op">(</tt><tt id="link-1172" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-1172', 'structure', 'link-2');">structure</a></tt><tt class="py-op">.</tt><tt id="link-1173" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-1173', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L1888"></a><tt class="py-lineno">1888</tt>  <tt class="py-line">         </tt>
<a name="L1889"></a><tt class="py-lineno">1889</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1174" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1174', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'MODEL     {0:&gt;4}'</tt><tt class="py-op">.</tt><tt id="link-1175" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1175', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-1176" class="py-name"><a title="csb.bio.structure.Structure.model_id" class="py-name" href="#" onclick="return doclink('link-1176', 'model_id', 'link-90');">model_id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt>        </tt>
<a name="L1890"></a><tt class="py-lineno">1890</tt>  <tt class="py-line">        <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-1177" class="py-name" targets="Class csb.bio.io.wwpdb.PDBEnsembleFileBuilder=csb.bio.io.wwpdb.PDBEnsembleFileBuilder-class.html"><a title="csb.bio.io.wwpdb.PDBEnsembleFileBuilder" class="py-name" href="#" onclick="return doclink('link-1177', 'PDBEnsembleFileBuilder', 'link-1177');">PDBEnsembleFileBuilder</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1178" class="py-name" targets="Method csb.bio.io.wwpdb.FileBuilder.add_structure()=csb.bio.io.wwpdb.FileBuilder-class.html#add_structure,Method csb.bio.io.wwpdb.PDBEnsembleFileBuilder.add_structure()=csb.bio.io.wwpdb.PDBEnsembleFileBuilder-class.html#add_structure,Method csb.bio.io.wwpdb.PDBFileBuilder.add_structure()=csb.bio.io.wwpdb.PDBFileBuilder-class.html#add_structure"><a title="csb.bio.io.wwpdb.FileBuilder.add_structure
csb.bio.io.wwpdb.PDBEnsembleFileBuilder.add_structure
csb.bio.io.wwpdb.PDBFileBuilder.add_structure" class="py-name" href="#" onclick="return doclink('link-1178', 'add_structure', 'link-1178');">add_structure</a></tt><tt class="py-op">(</tt><tt id="link-1179" class="py-name"><a title="csb.bio.fragments.BenchmarkAdapter.structure
csb.bio.hmm.ProfileHMM.structure
csb.bio.structure" class="py-name" href="#" onclick="return doclink('link-1179', 'structure', 'link-2');">structure</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1891"></a><tt class="py-lineno">1891</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1180" class="py-name"><a title="csb.bio.io.clans.ClansFileBuilder.writeline
csb.bio.io.fasta.OutputBuilder.writeline
csb.bio.io.hhpred.HHMFileBuilder.writeline
csb.bio.io.wwpdb.FileBuilder.writeline
csb.bio.io.wwpdb.PDBFileBuilder.writeline
csb.io.EntryWriter.writeline" class="py-name" href="#" onclick="return doclink('link-1180', 'writeline', 'link-1046');">writeline</a></tt><tt class="py-op">(</tt><tt class="py-string">'ENDMDL'</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1892"></a><tt class="py-lineno">1892</tt>  <tt class="py-line">         </tt>
<a name="StructureProvider"></a><div id="StructureProvider-def"><a name="L1893"></a><tt class="py-lineno">1893</tt>  <tt class="py-line"> </tt>
<a name="L1894"></a><tt class="py-lineno">1894</tt> <a class="py-toggle" href="#" id="StructureProvider-toggle" onclick="return toggle('StructureProvider');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureProvider-class.html">StructureProvider</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureProvider-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="StructureProvider-expanded"><a name="L1895"></a><tt class="py-lineno">1895</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1896"></a><tt class="py-lineno">1896</tt>  <tt class="py-line"><tt class="py-docstring">    Base class for all PDB data source providers.</tt> </tt>
<a name="L1897"></a><tt class="py-lineno">1897</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1898"></a><tt class="py-lineno">1898</tt>  <tt class="py-line"><tt class="py-docstring">    Concrete classes need to implement the C{find} method, which abstracts the</tt> </tt>
<a name="L1899"></a><tt class="py-lineno">1899</tt>  <tt class="py-line"><tt class="py-docstring">    retrieval of a PDB structure file by a structure identifier. This is a hook</tt> </tt>
<a name="L1900"></a><tt class="py-lineno">1900</tt>  <tt class="py-line"><tt class="py-docstring">    method called internally by C{get}, but subclasses can safely override both</tt> </tt>
<a name="L1901"></a><tt class="py-lineno">1901</tt>  <tt class="py-line"><tt class="py-docstring">    C{find} and {get} to in order to achieve completely custom behavior. </tt> </tt>
<a name="L1902"></a><tt class="py-lineno">1902</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1903"></a><tt class="py-lineno">1903</tt>  <tt class="py-line">     </tt>
<a name="L1904"></a><tt class="py-lineno">1904</tt>  <tt class="py-line">    <tt class="py-name">__metaclass__</tt> <tt class="py-op">=</tt> <tt id="link-1181" class="py-name"><a title="abc.ABCMeta" class="py-name" href="#" onclick="return doclink('link-1181', 'ABCMeta', 'link-15');">ABCMeta</a></tt> </tt>
<a name="L1905"></a><tt class="py-lineno">1905</tt>  <tt class="py-line">     </tt>
<a name="StructureProvider.__getitem__"></a><div id="StructureProvider.__getitem__-def"><a name="L1906"></a><tt class="py-lineno">1906</tt> <a class="py-toggle" href="#" id="StructureProvider.__getitem__-toggle" onclick="return toggle('StructureProvider.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureProvider-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureProvider.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StructureProvider.__getitem__-expanded"><a name="L1907"></a><tt class="py-lineno">1907</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1182" class="py-name" targets="Function csb.bio.io.dssp.get()=csb.bio.io.dssp-module.html#get,Method csb.bio.io.wwpdb.RemoteStructureProvider.get()=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#get,Method csb.bio.io.wwpdb.StructureProvider.get()=csb.bio.io.wwpdb.StructureProvider-class.html#get,Function csb.bio.io.wwpdb.get()=csb.bio.io.wwpdb-module.html#get,Static Method csb.bio.nmr.AtomConnectivity.get()=csb.bio.nmr.AtomConnectivity-class.html#get,Method csb.bio.nmr.NOEPeak.get()=csb.bio.nmr.NOEPeak-class.html#get,Static Method csb.bio.nmr.RandomCoil.get()=csb.bio.nmr.RandomCoil-class.html#get,Static Method csb.bio.sequence.SequenceAlphabets.get()=csb.bio.sequence.SequenceAlphabets-class.html#get,Static Method csb.build.BuildTypes.get()=csb.build.BuildTypes-class.html#get,Static Method csb.io.plots.Backend.get()=csb.io.plots.Backend-class.html#get,Method csb.statistics.maxent.MaxentModel.get()=csb.statistics.maxent.MaxentModel-class.html#get"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-1182', 'get', 'link-1182');">get</a></tt><tt class="py-op">(</tt><tt id="link-1183" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1183', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1908"></a><tt class="py-lineno">1908</tt>  <tt class="py-line">     </tt>
<a name="L1909"></a><tt class="py-lineno">1909</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">abstractmethod</tt> </tt>
<a name="StructureProvider.find"></a><div id="StructureProvider.find-def"><a name="L1910"></a><tt class="py-lineno">1910</tt> <a class="py-toggle" href="#" id="StructureProvider.find-toggle" onclick="return toggle('StructureProvider.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureProvider-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureProvider.find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StructureProvider.find-expanded"><a name="L1911"></a><tt class="py-lineno">1911</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1912"></a><tt class="py-lineno">1912</tt>  <tt class="py-line"><tt class="py-docstring">        Attempt to discover a PDB file, given a specific PDB C{id}.</tt> </tt>
<a name="L1913"></a><tt class="py-lineno">1913</tt>  <tt class="py-line"><tt class="py-docstring">            </tt> </tt>
<a name="L1914"></a><tt class="py-lineno">1914</tt>  <tt class="py-line"><tt class="py-docstring">        @param id: structure identifier (e.g. 1x80)</tt> </tt>
<a name="L1915"></a><tt class="py-lineno">1915</tt>  <tt class="py-line"><tt class="py-docstring">        @type id: str</tt> </tt>
<a name="L1916"></a><tt class="py-lineno">1916</tt>  <tt class="py-line"><tt class="py-docstring">        @return: path and file name on success, None otherwise</tt> </tt>
<a name="L1917"></a><tt class="py-lineno">1917</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str or None</tt> </tt>
<a name="L1918"></a><tt class="py-lineno">1918</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1919"></a><tt class="py-lineno">1919</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1920"></a><tt class="py-lineno">1920</tt>  <tt class="py-line">     </tt>
<a name="StructureProvider.get"></a><div id="StructureProvider.get-def"><a name="L1921"></a><tt class="py-lineno">1921</tt> <a class="py-toggle" href="#" id="StructureProvider.get-toggle" onclick="return toggle('StructureProvider.get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.StructureProvider-class.html#get">get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="StructureProvider.get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="StructureProvider.get-expanded"><a name="L1922"></a><tt class="py-lineno">1922</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1923"></a><tt class="py-lineno">1923</tt>  <tt class="py-line"><tt class="py-docstring">        Discover, parse and return the PDB structure, corresponding to the</tt> </tt>
<a name="L1924"></a><tt class="py-lineno">1924</tt>  <tt class="py-line"><tt class="py-docstring">        specified C{id}.</tt> </tt>
<a name="L1925"></a><tt class="py-lineno">1925</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1926"></a><tt class="py-lineno">1926</tt>  <tt class="py-line"><tt class="py-docstring">        @param id: structure identifier (e.g. 1x80)</tt> </tt>
<a name="L1927"></a><tt class="py-lineno">1927</tt>  <tt class="py-line"><tt class="py-docstring">        @type id: str</tt> </tt>
<a name="L1928"></a><tt class="py-lineno">1928</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: optional model identifier</tt> </tt>
<a name="L1929"></a><tt class="py-lineno">1929</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: str</tt> </tt>
<a name="L1930"></a><tt class="py-lineno">1930</tt>  <tt class="py-line"><tt class="py-docstring">                </tt> </tt>
<a name="L1931"></a><tt class="py-lineno">1931</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.bio.Structure}</tt> </tt>
<a name="L1932"></a><tt class="py-lineno">1932</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1933"></a><tt class="py-lineno">1933</tt>  <tt class="py-line"><tt class="py-docstring">        @raise StructureNotFoundError: when C{id} could not be found</tt> </tt>
<a name="L1934"></a><tt class="py-lineno">1934</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1935"></a><tt class="py-lineno">1935</tt>  <tt class="py-line">        <tt class="py-name">pdb</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1184" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-1184', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt id="link-1185" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1185', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1936"></a><tt class="py-lineno">1936</tt>  <tt class="py-line">         </tt>
<a name="L1937"></a><tt class="py-lineno">1937</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">pdb</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1938"></a><tt class="py-lineno">1938</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1186" class="py-name" targets="Class csb.bio.io.wwpdb.StructureNotFoundError=csb.bio.io.wwpdb.StructureNotFoundError-class.html"><a title="csb.bio.io.wwpdb.StructureNotFoundError" class="py-name" href="#" onclick="return doclink('link-1186', 'StructureNotFoundError', 'link-1186');">StructureNotFoundError</a></tt><tt class="py-op">(</tt><tt id="link-1187" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1187', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1939"></a><tt class="py-lineno">1939</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1940"></a><tt class="py-lineno">1940</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-1188" class="py-name"><a title="csb.bio.io.StructureParser" class="py-name" href="#" onclick="return doclink('link-1188', 'StructureParser', 'link-841');">StructureParser</a></tt><tt class="py-op">(</tt><tt class="py-name">pdb</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1189" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-1189', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt id="link-1190" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1190', 'model', 'link-153');">model</a></tt><tt class="py-op">=</tt><tt id="link-1191" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1191', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1941"></a><tt class="py-lineno">1941</tt>  <tt class="py-line">     </tt>
<a name="FileSystemStructureProvider"></a><div id="FileSystemStructureProvider-def"><a name="L1942"></a><tt class="py-lineno">1942</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider-toggle" onclick="return toggle('FileSystemStructureProvider');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html">FileSystemStructureProvider</a><tt class="py-op">(</tt><tt class="py-base-class">StructureProvider</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FileSystemStructureProvider-expanded"><a name="L1943"></a><tt class="py-lineno">1943</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1944"></a><tt class="py-lineno">1944</tt>  <tt class="py-line"><tt class="py-docstring">    Simple file system based PDB data source. Scans a list of local directories</tt> </tt>
<a name="L1945"></a><tt class="py-lineno">1945</tt>  <tt class="py-line"><tt class="py-docstring">    using pre-defined file name templates.</tt> </tt>
<a name="L1946"></a><tt class="py-lineno">1946</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L1947"></a><tt class="py-lineno">1947</tt>  <tt class="py-line"><tt class="py-docstring">    @param paths: a list of paths</tt> </tt>
<a name="L1948"></a><tt class="py-lineno">1948</tt>  <tt class="py-line"><tt class="py-docstring">    @type paths: iterable or str </tt> </tt>
<a name="L1949"></a><tt class="py-lineno">1949</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1950"></a><tt class="py-lineno">1950</tt>  <tt class="py-line">     </tt>
<a name="FileSystemStructureProvider.__init__"></a><div id="FileSystemStructureProvider.__init__-def"><a name="L1951"></a><tt class="py-lineno">1951</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.__init__-toggle" onclick="return toggle('FileSystemStructureProvider.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">paths</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.__init__-expanded"><a name="L1952"></a><tt class="py-lineno">1952</tt>  <tt class="py-line">         </tt>
<a name="L1953"></a><tt class="py-lineno">1953</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_templates</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'pdb{id}.ent'</tt><tt class="py-op">,</tt> <tt class="py-string">'pdb{id}.pdb'</tt><tt class="py-op">,</tt> <tt class="py-string">'{id}.pdb'</tt><tt class="py-op">,</tt> <tt class="py-string">'{id}.ent'</tt><tt class="py-op">]</tt> </tt>
<a name="L1954"></a><tt class="py-lineno">1954</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt> <tt class="py-op">=</tt> <tt id="link-1192" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1192', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1193" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-1193', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt id="link-1194" class="py-name"><a title="csb.core.OrderedDict" class="py-name" href="#" onclick="return doclink('link-1194', 'OrderedDict', 'link-714');">OrderedDict</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1955"></a><tt class="py-lineno">1955</tt>  <tt class="py-line">         </tt>
<a name="L1956"></a><tt class="py-lineno">1956</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-1195" class="py-name" targets="Variable csb.bio.io.wwpdb.CustomStructureProvider.paths=csb.bio.io.wwpdb.CustomStructureProvider-class.html#paths,Variable csb.bio.io.wwpdb.FileSystemStructureProvider.paths=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#paths"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1195', 'paths', 'link-1195');">paths</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1957"></a><tt class="py-lineno">1957</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-1196" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1196', 'paths', 'link-1195');">paths</a></tt><tt class="py-op">,</tt> <tt id="link-1197" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1197', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1198" class="py-name"><a title="csb.core" class="py-name" href="#" onclick="return doclink('link-1198', 'core', 'link-12');">core</a></tt><tt class="py-op">.</tt><tt class="py-name">string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1958"></a><tt class="py-lineno">1958</tt>  <tt class="py-line">                <tt id="link-1199" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1199', 'paths', 'link-1195');">paths</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-1200" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1200', 'paths', 'link-1195');">paths</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1959"></a><tt class="py-lineno">1959</tt>  <tt class="py-line">                             </tt>
<a name="L1960"></a><tt class="py-lineno">1960</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-1201" class="py-name" targets="Variable csb.build.RevisionHandler.path=csb.build.RevisionHandler-class.html#path"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1201', 'path', 'link-1201');">path</a></tt> <tt class="py-keyword">in</tt> <tt id="link-1202" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1202', 'paths', 'link-1195');">paths</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1961"></a><tt class="py-lineno">1961</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1203" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-1203', 'add', 'link-98');">add</a></tt><tt class="py-op">(</tt><tt id="link-1204" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1204', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1962"></a><tt class="py-lineno">1962</tt>  <tt class="py-line">                 </tt>
<a name="L1963"></a><tt class="py-lineno">1963</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="FileSystemStructureProvider.paths"></a><div id="FileSystemStructureProvider.paths-def"><a name="L1964"></a><tt class="py-lineno">1964</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.paths-toggle" onclick="return toggle('FileSystemStructureProvider.paths');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#paths">paths</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.paths-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.paths-expanded"><a name="L1965"></a><tt class="py-lineno">1965</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1966"></a><tt class="py-lineno">1966</tt>  <tt class="py-line"><tt class="py-docstring">        Current search paths</tt> </tt>
<a name="L1967"></a><tt class="py-lineno">1967</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: tuple</tt> </tt>
<a name="L1968"></a><tt class="py-lineno">1968</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1969"></a><tt class="py-lineno">1969</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1970"></a><tt class="py-lineno">1970</tt>  <tt class="py-line">     </tt>
<a name="L1971"></a><tt class="py-lineno">1971</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="FileSystemStructureProvider.templates"></a><div id="FileSystemStructureProvider.templates-def"><a name="L1972"></a><tt class="py-lineno">1972</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.templates-toggle" onclick="return toggle('FileSystemStructureProvider.templates');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#templates">templates</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.templates-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.templates-expanded"><a name="L1973"></a><tt class="py-lineno">1973</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1974"></a><tt class="py-lineno">1974</tt>  <tt class="py-line"><tt class="py-docstring">        Current file name match templates</tt> </tt>
<a name="L1975"></a><tt class="py-lineno">1975</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: tuple</tt> </tt>
<a name="L1976"></a><tt class="py-lineno">1976</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1977"></a><tt class="py-lineno">1977</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_templates</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1978"></a><tt class="py-lineno">1978</tt>  <tt class="py-line">         </tt>
<a name="FileSystemStructureProvider.add"></a><div id="FileSystemStructureProvider.add-def"><a name="L1979"></a><tt class="py-lineno">1979</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.add-toggle" onclick="return toggle('FileSystemStructureProvider.add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#add">add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">path</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.add-expanded"><a name="L1980"></a><tt class="py-lineno">1980</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1981"></a><tt class="py-lineno">1981</tt>  <tt class="py-line"><tt class="py-docstring">        Register a new local C{path}.</tt> </tt>
<a name="L1982"></a><tt class="py-lineno">1982</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1983"></a><tt class="py-lineno">1983</tt>  <tt class="py-line"><tt class="py-docstring">        @param path: directory name</tt> </tt>
<a name="L1984"></a><tt class="py-lineno">1984</tt>  <tt class="py-line"><tt class="py-docstring">        @type path: str</tt> </tt>
<a name="L1985"></a><tt class="py-lineno">1985</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1986"></a><tt class="py-lineno">1986</tt>  <tt class="py-line"><tt class="py-docstring">        @raise IOError: if C{path} is not a valid directory</tt> </tt>
<a name="L1987"></a><tt class="py-lineno">1987</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1988"></a><tt class="py-lineno">1988</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-1205" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1205', 'path', 'link-1201');">path</a></tt><tt class="py-op">.</tt><tt class="py-name">isdir</tt><tt class="py-op">(</tt><tt id="link-1206" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1206', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1989"></a><tt class="py-lineno">1989</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt><tt class="py-op">[</tt><tt id="link-1207" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1207', 'path', 'link-1201');">path</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-1208" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1208', 'path', 'link-1201');">path</a></tt> </tt>
<a name="L1990"></a><tt class="py-lineno">1990</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1991"></a><tt class="py-lineno">1991</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IOError</tt><tt class="py-op">(</tt><tt id="link-1209" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1209', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1992"></a><tt class="py-lineno">1992</tt>  <tt class="py-line">         </tt>
<a name="FileSystemStructureProvider.add_template"></a><div id="FileSystemStructureProvider.add_template-def"><a name="L1993"></a><tt class="py-lineno">1993</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.add_template-toggle" onclick="return toggle('FileSystemStructureProvider.add_template');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#add_template">add_template</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">template</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.add_template-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.add_template-expanded"><a name="L1994"></a><tt class="py-lineno">1994</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1995"></a><tt class="py-lineno">1995</tt>  <tt class="py-line"><tt class="py-docstring">        Register a custom file name name C{template}. The template must contain</tt> </tt>
<a name="L1996"></a><tt class="py-lineno">1996</tt>  <tt class="py-line"><tt class="py-docstring">        an E{lb}idE{rb} macro, e.g. pdbE{lb}idE{rb}.ent </tt> </tt>
<a name="L1997"></a><tt class="py-lineno">1997</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L1998"></a><tt class="py-lineno">1998</tt>  <tt class="py-line"><tt class="py-docstring">        @param template: pattern</tt> </tt>
<a name="L1999"></a><tt class="py-lineno">1999</tt>  <tt class="py-line"><tt class="py-docstring">        @type template: str </tt> </tt>
<a name="L2000"></a><tt class="py-lineno">2000</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2001"></a><tt class="py-lineno">2001</tt>  <tt class="py-line">         </tt>
<a name="L2002"></a><tt class="py-lineno">2002</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'{id}'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">template</tt><tt class="py-op">:</tt> </tt>
<a name="L2003"></a><tt class="py-lineno">2003</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'Template does not contain an "{id}" macro'</tt><tt class="py-op">)</tt> </tt>
<a name="L2004"></a><tt class="py-lineno">2004</tt>  <tt class="py-line">         </tt>
<a name="L2005"></a><tt class="py-lineno">2005</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">template</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_templates</tt><tt class="py-op">:</tt> </tt>
<a name="L2006"></a><tt class="py-lineno">2006</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_templates</tt><tt class="py-op">.</tt><tt id="link-1210" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1210', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">template</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2007"></a><tt class="py-lineno">2007</tt>  <tt class="py-line">             </tt>
<a name="FileSystemStructureProvider.remove"></a><div id="FileSystemStructureProvider.remove-def"><a name="L2008"></a><tt class="py-lineno">2008</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.remove-toggle" onclick="return toggle('FileSystemStructureProvider.remove');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#remove">remove</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">path</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.remove-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.remove-expanded"><a name="L2009"></a><tt class="py-lineno">2009</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2010"></a><tt class="py-lineno">2010</tt>  <tt class="py-line"><tt class="py-docstring">        Unregister an existing local C{path}.</tt> </tt>
<a name="L2011"></a><tt class="py-lineno">2011</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2012"></a><tt class="py-lineno">2012</tt>  <tt class="py-line"><tt class="py-docstring">        @param path: directory name</tt> </tt>
<a name="L2013"></a><tt class="py-lineno">2013</tt>  <tt class="py-line"><tt class="py-docstring">        @type path: str</tt> </tt>
<a name="L2014"></a><tt class="py-lineno">2014</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2015"></a><tt class="py-lineno">2015</tt>  <tt class="py-line"><tt class="py-docstring">        @raise ValueError: if C{path} had not been registered</tt> </tt>
<a name="L2016"></a><tt class="py-lineno">2016</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L2017"></a><tt class="py-lineno">2017</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-1211" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1211', 'path', 'link-1201');">path</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt><tt class="py-op">:</tt> </tt>
<a name="L2018"></a><tt class="py-lineno">2018</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">'path not found: {0}'</tt><tt class="py-op">.</tt><tt id="link-1212" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1212', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-1213" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1213', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2019"></a><tt class="py-lineno">2019</tt>  <tt class="py-line">         </tt>
<a name="L2020"></a><tt class="py-lineno">2020</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt><tt class="py-op">[</tt><tt id="link-1214" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1214', 'path', 'link-1201');">path</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L2021"></a><tt class="py-lineno">2021</tt>  <tt class="py-line">             </tt>
<a name="FileSystemStructureProvider.find"></a><div id="FileSystemStructureProvider.find-def"><a name="L2022"></a><tt class="py-lineno">2022</tt> <a class="py-toggle" href="#" id="FileSystemStructureProvider.find-toggle" onclick="return toggle('FileSystemStructureProvider.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FileSystemStructureProvider.find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FileSystemStructureProvider.find-expanded"><a name="L2023"></a><tt class="py-lineno">2023</tt>  <tt class="py-line">         </tt>
<a name="L2024"></a><tt class="py-lineno">2024</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1215" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1215', 'path', 'link-1201');">path</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_paths</tt><tt class="py-op">:</tt> </tt>
<a name="L2025"></a><tt class="py-lineno">2025</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">token</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1216" class="py-name" targets="Variable csb.bio.io.wwpdb.FileSystemStructureProvider.templates=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html#templates"><a title="csb.bio.io.wwpdb.FileSystemStructureProvider.templates" class="py-name" href="#" onclick="return doclink('link-1216', 'templates', 'link-1216');">templates</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2026"></a><tt class="py-lineno">2026</tt>  <tt class="py-line">                <tt class="py-name">fn</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-1217" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1217', 'path', 'link-1201');">path</a></tt><tt class="py-op">.</tt><tt id="link-1218" class="py-name"><a title="csb.bio.nmr.NOESpectrum.join" class="py-name" href="#" onclick="return doclink('link-1218', 'join', 'link-958');">join</a></tt><tt class="py-op">(</tt><tt id="link-1219" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1219', 'path', 'link-1201');">path</a></tt><tt class="py-op">,</tt> <tt class="py-name">token</tt><tt class="py-op">.</tt><tt id="link-1220" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1220', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt id="link-1221" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1221', 'id', 'link-29');">id</a></tt><tt class="py-op">=</tt><tt id="link-1222" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1222', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2027"></a><tt class="py-lineno">2027</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-1223" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1223', 'path', 'link-1201');">path</a></tt><tt class="py-op">.</tt><tt class="py-name">exists</tt><tt class="py-op">(</tt><tt class="py-name">fn</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2028"></a><tt class="py-lineno">2028</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">fn</tt> </tt>
<a name="L2029"></a><tt class="py-lineno">2029</tt>  <tt class="py-line">             </tt>
<a name="L2030"></a><tt class="py-lineno">2030</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div></div><a name="L2031"></a><tt class="py-lineno">2031</tt>  <tt class="py-line">     </tt>
<a name="RemoteStructureProvider"></a><div id="RemoteStructureProvider-def"><a name="L2032"></a><tt class="py-lineno">2032</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider-toggle" onclick="return toggle('RemoteStructureProvider');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html">RemoteStructureProvider</a><tt class="py-op">(</tt><tt class="py-base-class">StructureProvider</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="RemoteStructureProvider-expanded"><a name="L2033"></a><tt class="py-lineno">2033</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L2034"></a><tt class="py-lineno">2034</tt>  <tt class="py-line"><tt class="py-docstring">    Retrieves PDB structures from a specified remote URL.</tt> </tt>
<a name="L2035"></a><tt class="py-lineno">2035</tt>  <tt class="py-line"><tt class="py-docstring">    The URL requested from remote server takes the form: &lt;prefix&gt;/&lt;ID&gt;&lt;suffix&gt;</tt> </tt>
<a name="L2036"></a><tt class="py-lineno">2036</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2037"></a><tt class="py-lineno">2037</tt>  <tt class="py-line"><tt class="py-docstring">    @param prefix: URL prefix, including protocol</tt> </tt>
<a name="L2038"></a><tt class="py-lineno">2038</tt>  <tt class="py-line"><tt class="py-docstring">    @type prefix: str</tt> </tt>
<a name="L2039"></a><tt class="py-lineno">2039</tt>  <tt class="py-line"><tt class="py-docstring">    @param suffix: optional URL suffix (.ent by default)</tt> </tt>
<a name="L2040"></a><tt class="py-lineno">2040</tt>  <tt class="py-line"><tt class="py-docstring">    @type suffix: str</tt> </tt>
<a name="L2041"></a><tt class="py-lineno">2041</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2042"></a><tt class="py-lineno">2042</tt>  <tt class="py-line">     </tt>
<a name="RemoteStructureProvider.__init__"></a><div id="RemoteStructureProvider.__init__-def"><a name="L2043"></a><tt class="py-lineno">2043</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.__init__-toggle" onclick="return toggle('RemoteStructureProvider.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">=</tt><tt class="py-string">'https://files.rcsb.org/download/'</tt><tt class="py-op">,</tt> <tt class="py-param">suffix</tt><tt class="py-op">=</tt><tt class="py-string">'.pdb'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.__init__-expanded"><a name="L2044"></a><tt class="py-lineno">2044</tt>  <tt class="py-line">         </tt>
<a name="L2045"></a><tt class="py-lineno">2045</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_prefix</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L2046"></a><tt class="py-lineno">2046</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_suffix</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L2047"></a><tt class="py-lineno">2047</tt>  <tt class="py-line">         </tt>
<a name="L2048"></a><tt class="py-lineno">2048</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1224" class="py-name" targets="Variable csb.bio.io.wwpdb.RemoteStructureProvider.prefix=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#prefix"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.prefix" class="py-name" href="#" onclick="return doclink('link-1224', 'prefix', 'link-1224');">prefix</a></tt> <tt class="py-op">=</tt> <tt id="link-1225" class="py-name"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.prefix" class="py-name" href="#" onclick="return doclink('link-1225', 'prefix', 'link-1224');">prefix</a></tt> </tt>
<a name="L2049"></a><tt class="py-lineno">2049</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1226" class="py-name" targets="Variable csb.bio.io.wwpdb.RemoteStructureProvider.suffix=csb.bio.io.wwpdb.RemoteStructureProvider-class.html#suffix"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.suffix" class="py-name" href="#" onclick="return doclink('link-1226', 'suffix', 'link-1226');">suffix</a></tt> <tt class="py-op">=</tt> <tt id="link-1227" class="py-name"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.suffix" class="py-name" href="#" onclick="return doclink('link-1227', 'suffix', 'link-1226');">suffix</a></tt> </tt>
</div><a name="L2050"></a><tt class="py-lineno">2050</tt>  <tt class="py-line">         </tt>
<a name="L2051"></a><tt class="py-lineno">2051</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="RemoteStructureProvider.prefix"></a><div id="RemoteStructureProvider.prefix-def"><a name="L2052"></a><tt class="py-lineno">2052</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.prefix-toggle" onclick="return toggle('RemoteStructureProvider.prefix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#prefix">prefix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.prefix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.prefix-expanded"><a name="L2053"></a><tt class="py-lineno">2053</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2054"></a><tt class="py-lineno">2054</tt>  <tt class="py-line"><tt class="py-docstring">        Current URL prefix</tt> </tt>
<a name="L2055"></a><tt class="py-lineno">2055</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L2056"></a><tt class="py-lineno">2056</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2057"></a><tt class="py-lineno">2057</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_prefix</tt> </tt>
</div><a name="L2058"></a><tt class="py-lineno">2058</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">prefix</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="RemoteStructureProvider.prefix"></a><div id="RemoteStructureProvider.prefix-def"><a name="L2059"></a><tt class="py-lineno">2059</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.prefix-toggle" onclick="return toggle('RemoteStructureProvider.prefix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#prefix">prefix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.prefix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.prefix-expanded"><a name="L2060"></a><tt class="py-lineno">2060</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_prefix</tt> <tt class="py-op">=</tt> <tt id="link-1228" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-1228', 'value', 'link-77');">value</a></tt> </tt>
</div><a name="L2061"></a><tt class="py-lineno">2061</tt>  <tt class="py-line">         </tt>
<a name="L2062"></a><tt class="py-lineno">2062</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="RemoteStructureProvider.suffix"></a><div id="RemoteStructureProvider.suffix-def"><a name="L2063"></a><tt class="py-lineno">2063</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.suffix-toggle" onclick="return toggle('RemoteStructureProvider.suffix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#suffix">suffix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.suffix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.suffix-expanded"><a name="L2064"></a><tt class="py-lineno">2064</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2065"></a><tt class="py-lineno">2065</tt>  <tt class="py-line"><tt class="py-docstring">        Current URL suffix</tt> </tt>
<a name="L2066"></a><tt class="py-lineno">2066</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: str</tt> </tt>
<a name="L2067"></a><tt class="py-lineno">2067</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L2068"></a><tt class="py-lineno">2068</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_suffix</tt> </tt>
</div><a name="L2069"></a><tt class="py-lineno">2069</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">suffix</tt><tt class="py-op">.</tt><tt class="py-name">setter</tt> </tt>
<a name="RemoteStructureProvider.suffix"></a><div id="RemoteStructureProvider.suffix-def"><a name="L2070"></a><tt class="py-lineno">2070</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.suffix-toggle" onclick="return toggle('RemoteStructureProvider.suffix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#suffix">suffix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.suffix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.suffix-expanded"><a name="L2071"></a><tt class="py-lineno">2071</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_suffix</tt> <tt class="py-op">=</tt> <tt id="link-1229" class="py-name"><a title="csb.core.EnumItem.value
csb.statistics.pdf.parameterized.AbstractParameter.value" class="py-name" href="#" onclick="return doclink('link-1229', 'value', 'link-77');">value</a></tt>         </tt>
</div><a name="L2072"></a><tt class="py-lineno">2072</tt>  <tt class="py-line">     </tt>
<a name="RemoteStructureProvider._find"></a><div id="RemoteStructureProvider._find-def"><a name="L2073"></a><tt class="py-lineno">2073</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider._find-toggle" onclick="return toggle('RemoteStructureProvider._find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#_find">_find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider._find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider._find-expanded"><a name="L2074"></a><tt class="py-lineno">2074</tt>  <tt class="py-line">         </tt>
<a name="L2075"></a><tt class="py-lineno">2075</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2076"></a><tt class="py-lineno">2076</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-1230" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1230', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1231" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-1231', 'io', 'link-14');">io</a></tt><tt class="py-op">.</tt><tt class="py-name">urllib</tt><tt class="py-op">.</tt><tt class="py-name">urlopen</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1232" class="py-name"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.prefix" class="py-name" href="#" onclick="return doclink('link-1232', 'prefix', 'link-1224');">prefix</a></tt> <tt class="py-op">+</tt> <tt id="link-1233" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1233', 'id', 'link-29');">id</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1234" class="py-name"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.suffix" class="py-name" href="#" onclick="return doclink('link-1234', 'suffix', 'link-1226');">suffix</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2077"></a><tt class="py-lineno">2077</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">Exception</tt> <tt class="py-keyword">as</tt> <tt class="py-name">e</tt><tt class="py-op">:</tt> </tt>
<a name="L2078"></a><tt class="py-lineno">2078</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1235" class="py-name"><a title="csb.bio.io.wwpdb.StructureNotFoundError" class="py-name" href="#" onclick="return doclink('link-1235', 'StructureNotFoundError', 'link-1186');">StructureNotFoundError</a></tt><tt class="py-op">(</tt><tt id="link-1236" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1236', 'id', 'link-29');">id</a></tt> <tt class="py-op">+</tt> <tt class="py-string">": "</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2079"></a><tt class="py-lineno">2079</tt>  <tt class="py-line">         </tt>
<a name="RemoteStructureProvider.find"></a><div id="RemoteStructureProvider.find-def"><a name="L2080"></a><tt class="py-lineno">2080</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.find-toggle" onclick="return toggle('RemoteStructureProvider.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.find-expanded"><a name="L2081"></a><tt class="py-lineno">2081</tt>  <tt class="py-line">         </tt>
<a name="L2082"></a><tt class="py-lineno">2082</tt>  <tt class="py-line">        <tt id="link-1237" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1237', 'stream', 'link-81');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_find</tt><tt class="py-op">(</tt><tt id="link-1238" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1238', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2083"></a><tt class="py-lineno">2083</tt>  <tt class="py-line">         </tt>
<a name="L2084"></a><tt class="py-lineno">2084</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2085"></a><tt class="py-lineno">2085</tt>  <tt class="py-line">            <tt class="py-name">tmp</tt> <tt class="py-op">=</tt> <tt id="link-1239" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1239', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1240" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-1240', 'io', 'link-14');">io</a></tt><tt class="py-op">.</tt><tt id="link-1241" class="py-name" targets="Class csb.io.TempFile=csb.io.TempFile-class.html"><a title="csb.io.TempFile" class="py-name" href="#" onclick="return doclink('link-1241', 'TempFile', 'link-1241');">TempFile</a></tt><tt class="py-op">(</tt><tt id="link-1242" class="py-name" targets="Method csb.io.plots.Chart.dispose()=csb.io.plots.Chart-class.html#dispose"><a title="csb.io.plots.Chart.dispose" class="py-name" href="#" onclick="return doclink('link-1242', 'dispose', 'link-1242');">dispose</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L2086"></a><tt class="py-lineno">2086</tt>  <tt class="py-line">            <tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt id="link-1243" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1243', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt id="link-1244" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1244', 'stream', 'link-81');">stream</a></tt><tt class="py-op">.</tt><tt id="link-1245" class="py-name" targets="Static Method csb.bio.fragments.rosetta.RosettaFragmentMap.read()=csb.bio.fragments.rosetta.RosettaFragmentMap-class.html#read,Method csb.bio.io.fasta.BaseSequenceParser.read()=csb.bio.io.fasta.BaseSequenceParser-class.html#read,Method csb.bio.io.mrc.DensityMapReader.read()=csb.bio.io.mrc.DensityMapReader-class.html#read,Method csb.bio.io.noe.BasePeakListReader.read()=csb.bio.io.noe.BasePeakListReader-class.html#read,Method csb.bio.io.noe.SparkyPeakListReader.read()=csb.bio.io.noe.SparkyPeakListReader-class.html#read,Method csb.bio.io.noe.XeasyPeakListReader.read()=csb.bio.io.noe.XeasyPeakListReader-class.html#read,Method csb.build.GitHandler.read()=csb.build.GitHandler-class.html#read,Method csb.build.MercurialHandler.read()=csb.build.MercurialHandler-class.html#read,Method csb.build.RevisionHandler.read()=csb.build.RevisionHandler-class.html#read,Method csb.build.SubversionHandler.read()=csb.build.SubversionHandler-class.html#read"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.read
csb.bio.io.fasta.BaseSequenceParser.read
csb.bio.io.mrc.DensityMapReader.read
csb.bio.io.noe.BasePeakListReader.read
csb.bio.io.noe.SparkyPeakListReader.read
csb.bio.io.noe.XeasyPeakListReader.read
csb.build.GitHandler.read
csb.build.MercurialHandler.read
csb.build.RevisionHandler.read
csb.build.SubversionHandler.read" class="py-name" href="#" onclick="return doclink('link-1245', 'read', 'link-1245');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">decode</tt><tt class="py-op">(</tt><tt class="py-string">'utf-8'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2087"></a><tt class="py-lineno">2087</tt>  <tt class="py-line">            <tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt class="py-name">flush</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2088"></a><tt class="py-lineno">2088</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt id="link-1246" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-1246', 'name', 'link-82');">name</a></tt> </tt>
<a name="L2089"></a><tt class="py-lineno">2089</tt>  <tt class="py-line">                 </tt>
<a name="L2090"></a><tt class="py-lineno">2090</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt id="link-1247" class="py-name"><a title="csb.bio.io.wwpdb.StructureNotFoundError" class="py-name" href="#" onclick="return doclink('link-1247', 'StructureNotFoundError', 'link-1186');">StructureNotFoundError</a></tt><tt class="py-op">:</tt> </tt>
<a name="L2091"></a><tt class="py-lineno">2091</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt>             </tt>
<a name="L2092"></a><tt class="py-lineno">2092</tt>  <tt class="py-line">        <tt class="py-keyword">finally</tt><tt class="py-op">:</tt> </tt>
<a name="L2093"></a><tt class="py-lineno">2093</tt>  <tt class="py-line">            <tt id="link-1248" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1248', 'stream', 'link-81');">stream</a></tt><tt class="py-op">.</tt><tt id="link-1249" class="py-name"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-1249', 'close', 'link-76');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>          </tt>
</div><a name="L2094"></a><tt class="py-lineno">2094</tt>  <tt class="py-line">         </tt>
<a name="RemoteStructureProvider.get"></a><div id="RemoteStructureProvider.get-def"><a name="L2095"></a><tt class="py-lineno">2095</tt> <a class="py-toggle" href="#" id="RemoteStructureProvider.get-toggle" onclick="return toggle('RemoteStructureProvider.get');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.RemoteStructureProvider-class.html#get">get</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RemoteStructureProvider.get-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RemoteStructureProvider.get-expanded"><a name="L2096"></a><tt class="py-lineno">2096</tt>  <tt class="py-line">         </tt>
<a name="L2097"></a><tt class="py-lineno">2097</tt>  <tt class="py-line">        <tt id="link-1250" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1250', 'stream', 'link-81');">stream</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_find</tt><tt class="py-op">(</tt><tt id="link-1251" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1251', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2098"></a><tt class="py-lineno">2098</tt>  <tt class="py-line">         </tt>
<a name="L2099"></a><tt class="py-lineno">2099</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2100"></a><tt class="py-lineno">2100</tt>  <tt class="py-line">            <tt class="py-keyword">with</tt> <tt id="link-1252" class="py-name"><a title="csb" class="py-name" href="#" onclick="return doclink('link-1252', 'csb', 'link-0');">csb</a></tt><tt class="py-op">.</tt><tt id="link-1253" class="py-name"><a title="csb.bio.io
csb.io" class="py-name" href="#" onclick="return doclink('link-1253', 'io', 'link-14');">io</a></tt><tt class="py-op">.</tt><tt id="link-1254" class="py-name"><a title="csb.io.TempFile" class="py-name" href="#" onclick="return doclink('link-1254', 'TempFile', 'link-1241');">TempFile</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">tmp</tt><tt class="py-op">:</tt> </tt>
<a name="L2101"></a><tt class="py-lineno">2101</tt>  <tt class="py-line">                <tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt id="link-1255" class="py-name"><a title="csb.bio.io.clans.Clans.write
csb.bio.io.clans.ClansFileBuilder.write
csb.bio.io.fasta.OutputBuilder.write
csb.bio.io.hhpred.HHMFileBuilder.write
csb.bio.io.mrc.DensityMapWriter.write
csb.bio.io.wwpdb.FileBuilder.write
csb.build.RevisionHandler.write
csb.io.AutoFlushStream.write
csb.io.EntryWriter.write" class="py-name" href="#" onclick="return doclink('link-1255', 'write', 'link-1038');">write</a></tt><tt class="py-op">(</tt><tt id="link-1256" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1256', 'stream', 'link-81');">stream</a></tt><tt class="py-op">.</tt><tt id="link-1257" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.read
csb.bio.io.fasta.BaseSequenceParser.read
csb.bio.io.mrc.DensityMapReader.read
csb.bio.io.noe.BasePeakListReader.read
csb.bio.io.noe.SparkyPeakListReader.read
csb.bio.io.noe.XeasyPeakListReader.read
csb.build.GitHandler.read
csb.build.MercurialHandler.read
csb.build.RevisionHandler.read
csb.build.SubversionHandler.read" class="py-name" href="#" onclick="return doclink('link-1257', 'read', 'link-1245');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">decode</tt><tt class="py-op">(</tt><tt class="py-string">'utf-8'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L2102"></a><tt class="py-lineno">2102</tt>  <tt class="py-line">                <tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt class="py-name">flush</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2103"></a><tt class="py-lineno">2103</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-1258" class="py-name"><a title="csb.bio.io.StructureParser" class="py-name" href="#" onclick="return doclink('link-1258', 'StructureParser', 'link-841');">StructureParser</a></tt><tt class="py-op">(</tt><tt class="py-name">tmp</tt><tt class="py-op">.</tt><tt id="link-1259" class="py-name"><a title="csb.bio.hmm.ProfileHMM.name
csb.bio.io.clans.ClansEntry.name
csb.bio.io.clans.ClansSeqgroup.name
csb.bio.structure.Atom.name
csb.bio.structure.Chain.name
csb.core.EnumItem.name
csb.io.TempFile.name
csb.io.TempFolder.name
csb.io.tsv.ColumnInfo.name
csb.io.tsv.Table.name
csb.statistics.pdf.parameterized.AbstractParameter.name" class="py-name" href="#" onclick="return doclink('link-1259', 'name', 'link-82');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1260" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-1260', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt id="link-1261" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1261', 'model', 'link-153');">model</a></tt><tt class="py-op">=</tt><tt id="link-1262" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1262', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2104"></a><tt class="py-lineno">2104</tt>  <tt class="py-line">        <tt class="py-keyword">finally</tt><tt class="py-op">:</tt> </tt>
<a name="L2105"></a><tt class="py-lineno">2105</tt>  <tt class="py-line">            <tt id="link-1263" class="py-name"><a title="csb.io.EntryWriter.stream" class="py-name" href="#" onclick="return doclink('link-1263', 'stream', 'link-81');">stream</a></tt><tt class="py-op">.</tt><tt id="link-1264" class="py-name"><a title="csb.io.EntryWriter.close
csb.io.TempFile.close
csb.io.TempFolder.close
csb.io.tsv.RepositoryImp.close
csb.io.tsv.SQLiteRepository.close" class="py-name" href="#" onclick="return doclink('link-1264', 'close', 'link-76');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L2106"></a><tt class="py-lineno">2106</tt>  <tt class="py-line">         </tt>
<a name="CustomStructureProvider"></a><div id="CustomStructureProvider-def"><a name="L2107"></a><tt class="py-lineno">2107</tt> <a class="py-toggle" href="#" id="CustomStructureProvider-toggle" onclick="return toggle('CustomStructureProvider');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html">CustomStructureProvider</a><tt class="py-op">(</tt><tt class="py-base-class">StructureProvider</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  </tt>
</div><a name="L2108"></a><tt class="py-lineno">2108</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L2109"></a><tt class="py-lineno">2109</tt>  <tt class="py-line"><tt class="py-docstring">    A custom PDB data source. Functions as a user-defined map of structure</tt> </tt>
<a name="L2110"></a><tt class="py-lineno">2110</tt>  <tt class="py-line"><tt class="py-docstring">    identifiers and their corresponding local file names. </tt> </tt>
<a name="L2111"></a><tt class="py-lineno">2111</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2112"></a><tt class="py-lineno">2112</tt>  <tt class="py-line"><tt class="py-docstring">    @param files: initialization dictionary of id:file pairs</tt> </tt>
<a name="L2113"></a><tt class="py-lineno">2113</tt>  <tt class="py-line"><tt class="py-docstring">    @type files: dict-like</tt> </tt>
<a name="L2114"></a><tt class="py-lineno">2114</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2115"></a><tt class="py-lineno">2115</tt>  <tt class="py-line">         </tt>
<a name="CustomStructureProvider.__init__"></a><div id="CustomStructureProvider.__init__-def"><a name="L2116"></a><tt class="py-lineno">2116</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.__init__-toggle" onclick="return toggle('CustomStructureProvider.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">files</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.__init__-expanded"><a name="L2117"></a><tt class="py-lineno">2117</tt>  <tt class="py-line">         </tt>
<a name="L2118"></a><tt class="py-lineno">2118</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>         </tt>
<a name="L2119"></a><tt class="py-lineno">2119</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1265" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1265', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">files</tt><tt class="py-op">:</tt> </tt>
<a name="L2120"></a><tt class="py-lineno">2120</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1266" class="py-name"><a title="csb.apps.csfrag.MatchTable.add
csb.apps.hhfrag.PredictionBuilder.add
csb.bio.io.clans.ClansSeqgroup.add
csb.bio.io.wwpdb.CustomStructureProvider.add
csb.bio.io.wwpdb.FileSystemStructureProvider.add
csb.bio.nmr.NOESpectrum.add
csb.bio.sequence.AbstractAlignment.add
csb.io.plots.Backend.add" class="py-name" href="#" onclick="return doclink('link-1266', 'add', 'link-98');">add</a></tt><tt class="py-op">(</tt><tt id="link-1267" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1267', 'id', 'link-29');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">files</tt><tt class="py-op">[</tt><tt id="link-1268" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1268', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2121"></a><tt class="py-lineno">2121</tt>  <tt class="py-line">             </tt>
<a name="L2122"></a><tt class="py-lineno">2122</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="CustomStructureProvider.paths"></a><div id="CustomStructureProvider.paths-def"><a name="L2123"></a><tt class="py-lineno">2123</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.paths-toggle" onclick="return toggle('CustomStructureProvider.paths');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#paths">paths</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.paths-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.paths-expanded"><a name="L2124"></a><tt class="py-lineno">2124</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2125"></a><tt class="py-lineno">2125</tt>  <tt class="py-line"><tt class="py-docstring">        List of currently registered file names</tt> </tt>
<a name="L2126"></a><tt class="py-lineno">2126</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: tuple</tt> </tt>
<a name="L2127"></a><tt class="py-lineno">2127</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>         </tt>
<a name="L2128"></a><tt class="py-lineno">2128</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">.</tt><tt id="link-1269" class="py-name" targets="Static Method csb.core.Enum.values()=csb.core.Enum-class.html#values"><a title="csb.core.Enum.values" class="py-name" href="#" onclick="return doclink('link-1269', 'values', 'link-1269');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>             </tt>
</div><a name="L2129"></a><tt class="py-lineno">2129</tt>  <tt class="py-line"> </tt>
<a name="L2130"></a><tt class="py-lineno">2130</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="CustomStructureProvider.identifiers"></a><div id="CustomStructureProvider.identifiers-def"><a name="L2131"></a><tt class="py-lineno">2131</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.identifiers-toggle" onclick="return toggle('CustomStructureProvider.identifiers');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#identifiers">identifiers</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.identifiers-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.identifiers-expanded"><a name="L2132"></a><tt class="py-lineno">2132</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2133"></a><tt class="py-lineno">2133</tt>  <tt class="py-line"><tt class="py-docstring">        List of currently registered structure identifiers</tt> </tt>
<a name="L2134"></a><tt class="py-lineno">2134</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: tuple</tt> </tt>
<a name="L2135"></a><tt class="py-lineno">2135</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt>           </tt>
<a name="L2136"></a><tt class="py-lineno">2136</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2137"></a><tt class="py-lineno">2137</tt>  <tt class="py-line">             </tt>
<a name="CustomStructureProvider.add"></a><div id="CustomStructureProvider.add-def"><a name="L2138"></a><tt class="py-lineno">2138</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.add-toggle" onclick="return toggle('CustomStructureProvider.add');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#add">add</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">path</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.add-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.add-expanded"><a name="L2139"></a><tt class="py-lineno">2139</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2140"></a><tt class="py-lineno">2140</tt>  <tt class="py-line"><tt class="py-docstring">        Register a new local C{id}:C{path} pair.</tt> </tt>
<a name="L2141"></a><tt class="py-lineno">2141</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2142"></a><tt class="py-lineno">2142</tt>  <tt class="py-line"><tt class="py-docstring">        @param id: structure identifier</tt> </tt>
<a name="L2143"></a><tt class="py-lineno">2143</tt>  <tt class="py-line"><tt class="py-docstring">        @type id: str        </tt> </tt>
<a name="L2144"></a><tt class="py-lineno">2144</tt>  <tt class="py-line"><tt class="py-docstring">        @param path: path and file name</tt> </tt>
<a name="L2145"></a><tt class="py-lineno">2145</tt>  <tt class="py-line"><tt class="py-docstring">        @type path: str</tt> </tt>
<a name="L2146"></a><tt class="py-lineno">2146</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2147"></a><tt class="py-lineno">2147</tt>  <tt class="py-line"><tt class="py-docstring">        @raise IOError: if C{path} is not a valid file name</tt> </tt>
<a name="L2148"></a><tt class="py-lineno">2148</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2149"></a><tt class="py-lineno">2149</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-1270" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1270', 'path', 'link-1201');">path</a></tt><tt class="py-op">.</tt><tt class="py-name">isfile</tt><tt class="py-op">(</tt><tt id="link-1271" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1271', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2150"></a><tt class="py-lineno">2150</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">[</tt><tt id="link-1272" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1272', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-1273" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1273', 'path', 'link-1201');">path</a></tt> </tt>
<a name="L2151"></a><tt class="py-lineno">2151</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2152"></a><tt class="py-lineno">2152</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IOError</tt><tt class="py-op">(</tt><tt id="link-1274" class="py-name"><a title="csb.build.RevisionHandler.path" class="py-name" href="#" onclick="return doclink('link-1274', 'path', 'link-1201');">path</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2153"></a><tt class="py-lineno">2153</tt>  <tt class="py-line">         </tt>
<a name="CustomStructureProvider.remove"></a><div id="CustomStructureProvider.remove-def"><a name="L2154"></a><tt class="py-lineno">2154</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.remove-toggle" onclick="return toggle('CustomStructureProvider.remove');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#remove">remove</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.remove-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.remove-expanded"><a name="L2155"></a><tt class="py-lineno">2155</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2156"></a><tt class="py-lineno">2156</tt>  <tt class="py-line"><tt class="py-docstring">        Unregister an existing structure C{id}.</tt> </tt>
<a name="L2157"></a><tt class="py-lineno">2157</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2158"></a><tt class="py-lineno">2158</tt>  <tt class="py-line"><tt class="py-docstring">        @param id: structure identifier</tt> </tt>
<a name="L2159"></a><tt class="py-lineno">2159</tt>  <tt class="py-line"><tt class="py-docstring">        @type id: str</tt> </tt>
<a name="L2160"></a><tt class="py-lineno">2160</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2161"></a><tt class="py-lineno">2161</tt>  <tt class="py-line"><tt class="py-docstring">        @raise ValueError: if C{id} had not been registered</tt> </tt>
<a name="L2162"></a><tt class="py-lineno">2162</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2163"></a><tt class="py-lineno">2163</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-1275" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1275', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">:</tt> </tt>
<a name="L2164"></a><tt class="py-lineno">2164</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt id="link-1276" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1276', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2165"></a><tt class="py-lineno">2165</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2166"></a><tt class="py-lineno">2166</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">[</tt><tt id="link-1277" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1277', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L2167"></a><tt class="py-lineno">2167</tt>  <tt class="py-line">         </tt>
<a name="CustomStructureProvider.find"></a><div id="CustomStructureProvider.find-def"><a name="L2168"></a><tt class="py-lineno">2168</tt> <a class="py-toggle" href="#" id="CustomStructureProvider.find-toggle" onclick="return toggle('CustomStructureProvider.find');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.CustomStructureProvider-class.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CustomStructureProvider.find-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CustomStructureProvider.find-expanded"><a name="L2169"></a><tt class="py-lineno">2169</tt>  <tt class="py-line">         </tt>
<a name="L2170"></a><tt class="py-lineno">2170</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-1278" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1278', 'id', 'link-29');">id</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">:</tt> </tt>
<a name="L2171"></a><tt class="py-lineno">2171</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_files</tt><tt class="py-op">[</tt><tt id="link-1279" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1279', 'id', 'link-29');">id</a></tt><tt class="py-op">]</tt> </tt>
<a name="L2172"></a><tt class="py-lineno">2172</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2173"></a><tt class="py-lineno">2173</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div></div><a name="L2174"></a><tt class="py-lineno">2174</tt>  <tt class="py-line">     </tt>
<a name="get"></a><div id="get-def"><a name="L2175"></a><tt class="py-lineno">2175</tt> <a class="py-toggle" href="#" id="get-toggle" onclick="return toggle('get');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb-module.html#get">get</a><tt class="py-op">(</tt><tt class="py-param">accession</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">prefix</tt><tt class="py-op">=</tt><tt class="py-string">'https://files.rcsb.org/download/'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="get-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="get-expanded"><a name="L2176"></a><tt class="py-lineno">2176</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L2177"></a><tt class="py-lineno">2177</tt>  <tt class="py-line"><tt class="py-docstring">    Download and parse a PDB entry.</tt> </tt>
<a name="L2178"></a><tt class="py-lineno">2178</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2179"></a><tt class="py-lineno">2179</tt>  <tt class="py-line"><tt class="py-docstring">    @param accession: accession number of the entry</tt> </tt>
<a name="L2180"></a><tt class="py-lineno">2180</tt>  <tt class="py-line"><tt class="py-docstring">    @type accession: str</tt> </tt>
<a name="L2181"></a><tt class="py-lineno">2181</tt>  <tt class="py-line"><tt class="py-docstring">    @param model: model identifier</tt> </tt>
<a name="L2182"></a><tt class="py-lineno">2182</tt>  <tt class="py-line"><tt class="py-docstring">    @type model: str</tt> </tt>
<a name="L2183"></a><tt class="py-lineno">2183</tt>  <tt class="py-line"><tt class="py-docstring">    @param prefix: download URL prefix</tt> </tt>
<a name="L2184"></a><tt class="py-lineno">2184</tt>  <tt class="py-line"><tt class="py-docstring">    @type prefix: str</tt> </tt>
<a name="L2185"></a><tt class="py-lineno">2185</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L2186"></a><tt class="py-lineno">2186</tt>  <tt class="py-line"><tt class="py-docstring">    @return: object representation of the selected model</tt> </tt>
<a name="L2187"></a><tt class="py-lineno">2187</tt>  <tt class="py-line"><tt class="py-docstring">    @rtype: L{Structure}</tt> </tt>
<a name="L2188"></a><tt class="py-lineno">2188</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2189"></a><tt class="py-lineno">2189</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-1280" class="py-name" targets="Class csb.bio.io.wwpdb.RemoteStructureProvider=csb.bio.io.wwpdb.RemoteStructureProvider-class.html"><a title="csb.bio.io.wwpdb.RemoteStructureProvider" class="py-name" href="#" onclick="return doclink('link-1280', 'RemoteStructureProvider', 'link-1280');">RemoteStructureProvider</a></tt><tt class="py-op">(</tt><tt id="link-1281" class="py-name"><a title="csb.bio.io.wwpdb.RemoteStructureProvider.prefix" class="py-name" href="#" onclick="return doclink('link-1281', 'prefix', 'link-1224');">prefix</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1282" class="py-name"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-1282', 'get', 'link-1182');">get</a></tt><tt class="py-op">(</tt><tt id="link-1283" class="py-name"><a title="csb.bio.fragments.Target.accession
csb.bio.fragments.rosetta.RosettaFragment.accession
csb.bio.io.wwpdb.EntryID.accession
csb.bio.structure.Chain.accession
csb.bio.structure.Structure.accession" class="py-name" href="#" onclick="return doclink('link-1283', 'accession', 'link-212');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-1284" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1284', 'model', 'link-153');">model</a></tt><tt class="py-op">=</tt><tt id="link-1285" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1285', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2190"></a><tt class="py-lineno">2190</tt>  <tt class="py-line"> </tt>
<a name="find"></a><div id="find-def"><a name="L2191"></a><tt class="py-lineno">2191</tt> <a class="py-toggle" href="#" id="find-toggle" onclick="return toggle('find');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb-module.html#find">find</a><tt class="py-op">(</tt><tt class="py-param">id</tt><tt class="py-op">,</tt> <tt class="py-param">paths</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="find-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="find-expanded"><a name="L2192"></a><tt class="py-lineno">2192</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L2193"></a><tt class="py-lineno">2193</tt>  <tt class="py-line"><tt class="py-docstring">    Try to discover a PDB file for PDB C{id} in C{paths}.</tt> </tt>
<a name="L2194"></a><tt class="py-lineno">2194</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2195"></a><tt class="py-lineno">2195</tt>  <tt class="py-line"><tt class="py-docstring">    @param id: PDB ID of the entry</tt> </tt>
<a name="L2196"></a><tt class="py-lineno">2196</tt>  <tt class="py-line"><tt class="py-docstring">    @type id: str</tt> </tt>
<a name="L2197"></a><tt class="py-lineno">2197</tt>  <tt class="py-line"><tt class="py-docstring">    @param paths: a list of directories to scan</tt> </tt>
<a name="L2198"></a><tt class="py-lineno">2198</tt>  <tt class="py-line"><tt class="py-docstring">    @type paths: list of str</tt> </tt>
<a name="L2199"></a><tt class="py-lineno">2199</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2200"></a><tt class="py-lineno">2200</tt>  <tt class="py-line"><tt class="py-docstring">    @return: path and file name on success, None otherwise</tt> </tt>
<a name="L2201"></a><tt class="py-lineno">2201</tt>  <tt class="py-line"><tt class="py-docstring">    @rtype: str</tt> </tt>
<a name="L2202"></a><tt class="py-lineno">2202</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2203"></a><tt class="py-lineno">2203</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-1286" class="py-name" targets="Class csb.bio.io.wwpdb.FileSystemStructureProvider=csb.bio.io.wwpdb.FileSystemStructureProvider-class.html"><a title="csb.bio.io.wwpdb.FileSystemStructureProvider" class="py-name" href="#" onclick="return doclink('link-1286', 'FileSystemStructureProvider', 'link-1286');">FileSystemStructureProvider</a></tt><tt class="py-op">(</tt><tt id="link-1287" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.paths
csb.bio.io.wwpdb.FileSystemStructureProvider.paths" class="py-name" href="#" onclick="return doclink('link-1287', 'paths', 'link-1195');">paths</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-1288" class="py-name"><a title="csb.bio.io.wwpdb.CustomStructureProvider.find
csb.bio.io.wwpdb.FileSystemStructureProvider.find
csb.bio.io.wwpdb.RemoteStructureProvider.find
csb.bio.io.wwpdb.StructureProvider.find
csb.bio.io.wwpdb.find
csb.bio.structure.Chain.find
csb.bio.structure.DisorderedAtom.find" class="py-name" href="#" onclick="return doclink('link-1288', 'find', 'link-500');">find</a></tt><tt class="py-op">(</tt><tt id="link-1289" class="py-name"><a title="csb.bio.fragments.FragmentMatch.id
csb.bio.fragments.Target.id
csb.bio.fragments.rosetta.RosettaFragment.id
csb.bio.hmm.HHpredHit.id
csb.bio.hmm.ProfileHMM.id
csb.bio.io.wwpdb.ResidueInfo.id
csb.bio.sequence.AbstractSequence.id
csb.bio.sequence.SequenceAdapter.id
csb.bio.structure.Chain.id
csb.bio.structure.Residue.id" class="py-name" href="#" onclick="return doclink('link-1289', 'id', 'link-29');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2204"></a><tt class="py-lineno">2204</tt>  <tt class="py-line">     </tt>
<a name="AsyncParseResult"></a><div id="AsyncParseResult-def"><a name="L2205"></a><tt class="py-lineno">2205</tt>  <tt class="py-line">     </tt>
<a name="L2206"></a><tt class="py-lineno">2206</tt> <a class="py-toggle" href="#" id="AsyncParseResult-toggle" onclick="return toggle('AsyncParseResult');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncParseResult-class.html">AsyncParseResult</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncParseResult-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AsyncParseResult-expanded"><a name="L2207"></a><tt class="py-lineno">2207</tt>  <tt class="py-line">     </tt>
<a name="AsyncParseResult.__init__"></a><div id="AsyncParseResult.__init__-def"><a name="L2208"></a><tt class="py-lineno">2208</tt> <a class="py-toggle" href="#" id="AsyncParseResult.__init__-toggle" onclick="return toggle('AsyncParseResult.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncParseResult-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">result</tt><tt class="py-op">,</tt> <tt class="py-param">exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncParseResult.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncParseResult.__init__-expanded"><a name="L2209"></a><tt class="py-lineno">2209</tt>  <tt class="py-line">         </tt>
<a name="L2210"></a><tt class="py-lineno">2210</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1290" class="py-name" targets="Variable csb.apps.hhsearch.Context.result=csb.apps.hhsearch.Context-class.html#result,Variable csb.core.InterruptibleThread.result=csb.core.InterruptibleThread-class.html#result"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1290', 'result', 'link-1290');">result</a></tt> <tt class="py-op">=</tt> <tt id="link-1291" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1291', 'result', 'link-1290');">result</a></tt> </tt>
<a name="L2211"></a><tt class="py-lineno">2211</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">exception</tt> <tt class="py-op">=</tt> <tt class="py-name">exception</tt> </tt>
</div><a name="L2212"></a><tt class="py-lineno">2212</tt>  <tt class="py-line">     </tt>
<a name="AsyncParseResult.__repr__"></a><div id="AsyncParseResult.__repr__-def"><a name="L2213"></a><tt class="py-lineno">2213</tt> <a class="py-toggle" href="#" id="AsyncParseResult.__repr__-toggle" onclick="return toggle('AsyncParseResult.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncParseResult-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncParseResult.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncParseResult.__repr__-expanded"><a name="L2214"></a><tt class="py-lineno">2214</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'&lt;AsyncParseResult: result={0.result}, error={0.exception.__class__.__name__}&gt;'</tt><tt class="py-op">.</tt><tt id="link-1292" class="py-name"><a title="csb.bio.io.wwpdb.EntryID.format
csb.bio.io.wwpdb.SeqResID.format
csb.bio.sequence.A3MAlignment.format
csb.bio.sequence.AbstractAlignment.format" class="py-name" href="#" onclick="return doclink('link-1292', 'format', 'link-44');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L2215"></a><tt class="py-lineno">2215</tt>  <tt class="py-line"> </tt>
<a name="_parse_async"></a><div id="_parse_async-def"><a name="L2216"></a><tt class="py-lineno">2216</tt>  <tt class="py-line"> </tt>
<a name="L2217"></a><tt class="py-lineno">2217</tt> <a class="py-toggle" href="#" id="_parse_async-toggle" onclick="return toggle('_parse_async');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb-module.html#_parse_async">_parse_async</a><tt class="py-op">(</tt><tt class="py-param">parser</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_parse_async-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_parse_async-expanded"><a name="L2218"></a><tt class="py-lineno">2218</tt>  <tt class="py-line">    <tt id="link-1293" class="py-name" targets="Variable csb.statistics.pdf.GeneralizedInverseGaussian.p=csb.statistics.pdf.GeneralizedInverseGaussian-class.html#p"><a title="csb.statistics.pdf.GeneralizedInverseGaussian.p" class="py-name" href="#" onclick="return doclink('link-1293', 'p', 'link-1293');">p</a></tt> <tt class="py-op">=</tt> <tt id="link-1294" class="py-name" targets="Variable csb.apps.ArgHandler.parser=csb.apps.ArgHandler-class.html#parser,Variable csb.apps.hhsearch.HHsearch.parser=csb.apps.hhsearch.HHsearch-class.html#parser"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-1294', 'parser', 'link-1294');">parser</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">)</tt> </tt>
<a name="L2219"></a><tt class="py-lineno">2219</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-1295" class="py-name"><a title="csb.statistics.pdf.GeneralizedInverseGaussian.p" class="py-name" href="#" onclick="return doclink('link-1295', 'p', 'link-1293');">p</a></tt><tt class="py-op">.</tt><tt id="link-1296" class="py-name"><a title="csb.bio.io.wwpdb.AbstractStructureParser.parse_structure
csb.bio.io.wwpdb.AsyncStructureParser.parse_structure" class="py-name" href="#" onclick="return doclink('link-1296', 'parse_structure', 'link-152');">parse_structure</a></tt><tt class="py-op">(</tt><tt id="link-1297" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1297', 'model', 'link-153');">model</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L2220"></a><tt class="py-lineno">2220</tt>  <tt class="py-line"> </tt>
<a name="AsyncStructureParser"></a><div id="AsyncStructureParser-def"><a name="L2221"></a><tt class="py-lineno">2221</tt>  <tt class="py-line"> </tt>
<a name="L2222"></a><tt class="py-lineno">2222</tt> <a class="py-toggle" href="#" id="AsyncStructureParser-toggle" onclick="return toggle('AsyncStructureParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncStructureParser-class.html">AsyncStructureParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncStructureParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AsyncStructureParser-expanded"><a name="L2223"></a><tt class="py-lineno">2223</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L2224"></a><tt class="py-lineno">2224</tt>  <tt class="py-line"><tt class="py-docstring">    Wraps StructureParser in an asynchronous call. Since a new process is</tt> </tt>
<a name="L2225"></a><tt class="py-lineno">2225</tt>  <tt class="py-line"><tt class="py-docstring">    started by Python internally (as opposed to only starting a new thread),</tt> </tt>
<a name="L2226"></a><tt class="py-lineno">2226</tt>  <tt class="py-line"><tt class="py-docstring">    this makes the parser slower, but provides a way to set a parse timeout</tt> </tt>
<a name="L2227"></a><tt class="py-lineno">2227</tt>  <tt class="py-line"><tt class="py-docstring">    limit.</tt> </tt>
<a name="L2228"></a><tt class="py-lineno">2228</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2229"></a><tt class="py-lineno">2229</tt>  <tt class="py-line"><tt class="py-docstring">    If initialized with more than one worker, supports parallel parsing</tt> </tt>
<a name="L2230"></a><tt class="py-lineno">2230</tt>  <tt class="py-line"><tt class="py-docstring">    through the C{self.parse_async} method. </tt> </tt>
<a name="L2231"></a><tt class="py-lineno">2231</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L2232"></a><tt class="py-lineno">2232</tt>  <tt class="py-line"><tt class="py-docstring">    @param workers: number of worker threads (1 by default)</tt> </tt>
<a name="L2233"></a><tt class="py-lineno">2233</tt>  <tt class="py-line"><tt class="py-docstring">    @type workers: int</tt> </tt>
<a name="L2234"></a><tt class="py-lineno">2234</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L2235"></a><tt class="py-lineno">2235</tt>  <tt class="py-line"> </tt>
<a name="AsyncStructureParser.__init__"></a><div id="AsyncStructureParser.__init__-def"><a name="L2236"></a><tt class="py-lineno">2236</tt> <a class="py-toggle" href="#" id="AsyncStructureParser.__init__-toggle" onclick="return toggle('AsyncStructureParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncStructureParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">workers</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncStructureParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncStructureParser.__init__-expanded"><a name="L2237"></a><tt class="py-lineno">2237</tt>  <tt class="py-line"> </tt>
<a name="L2238"></a><tt class="py-lineno">2238</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pool</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L2239"></a><tt class="py-lineno">2239</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_workers</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L2240"></a><tt class="py-lineno">2240</tt>  <tt class="py-line">         </tt>
<a name="L2241"></a><tt class="py-lineno">2241</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">workers</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L2242"></a><tt class="py-lineno">2242</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_workers</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">workers</tt><tt class="py-op">)</tt> </tt>
<a name="L2243"></a><tt class="py-lineno">2243</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2244"></a><tt class="py-lineno">2244</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-name">workers</tt><tt class="py-op">)</tt> </tt>
<a name="L2245"></a><tt class="py-lineno">2245</tt>  <tt class="py-line">         </tt>
<a name="L2246"></a><tt class="py-lineno">2246</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_recycle</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2247"></a><tt class="py-lineno">2247</tt>  <tt class="py-line">         </tt>
<a name="AsyncStructureParser._recycle"></a><div id="AsyncStructureParser._recycle-def"><a name="L2248"></a><tt class="py-lineno">2248</tt> <a class="py-toggle" href="#" id="AsyncStructureParser._recycle-toggle" onclick="return toggle('AsyncStructureParser._recycle');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncStructureParser-class.html#_recycle">_recycle</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncStructureParser._recycle-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncStructureParser._recycle-expanded"><a name="L2249"></a><tt class="py-lineno">2249</tt>  <tt class="py-line"> </tt>
<a name="L2250"></a><tt class="py-lineno">2250</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pool</tt><tt class="py-op">:</tt> </tt>
<a name="L2251"></a><tt class="py-lineno">2251</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pool</tt><tt class="py-op">.</tt><tt class="py-name">terminate</tt><tt class="py-op">(</tt><tt class="py-op">)</tt>          </tt>
<a name="L2252"></a><tt class="py-lineno">2252</tt>  <tt class="py-line">                </tt>
<a name="L2253"></a><tt class="py-lineno">2253</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pool</tt> <tt class="py-op">=</tt> <tt class="py-name">multiprocessing</tt><tt class="py-op">.</tt><tt class="py-name">Pool</tt><tt class="py-op">(</tt><tt class="py-name">processes</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_workers</tt><tt class="py-op">)</tt> </tt>
</div><a name="L2254"></a><tt class="py-lineno">2254</tt>  <tt class="py-line"> </tt>
<a name="AsyncStructureParser.parse_structure"></a><div id="AsyncStructureParser.parse_structure-def"><a name="L2255"></a><tt class="py-lineno">2255</tt> <a class="py-toggle" href="#" id="AsyncStructureParser.parse_structure-toggle" onclick="return toggle('AsyncStructureParser.parse_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncStructureParser-class.html#parse_structure">parse_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure_file</tt><tt class="py-op">,</tt> <tt class="py-param">timeout</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L2256"></a><tt class="py-lineno">2256</tt>  <tt class="py-line">                        <tt class="py-param">parser</tt><tt class="py-op">=</tt><tt id="link-1298" class="py-name"><a title="csb.bio.io.wwpdb.RegularStructureParser" class="py-name" href="#" onclick="return doclink('link-1298', 'RegularStructureParser', 'link-67');">RegularStructureParser</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncStructureParser.parse_structure-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncStructureParser.parse_structure-expanded"><a name="L2257"></a><tt class="py-lineno">2257</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2258"></a><tt class="py-lineno">2258</tt>  <tt class="py-line"><tt class="py-docstring">        Call StructureParser.parse_structure() in a separate process and return</tt> </tt>
<a name="L2259"></a><tt class="py-lineno">2259</tt>  <tt class="py-line"><tt class="py-docstring">        the output. Raise TimeoutError if the parser does not respond within</tt> </tt>
<a name="L2260"></a><tt class="py-lineno">2260</tt>  <tt class="py-line"><tt class="py-docstring">        C{timeout} seconds.</tt> </tt>
<a name="L2261"></a><tt class="py-lineno">2261</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2262"></a><tt class="py-lineno">2262</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure_file: structure file to parse</tt> </tt>
<a name="L2263"></a><tt class="py-lineno">2263</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure_file: str</tt> </tt>
<a name="L2264"></a><tt class="py-lineno">2264</tt>  <tt class="py-line"><tt class="py-docstring">        @param timeout: raise multiprocessing.TimeoutError if C{timeout} seconds</tt> </tt>
<a name="L2265"></a><tt class="py-lineno">2265</tt>  <tt class="py-line"><tt class="py-docstring">                        elapse before the parser completes its job</tt> </tt>
<a name="L2266"></a><tt class="py-lineno">2266</tt>  <tt class="py-line"><tt class="py-docstring">        @type timeout: int</tt> </tt>
<a name="L2267"></a><tt class="py-lineno">2267</tt>  <tt class="py-line"><tt class="py-docstring">        @param parser: any implementing L{AbstractStructureParser} class</tt> </tt>
<a name="L2268"></a><tt class="py-lineno">2268</tt>  <tt class="py-line"><tt class="py-docstring">        @type parser: type  </tt> </tt>
<a name="L2269"></a><tt class="py-lineno">2269</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2270"></a><tt class="py-lineno">2270</tt>  <tt class="py-line"><tt class="py-docstring">        @return: parsed structure</tt> </tt>
<a name="L2271"></a><tt class="py-lineno">2271</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{csb.structure.Structure}    </tt> </tt>
<a name="L2272"></a><tt class="py-lineno">2272</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2273"></a><tt class="py-lineno">2273</tt>  <tt class="py-line"> </tt>
<a name="L2274"></a><tt class="py-lineno">2274</tt>  <tt class="py-line">        <tt id="link-1299" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1299', 'r', 'link-939');">r</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1300" class="py-name" targets="Method csb.bio.io.wwpdb.AsyncStructureParser.parse_async()=csb.bio.io.wwpdb.AsyncStructureParser-class.html#parse_async"><a title="csb.bio.io.wwpdb.AsyncStructureParser.parse_async" class="py-name" href="#" onclick="return doclink('link-1300', 'parse_async', 'link-1300');">parse_async</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">structure_file</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">timeout</tt><tt class="py-op">,</tt> <tt id="link-1301" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1301', 'model', 'link-153');">model</a></tt><tt class="py-op">,</tt> <tt id="link-1302" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-1302', 'parser', 'link-1294');">parser</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2275"></a><tt class="py-lineno">2275</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-1303" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1303', 'r', 'link-939');">r</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L2276"></a><tt class="py-lineno">2276</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-1304" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1304', 'r', 'link-939');">r</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">exception</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L2277"></a><tt class="py-lineno">2277</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-1305" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1305', 'r', 'link-939');">r</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">exception</tt> </tt>
<a name="L2278"></a><tt class="py-lineno">2278</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L2279"></a><tt class="py-lineno">2279</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt id="link-1306" class="py-name"><a title="csb.bio.io.clans.Color.r" class="py-name" href="#" onclick="return doclink('link-1306', 'r', 'link-939');">r</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-1307" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1307', 'result', 'link-1290');">result</a></tt> </tt>
<a name="L2280"></a><tt class="py-lineno">2280</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L2281"></a><tt class="py-lineno">2281</tt>  <tt class="py-line">     </tt>
<a name="AsyncStructureParser.parse_async"></a><div id="AsyncStructureParser.parse_async-def"><a name="L2282"></a><tt class="py-lineno">2282</tt> <a class="py-toggle" href="#" id="AsyncStructureParser.parse_async-toggle" onclick="return toggle('AsyncStructureParser.parse_async');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="csb.bio.io.wwpdb.AsyncStructureParser-class.html#parse_async">parse_async</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure_files</tt><tt class="py-op">,</tt> <tt class="py-param">timeout</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L2283"></a><tt class="py-lineno">2283</tt>  <tt class="py-line">                        <tt class="py-param">parser</tt><tt class="py-op">=</tt><tt id="link-1308" class="py-name"><a title="csb.bio.io.wwpdb.RegularStructureParser" class="py-name" href="#" onclick="return doclink('link-1308', 'RegularStructureParser', 'link-67');">RegularStructureParser</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AsyncStructureParser.parse_async-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AsyncStructureParser.parse_async-expanded"><a name="L2284"></a><tt class="py-lineno">2284</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L2285"></a><tt class="py-lineno">2285</tt>  <tt class="py-line"><tt class="py-docstring">        Call C{self.parse_structure} for a list of structure files</tt> </tt>
<a name="L2286"></a><tt class="py-lineno">2286</tt>  <tt class="py-line"><tt class="py-docstring">        simultaneously. The actual degree of parallelism will depend on the</tt> </tt>
<a name="L2287"></a><tt class="py-lineno">2287</tt>  <tt class="py-line"><tt class="py-docstring">        number of workers specified while constructing the parser object.</tt> </tt>
<a name="L2288"></a><tt class="py-lineno">2288</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2289"></a><tt class="py-lineno">2289</tt>  <tt class="py-line"><tt class="py-docstring">        @note: Don't be tempted to pass a large list of structures to this </tt> </tt>
<a name="L2290"></a><tt class="py-lineno">2290</tt>  <tt class="py-line"><tt class="py-docstring">               method. Every time a C{TimeoutError} is encountered, the </tt> </tt>
<a name="L2291"></a><tt class="py-lineno">2291</tt>  <tt class="py-line"><tt class="py-docstring">               corresponding worker process in the pool will hang until the</tt> </tt>
<a name="L2292"></a><tt class="py-lineno">2292</tt>  <tt class="py-line"><tt class="py-docstring">               process terminates on its own. During that time, this worker is</tt> </tt>
<a name="L2293"></a><tt class="py-lineno">2293</tt>  <tt class="py-line"><tt class="py-docstring">               unusable. If a sufficiently high number of timeouts occur, the </tt> </tt>
<a name="L2294"></a><tt class="py-lineno">2294</tt>  <tt class="py-line"><tt class="py-docstring">               whole pool of workers will be unsable. At the end of the method</tt> </tt>
<a name="L2295"></a><tt class="py-lineno">2295</tt>  <tt class="py-line"><tt class="py-docstring">               however a pool cleanup is performed and any unusable workers</tt> </tt>
<a name="L2296"></a><tt class="py-lineno">2296</tt>  <tt class="py-line"><tt class="py-docstring">               are 'reactivated'. However, that only happens at B{the end} of</tt> </tt>
<a name="L2297"></a><tt class="py-lineno">2297</tt>  <tt class="py-line"><tt class="py-docstring">               C{parse_async}.</tt> </tt>
<a name="L2298"></a><tt class="py-lineno">2298</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2299"></a><tt class="py-lineno">2299</tt>  <tt class="py-line"><tt class="py-docstring">        @param structure_files: a list of structure files</tt> </tt>
<a name="L2300"></a><tt class="py-lineno">2300</tt>  <tt class="py-line"><tt class="py-docstring">        @type structure_files: tuple of str</tt> </tt>
<a name="L2301"></a><tt class="py-lineno">2301</tt>  <tt class="py-line"><tt class="py-docstring">        @param timeout: raise multiprocessing.TimeoutError if C{timeout} seconds</tt> </tt>
<a name="L2302"></a><tt class="py-lineno">2302</tt>  <tt class="py-line"><tt class="py-docstring">                        elapse before the parser completes its job</tt> </tt>
<a name="L2303"></a><tt class="py-lineno">2303</tt>  <tt class="py-line"><tt class="py-docstring">        @type timeout: int</tt> </tt>
<a name="L2304"></a><tt class="py-lineno">2304</tt>  <tt class="py-line"><tt class="py-docstring">        @param parser: any implementing L{AbstractStructureParser} class</tt> </tt>
<a name="L2305"></a><tt class="py-lineno">2305</tt>  <tt class="py-line"><tt class="py-docstring">        @type parser: type</tt> </tt>
<a name="L2306"></a><tt class="py-lineno">2306</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L2307"></a><tt class="py-lineno">2307</tt>  <tt class="py-line"><tt class="py-docstring">        @return: a list of L{AsyncParseResult} objects</tt> </tt>
<a name="L2308"></a><tt class="py-lineno">2308</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: list     </tt> </tt>
<a name="L2309"></a><tt class="py-lineno">2309</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L2310"></a><tt class="py-lineno">2310</tt>  <tt class="py-line"> </tt>
<a name="L2311"></a><tt class="py-lineno">2311</tt>  <tt class="py-line">        <tt class="py-name">pool</tt> <tt class="py-op">=</tt>  <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_pool</tt> </tt>
<a name="L2312"></a><tt class="py-lineno">2312</tt>  <tt class="py-line">        <tt class="py-name">workers</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L2313"></a><tt class="py-lineno">2313</tt>  <tt class="py-line">        <tt class="py-name">results</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L2314"></a><tt class="py-lineno">2314</tt>  <tt class="py-line">         </tt>
<a name="L2315"></a><tt class="py-lineno">2315</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">file</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">structure_files</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2316"></a><tt class="py-lineno">2316</tt>  <tt class="py-line">            <tt id="link-1309" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1309', 'result', 'link-1290');">result</a></tt> <tt class="py-op">=</tt> <tt class="py-name">pool</tt><tt class="py-op">.</tt><tt class="py-name">apply_async</tt><tt class="py-op">(</tt><tt class="py-name">_parse_async</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt id="link-1310" class="py-name"><a title="csb.apps.ArgHandler.parser
csb.apps.hhsearch.HHsearch.parser" class="py-name" href="#" onclick="return doclink('link-1310', 'parser', 'link-1294');">parser</a></tt><tt class="py-op">,</tt> <tt class="py-name">file</tt><tt class="py-op">,</tt> <tt id="link-1311" class="py-name"><a title="csb.statistics.maxent.MaxentPosterior.model" class="py-name" href="#" onclick="return doclink('link-1311', 'model', 'link-153');">model</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L2317"></a><tt class="py-lineno">2317</tt>  <tt class="py-line">            <tt class="py-name">workers</tt><tt class="py-op">.</tt><tt id="link-1312" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1312', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-1313" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1313', 'result', 'link-1290');">result</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2318"></a><tt class="py-lineno">2318</tt>  <tt class="py-line">         </tt>
<a name="L2319"></a><tt class="py-lineno">2319</tt>  <tt class="py-line">        <tt class="py-name">hanging</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L2320"></a><tt class="py-lineno">2320</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-1314" class="py-name" targets="Variable csb.statistics.mixtures.GaussianMixture.w=csb.statistics.mixtures.GaussianMixture-class.html#w"><a title="csb.statistics.mixtures.GaussianMixture.w" class="py-name" href="#" onclick="return doclink('link-1314', 'w', 'link-1314');">w</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">workers</tt><tt class="py-op">:</tt> </tt>
<a name="L2321"></a><tt class="py-lineno">2321</tt>  <tt class="py-line">            <tt id="link-1315" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1315', 'result', 'link-1290');">result</a></tt> <tt class="py-op">=</tt> <tt id="link-1316" class="py-name" targets="Class csb.bio.io.wwpdb.AsyncParseResult=csb.bio.io.wwpdb.AsyncParseResult-class.html"><a title="csb.bio.io.wwpdb.AsyncParseResult" class="py-name" href="#" onclick="return doclink('link-1316', 'AsyncParseResult', 'link-1316');">AsyncParseResult</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L2322"></a><tt class="py-lineno">2322</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L2323"></a><tt class="py-lineno">2323</tt>  <tt class="py-line">                <tt id="link-1317" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1317', 'result', 'link-1290');">result</a></tt><tt class="py-op">.</tt><tt id="link-1318" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1318', 'result', 'link-1290');">result</a></tt> <tt class="py-op">=</tt> <tt id="link-1319" class="py-name"><a title="csb.statistics.mixtures.GaussianMixture.w" class="py-name" href="#" onclick="return doclink('link-1319', 'w', 'link-1314');">w</a></tt><tt class="py-op">.</tt><tt id="link-1320" class="py-name"><a title="csb.bio.io.dssp.get
csb.bio.io.wwpdb.RemoteStructureProvider.get
csb.bio.io.wwpdb.StructureProvider.get
csb.bio.io.wwpdb.get
csb.bio.nmr.AtomConnectivity.get
csb.bio.nmr.NOEPeak.get
csb.bio.nmr.RandomCoil.get
csb.bio.sequence.SequenceAlphabets.get
csb.build.BuildTypes.get
csb.io.plots.Backend.get
csb.statistics.maxent.MaxentModel.get" class="py-name" href="#" onclick="return doclink('link-1320', 'get', 'link-1182');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">timeout</tt><tt class="py-op">=</tt><tt class="py-name">timeout</tt><tt class="py-op">)</tt> </tt>
<a name="L2324"></a><tt class="py-lineno">2324</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyboardInterrupt</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ki</tt><tt class="py-op">:</tt> </tt>
<a name="L2325"></a><tt class="py-lineno">2325</tt>  <tt class="py-line">                <tt class="py-name">pool</tt><tt class="py-op">.</tt><tt class="py-name">terminate</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2326"></a><tt class="py-lineno">2326</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ki</tt> </tt>
<a name="L2327"></a><tt class="py-lineno">2327</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">Exception</tt> <tt class="py-keyword">as</tt> <tt class="py-name">ex</tt><tt class="py-op">:</tt> </tt>
<a name="L2328"></a><tt class="py-lineno">2328</tt>  <tt class="py-line">                <tt id="link-1321" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1321', 'result', 'link-1290');">result</a></tt><tt class="py-op">.</tt><tt class="py-name">exception</tt> <tt class="py-op">=</tt> <tt class="py-name">ex</tt> </tt>
<a name="L2329"></a><tt class="py-lineno">2329</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">ex</tt><tt class="py-op">,</tt> <tt class="py-name">multiprocessing</tt><tt class="py-op">.</tt><tt id="link-1322" class="py-name" targets="Class csb.io.TimeoutError=csb.io.TimeoutError-class.html"><a title="csb.io.TimeoutError" class="py-name" href="#" onclick="return doclink('link-1322', 'TimeoutError', 'link-1322');">TimeoutError</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L2330"></a><tt class="py-lineno">2330</tt>  <tt class="py-line">                    <tt class="py-name">hanging</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt>                     </tt>
<a name="L2331"></a><tt class="py-lineno">2331</tt>  <tt class="py-line">            <tt class="py-name">results</tt><tt class="py-op">.</tt><tt id="link-1323" class="py-name"><a title="csb.bio.fragments.rosetta.RosettaFragmentMap.append
csb.bio.hmm.EmissionTable.append
csb.bio.hmm.HMMLayer.append
csb.bio.hmm.TransitionTable.append
csb.bio.structure.ChainResiduesCollection.append
csb.bio.structure.DisorderedAtom.append
csb.bio.structure.EnsembleModelsCollection.append
csb.bio.structure.ResidueAtomsTable.append
csb.bio.structure.SecondaryStructure.append
csb.bio.structure.StructureChainsTable.append
csb.core.CollectionContainer.append
csb.core.DictionaryContainer.append" class="py-name" href="#" onclick="return doclink('link-1323', 'append', 'link-96');">append</a></tt><tt class="py-op">(</tt><tt id="link-1324" class="py-name"><a title="csb.apps.hhsearch.Context.result
csb.core.InterruptibleThread.result" class="py-name" href="#" onclick="return doclink('link-1324', 'result', 'link-1290');">result</a></tt><tt class="py-op">)</tt> </tt>
<a name="L2332"></a><tt class="py-lineno">2332</tt>  <tt class="py-line">         </tt>
<a name="L2333"></a><tt class="py-lineno">2333</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hanging</tt><tt class="py-op">:</tt> </tt>
<a name="L2334"></a><tt class="py-lineno">2334</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_recycle</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L2335"></a><tt class="py-lineno">2335</tt>  <tt class="py-line">             </tt>
<a name="L2336"></a><tt class="py-lineno">2336</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">results</tt> </tt>
</div></div><a name="L2337"></a><tt class="py-lineno">2337</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="csb-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            >CSB v1.2.5.d7fd331</th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul  4 20:19:09 2017
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
